--- a/optifine/OptiFineClassTransformer.java
+++ b/optifine/OptiFineClassTransformer.java
@@ -75,8 +75,19 @@
 
     public byte[] transform(String name, String transformedName, byte[] bytes)
     {
-        byte[] ofBytes = this.getOptiFineClass(name);
-        return ofBytes != null ? ofBytes : bytes;
+        if (name.length() > 3)
+        {
+            return bytes;
+        }
+        else if (!name.toLowerCase().equals(name))
+        {
+            return bytes;
+        }
+        else
+        {
+            byte[] ofBytes = this.getOptiFineClass(name);
+            return ofBytes != null ? ofBytes : bytes;
+        }
     }
 
     private byte[] getOptiFineClass(String name)
