--- a/net/minecraft/client/shader/Framebuffer.java
+++ b/net/minecraft/client/shader/Framebuffer.java
@@ -1,10 +1,18 @@
 package net.minecraft.client.shader;
 
 import java.nio.ByteBuffer;
+
 import net.minecraft.client.renderer.OpenGlHelper;
 import net.minecraft.client.renderer.Tessellator;
 import net.minecraft.client.renderer.texture.TextureUtil;
+import net.minecraft.src.Reflector;
+import org.lwjgl.opengl.ARBFramebufferObject;
+import org.lwjgl.opengl.ContextCapabilities;
+import org.lwjgl.opengl.EXTFramebufferObject;
 import org.lwjgl.opengl.GL11;
+import org.lwjgl.opengl.GL30;
+import org.lwjgl.opengl.GL32;
+import org.lwjgl.opengl.GLContext;
 
 public class Framebuffer
 {
@@ -20,8 +28,26 @@
     public int framebufferFilter;
     private static final String __OBFID = "CL_00000959";
 
-    public Framebuffer(int p_i45078_1_, int p_i45078_2_, boolean p_i45078_3_)
+    /** Minecrift **/
+    public String name;
+    public boolean genMipMaps = false;
+    public boolean multiSample = false;
+    public int multiSampleCount = 0;
+    public int textureType = GL11.GL_TEXTURE_2D;
+
+    public Framebuffer(/*String name,*/ int p_i45078_1_, int p_i45078_2_, boolean p_i45078_3_)
+    {
+        this(p_i45078_1_, p_i45078_2_, p_i45078_3_, false, false, 0);
+    }
+
+    public Framebuffer(/*String name,*/ int p_i45078_1_, int p_i45078_2_, boolean p_i45078_3_, boolean generateMipMaps)
     {
+        this(p_i45078_1_, p_i45078_2_, p_i45078_3_, generateMipMaps, false, 0);
+    }
+
+    public Framebuffer(/*String name,*/ int p_i45078_1_, int p_i45078_2_, boolean p_i45078_3_, boolean generateMipMaps, boolean multisample, int multisamplecount)
+    {
+        //this.name = name;
         this.useDepth = p_i45078_3_;
         this.framebufferObject = -1;
         this.framebufferTexture = -1;
@@ -31,6 +57,15 @@
         this.framebufferColor[1] = 1.0F;
         this.framebufferColor[2] = 1.0F;
         this.framebufferColor[3] = 0.0F;
+
+        /** Minecrift **/
+        this.genMipMaps = generateMipMaps;
+        this.multiSample = multisample;
+        if (this.multiSample) {
+            this.multiSampleCount = multisamplecount;
+            this.textureType = GL32.GL_TEXTURE_2D_MULTISAMPLE;
+        }
+
         this.createBindFramebuffer(p_i45078_1_, p_i45078_2_);
     }
 
@@ -51,7 +86,6 @@
             }
 
             this.createFramebuffer(p_147613_1_, p_147613_2_);
-            this.checkFramebufferComplete();
             OpenGlHelper.func_153171_g(OpenGlHelper.field_153198_e, 0);
         }
     }
@@ -97,43 +131,72 @@
         }
         else
         {
-            this.framebufferObject = OpenGlHelper.func_153165_e();
-            this.framebufferTexture = TextureUtil.glGenTextures();
+            this.framebufferObject = OpenGlHelper.func_153165_e();      // GL30.glGenFramebuffers()
+            this.framebufferTexture = TextureUtil.glGenTextures();      // GL11.glGenTextures()
 
             if (this.useDepth)
             {
-                this.depthBuffer = OpenGlHelper.func_153185_f();
+                this.depthBuffer = OpenGlHelper.func_153185_f();        // GL30.glGenRenderbuffers()
             }
 
-            this.setFramebufferFilter(9728);
-            GL11.glBindTexture(GL11.GL_TEXTURE_2D, this.framebufferTexture);
-            GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, GL11.GL_RGBA8, this.framebufferTextureWidth, this.framebufferTextureHeight, 0, GL11.GL_RGBA, GL11.GL_UNSIGNED_BYTE, (ByteBuffer)null);
-            OpenGlHelper.func_153171_g(OpenGlHelper.field_153198_e, this.framebufferObject);
-            OpenGlHelper.func_153188_a(OpenGlHelper.field_153198_e, OpenGlHelper.field_153200_g, 3553, this.framebufferTexture, 0);
+            this.setFramebufferFilter(9728);        // GL11.GL_NEAREST
+            GL11.glBindTexture(this.textureType, this.framebufferTexture);
+            if (!this.multiSample) {
+                GL11.glTexImage2D(this.textureType, 0, GL11.GL_RGBA8, this.framebufferTextureWidth, this.framebufferTextureHeight, 0, GL11.GL_RGBA, GL11.GL_UNSIGNED_BYTE, (ByteBuffer) null);
+            }
+            else {
+                // TODO: Check GLContext capabilities
+                GL32.glTexImage2DMultisample(this.textureType, this.multiSampleCount, GL11.GL_RGBA8, this.framebufferTextureWidth, this.framebufferTextureHeight, false);
+            }
+
+            if (this.genMipMaps) {
+                // Allocate mip map storage
+                genMipMaps();                // TODO: Check GLContext capabilities
+            }
+
+            OpenGlHelper.func_153171_g(OpenGlHelper.field_153198_e, this.framebufferObject);   // GL30.glBindFramebuffer
+            OpenGlHelper.func_153188_a(OpenGlHelper.field_153198_e, OpenGlHelper.field_153200_g, this.textureType, this.framebufferTexture, 0);    // GL30.glFramebufferTexture2D
 
             if (this.useDepth)
             {
-                OpenGlHelper.func_153176_h(OpenGlHelper.field_153199_f, this.depthBuffer);
-                OpenGlHelper.func_153186_a(OpenGlHelper.field_153199_f, 33190, this.framebufferTextureWidth, this.framebufferTextureHeight);
+                OpenGlHelper.func_153176_h(OpenGlHelper.field_153199_f, this.depthBuffer);   // GL30.glBindRenderbuffer
+                if (!Reflector.MinecraftForgeClient_getStencilBits.exists() || Reflector.callInt(Reflector.MinecraftForgeClient_getStencilBits, new Object[0]) == 0)
+                {
+                func_153186_a(OpenGlHelper.field_153199_f, 33190, this.framebufferTextureWidth, this.framebufferTextureHeight); // GL30.glRenderbufferStorage
                 OpenGlHelper.func_153190_b(OpenGlHelper.field_153198_e, OpenGlHelper.field_153201_h, OpenGlHelper.field_153199_f, this.depthBuffer);
+                }
+                else
+                {
+                    func_153186_a(OpenGlHelper.field_153199_f, org.lwjgl.opengl.EXTPackedDepthStencil.GL_DEPTH24_STENCIL8_EXT, this.framebufferTextureWidth, this.framebufferTextureHeight); // GL30.glRenderbufferStorage
+                    OpenGlHelper.func_153190_b(OpenGlHelper.field_153198_e, org.lwjgl.opengl.EXTFramebufferObject.GL_DEPTH_ATTACHMENT_EXT, OpenGlHelper.field_153199_f, this.depthBuffer); // GL30.glFramebufferRenderbuffer
+                    OpenGlHelper.func_153190_b(OpenGlHelper.field_153198_e, org.lwjgl.opengl.EXTFramebufferObject.GL_STENCIL_ATTACHMENT_EXT, OpenGlHelper.field_153199_f, this.depthBuffer);  // GL30.glFramebufferRenderbuffer
+                }
             }
 
+            this.checkFramebufferComplete();
             this.framebufferClear();
             this.unbindFramebufferTexture();
         }
     }
 
+    public void genMipMaps()
+    {
+        GL30.glGenerateMipmap(this.textureType);    // TODO: Minecrift - Check GLContext capabilities
+    }
+
     public void setFramebufferFilter(int p_147607_1_)
     {
         if (OpenGlHelper.isFramebufferEnabled())
         {
             this.framebufferFilter = p_147607_1_;
-            GL11.glBindTexture(GL11.GL_TEXTURE_2D, this.framebufferTexture);
-            GL11.glTexParameterf(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MIN_FILTER, (float)p_147607_1_);
-            GL11.glTexParameterf(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MAG_FILTER, (float)p_147607_1_);
-            GL11.glTexParameterf(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_WRAP_S, 10496.0F);
-            GL11.glTexParameterf(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_WRAP_T, 10496.0F);
-            GL11.glBindTexture(GL11.GL_TEXTURE_2D, 0);
+            GL11.glBindTexture(this.textureType, this.framebufferTexture);
+            if (!this.multiSample) {
+                GL11.glTexParameterf(this.textureType, GL11.GL_TEXTURE_MIN_FILTER, (float) p_147607_1_);
+                GL11.glTexParameterf(this.textureType, GL11.GL_TEXTURE_MAG_FILTER, (float) p_147607_1_);
+                GL11.glTexParameterf(this.textureType, GL11.GL_TEXTURE_WRAP_S, 10496.0F);
+                GL11.glTexParameterf(this.textureType, GL11.GL_TEXTURE_WRAP_T, 10496.0F);
+            }
+            GL11.glBindTexture(this.textureType, 0);
         }
     }
 
@@ -159,6 +222,10 @@
             {
                 throw new RuntimeException("GL_FRAMEBUFFER_INCOMPLETE_READ_BUFFER");
             }
+            else if (var1 == GL30.GL_FRAMEBUFFER_INCOMPLETE_MULTISAMPLE) // TODO: Minecrift - need to handle older OpenGL versions etc
+            {
+                throw new RuntimeException("GL_FRAMEBUFFER_INCOMPLETE_MULTISAMPLE");
+            }
             else
             {
                 throw new RuntimeException("glCheckFramebufferStatus returned unknown status:" + var1);
@@ -168,22 +235,26 @@
 
     public void bindFramebufferTexture()
     {
+        //System.out.println("Binding texture of: " + name);
         if (OpenGlHelper.isFramebufferEnabled())
         {
-            GL11.glBindTexture(GL11.GL_TEXTURE_2D, this.framebufferTexture);
+            GL11.glBindTexture(this.textureType, this.framebufferTexture);
         }
     }
 
     public void unbindFramebufferTexture()
     {
+        //System.out.println("Unbinding texture of: " + name);
         if (OpenGlHelper.isFramebufferEnabled())
         {
-            GL11.glBindTexture(GL11.GL_TEXTURE_2D, 0);
+            GL11.glBindTexture(this.textureType, 0);
         }
     }
 
     public void bindFramebuffer(boolean p_147610_1_)
     {
+        //Minecraft.getMinecraft().lastFB = name;
+        //System.out.println("Binding framebuffer of: " + name);
         if (OpenGlHelper.isFramebufferEnabled())
         {
             OpenGlHelper.func_153171_g(OpenGlHelper.field_153198_e, this.framebufferObject);
@@ -197,6 +268,8 @@
 
     public void unbindFramebuffer()
     {
+        //Minecraft.getMinecraft().lastFB = "main";
+        //System.out.println("Unbinding framebuffer " + name);
         if (OpenGlHelper.isFramebufferEnabled())
         {
             OpenGlHelper.func_153171_g(OpenGlHelper.field_153198_e, 0);
@@ -213,6 +286,12 @@
 
     public void framebufferRender(int p_147615_1_, int p_147615_2_)
     {
+		/** Minecrift **/
+        framebufferRender(0, p_147615_1_, p_147615_2_, 0);
+    }
+
+    public void framebufferRender(int left, int p_147615_1_, int p_147615_2_, int top)
+    {
         if (OpenGlHelper.isFramebufferEnabled())
         {
             GL11.glColorMask(true, true, true, false);
@@ -224,8 +303,8 @@
             GL11.glMatrixMode(GL11.GL_MODELVIEW);
             GL11.glLoadIdentity();
             GL11.glTranslatef(0.0F, 0.0F, -2000.0F);
-            GL11.glViewport(0, 0, p_147615_1_, p_147615_2_);
-            GL11.glEnable(GL11.GL_TEXTURE_2D);
+            GL11.glViewport(left, top, p_147615_1_, p_147615_2_);
+            GL11.glEnable(this.textureType);
             GL11.glDisable(GL11.GL_LIGHTING);
             GL11.glDisable(GL11.GL_ALPHA_TEST);
             GL11.glDisable(GL11.GL_BLEND);
@@ -243,12 +322,14 @@
             var7.addVertexWithUV((double)var3, (double)var4, 0.0D, (double)var5, 0.0D);
             var7.addVertexWithUV((double)var3, 0.0D, 0.0D, (double)var5, (double)var6);
             var7.addVertexWithUV(0.0D, 0.0D, 0.0D, 0.0D, (double)var6);
+            //System.out.println("framebufferRender of " + name + " -> " + Minecraft.getMinecraft().lastFB);
             var7.draw();
             this.unbindFramebufferTexture();
             GL11.glDepthMask(true);
             GL11.glColorMask(true, true, true, true);
         }
     }
+	/** end Minecrift **/
 
     public void framebufferClear()
     {
@@ -265,4 +346,42 @@
         GL11.glClear(var1);
         this.unbindFramebuffer();
     }
+
+    public void func_153186_a(int p_153186_0_, int p_153186_1_, int p_153186_2_, int p_153186_3_)
+    {
+        if (OpenGlHelper.framebufferSupported)
+        {
+            ContextCapabilities contextcapabilities = GLContext.getCapabilities();
+            int callType = 0;
+            if (contextcapabilities.OpenGL30)
+            {
+                callType = 0;
+            }
+            else if (contextcapabilities.GL_ARB_framebuffer_object)
+            {
+                callType = 1;
+            }
+            else if (contextcapabilities.GL_EXT_framebuffer_object)
+            {
+                callType = 2;
+            }
+
+            switch (callType)
+            {
+                case 0:
+                    if (!this.multiSample) {
+                        GL30.glRenderbufferStorage(p_153186_0_, p_153186_1_, p_153186_2_, p_153186_3_);
+                    }
+                    else {
+                        GL30.glRenderbufferStorageMultisample(p_153186_0_, this.multiSampleCount, p_153186_1_, p_153186_2_, p_153186_3_);
+                    }
+                    break;
+                case 1:
+                    ARBFramebufferObject.glRenderbufferStorage(p_153186_0_, p_153186_1_, p_153186_2_, p_153186_3_);
+                    break;
+                case 2:
+                    EXTFramebufferObject.glRenderbufferStorageEXT(p_153186_0_, p_153186_1_, p_153186_2_, p_153186_3_);
+            }
+        }
+    }
 }
