--- a/net/minecraft/src/AbstractClientPlayer.java
+++ b/net/minecraft/src/AbstractClientPlayer.java
@@ -24,7 +24,6 @@
             this.locationCape = getLocationCape(this.username);
             this.downloadImageSkin = getDownloadImageSkin(this.locationSkin, this.username);
             this.downloadImageCape = getDownloadImageCape(this.locationCape, this.username);
-            this.downloadImageCape.enabled = Config.isShowCapes();
         }
     }
 
