--- a/net/minecraft/src/AbstractResourcePack.java
+++ b/net/minecraft/src/AbstractResourcePack.java
@@ -13,49 +13,49 @@
 
 public abstract class AbstractResourcePack implements ResourcePack
 {
-    protected static final ILogAgent resourceLog = Minecraft.getMinecraft().getLogAgent();
-    protected final File resourcePackFile;
+    protected static final ILogAgent field_110598_a = Minecraft.getMinecraft().getLogAgent();
+    protected final File field_110597_b;
 
     public AbstractResourcePack(File par1File)
     {
-        this.resourcePackFile = par1File;
+        this.field_110597_b = par1File;
     }
 
-    private static String locationToName(ResourceLocation par0ResourceLocation)
+    private static String func_110592_c(ResourceLocation par0ResourceLocation)
     {
-        return String.format("%s/%s/%s", new Object[] {"assets", par0ResourceLocation.getResourceDomain(), par0ResourceLocation.getResourcePath()});
+        return String.format("%s/%s/%s", new Object[] {"assets", par0ResourceLocation.func_110624_b(), par0ResourceLocation.func_110623_a()});
     }
 
-    protected static String getRelativeName(File par0File, File par1File)
+    protected static String func_110595_a(File par0File, File par1File)
     {
         return par0File.toURI().relativize(par1File.toURI()).getPath();
     }
 
-    public InputStream getInputStream(ResourceLocation par1ResourceLocation) throws IOException
+    public InputStream func_110590_a(ResourceLocation par1ResourceLocation) throws IOException
     {
-        return this.getInputStreamByName(locationToName(par1ResourceLocation));
+        return this.func_110591_a(func_110592_c(par1ResourceLocation));
     }
 
-    public boolean resourceExists(ResourceLocation par1ResourceLocation)
+    public boolean func_110589_b(ResourceLocation par1ResourceLocation)
     {
-        return this.hasResourceName(locationToName(par1ResourceLocation));
+        return this.func_110593_b(func_110592_c(par1ResourceLocation));
     }
 
-    protected abstract InputStream getInputStreamByName(String var1) throws IOException;
+    protected abstract InputStream func_110591_a(String var1) throws IOException;
 
-    protected abstract boolean hasResourceName(String var1);
+    protected abstract boolean func_110593_b(String var1);
 
-    protected void logNameNotLowercase(String par1Str)
+    protected void func_110594_c(String par1Str)
     {
-        resourceLog.logWarningFormatted("ResourcePack: ignored non-lowercase namespace: %s in %s", new Object[] {par1Str, this.resourcePackFile});
+        field_110598_a.logWarningFormatted("ResourcePack: ignored non-lowercase namespace: %s in %s", new Object[] {par1Str, this.field_110597_b});
     }
 
-    public MetadataSection getPackMetadata(MetadataSerializer par1MetadataSerializer, String par2Str) throws IOException
+    public MetadataSection func_135058_a(MetadataSerializer par1MetadataSerializer, String par2Str) throws IOException
     {
-        return readMetadata(par1MetadataSerializer, this.getInputStreamByName("pack.mcmeta"), par2Str);
+        return func_110596_a(par1MetadataSerializer, this.func_110591_a("pack.mcmeta"), par2Str);
     }
 
-    static MetadataSection readMetadata(MetadataSerializer par0MetadataSerializer, InputStream par1InputStream, String par2Str)
+    static MetadataSection func_110596_a(MetadataSerializer par0MetadataSerializer, InputStream par1InputStream, String par2Str)
     {
         JsonObject var3 = null;
         BufferedReader var4 = null;
@@ -70,16 +70,16 @@
             IOUtils.closeQuietly(var4);
         }
 
-        return par0MetadataSerializer.parseMetadataSection(par2Str, var3);
+        return par0MetadataSerializer.func_110503_a(par2Str, var3);
     }
 
-    public BufferedImage getPackImage() throws IOException
+    public BufferedImage func_110586_a() throws IOException
     {
-        return ImageIO.read(this.getInputStreamByName("pack.png"));
+        return ImageIO.read(this.func_110591_a("pack.png"));
     }
 
-    public String getPackName()
+    public String func_130077_b()
     {
-        return this.resourcePackFile.getName();
+        return this.field_110597_b.getName();
     }
 }
