--- a/net/minecraft/src/AttributeInstance.java
+++ b/net/minecraft/src/AttributeInstance.java
@@ -7,22 +7,19 @@
 {
     Attribute func_111123_a();
 
-    double getBaseValue();
+    double func_111125_b();
 
-    void setAttribute(double var1);
+    void func_111128_a(double var1);
 
     Collection func_111122_c();
 
-    /**
-     * Returns attribute modifier, if any, by the given UUID
-     */
-    AttributeModifier getModifier(UUID var1);
+    AttributeModifier func_111127_a(UUID var1);
 
-    void applyModifier(AttributeModifier var1);
+    void func_111121_a(AttributeModifier var1);
 
-    void removeModifier(AttributeModifier var1);
+    void func_111124_b(AttributeModifier var1);
 
     void func_142049_d();
 
-    double getAttributeValue();
+    double func_111126_e();
 }
