--- a/net/minecraft/src/AttributeModifier.java
+++ b/net/minecraft/src/AttributeModifier.java
@@ -5,15 +5,11 @@
 
 public class AttributeModifier
 {
-    private final double amount;
-    private final int operation;
-    private final String name;
-    private final UUID id;
-
-    /**
-     * If false, this modifier is not saved in NBT. Used for "natural" modifiers like speed boost from sprinting
-     */
-    private boolean isSaved;
+    private final double field_111174_a;
+    private final int field_111172_b;
+    private final String field_111173_c;
+    private final UUID field_111170_d;
+    private boolean field_111171_e;
 
     public AttributeModifier(String par1Str, double par2, int par4)
     {
@@ -22,49 +18,43 @@
 
     public AttributeModifier(UUID par1UUID, String par2Str, double par3, int par5)
     {
-        this.isSaved = true;
-        this.id = par1UUID;
-        this.name = par2Str;
-        this.amount = par3;
-        this.operation = par5;
+        this.field_111171_e = true;
+        this.field_111170_d = par1UUID;
+        this.field_111173_c = par2Str;
+        this.field_111174_a = par3;
+        this.field_111172_b = par5;
         Validate.notEmpty(par2Str, "Modifier name cannot be empty", new Object[0]);
         Validate.inclusiveBetween(Integer.valueOf(0), Integer.valueOf(2), Integer.valueOf(par5), "Invalid operation", new Object[0]);
     }
 
-    public UUID getID()
+    public UUID func_111167_a()
     {
-        return this.id;
+        return this.field_111170_d;
     }
 
-    public String getName()
+    public String func_111166_b()
     {
-        return this.name;
+        return this.field_111173_c;
     }
 
-    public int getOperation()
+    public int func_111169_c()
     {
-        return this.operation;
+        return this.field_111172_b;
     }
 
-    public double getAmount()
+    public double func_111164_d()
     {
-        return this.amount;
+        return this.field_111174_a;
     }
 
-    /**
-     * @see #isSaved
-     */
-    public boolean isSaved()
+    public boolean func_111165_e()
     {
-        return this.isSaved;
+        return this.field_111171_e;
     }
 
-    /**
-     * @see #isSaved
-     */
-    public AttributeModifier setSaved(boolean par1)
+    public AttributeModifier func_111168_a(boolean par1)
     {
-        this.isSaved = par1;
+        this.field_111171_e = par1;
         return this;
     }
 
@@ -78,14 +68,14 @@
         {
             AttributeModifier var2 = (AttributeModifier)par1Obj;
 
-            if (this.id != null)
+            if (this.field_111170_d != null)
             {
-                if (!this.id.equals(var2.id))
+                if (!this.field_111170_d.equals(var2.field_111170_d))
                 {
                     return false;
                 }
             }
-            else if (var2.id != null)
+            else if (var2.field_111170_d != null)
             {
                 return false;
             }
@@ -100,11 +90,11 @@
 
     public int hashCode()
     {
-        return this.id != null ? this.id.hashCode() : 0;
+        return this.field_111170_d != null ? this.field_111170_d.hashCode() : 0;
     }
 
     public String toString()
     {
-        return "AttributeModifier{amount=" + this.amount + ", operation=" + this.operation + ", name=\'" + this.name + '\'' + ", id=" + this.id + ", serialize=" + this.isSaved + '}';
+        return "AttributeModifier{amount=" + this.field_111174_a + ", operation=" + this.field_111172_b + ", name=\'" + this.field_111173_c + '\'' + ", id=" + this.field_111170_d + ", serialize=" + this.field_111171_e + '}';
     }
 }
