--- a/net/minecraft/src/BanList.java
+++ b/net/minecraft/src/BanList.java
@@ -145,7 +145,7 @@
 
             if (par1)
             {
-                var2.println("# Updated " + (new SimpleDateFormat()).format(new Date()) + " by Minecraft " + "1.6.4");
+                var2.println("# Updated " + (new SimpleDateFormat()).format(new Date()) + " by Minecraft " + "1.6.2");
                 var2.println("# victim name | ban date | banned by | banned until | reason");
                 var2.println();
             }
