--- a/net/minecraft/src/BiomeCache.java
+++ b/net/minecraft/src/BiomeCache.java
@@ -42,7 +42,7 @@
             this.cache.add(var5);
         }
 
-        var5.lastAccessTime = MinecraftServer.getSystemTimeMillis();
+        var5.lastAccessTime = MinecraftServer.func_130071_aq();
         return var5;
     }
 
@@ -59,7 +59,7 @@
      */
     public void cleanupCache()
     {
-        long var1 = MinecraftServer.getSystemTimeMillis();
+        long var1 = MinecraftServer.func_130071_aq();
         long var3 = var1 - this.lastCleanupTime;
 
         if (var3 > 7500L || var3 < 0L)
