--- a/net/minecraft/src/BlockStairs.java
+++ b/net/minecraft/src/BlockStairs.java
@@ -87,11 +87,7 @@
         return par0 > 0 && Block.blocksList[par0] instanceof BlockStairs;
     }
 
-    /**
-     * returns true if the given block is a stairs block and is in the given direction of par5.  Parameters are
-     * IBlockAccess, x, y, z, direction
-     */
-    private boolean isBlockStairsDirection(IBlockAccess par1IBlockAccess, int par2, int par3, int par4, int par5)
+    private boolean func_82540_f(IBlockAccess par1IBlockAccess, int par2, int par3, int par4, int par5)
     {
         int var6 = par1IBlockAccess.getBlockId(par2, par3, par4);
         return isBlockStairsID(var6) && par1IBlockAccess.getBlockMetadata(par2, par3, par4) == par5;
@@ -130,12 +126,12 @@
             {
                 var16 = var15 & 3;
 
-                if (var16 == 3 && !this.isBlockStairsDirection(par1IBlockAccess, par2, par3, par4 + 1, var5))
+                if (var16 == 3 && !this.func_82540_f(par1IBlockAccess, par2, par3, par4 + 1, var5))
                 {
                     var12 = 0.5F;
                     var13 = false;
                 }
-                else if (var16 == 2 && !this.isBlockStairsDirection(par1IBlockAccess, par2, par3, par4 - 1, var5))
+                else if (var16 == 2 && !this.func_82540_f(par1IBlockAccess, par2, par3, par4 - 1, var5))
                 {
                     var11 = 0.5F;
                     var13 = false;
@@ -153,12 +149,12 @@
             {
                 var16 = var15 & 3;
 
-                if (var16 == 3 && !this.isBlockStairsDirection(par1IBlockAccess, par2, par3, par4 + 1, var5))
+                if (var16 == 3 && !this.func_82540_f(par1IBlockAccess, par2, par3, par4 + 1, var5))
                 {
                     var12 = 0.5F;
                     var13 = false;
                 }
-                else if (var16 == 2 && !this.isBlockStairsDirection(par1IBlockAccess, par2, par3, par4 - 1, var5))
+                else if (var16 == 2 && !this.func_82540_f(par1IBlockAccess, par2, par3, par4 - 1, var5))
                 {
                     var11 = 0.5F;
                     var13 = false;
@@ -176,12 +172,12 @@
             {
                 var16 = var15 & 3;
 
-                if (var16 == 1 && !this.isBlockStairsDirection(par1IBlockAccess, par2 + 1, par3, par4, var5))
+                if (var16 == 1 && !this.func_82540_f(par1IBlockAccess, par2 + 1, par3, par4, var5))
                 {
                     var10 = 0.5F;
                     var13 = false;
                 }
-                else if (var16 == 0 && !this.isBlockStairsDirection(par1IBlockAccess, par2 - 1, par3, par4, var5))
+                else if (var16 == 0 && !this.func_82540_f(par1IBlockAccess, par2 - 1, par3, par4, var5))
                 {
                     var9 = 0.5F;
                     var13 = false;
@@ -197,12 +193,12 @@
             {
                 var16 = var15 & 3;
 
-                if (var16 == 1 && !this.isBlockStairsDirection(par1IBlockAccess, par2 + 1, par3, par4, var5))
+                if (var16 == 1 && !this.func_82540_f(par1IBlockAccess, par2 + 1, par3, par4, var5))
                 {
                     var10 = 0.5F;
                     var13 = false;
                 }
-                else if (var16 == 0 && !this.isBlockStairsDirection(par1IBlockAccess, par2 - 1, par3, par4, var5))
+                else if (var16 == 0 && !this.func_82540_f(par1IBlockAccess, par2 - 1, par3, par4, var5))
                 {
                     var9 = 0.5F;
                     var13 = false;
@@ -245,13 +241,13 @@
             {
                 var16 = var15 & 3;
 
-                if (var16 == 3 && !this.isBlockStairsDirection(par1IBlockAccess, par2, par3, par4 - 1, var5))
+                if (var16 == 3 && !this.func_82540_f(par1IBlockAccess, par2, par3, par4 - 1, var5))
                 {
                     var11 = 0.0F;
                     var12 = 0.5F;
                     var13 = true;
                 }
-                else if (var16 == 2 && !this.isBlockStairsDirection(par1IBlockAccess, par2, par3, par4 + 1, var5))
+                else if (var16 == 2 && !this.func_82540_f(par1IBlockAccess, par2, par3, par4 + 1, var5))
                 {
                     var11 = 0.5F;
                     var12 = 1.0F;
@@ -270,13 +266,13 @@
                 var10 = 1.0F;
                 var16 = var15 & 3;
 
-                if (var16 == 3 && !this.isBlockStairsDirection(par1IBlockAccess, par2, par3, par4 - 1, var5))
+                if (var16 == 3 && !this.func_82540_f(par1IBlockAccess, par2, par3, par4 - 1, var5))
                 {
                     var11 = 0.0F;
                     var12 = 0.5F;
                     var13 = true;
                 }
-                else if (var16 == 2 && !this.isBlockStairsDirection(par1IBlockAccess, par2, par3, par4 + 1, var5))
+                else if (var16 == 2 && !this.func_82540_f(par1IBlockAccess, par2, par3, par4 + 1, var5))
                 {
                     var11 = 0.5F;
                     var12 = 1.0F;
@@ -295,11 +291,11 @@
                 var12 = 0.5F;
                 var16 = var15 & 3;
 
-                if (var16 == 1 && !this.isBlockStairsDirection(par1IBlockAccess, par2 - 1, par3, par4, var5))
+                if (var16 == 1 && !this.func_82540_f(par1IBlockAccess, par2 - 1, par3, par4, var5))
                 {
                     var13 = true;
                 }
-                else if (var16 == 0 && !this.isBlockStairsDirection(par1IBlockAccess, par2 + 1, par3, par4, var5))
+                else if (var16 == 0 && !this.func_82540_f(par1IBlockAccess, par2 + 1, par3, par4, var5))
                 {
                     var9 = 0.5F;
                     var10 = 1.0F;
@@ -316,11 +312,11 @@
             {
                 var16 = var15 & 3;
 
-                if (var16 == 1 && !this.isBlockStairsDirection(par1IBlockAccess, par2 - 1, par3, par4, var5))
+                if (var16 == 1 && !this.func_82540_f(par1IBlockAccess, par2 - 1, par3, par4, var5))
                 {
                     var13 = true;
                 }
-                else if (var16 == 0 && !this.isBlockStairsDirection(par1IBlockAccess, par2 + 1, par3, par4, var5))
+                else if (var16 == 0 && !this.func_82540_f(par1IBlockAccess, par2 + 1, par3, par4, var5))
                 {
                     var9 = 0.5F;
                     var10 = 1.0F;
@@ -479,9 +475,7 @@
     }
 
     /**
-     * Called on server worlds only when the block has been replaced by a different block ID, or the same block with a
-     * different metadata value, but before the new metadata value is set. Args: World, x, y, z, old block ID, old
-     * metadata
+     * ejects contained items into the world, and notifies neighbours of an update, as appropriate
      */
     public void breakBlock(World par1World, int par2, int par3, int par4, int par5, int par6)
     {
