--- a/net/minecraft/src/BlockFluid.java
+++ b/net/minecraft/src/BlockFluid.java
@@ -618,7 +618,7 @@
         }
     }
 
-    public static Icon getFluidIcon(String par0Str)
+    public static Icon func_94424_b(String par0Str)
     {
         return par0Str == "water_still" ? Block.waterMoving.theIcon[0] : (par0Str == "water_flow" ? Block.waterMoving.theIcon[1] : (par0Str == "lava_still" ? Block.lavaMoving.theIcon[0] : (par0Str == "lava_flow" ? Block.lavaMoving.theIcon[1] : null)));
     }
