--- a/net/minecraft/src/BlockLog.java
+++ b/net/minecraft/src/BlockLog.java
@@ -33,9 +33,7 @@
     }
 
     /**
-     * Called on server worlds only when the block has been replaced by a different block ID, or the same block with a
-     * different metadata value, but before the new metadata value is set. Args: World, x, y, z, old block ID, old
-     * metadata
+     * ejects contained items into the world, and notifies neighbours of an update, as appropriate
      */
     public void breakBlock(World par1World, int par2, int par3, int par4, int par5, int par6)
     {
@@ -67,18 +65,12 @@
         }
     }
 
-    /**
-     * The icon for the side of the block.
-     */
-    protected Icon getSideIcon(int par1)
+    protected Icon func_111048_c(int par1)
     {
         return this.field_111052_c[par1];
     }
 
-    /**
-     * The icon for the tops and bottoms of the block.
-     */
-    protected Icon getEndIcon(int par1)
+    protected Icon func_111049_d(int par1)
     {
         return this.tree_top[par1];
     }
@@ -113,8 +105,8 @@
 
         for (int var2 = 0; var2 < this.field_111052_c.length; ++var2)
         {
-            this.field_111052_c[var2] = par1IconRegister.registerIcon(this.getTextureName() + "_" + woodType[var2]);
-            this.tree_top[var2] = par1IconRegister.registerIcon(this.getTextureName() + "_" + woodType[var2] + "_top");
+            this.field_111052_c[var2] = par1IconRegister.registerIcon(this.func_111023_E() + "_" + woodType[var2]);
+            this.tree_top[var2] = par1IconRegister.registerIcon(this.func_111023_E() + "_" + woodType[var2] + "_top");
         }
     }
 }
