--- a/net/minecraft/src/BlockRailPowered.java
+++ b/net/minecraft/src/BlockRailPowered.java
@@ -24,7 +24,7 @@
     public void registerIcons(IconRegister par1IconRegister)
     {
         super.registerIcons(par1IconRegister);
-        this.theIcon = par1IconRegister.registerIcon(this.getTextureName() + "_powered");
+        this.theIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_powered");
     }
 
     protected boolean func_94360_a(World par1World, int par2, int par3, int par4, int par5, boolean par6, int par7)
@@ -41,6 +41,7 @@
             switch (var8)
             {
                 case 0:
+
                     if (par6)
                     {
                         ++par4;
@@ -51,8 +52,8 @@
                     }
 
                     break;
-
                 case 1:
+
                     if (par6)
                     {
                         --par2;
@@ -63,8 +64,8 @@
                     }
 
                     break;
-
                 case 2:
+
                     if (par6)
                     {
                         --par2;
@@ -78,8 +79,8 @@
 
                     var8 = 1;
                     break;
-
                 case 3:
+
                     if (par6)
                     {
                         --par2;
@@ -93,8 +94,8 @@
 
                     var8 = 1;
                     break;
-
                 case 4:
+
                     if (par6)
                     {
                         ++par4;
@@ -108,8 +109,8 @@
 
                     var8 = 0;
                     break;
-
                 case 5:
+
                     if (par6)
                     {
                         ++par4;
