--- a/net/minecraft/src/BlockEndPortalFrame.java
+++ b/net/minecraft/src/BlockEndPortalFrame.java
@@ -27,9 +27,9 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.blockIcon = par1IconRegister.registerIcon(this.getTextureName() + "_side");
-        this.field_94400_a = par1IconRegister.registerIcon(this.getTextureName() + "_top");
-        this.field_94399_b = par1IconRegister.registerIcon(this.getTextureName() + "_eye");
+        this.blockIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_side");
+        this.field_94400_a = par1IconRegister.registerIcon(this.func_111023_E() + "_top");
+        this.field_94399_b = par1IconRegister.registerIcon(this.func_111023_E() + "_eye");
     }
 
     public Icon func_94398_p()
