--- a/net/minecraft/src/BlockLeaves.java
+++ b/net/minecraft/src/BlockLeaves.java
@@ -7,9 +7,7 @@
 {
     public static final String[] LEAF_TYPES = new String[] {"oak", "spruce", "birch", "jungle"};
     public static final String[][] field_94396_b = new String[][] {{"leaves_oak", "leaves_spruce", "leaves_birch", "leaves_jungle"}, {"leaves_oak_opaque", "leaves_spruce_opaque", "leaves_birch_opaque", "leaves_jungle_opaque"}};
-
-    /** 1 for fast graphic. 0 for fancy graphics. used in iconArray. */
-    private int iconType;
+    private int field_94394_cP;
     private Icon[][] iconArray = new Icon[2][];
     int[] adjacentTreeBlocks;
 
@@ -73,9 +71,7 @@
     }
 
     /**
-     * Called on server worlds only when the block has been replaced by a different block ID, or the same block with a
-     * different metadata value, but before the new metadata value is set. Args: World, x, y, z, old block ID, old
-     * metadata
+     * ejects contained items into the world, and notifies neighbours of an update, as appropriate
      */
     public void breakBlock(World par1World, int par2, int par3, int par4, int par5, int par6)
     {
@@ -341,7 +337,7 @@
      */
     public Icon getIcon(int par1, int par2)
     {
-        return (par2 & 3) == 1 ? this.iconArray[this.iconType][1] : ((par2 & 3) == 3 ? this.iconArray[this.iconType][3] : ((par2 & 3) == 2 ? this.iconArray[this.iconType][2] : this.iconArray[this.iconType][0]));
+        return (par2 & 3) == 1 ? this.iconArray[this.field_94394_cP][1] : ((par2 & 3) == 3 ? this.iconArray[this.field_94394_cP][3] : ((par2 & 3) == 2 ? this.iconArray[this.field_94394_cP][2] : this.iconArray[this.field_94394_cP][0]));
     }
 
     /**
@@ -350,7 +346,7 @@
     public void setGraphicsLevel(boolean par1)
     {
         this.graphicsLevel = par1;
-        this.iconType = par1 ? 0 : 1;
+        this.field_94394_cP = par1 ? 0 : 1;
     }
 
     /**
