--- a/net/minecraft/src/BlockWorkbench.java
+++ b/net/minecraft/src/BlockWorkbench.java
@@ -25,9 +25,9 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.blockIcon = par1IconRegister.registerIcon(this.getTextureName() + "_side");
-        this.workbenchIconTop = par1IconRegister.registerIcon(this.getTextureName() + "_top");
-        this.workbenchIconFront = par1IconRegister.registerIcon(this.getTextureName() + "_front");
+        this.blockIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_side");
+        this.workbenchIconTop = par1IconRegister.registerIcon(this.func_111023_E() + "_top");
+        this.workbenchIconFront = par1IconRegister.registerIcon(this.func_111023_E() + "_front");
     }
 
     /**
