--- a/net/minecraft/src/BlockMelon.java
+++ b/net/minecraft/src/BlockMelon.java
@@ -57,7 +57,7 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.blockIcon = par1IconRegister.registerIcon(this.getTextureName() + "_side");
-        this.theIcon = par1IconRegister.registerIcon(this.getTextureName() + "_top");
+        this.blockIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_side");
+        this.theIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_top");
     }
 }
