--- a/net/minecraft/src/BlockDetectorRail.java
+++ b/net/minecraft/src/BlockDetectorRail.java
@@ -164,8 +164,8 @@
     public void registerIcons(IconRegister par1IconRegister)
     {
         this.iconArray = new Icon[2];
-        this.iconArray[0] = par1IconRegister.registerIcon(this.getTextureName());
-        this.iconArray[1] = par1IconRegister.registerIcon(this.getTextureName() + "_powered");
+        this.iconArray[0] = par1IconRegister.registerIcon(this.func_111023_E());
+        this.iconArray[1] = par1IconRegister.registerIcon(this.func_111023_E() + "_powered");
     }
 
     /**
