--- a/net/minecraft/src/BlockBed.java
+++ b/net/minecraft/src/BlockBed.java
@@ -144,9 +144,9 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.bedTopIcons = new Icon[] {par1IconRegister.registerIcon(this.getTextureName() + "_feet_top"), par1IconRegister.registerIcon(this.getTextureName() + "_head_top")};
-        this.field_94472_b = new Icon[] {par1IconRegister.registerIcon(this.getTextureName() + "_feet_end"), par1IconRegister.registerIcon(this.getTextureName() + "_head_end")};
-        this.bedSideIcons = new Icon[] {par1IconRegister.registerIcon(this.getTextureName() + "_feet_side"), par1IconRegister.registerIcon(this.getTextureName() + "_head_side")};
+        this.bedTopIcons = new Icon[] {par1IconRegister.registerIcon(this.func_111023_E() + "_feet_top"), par1IconRegister.registerIcon(this.func_111023_E() + "_head_top")};
+        this.field_94472_b = new Icon[] {par1IconRegister.registerIcon(this.func_111023_E() + "_feet_end"), par1IconRegister.registerIcon(this.func_111023_E() + "_head_end")};
+        this.bedSideIcons = new Icon[] {par1IconRegister.registerIcon(this.func_111023_E() + "_feet_side"), par1IconRegister.registerIcon(this.func_111023_E() + "_head_side")};
     }
 
     /**
