--- a/net/minecraft/src/BlockQuartz.java
+++ b/net/minecraft/src/BlockQuartz.java
@@ -65,12 +65,10 @@
                 case 1:
                     par9 = 2;
                     break;
-
                 case 2:
                 case 3:
                     par9 = 4;
                     break;
-
                 case 4:
                 case 5:
                     par9 = 3;
@@ -131,13 +129,13 @@
             }
             else
             {
-                this.quartzblockIcons[var2] = par1IconRegister.registerIcon(this.getTextureName() + "_" + quartzBlockTextureTypes[var2]);
+                this.quartzblockIcons[var2] = par1IconRegister.registerIcon(this.func_111023_E() + "_" + quartzBlockTextureTypes[var2]);
             }
         }
 
-        this.quartzblock_top = par1IconRegister.registerIcon(this.getTextureName() + "_" + "top");
-        this.quartzblock_chiseled_top = par1IconRegister.registerIcon(this.getTextureName() + "_" + "chiseled_top");
-        this.quartzblock_lines_top = par1IconRegister.registerIcon(this.getTextureName() + "_" + "lines_top");
-        this.quartzblock_bottom = par1IconRegister.registerIcon(this.getTextureName() + "_" + "bottom");
+        this.quartzblock_top = par1IconRegister.registerIcon(this.func_111023_E() + "_" + "top");
+        this.quartzblock_chiseled_top = par1IconRegister.registerIcon(this.func_111023_E() + "_" + "chiseled_top");
+        this.quartzblock_lines_top = par1IconRegister.registerIcon(this.func_111023_E() + "_" + "lines_top");
+        this.quartzblock_bottom = par1IconRegister.registerIcon(this.func_111023_E() + "_" + "bottom");
     }
 }
