--- a/net/minecraft/src/BlockDaylightDetector.java
+++ b/net/minecraft/src/BlockDaylightDetector.java
@@ -130,7 +130,7 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.iconArray[0] = par1IconRegister.registerIcon(this.getTextureName() + "_top");
-        this.iconArray[1] = par1IconRegister.registerIcon(this.getTextureName() + "_side");
+        this.iconArray[0] = par1IconRegister.registerIcon(this.func_111023_E() + "_top");
+        this.iconArray[1] = par1IconRegister.registerIcon(this.func_111023_E() + "_side");
     }
 }
