--- a/net/minecraft/src/BlockFarmland.java
+++ b/net/minecraft/src/BlockFarmland.java
@@ -171,7 +171,7 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.field_94441_a = par1IconRegister.registerIcon(this.getTextureName() + "_wet");
-        this.field_94440_b = par1IconRegister.registerIcon(this.getTextureName() + "_dry");
+        this.field_94441_a = par1IconRegister.registerIcon(this.func_111023_E() + "_wet");
+        this.field_94440_b = par1IconRegister.registerIcon(this.func_111023_E() + "_dry");
     }
 }
