--- a/net/minecraft/src/BlockChest.java
+++ b/net/minecraft/src/BlockChest.java
@@ -7,13 +7,13 @@
 {
     private final Random random = new Random();
 
-    /** 1 for trapped chests, 0 for normal chests. */
-    public final int chestType;
+    /** Determines whether of not the chest is trapped. */
+    public final int isTrapped;
 
     protected BlockChest(int par1, int par2)
     {
         super(par1, Material.wood);
-        this.chestType = par2;
+        this.isTrapped = par2;
         this.setCreativeTab(CreativeTabs.tabDecorations);
         this.setBlockBounds(0.0625F, 0.0F, 0.0625F, 0.9375F, 0.875F, 0.9375F);
     }
@@ -342,9 +342,7 @@
     }
 
     /**
-     * Called on server worlds only when the block has been replaced by a different block ID, or the same block with a
-     * different metadata value, but before the new metadata value is set. Args: World, x, y, z, old block ID, old
-     * metadata
+     * ejects contained items into the world, and notifies neighbours of an update, as appropriate
      */
     public void breakBlock(World par1World, int par2, int par3, int par4, int par5, int par6)
     {
@@ -490,7 +488,7 @@
      */
     public boolean canProvidePower()
     {
-        return this.chestType == 1;
+        return this.isTrapped == 1;
     }
 
     /**
