--- a/net/minecraft/src/BlockFire.java
+++ b/net/minecraft/src/BlockFire.java
@@ -41,8 +41,8 @@
         this.setBurnRate(Block.tallGrass.blockID, 60, 100);
         this.setBurnRate(Block.cloth.blockID, 30, 60);
         this.setBurnRate(Block.vine.blockID, 15, 100);
-        this.setBurnRate(Block.coalBlock.blockID, 5, 5);
-        this.setBurnRate(Block.hay.blockID, 60, 20);
+        this.setBurnRate(Block.field_111034_cE.blockID, 5, 5);
+        this.setBurnRate(Block.field_111038_cB.blockID, 60, 20);
     }
 
     /**
@@ -437,10 +437,10 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.iconArray = new Icon[] {par1IconRegister.registerIcon(this.getTextureName() + "_layer_0"), par1IconRegister.registerIcon(this.getTextureName() + "_layer_1")};
+        this.iconArray = new Icon[] {par1IconRegister.registerIcon(this.func_111023_E() + "_layer_0"), par1IconRegister.registerIcon(this.func_111023_E() + "_layer_1")};
     }
 
-    public Icon getFireIcon(int par1)
+    public Icon func_94438_c(int par1)
     {
         return this.iconArray[par1];
     }
