--- a/net/minecraft/src/BlockColored.java
+++ b/net/minecraft/src/BlockColored.java
@@ -65,7 +65,7 @@
 
         for (int var2 = 0; var2 < this.iconArray.length; ++var2)
         {
-            this.iconArray[var2] = par1IconRegister.registerIcon(this.getTextureName() + "_" + ItemDye.dyeItemNames[getDyeFromBlock(var2)]);
+            this.iconArray[var2] = par1IconRegister.registerIcon(this.func_111023_E() + "_" + ItemDye.field_94595_b[getDyeFromBlock(var2)]);
         }
     }
 }
