--- a/net/minecraft/src/BlockCarrot.java
+++ b/net/minecraft/src/BlockCarrot.java
@@ -55,7 +55,7 @@
 
         for (int var2 = 0; var2 < this.iconArray.length; ++var2)
         {
-            this.iconArray[var2] = par1IconRegister.registerIcon(this.getTextureName() + "_stage_" + var2);
+            this.iconArray[var2] = par1IconRegister.registerIcon(this.func_111023_E() + "_stage_" + var2);
         }
     }
 }
