--- a/net/minecraft/src/BlockDispenser.java
+++ b/net/minecraft/src/BlockDispenser.java
@@ -89,8 +89,8 @@
     {
         this.blockIcon = par1IconRegister.registerIcon("furnace_side");
         this.furnaceTopIcon = par1IconRegister.registerIcon("furnace_top");
-        this.furnaceFrontIcon = par1IconRegister.registerIcon(this.getTextureName() + "_front_horizontal");
-        this.field_96473_e = par1IconRegister.registerIcon(this.getTextureName() + "_front_vertical");
+        this.furnaceFrontIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_front_horizontal");
+        this.field_96473_e = par1IconRegister.registerIcon(this.func_111023_E() + "_front_vertical");
     }
 
     /**
@@ -147,7 +147,7 @@
      */
     protected IBehaviorDispenseItem getBehaviorForItemStack(ItemStack par1ItemStack)
     {
-        return (IBehaviorDispenseItem)dispenseBehaviorRegistry.getObject(par1ItemStack.getItem());
+        return (IBehaviorDispenseItem)dispenseBehaviorRegistry.func_82594_a(par1ItemStack.getItem());
     }
 
     /**
@@ -205,9 +205,7 @@
     }
 
     /**
-     * Called on server worlds only when the block has been replaced by a different block ID, or the same block with a
-     * different metadata value, but before the new metadata value is set. Args: World, x, y, z, old block ID, old
-     * metadata
+     * ejects contained items into the world, and notifies neighbours of an update, as appropriate
      */
     public void breakBlock(World par1World, int par2, int par3, int par4, int par5, int par6)
     {
