--- a/net/minecraft/src/BlockStem.java
+++ b/net/minecraft/src/BlockStem.java
@@ -283,11 +283,11 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.blockIcon = par1IconRegister.registerIcon(this.getTextureName() + "_disconnected");
-        this.theIcon = par1IconRegister.registerIcon(this.getTextureName() + "_connected");
+        this.blockIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_disconnected");
+        this.theIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_connected");
     }
 
-    public Icon getStemIcon()
+    public Icon func_94368_p()
     {
         return this.theIcon;
     }
