--- a/net/minecraft/src/BlockTNT.java
+++ b/net/minecraft/src/BlockTNT.java
@@ -63,7 +63,7 @@
     {
         if (!par1World.isRemote)
         {
-            EntityTNTPrimed var6 = new EntityTNTPrimed(par1World, (double)((float)par2 + 0.5F), (double)((float)par3 + 0.5F), (double)((float)par4 + 0.5F), par5Explosion.getExplosivePlacedBy());
+            EntityTNTPrimed var6 = new EntityTNTPrimed(par1World, (double)((float)par2 + 0.5F), (double)((float)par3 + 0.5F), (double)((float)par4 + 0.5F), par5Explosion.func_94613_c());
             var6.fuse = par1World.rand.nextInt(var6.fuse / 4) + var6.fuse / 8;
             par1World.spawnEntityInWorld(var6);
         }
@@ -74,13 +74,10 @@
      */
     public void onBlockDestroyedByPlayer(World par1World, int par2, int par3, int par4, int par5)
     {
-        this.primeTnt(par1World, par2, par3, par4, par5, (EntityLivingBase)null);
+        this.func_94391_a(par1World, par2, par3, par4, par5, (EntityLivingBase)null);
     }
 
-    /**
-     * spawns the primed tnt and plays the fuse sound.
-     */
-    public void primeTnt(World par1World, int par2, int par3, int par4, int par5, EntityLivingBase par6EntityLivingBase)
+    public void func_94391_a(World par1World, int par2, int par3, int par4, int par5, EntityLivingBase par6EntityLivingBase)
     {
         if (!par1World.isRemote)
         {
@@ -100,7 +97,7 @@
     {
         if (par5EntityPlayer.getCurrentEquippedItem() != null && par5EntityPlayer.getCurrentEquippedItem().itemID == Item.flintAndSteel.itemID)
         {
-            this.primeTnt(par1World, par2, par3, par4, 1, par5EntityPlayer);
+            this.func_94391_a(par1World, par2, par3, par4, 1, par5EntityPlayer);
             par1World.setBlockToAir(par2, par3, par4);
             par5EntityPlayer.getCurrentEquippedItem().damageItem(1, par5EntityPlayer);
             return true;
@@ -122,7 +119,7 @@
 
             if (var6.isBurning())
             {
-                this.primeTnt(par1World, par2, par3, par4, 1, var6.shootingEntity instanceof EntityLivingBase ? (EntityLivingBase)var6.shootingEntity : null);
+                this.func_94391_a(par1World, par2, par3, par4, 1, var6.shootingEntity instanceof EntityLivingBase ? (EntityLivingBase)var6.shootingEntity : null);
                 par1World.setBlockToAir(par2, par3, par4);
             }
         }
@@ -142,8 +139,8 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.blockIcon = par1IconRegister.registerIcon(this.getTextureName() + "_side");
-        this.field_94393_a = par1IconRegister.registerIcon(this.getTextureName() + "_top");
-        this.field_94392_b = par1IconRegister.registerIcon(this.getTextureName() + "_bottom");
+        this.blockIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_side");
+        this.field_94393_a = par1IconRegister.registerIcon(this.func_111023_E() + "_top");
+        this.field_94392_b = par1IconRegister.registerIcon(this.func_111023_E() + "_bottom");
     }
 }
