--- a/net/minecraft/src/BlockRedstoneRepeater.java
+++ b/net/minecraft/src/BlockRedstoneRepeater.java
@@ -98,15 +98,12 @@
                     case 0:
                         var16 = -0.3125D;
                         break;
-
                     case 1:
                         var14 = 0.3125D;
                         break;
-
                     case 2:
                         var16 = 0.3125D;
                         break;
-
                     case 3:
                         var14 = -0.3125D;
                 }
@@ -120,15 +117,12 @@
                     case 0:
                         var16 = repeaterTorchOffset[var18];
                         break;
-
                     case 1:
                         var14 = -repeaterTorchOffset[var18];
                         break;
-
                     case 2:
                         var16 = -repeaterTorchOffset[var18];
                         break;
-
                     case 3:
                         var14 = repeaterTorchOffset[var18];
                 }
