--- a/net/minecraft/src/BlockMushroomCap.java
+++ b/net/minecraft/src/BlockMushroomCap.java
@@ -67,10 +67,10 @@
 
         for (int var2 = 0; var2 < this.iconArray.length; ++var2)
         {
-            this.iconArray[var2] = par1IconRegister.registerIcon(this.getTextureName() + "_" + field_94429_a[var2]);
+            this.iconArray[var2] = par1IconRegister.registerIcon(this.func_111023_E() + "_" + field_94429_a[var2]);
         }
 
-        this.field_94427_cP = par1IconRegister.registerIcon(this.getTextureName() + "_" + "inside");
-        this.field_94426_cO = par1IconRegister.registerIcon(this.getTextureName() + "_" + "skin_stem");
+        this.field_94427_cP = par1IconRegister.registerIcon(this.func_111023_E() + "_" + "inside");
+        this.field_94426_cO = par1IconRegister.registerIcon(this.func_111023_E() + "_" + "skin_stem");
     }
 }
