--- a/net/minecraft/src/BlockJukeBox.java
+++ b/net/minecraft/src/BlockJukeBox.java
@@ -84,9 +84,7 @@
     }
 
     /**
-     * Called on server worlds only when the block has been replaced by a different block ID, or the same block with a
-     * different metadata value, but before the new metadata value is set. Args: World, x, y, z, old block ID, old
-     * metadata
+     * ejects contained items into the world, and notifies neighbours of an update, as appropriate
      */
     public void breakBlock(World par1World, int par2, int par3, int par4, int par5, int par6)
     {
@@ -119,8 +117,8 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.blockIcon = par1IconRegister.registerIcon(this.getTextureName() + "_side");
-        this.theIcon = par1IconRegister.registerIcon(this.getTextureName() + "_top");
+        this.blockIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_side");
+        this.theIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_top");
     }
 
     /**
