--- a/net/minecraft/src/BlockHay.java
+++ b/net/minecraft/src/BlockHay.java
@@ -16,10 +16,7 @@
         return 31;
     }
 
-    /**
-     * The icon for the side of the block.
-     */
-    protected Icon getSideIcon(int par1)
+    protected Icon func_111048_c(int par1)
     {
         return this.blockIcon;
     }
@@ -30,7 +27,7 @@
      */
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.field_111051_a = par1IconRegister.registerIcon(this.getTextureName() + "_top");
-        this.blockIcon = par1IconRegister.registerIcon(this.getTextureName() + "_side");
+        this.field_111051_a = par1IconRegister.registerIcon(this.func_111023_E() + "_top");
+        this.blockIcon = par1IconRegister.registerIcon(this.func_111023_E() + "_side");
     }
 }
