--- a/net/minecraft/src/BlockFlowerPot.java
+++ b/net/minecraft/src/BlockFlowerPot.java
@@ -164,37 +164,26 @@
         {
             case 1:
                 return new ItemStack(Block.plantRed);
-
             case 2:
                 return new ItemStack(Block.plantYellow);
-
             case 3:
                 return new ItemStack(Block.sapling, 1, 0);
-
             case 4:
                 return new ItemStack(Block.sapling, 1, 1);
-
             case 5:
                 return new ItemStack(Block.sapling, 1, 2);
-
             case 6:
                 return new ItemStack(Block.sapling, 1, 3);
-
             case 7:
                 return new ItemStack(Block.mushroomRed);
-
             case 8:
                 return new ItemStack(Block.mushroomBrown);
-
             case 9:
                 return new ItemStack(Block.cactus);
-
             case 10:
                 return new ItemStack(Block.deadBush);
-
             case 11:
                 return new ItemStack(Block.tallGrass, 1, 2);
-
             default:
                 return null;
         }
@@ -239,13 +228,10 @@
                 {
                     case 0:
                         return 3;
-
                     case 1:
                         return 4;
-
                     case 2:
                         return 5;
-
                     case 3:
                         return 6;
                 }
