--- a/net/minecraft/src/ChatMessageComponent.java
+++ b/net/minecraft/src/ChatMessageComponent.java
@@ -9,109 +9,109 @@
 public class ChatMessageComponent
 {
     private static final Gson field_111089_a = (new GsonBuilder()).registerTypeAdapter(ChatMessageComponent.class, new MessageComponentSerializer()).create();
-    private EnumChatFormatting color;
-    private Boolean bold;
-    private Boolean italic;
-    private Boolean underline;
-    private Boolean obfuscated;
-    private String text;
-    private String translationKey;
+    private EnumChatFormatting field_111087_b;
+    private Boolean field_111088_c;
+    private Boolean field_111085_d;
+    private Boolean field_111086_e;
+    private Boolean field_111083_f;
+    private String field_111084_g;
+    private String field_111090_h;
     private List field_111091_i;
 
     public ChatMessageComponent() {}
 
     public ChatMessageComponent(ChatMessageComponent par1ChatMessageComponent)
     {
-        this.color = par1ChatMessageComponent.color;
-        this.bold = par1ChatMessageComponent.bold;
-        this.italic = par1ChatMessageComponent.italic;
-        this.underline = par1ChatMessageComponent.underline;
-        this.obfuscated = par1ChatMessageComponent.obfuscated;
-        this.text = par1ChatMessageComponent.text;
-        this.translationKey = par1ChatMessageComponent.translationKey;
+        this.field_111087_b = par1ChatMessageComponent.field_111087_b;
+        this.field_111088_c = par1ChatMessageComponent.field_111088_c;
+        this.field_111085_d = par1ChatMessageComponent.field_111085_d;
+        this.field_111086_e = par1ChatMessageComponent.field_111086_e;
+        this.field_111083_f = par1ChatMessageComponent.field_111083_f;
+        this.field_111084_g = par1ChatMessageComponent.field_111084_g;
+        this.field_111090_h = par1ChatMessageComponent.field_111090_h;
         this.field_111091_i = par1ChatMessageComponent.field_111091_i == null ? null : Lists.newArrayList(par1ChatMessageComponent.field_111091_i);
     }
 
-    public ChatMessageComponent setColor(EnumChatFormatting par1EnumChatFormatting)
+    public ChatMessageComponent func_111059_a(EnumChatFormatting par1EnumChatFormatting)
     {
-        if (par1EnumChatFormatting != null && !par1EnumChatFormatting.isColor())
+        if (par1EnumChatFormatting != null && !par1EnumChatFormatting.func_96302_c())
         {
             throw new IllegalArgumentException("Argument is not a valid color!");
         }
         else
         {
-            this.color = par1EnumChatFormatting;
+            this.field_111087_b = par1EnumChatFormatting;
             return this;
         }
     }
 
-    public EnumChatFormatting getColor()
+    public EnumChatFormatting func_111065_a()
     {
-        return this.color;
+        return this.field_111087_b;
     }
 
-    public ChatMessageComponent setBold(Boolean par1)
+    public ChatMessageComponent func_111071_a(Boolean par1)
     {
-        this.bold = par1;
+        this.field_111088_c = par1;
         return this;
     }
 
-    public Boolean isBold()
+    public Boolean func_111058_b()
     {
-        return this.bold;
+        return this.field_111088_c;
     }
 
-    public ChatMessageComponent setItalic(Boolean par1)
+    public ChatMessageComponent func_111063_b(Boolean par1)
     {
-        this.italic = par1;
+        this.field_111085_d = par1;
         return this;
     }
 
-    public Boolean isItalic()
+    public Boolean func_111064_c()
     {
-        return this.italic;
+        return this.field_111085_d;
     }
 
-    public ChatMessageComponent setUnderline(Boolean par1)
+    public ChatMessageComponent func_111081_c(Boolean par1)
     {
-        this.underline = par1;
+        this.field_111086_e = par1;
         return this;
     }
 
-    public Boolean isUnderline()
+    public Boolean func_111067_d()
     {
-        return this.underline;
+        return this.field_111086_e;
     }
 
-    public ChatMessageComponent setObfuscated(Boolean par1)
+    public ChatMessageComponent func_111061_d(Boolean par1)
     {
-        this.obfuscated = par1;
+        this.field_111083_f = par1;
         return this;
     }
 
-    public Boolean isObfuscated()
+    public Boolean func_111076_e()
     {
-        return this.obfuscated;
+        return this.field_111083_f;
     }
 
-    protected String getText()
+    protected String func_111075_f()
     {
-        return this.text;
+        return this.field_111084_g;
     }
 
-    protected String getTranslationKey()
+    protected String func_111074_g()
     {
-        return this.translationKey;
+        return this.field_111090_h;
     }
 
-    protected List getSubComponents()
+    protected List func_111069_h()
     {
         return this.field_111091_i;
     }
 
-    public ChatMessageComponent appendComponent(ChatMessageComponent par1ChatMessageComponent)
+    public ChatMessageComponent func_111073_a(ChatMessageComponent par1ChatMessageComponent)
     {
-        if (this.text == null && this.translationKey == null)
+        if (this.field_111084_g == null && this.field_111090_h == null)
         {
             if (this.field_111091_i != null)
             {
@@ -125,77 +125,70 @@
         else
         {
             this.field_111091_i = Lists.newArrayList(new ChatMessageComponent[] {new ChatMessageComponent(this), par1ChatMessageComponent});
-            this.text = null;
-            this.translationKey = null;
+            this.field_111084_g = null;
+            this.field_111090_h = null;
         }
 
         return this;
     }
 
-    public ChatMessageComponent addText(String par1Str)
+    public ChatMessageComponent func_111079_a(String par1Str)
     {
-        if (this.text == null && this.translationKey == null)
+        if (this.field_111084_g == null && this.field_111090_h == null)
         {
             if (this.field_111091_i != null)
             {
-                this.field_111091_i.add(createFromText(par1Str));
+                this.field_111091_i.add(func_111066_d(par1Str));
             }
             else
             {
-                this.text = par1Str;
+                this.field_111084_g = par1Str;
             }
         }
         else
         {
-            this.field_111091_i = Lists.newArrayList(new ChatMessageComponent[] {new ChatMessageComponent(this), createFromText(par1Str)});
-            this.text = null;
-            this.translationKey = null;
+            this.field_111091_i = Lists.newArrayList(new ChatMessageComponent[] {new ChatMessageComponent(this), func_111066_d(par1Str)});
+            this.field_111084_g = null;
+            this.field_111090_h = null;
         }
 
         return this;
     }
 
-    /**
-     * Appends a translated string.
-     */
-    public ChatMessageComponent addKey(String par1Str)
+    public ChatMessageComponent func_111072_b(String par1Str)
     {
-        if (this.text == null && this.translationKey == null)
+        if (this.field_111084_g == null && this.field_111090_h == null)
         {
             if (this.field_111091_i != null)
             {
-                this.field_111091_i.add(createFromTranslationKey(par1Str));
+                this.field_111091_i.add(func_111077_e(par1Str));
             }
             else
             {
-                this.translationKey = par1Str;
+                this.field_111090_h = par1Str;
             }
         }
         else
         {
-            this.field_111091_i = Lists.newArrayList(new ChatMessageComponent[] {new ChatMessageComponent(this), createFromTranslationKey(par1Str)});
-            this.text = null;
-            this.translationKey = null;
+            this.field_111091_i = Lists.newArrayList(new ChatMessageComponent[] {new ChatMessageComponent(this), func_111077_e(par1Str)});
+            this.field_111084_g = null;
+            this.field_111090_h = null;
         }
 
         return this;
     }
 
-    /**
-     * Appends a formatted translation key. Args: key, params. The text ultimately displayed is
-     * String.format(translate(key), params)
-     */
-    public ChatMessageComponent addFormatted(String par1Str, Object ... par2ArrayOfObj)
+    public ChatMessageComponent func_111080_a(String par1Str, Object ... par2ArrayOfObj)
     {
-        if (this.text == null && this.translationKey == null)
+        if (this.field_111084_g == null && this.field_111090_h == null)
         {
             if (this.field_111091_i != null)
             {
-                this.field_111091_i.add(createFromTranslationWithSubstitutions(par1Str, par2ArrayOfObj));
+                this.field_111091_i.add(func_111082_b(par1Str, par2ArrayOfObj));
             }
             else
             {
-                this.translationKey = par1Str;
+                this.field_111090_h = par1Str;
                 this.field_111091_i = Lists.newArrayList();
                 Object[] var3 = par2ArrayOfObj;
                 int var4 = par2ArrayOfObj.length;
@@ -210,16 +203,16 @@
                     }
                     else
                     {
-                        this.field_111091_i.add(createFromText(var6.toString()));
+                        this.field_111091_i.add(func_111066_d(var6.toString()));
                     }
                 }
             }
         }
         else
         {
-            this.field_111091_i = Lists.newArrayList(new ChatMessageComponent[] {new ChatMessageComponent(this), createFromTranslationWithSubstitutions(par1Str, par2ArrayOfObj)});
-            this.text = null;
-            this.translationKey = null;
+            this.field_111091_i = Lists.newArrayList(new ChatMessageComponent[] {new ChatMessageComponent(this), func_111082_b(par1Str, par2ArrayOfObj)});
+            this.field_111084_g = null;
+            this.field_111090_h = null;
         }
 
         return this;
@@ -227,31 +220,28 @@
 
     public String toString()
     {
-        return this.toStringWithFormatting(false);
+        return this.func_111068_a(false);
     }
 
-    public String toStringWithFormatting(boolean par1)
+    public String func_111068_a(boolean par1)
     {
-        return this.toStringWithDefaultFormatting(par1, (EnumChatFormatting)null, false, false, false, false);
+        return this.func_111070_a(par1, (EnumChatFormatting)null, false, false, false, false);
     }
 
-    /**
-     * args: enableFormat, defaultColor, defaultBold, defaultItalic, defaultUnderline, defaultObfuscated
-     */
-    public String toStringWithDefaultFormatting(boolean par1, EnumChatFormatting par2EnumChatFormatting, boolean par3, boolean par4, boolean par5, boolean par6)
+    public String func_111070_a(boolean par1, EnumChatFormatting par2EnumChatFormatting, boolean par3, boolean par4, boolean par5, boolean par6)
     {
         StringBuilder var7 = new StringBuilder();
-        EnumChatFormatting var8 = this.color == null ? par2EnumChatFormatting : this.color;
-        boolean var9 = this.bold == null ? par3 : this.bold.booleanValue();
-        boolean var10 = this.italic == null ? par4 : this.italic.booleanValue();
-        boolean var11 = this.underline == null ? par5 : this.underline.booleanValue();
-        boolean var12 = this.obfuscated == null ? par6 : this.obfuscated.booleanValue();
+        EnumChatFormatting var8 = this.field_111087_b == null ? par2EnumChatFormatting : this.field_111087_b;
+        boolean var9 = this.field_111088_c == null ? par3 : this.field_111088_c.booleanValue();
+        boolean var10 = this.field_111085_d == null ? par4 : this.field_111085_d.booleanValue();
+        boolean var11 = this.field_111086_e == null ? par5 : this.field_111086_e.booleanValue();
+        boolean var12 = this.field_111083_f == null ? par6 : this.field_111083_f.booleanValue();
 
-        if (this.translationKey != null)
+        if (this.field_111090_h != null)
         {
             if (par1)
             {
-                appendFormattingToString(var7, var8, var9, var10, var11, var12);
+                func_111060_a(var7, var8, var9, var10, var11, var12);
             }
 
             if (this.field_111091_i != null)
@@ -260,24 +250,24 @@
 
                 for (int var14 = 0; var14 < this.field_111091_i.size(); ++var14)
                 {
-                    var13[var14] = ((ChatMessageComponent)this.field_111091_i.get(var14)).toStringWithDefaultFormatting(par1, var8, var9, var10, var11, var12);
+                    var13[var14] = ((ChatMessageComponent)this.field_111091_i.get(var14)).func_111070_a(par1, var8, var9, var10, var11, var12);
                 }
 
-                var7.append(StatCollector.translateToLocalFormatted(this.translationKey, var13));
+                var7.append(StatCollector.translateToLocalFormatted(this.field_111090_h, var13));
             }
             else
             {
-                var7.append(StatCollector.translateToLocal(this.translationKey));
+                var7.append(StatCollector.translateToLocal(this.field_111090_h));
             }
         }
-        else if (this.text != null)
+        else if (this.field_111084_g != null)
         {
             if (par1)
             {
-                appendFormattingToString(var7, var8, var9, var10, var11, var12);
+                func_111060_a(var7, var8, var9, var10, var11, var12);
             }
 
-            var7.append(this.text);
+            var7.append(this.field_111084_g);
         }
         else
         {
@@ -285,13 +275,13 @@
 
             if (this.field_111091_i != null)
             {
-                for (Iterator var15 = this.field_111091_i.iterator(); var15.hasNext(); var7.append(var16.toStringWithDefaultFormatting(par1, var8, var9, var10, var11, var12)))
+                for (Iterator var15 = this.field_111091_i.iterator(); var15.hasNext(); var7.append(var16.func_111070_a(par1, var8, var9, var10, var11, var12)))
                 {
                     var16 = (ChatMessageComponent)var15.next();
 
                     if (par1)
                     {
-                        appendFormattingToString(var7, var8, var9, var10, var11, var12);
+                        func_111060_a(var7, var8, var9, var10, var11, var12);
                     }
                 }
             }
@@ -300,7 +290,7 @@
         return var7.toString();
     }
 
-    private static void appendFormattingToString(StringBuilder par0StringBuilder, EnumChatFormatting par1EnumChatFormatting, boolean par2, boolean par3, boolean par4, boolean par5)
+    private static void func_111060_a(StringBuilder par0StringBuilder, EnumChatFormatting par1EnumChatFormatting, boolean par2, boolean par3, boolean par4, boolean par5)
     {
         if (par1EnumChatFormatting != null)
         {
@@ -332,7 +322,7 @@
         }
     }
 
-    public static ChatMessageComponent createFromJson(String par0Str)
+    public static ChatMessageComponent func_111078_c(String par0Str)
     {
         try
         {
@@ -347,28 +337,28 @@
         }
     }
 
-    public static ChatMessageComponent createFromText(String par0Str)
+    public static ChatMessageComponent func_111066_d(String par0Str)
     {
         ChatMessageComponent var1 = new ChatMessageComponent();
-        var1.addText(par0Str);
+        var1.func_111079_a(par0Str);
         return var1;
     }
 
-    public static ChatMessageComponent createFromTranslationKey(String par0Str)
+    public static ChatMessageComponent func_111077_e(String par0Str)
     {
         ChatMessageComponent var1 = new ChatMessageComponent();
-        var1.addKey(par0Str);
+        var1.func_111072_b(par0Str);
         return var1;
     }
 
-    public static ChatMessageComponent createFromTranslationWithSubstitutions(String par0Str, Object ... par1ArrayOfObj)
+    public static ChatMessageComponent func_111082_b(String par0Str, Object ... par1ArrayOfObj)
     {
         ChatMessageComponent var2 = new ChatMessageComponent();
-        var2.addFormatted(par0Str, par1ArrayOfObj);
+        var2.func_111080_a(par0Str, par1ArrayOfObj);
         return var2;
     }
 
-    public String toJson()
+    public String func_111062_i()
     {
         return field_111089_a.toJson(this);
     }
