--- a/net/minecraft/src/CombatTracker.java
+++ b/net/minecraft/src/CombatTracker.java
@@ -7,9 +7,7 @@
 public class CombatTracker
 {
     private final List field_94556_a = new ArrayList();
-
-    /** The entity tracked. */
-    private final EntityLivingBase fighter;
+    private final EntityLivingBase field_94554_b;
     private int field_94555_c;
     private boolean field_94552_d;
     private boolean field_94553_e;
@@ -17,16 +15,16 @@
 
     public CombatTracker(EntityLivingBase par1EntityLivingBase)
     {
-        this.fighter = par1EntityLivingBase;
+        this.field_94554_b = par1EntityLivingBase;
     }
 
     public void func_94545_a()
     {
         this.func_94542_g();
 
-        if (this.fighter.isOnLadder())
+        if (this.field_94554_b.isOnLadder())
         {
-            int var1 = this.fighter.worldObj.getBlockId(MathHelper.floor_double(this.fighter.posX), MathHelper.floor_double(this.fighter.boundingBox.minY), MathHelper.floor_double(this.fighter.posZ));
+            int var1 = this.field_94554_b.worldObj.getBlockId(MathHelper.floor_double(this.field_94554_b.posX), MathHelper.floor_double(this.field_94554_b.boundingBox.minY), MathHelper.floor_double(this.field_94554_b.posZ));
 
             if (var1 == Block.ladder.blockID)
             {
@@ -37,7 +35,7 @@
                 this.field_94551_f = "vines";
             }
         }
-        else if (this.fighter.isInWater())
+        else if (this.field_94554_b.isInWater())
         {
             this.field_94551_f = "water";
         }
@@ -47,9 +45,9 @@
     {
         this.func_94549_h();
         this.func_94545_a();
-        CombatEntry var4 = new CombatEntry(par1DamageSource, this.fighter.ticksExisted, par2, par3, this.field_94551_f, this.fighter.fallDistance);
+        CombatEntry var4 = new CombatEntry(par1DamageSource, this.field_94554_b.ticksExisted, par2, par3, this.field_94551_f, this.field_94554_b.fallDistance);
         this.field_94556_a.add(var4);
-        this.field_94555_c = this.fighter.ticksExisted;
+        this.field_94555_c = this.field_94554_b.ticksExisted;
         this.field_94553_e = true;
         this.field_94552_d |= var4.func_94559_f();
     }
@@ -58,34 +56,34 @@
     {
         if (this.field_94556_a.size() == 0)
         {
-            return ChatMessageComponent.createFromTranslationWithSubstitutions("death.attack.generic", new Object[] {this.fighter.getTranslatedEntityName()});
+            return ChatMessageComponent.func_111082_b("death.attack.generic", new Object[] {this.field_94554_b.getTranslatedEntityName()});
         }
         else
         {
             CombatEntry var1 = this.func_94544_f();
             CombatEntry var2 = (CombatEntry)this.field_94556_a.get(this.field_94556_a.size() - 1);
             String var4 = var2.func_94558_h();
-            Entity var5 = var2.getDamageSrc().getEntity();
+            Entity var5 = var2.func_94560_a().getEntity();
             ChatMessageComponent var3;
 
-            if (var1 != null && var2.getDamageSrc() == DamageSource.fall)
+            if (var1 != null && var2.func_94560_a() == DamageSource.fall)
             {
                 String var6 = var1.func_94558_h();
 
-                if (var1.getDamageSrc() != DamageSource.fall && var1.getDamageSrc() != DamageSource.outOfWorld)
+                if (var1.func_94560_a() != DamageSource.fall && var1.func_94560_a() != DamageSource.outOfWorld)
                 {
                     if (var6 != null && (var4 == null || !var6.equals(var4)))
                     {
-                        Entity var9 = var1.getDamageSrc().getEntity();
+                        Entity var9 = var1.func_94560_a().getEntity();
                         ItemStack var8 = var9 instanceof EntityLivingBase ? ((EntityLivingBase)var9).getHeldItem() : null;
 
                         if (var8 != null && var8.hasDisplayName())
                         {
-                            var3 = ChatMessageComponent.createFromTranslationWithSubstitutions("death.fell.assist.item", new Object[] {this.fighter.getTranslatedEntityName(), var6, var8.getDisplayName()});
+                            var3 = ChatMessageComponent.func_111082_b("death.fell.assist.item", new Object[] {this.field_94554_b.getTranslatedEntityName(), var6, var8.getDisplayName()});
                         }
                         else
                         {
-                            var3 = ChatMessageComponent.createFromTranslationWithSubstitutions("death.fell.assist", new Object[] {this.fighter.getTranslatedEntityName(), var6});
+                            var3 = ChatMessageComponent.func_111082_b("death.fell.assist", new Object[] {this.field_94554_b.getTranslatedEntityName(), var6});
                         }
                     }
                     else if (var4 != null)
@@ -94,26 +92,26 @@
 
                         if (var7 != null && var7.hasDisplayName())
                         {
-                            var3 = ChatMessageComponent.createFromTranslationWithSubstitutions("death.fell.finish.item", new Object[] {this.fighter.getTranslatedEntityName(), var4, var7.getDisplayName()});
+                            var3 = ChatMessageComponent.func_111082_b("death.fell.finish.item", new Object[] {this.field_94554_b.getTranslatedEntityName(), var4, var7.getDisplayName()});
                         }
                         else
                         {
-                            var3 = ChatMessageComponent.createFromTranslationWithSubstitutions("death.fell.finish", new Object[] {this.fighter.getTranslatedEntityName(), var4});
+                            var3 = ChatMessageComponent.func_111082_b("death.fell.finish", new Object[] {this.field_94554_b.getTranslatedEntityName(), var4});
                         }
                     }
                     else
                     {
-                        var3 = ChatMessageComponent.createFromTranslationWithSubstitutions("death.fell.killer", new Object[] {this.fighter.getTranslatedEntityName()});
+                        var3 = ChatMessageComponent.func_111082_b("death.fell.killer", new Object[] {this.field_94554_b.getTranslatedEntityName()});
                     }
                 }
                 else
                 {
-                    var3 = ChatMessageComponent.createFromTranslationWithSubstitutions("death.fell.accident." + this.func_94548_b(var1), new Object[] {this.fighter.getTranslatedEntityName()});
+                    var3 = ChatMessageComponent.func_111082_b("death.fell.accident." + this.func_94548_b(var1), new Object[] {this.field_94554_b.getTranslatedEntityName()});
                 }
             }
             else
             {
-                var3 = var2.getDamageSrc().getDeathMessage(this.fighter);
+                var3 = var2.func_94560_a().getDeathMessage(this.field_94554_b);
             }
 
             return var3;
@@ -132,16 +130,16 @@
         {
             CombatEntry var6 = (CombatEntry)var5.next();
 
-            if (var6.getDamageSrc().getEntity() instanceof EntityPlayer && (var2 == null || var6.func_94563_c() > var4))
+            if (var6.func_94560_a().getEntity() instanceof EntityPlayer && (var2 == null || var6.func_94563_c() > var4))
             {
                 var4 = var6.func_94563_c();
-                var2 = (EntityPlayer)var6.getDamageSrc().getEntity();
+                var2 = (EntityPlayer)var6.func_94560_a().getEntity();
             }
 
-            if (var6.getDamageSrc().getEntity() instanceof EntityLivingBase && (var1 == null || var6.func_94563_c() > var3))
+            if (var6.func_94560_a().getEntity() instanceof EntityLivingBase && (var1 == null || var6.func_94563_c() > var3))
             {
                 var3 = var6.func_94563_c();
-                var1 = (EntityLivingBase)var6.getDamageSrc().getEntity();
+                var1 = (EntityLivingBase)var6.func_94560_a().getEntity();
             }
         }
 
@@ -167,7 +165,7 @@
             CombatEntry var6 = (CombatEntry)this.field_94556_a.get(var5);
             CombatEntry var7 = var5 > 0 ? (CombatEntry)this.field_94556_a.get(var5 - 1) : null;
 
-            if ((var6.getDamageSrc() == DamageSource.fall || var6.getDamageSrc() == DamageSource.outOfWorld) && var6.func_94561_i() > 0.0F && (var1 == null || var6.func_94561_i() > var4))
+            if ((var6.func_94560_a() == DamageSource.fall || var6.func_94560_a() == DamageSource.outOfWorld) && var6.func_94561_i() > 0.0F && (var1 == null || var6.func_94561_i() > var4))
             {
                 if (var5 > 0)
                 {
@@ -215,7 +213,7 @@
     {
         int var1 = this.field_94552_d ? 300 : 100;
 
-        if (this.field_94553_e && this.fighter.ticksExisted - this.field_94555_c > var1)
+        if (this.field_94553_e && this.field_94554_b.ticksExisted - this.field_94555_c > var1)
         {
             this.field_94556_a.clear();
             this.field_94553_e = false;
