--- a/net/minecraft/src/CombatEntry.java
+++ b/net/minecraft/src/CombatEntry.java
@@ -2,7 +2,7 @@
 
 public class CombatEntry
 {
-    private final DamageSource damageSrc;
+    private final DamageSource field_94569_a;
     private final int field_94567_b;
     private final float field_94568_c;
     private final float field_94565_d;
@@ -11,7 +11,7 @@
 
     public CombatEntry(DamageSource par1DamageSource, int par2, float par3, float par4, String par5Str, float par6)
     {
-        this.damageSrc = par1DamageSource;
+        this.field_94569_a = par1DamageSource;
         this.field_94567_b = par2;
         this.field_94568_c = par4;
         this.field_94565_d = par3;
@@ -19,12 +19,9 @@
         this.field_94564_f = par6;
     }
 
-    /**
-     * Get the DamageSource of the CombatEntry instance.
-     */
-    public DamageSource getDamageSrc()
+    public DamageSource func_94560_a()
     {
-        return this.damageSrc;
+        return this.field_94569_a;
     }
 
     public float func_94563_c()
@@ -34,7 +31,7 @@
 
     public boolean func_94559_f()
     {
-        return this.damageSrc.getEntity() instanceof EntityLivingBase;
+        return this.field_94569_a.getEntity() instanceof EntityLivingBase;
     }
 
     public String func_94562_g()
@@ -44,11 +41,11 @@
 
     public String func_94558_h()
     {
-        return this.getDamageSrc().getEntity() == null ? null : this.getDamageSrc().getEntity().getTranslatedEntityName();
+        return this.func_94560_a().getEntity() == null ? null : this.func_94560_a().getEntity().getTranslatedEntityName();
     }
 
     public float func_94561_i()
     {
-        return this.damageSrc == DamageSource.outOfWorld ? Float.MAX_VALUE : this.field_94564_f;
+        return this.field_94569_a == DamageSource.outOfWorld ? Float.MAX_VALUE : this.field_94564_f;
     }
 }
