--- a/net/minecraft/src/CommandShowSeed.java
+++ b/net/minecraft/src/CommandShowSeed.java
@@ -33,6 +33,6 @@
     public void processCommand(ICommandSender par1ICommandSender, String[] par2ArrayOfStr)
     {
         Object var3 = par1ICommandSender instanceof EntityPlayer ? ((EntityPlayer)par1ICommandSender).worldObj : MinecraftServer.getServer().worldServerForDimension(0);
-        par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.seed.success", new Object[] {Long.valueOf(((World)var3).getSeed())}));
+        par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.seed.success", new Object[] {Long.valueOf(((World)var3).getSeed())}));
     }
 }
