--- a/net/minecraft/src/CommandGive.java
+++ b/net/minecraft/src/CommandGive.java
@@ -27,7 +27,7 @@
     {
         if (par2ArrayOfStr.length >= 2)
         {
-            EntityPlayerMP var3 = getPlayer(par1ICommandSender, par2ArrayOfStr[0]);
+            EntityPlayerMP var3 = func_82359_c(par1ICommandSender, par2ArrayOfStr[0]);
             int var4 = parseIntWithMin(par1ICommandSender, par2ArrayOfStr[1], 1);
             int var5 = 1;
             int var6 = 0;
