--- a/net/minecraft/src/CommandBase.java
+++ b/net/minecraft/src/CommandBase.java
@@ -165,7 +165,7 @@
         }
     }
 
-    public static EntityPlayerMP getPlayer(ICommandSender par0ICommandSender, String par1Str)
+    public static EntityPlayerMP func_82359_c(ICommandSender par0ICommandSender, String par1Str)
     {
         EntityPlayerMP var2 = PlayerSelector.matchOnePlayer(par0ICommandSender, par1Str);
 
