--- a/net/minecraft/src/CommandDifficulty.java
+++ b/net/minecraft/src/CommandDifficulty.java
@@ -31,7 +31,7 @@
         {
             int var3 = this.getDifficultyForName(par1ICommandSender, par2ArrayOfStr[0]);
             MinecraftServer.getServer().setDifficultyForAllWorlds(var3);
-            notifyAdmins(par1ICommandSender, "commands.difficulty.success", new Object[] {ChatMessageComponent.createFromTranslationKey(difficulties[var3])});
+            notifyAdmins(par1ICommandSender, "commands.difficulty.success", new Object[] {ChatMessageComponent.func_111077_e(difficulties[var3])});
         }
         else
         {
