--- a/net/minecraft/src/CommandServerOp.java
+++ b/net/minecraft/src/CommandServerOp.java
@@ -53,7 +53,7 @@
             {
                 String var8 = var5[var7];
 
-                if (!MinecraftServer.getServer().getConfigurationManager().isPlayerOpped(var8) && doesStringStartWith(var3, var8))
+                if (!MinecraftServer.getServer().getConfigurationManager().areCommandsAllowed(var8) && doesStringStartWith(var3, var8))
                 {
                     var4.add(var8);
                 }
