--- a/net/minecraft/src/CommandSetSpawnpoint.java
+++ b/net/minecraft/src/CommandSetSpawnpoint.java
@@ -25,7 +25,7 @@
 
     public void processCommand(ICommandSender par1ICommandSender, String[] par2ArrayOfStr)
     {
-        EntityPlayerMP var3 = par2ArrayOfStr.length == 0 ? getCommandSenderAsPlayer(par1ICommandSender) : getPlayer(par1ICommandSender, par2ArrayOfStr[0]);
+        EntityPlayerMP var3 = par2ArrayOfStr.length == 0 ? getCommandSenderAsPlayer(par1ICommandSender) : func_82359_c(par1ICommandSender, par2ArrayOfStr[0]);
 
         if (par2ArrayOfStr.length == 4)
         {
