--- a/net/minecraft/src/CommandPlaySound.java
+++ b/net/minecraft/src/CommandPlaySound.java
@@ -31,7 +31,7 @@
             byte var3 = 0;
             int var36 = var3 + 1;
             String var4 = par2ArrayOfStr[var3];
-            EntityPlayerMP var5 = getPlayer(par1ICommandSender, par2ArrayOfStr[var36++]);
+            EntityPlayerMP var5 = func_82359_c(par1ICommandSender, par2ArrayOfStr[var36++]);
             double var6 = (double)var5.getPlayerCoordinates().posX;
             double var8 = (double)var5.getPlayerCoordinates().posY;
             double var10 = (double)var5.getPlayerCoordinates().posZ;
