--- a/net/minecraft/src/CommandGameMode.java
+++ b/net/minecraft/src/CommandGameMode.java
@@ -28,10 +28,10 @@
         if (par2ArrayOfStr.length > 0)
         {
             EnumGameType var3 = this.getGameModeFromCommand(par1ICommandSender, par2ArrayOfStr[0]);
-            EntityPlayerMP var4 = par2ArrayOfStr.length >= 2 ? getPlayer(par1ICommandSender, par2ArrayOfStr[1]) : getCommandSenderAsPlayer(par1ICommandSender);
+            EntityPlayerMP var4 = par2ArrayOfStr.length >= 2 ? func_82359_c(par1ICommandSender, par2ArrayOfStr[1]) : getCommandSenderAsPlayer(par1ICommandSender);
             var4.setGameType(var3);
             var4.fallDistance = 0.0F;
-            ChatMessageComponent var5 = ChatMessageComponent.createFromTranslationKey("gameMode." + var3.getName());
+            ChatMessageComponent var5 = ChatMessageComponent.func_111077_e("gameMode." + var3.getName());
 
             if (var4 != par1ICommandSender)
             {
