--- a/net/minecraft/src/CommandHandler.java
+++ b/net/minecraft/src/CommandHandler.java
@@ -61,7 +61,7 @@
                         }
                         catch (CommandException var15)
                         {
-                            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions(var15.getMessage(), var15.getErrorOjbects()).setColor(EnumChatFormatting.RED));
+                            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b(var15.getMessage(), var15.getErrorOjbects()).func_111059_a(EnumChatFormatting.RED));
                         }
                     }
 
@@ -75,20 +75,20 @@
             }
             else
             {
-                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationKey("commands.generic.permission").setColor(EnumChatFormatting.RED));
+                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111077_e("commands.generic.permission").func_111059_a(EnumChatFormatting.RED));
             }
         }
         catch (WrongUsageException var16)
         {
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.generic.usage", new Object[] {ChatMessageComponent.createFromTranslationWithSubstitutions(var16.getMessage(), var16.getErrorOjbects())}).setColor(EnumChatFormatting.RED));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.generic.usage", new Object[] {ChatMessageComponent.func_111082_b(var16.getMessage(), var16.getErrorOjbects())}).func_111059_a(EnumChatFormatting.RED));
         }
         catch (CommandException var17)
         {
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions(var17.getMessage(), var17.getErrorOjbects()).setColor(EnumChatFormatting.RED));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b(var17.getMessage(), var17.getErrorOjbects()).func_111059_a(EnumChatFormatting.RED));
         }
         catch (Throwable var18)
         {
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationKey("commands.generic.exception").setColor(EnumChatFormatting.RED));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111077_e("commands.generic.exception").func_111059_a(EnumChatFormatting.RED));
             var18.printStackTrace();
         }
 
