--- a/net/minecraft/src/CommandHelp.java
+++ b/net/minecraft/src/CommandHelp.java
@@ -58,17 +58,17 @@
         }
 
         int var7 = Math.min((var11 + 1) * var4, var3.size());
-        par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.help.header", new Object[] {Integer.valueOf(var11 + 1), Integer.valueOf(var5 + 1)}).setColor(EnumChatFormatting.DARK_GREEN));
+        par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.help.header", new Object[] {Integer.valueOf(var11 + 1), Integer.valueOf(var5 + 1)}).func_111059_a(EnumChatFormatting.DARK_GREEN));
 
         for (int var12 = var11 * var4; var12 < var7; ++var12)
         {
             var9 = (ICommand)var3.get(var12);
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationKey(var9.getCommandUsage(par1ICommandSender)));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111077_e(var9.getCommandUsage(par1ICommandSender)));
         }
 
         if (var11 == 0 && par1ICommandSender instanceof EntityPlayer)
         {
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationKey("commands.help.footer").setColor(EnumChatFormatting.GREEN));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111077_e("commands.help.footer").func_111059_a(EnumChatFormatting.GREEN));
         }
     }
 
