--- a/net/minecraft/src/CommandServerWhitelist.java
+++ b/net/minecraft/src/CommandServerWhitelist.java
@@ -45,9 +45,9 @@
 
             if (par2ArrayOfStr[0].equals("list"))
             {
-                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.whitelist.list", new Object[] {Integer.valueOf(MinecraftServer.getServer().getConfigurationManager().getWhiteListedPlayers().size()), Integer.valueOf(MinecraftServer.getServer().getConfigurationManager().getAvailablePlayerDat().length)}));
+                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.whitelist.list", new Object[] {Integer.valueOf(MinecraftServer.getServer().getConfigurationManager().getWhiteListedPlayers().size()), Integer.valueOf(MinecraftServer.getServer().getConfigurationManager().getAvailablePlayerDat().length)}));
                 Set var3 = MinecraftServer.getServer().getConfigurationManager().getWhiteListedPlayers();
-                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromText(joinNiceString(var3.toArray(new String[var3.size()]))));
+                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111066_d(joinNiceString(var3.toArray(new String[var3.size()]))));
                 return;
             }
 
