--- a/net/minecraft/src/ComponentStrongholdPortalRoom.java
+++ b/net/minecraft/src/ComponentStrongholdPortalRoom.java
@@ -7,8 +7,6 @@
 {
     private boolean hasSpawner;
 
-    public ComponentStrongholdPortalRoom() {}
-
     public ComponentStrongholdPortalRoom(int par1, Random par2Random, StructureBoundingBox par3StructureBoundingBox, int par4)
     {
         super(par1);
@@ -16,18 +14,6 @@
         this.boundingBox = par3StructureBoundingBox;
     }
 
-    protected void func_143012_a(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143012_a(par1NBTTagCompound);
-        par1NBTTagCompound.setBoolean("Mob", this.hasSpawner);
-    }
-
-    protected void func_143011_b(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143011_b(par1NBTTagCompound);
-        this.hasSpawner = par1NBTTagCompound.getBoolean("Mob");
-    }
-
     /**
      * Initiates construction of the Structure Component picked, at the current Location of StructGen
      */
@@ -100,17 +86,14 @@
                 var14 = 0;
                 var7 = 2;
                 break;
-
             case 1:
                 var14 = 1;
                 var7 = 3;
                 var8 = 0;
                 var9 = 2;
-
             case 2:
             default:
                 break;
-
             case 3:
                 var14 = 3;
                 var7 = 1;
