--- a/net/minecraft/src/ComponentVillage.java
+++ b/net/minecraft/src/ComponentVillage.java
@@ -5,36 +5,16 @@
 
 abstract class ComponentVillage extends StructureComponent
 {
-    protected int field_143015_k = -1;
-
     /** The number of villagers that have been spawned in this component. */
     private int villagersSpawned;
-    private boolean field_143014_b;
 
-    public ComponentVillage() {}
+    /** The starting piece of the village. */
+    protected ComponentVillageStartPiece startPiece;
 
     protected ComponentVillage(ComponentVillageStartPiece par1ComponentVillageStartPiece, int par2)
     {
         super(par2);
-
-        if (par1ComponentVillageStartPiece != null)
-        {
-            this.field_143014_b = par1ComponentVillageStartPiece.inDesert;
-        }
-    }
-
-    protected void func_143012_a(NBTTagCompound par1NBTTagCompound)
-    {
-        par1NBTTagCompound.setInteger("HPos", this.field_143015_k);
-        par1NBTTagCompound.setInteger("VCount", this.villagersSpawned);
-        par1NBTTagCompound.setBoolean("Desert", this.field_143014_b);
-    }
-
-    protected void func_143011_b(NBTTagCompound par1NBTTagCompound)
-    {
-        this.field_143015_k = par1NBTTagCompound.getInteger("HPos");
-        this.villagersSpawned = par1NBTTagCompound.getInteger("VCount");
-        this.field_143014_b = par1NBTTagCompound.getBoolean("Desert");
+        this.startPiece = par1ComponentVillageStartPiece;
     }
 
     /**
@@ -46,16 +26,12 @@
         {
             case 0:
                 return StructureVillagePieces.getNextStructureComponent(par1ComponentVillageStartPiece, par2List, par3Random, this.boundingBox.minX - 1, this.boundingBox.minY + par4, this.boundingBox.minZ + par5, 1, this.getComponentType());
-
             case 1:
                 return StructureVillagePieces.getNextStructureComponent(par1ComponentVillageStartPiece, par2List, par3Random, this.boundingBox.minX + par5, this.boundingBox.minY + par4, this.boundingBox.minZ - 1, 2, this.getComponentType());
-
             case 2:
                 return StructureVillagePieces.getNextStructureComponent(par1ComponentVillageStartPiece, par2List, par3Random, this.boundingBox.minX - 1, this.boundingBox.minY + par4, this.boundingBox.minZ + par5, 1, this.getComponentType());
-
             case 3:
                 return StructureVillagePieces.getNextStructureComponent(par1ComponentVillageStartPiece, par2List, par3Random, this.boundingBox.minX + par5, this.boundingBox.minY + par4, this.boundingBox.minZ - 1, 2, this.getComponentType());
-
             default:
                 return null;
         }
@@ -70,16 +46,12 @@
         {
             case 0:
                 return StructureVillagePieces.getNextStructureComponent(par1ComponentVillageStartPiece, par2List, par3Random, this.boundingBox.maxX + 1, this.boundingBox.minY + par4, this.boundingBox.minZ + par5, 3, this.getComponentType());
-
             case 1:
                 return StructureVillagePieces.getNextStructureComponent(par1ComponentVillageStartPiece, par2List, par3Random, this.boundingBox.minX + par5, this.boundingBox.minY + par4, this.boundingBox.maxZ + 1, 0, this.getComponentType());
-
             case 2:
                 return StructureVillagePieces.getNextStructureComponent(par1ComponentVillageStartPiece, par2List, par3Random, this.boundingBox.maxX + 1, this.boundingBox.minY + par4, this.boundingBox.minZ + par5, 3, this.getComponentType());
-
             case 3:
                 return StructureVillagePieces.getNextStructureComponent(par1ComponentVillageStartPiece, par2List, par3Random, this.boundingBox.minX + par5, this.boundingBox.minY + par4, this.boundingBox.maxZ + 1, 0, this.getComponentType());
-
             default:
                 return null;
         }
@@ -161,7 +133,7 @@
      */
     protected int getBiomeSpecificBlock(int par1, int par2)
     {
-        if (this.field_143014_b)
+        if (this.startPiece.inDesert)
         {
             if (par1 == Block.wood.blockID)
             {
@@ -202,7 +174,7 @@
      */
     protected int getBiomeSpecificBlockMetadata(int par1, int par2)
     {
-        if (this.field_143014_b)
+        if (this.startPiece.inDesert)
         {
             if (par1 == Block.wood.blockID)
             {
