--- a/net/minecraft/src/ComponentStrongholdStairs2.java
+++ b/net/minecraft/src/ComponentStrongholdStairs2.java
@@ -10,8 +10,6 @@
     public ComponentStrongholdPortalRoom strongholdPortalRoom;
     public List field_75026_c = new ArrayList();
 
-    public ComponentStrongholdStairs2() {}
-
     public ComponentStrongholdStairs2(int par1, Random par2Random, int par3, int par4)
     {
         super(0, par2Random, par3, par4);
