--- a/net/minecraft/src/ComponentVillageWoodHut.java
+++ b/net/minecraft/src/ComponentVillageWoodHut.java
@@ -5,10 +5,9 @@
 
 public class ComponentVillageWoodHut extends ComponentVillage
 {
-    private boolean isTallHouse;
-    private int tablePosition;
-
-    public ComponentVillageWoodHut() {}
+    private int averageGroundLevel = -1;
+    private final boolean isTallHouse;
+    private final int tablePosition;
 
     public ComponentVillageWoodHut(ComponentVillageStartPiece par1ComponentVillageStartPiece, int par2, Random par3Random, StructureBoundingBox par4StructureBoundingBox, int par5)
     {
@@ -19,20 +18,6 @@
         this.tablePosition = par3Random.nextInt(3);
     }
 
-    protected void func_143012_a(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143012_a(par1NBTTagCompound);
-        par1NBTTagCompound.setInteger("T", this.tablePosition);
-        par1NBTTagCompound.setBoolean("C", this.isTallHouse);
-    }
-
-    protected void func_143011_b(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143011_b(par1NBTTagCompound);
-        this.tablePosition = par1NBTTagCompound.getInteger("T");
-        this.isTallHouse = par1NBTTagCompound.getBoolean("C");
-    }
-
     public static ComponentVillageWoodHut func_74908_a(ComponentVillageStartPiece par0ComponentVillageStartPiece, List par1List, Random par2Random, int par3, int par4, int par5, int par6, int par7)
     {
         StructureBoundingBox var8 = StructureBoundingBox.getComponentToAddBoundingBox(par3, par4, par5, 0, 0, 0, 4, 6, 5, par6);
@@ -45,16 +30,16 @@
      */
     public boolean addComponentParts(World par1World, Random par2Random, StructureBoundingBox par3StructureBoundingBox)
     {
-        if (this.field_143015_k < 0)
+        if (this.averageGroundLevel < 0)
         {
-            this.field_143015_k = this.getAverageGroundLevel(par1World, par3StructureBoundingBox);
+            this.averageGroundLevel = this.getAverageGroundLevel(par1World, par3StructureBoundingBox);
 
-            if (this.field_143015_k < 0)
+            if (this.averageGroundLevel < 0)
             {
                 return true;
             }
 
-            this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 6 - 1, 0);
+            this.boundingBox.offset(0, this.averageGroundLevel - this.boundingBox.maxY + 6 - 1, 0);
         }
 
         this.fillWithBlocks(par1World, par3StructureBoundingBox, 1, 1, 1, 3, 5, 4, 0, 0, false);
