--- a/net/minecraft/src/ComponentScatteredFeatureJunglePyramid.java
+++ b/net/minecraft/src/ComponentScatteredFeatureJunglePyramid.java
@@ -10,7 +10,7 @@
     private boolean field_74946_k;
 
     /** List of Chest contents to be generated in the Jungle Pyramid chests. */
-    private static final WeightedRandomChestContent[] junglePyramidsChestContents = new WeightedRandomChestContent[] {new WeightedRandomChestContent(Item.diamond.itemID, 0, 1, 3, 3), new WeightedRandomChestContent(Item.ingotIron.itemID, 0, 1, 5, 10), new WeightedRandomChestContent(Item.ingotGold.itemID, 0, 2, 7, 15), new WeightedRandomChestContent(Item.emerald.itemID, 0, 1, 3, 2), new WeightedRandomChestContent(Item.bone.itemID, 0, 4, 6, 20), new WeightedRandomChestContent(Item.rottenFlesh.itemID, 0, 3, 7, 16), new WeightedRandomChestContent(Item.saddle.itemID, 0, 1, 1, 3), new WeightedRandomChestContent(Item.horseArmorIron.itemID, 0, 1, 1, 1), new WeightedRandomChestContent(Item.horseArmorGold.itemID, 0, 1, 1, 1), new WeightedRandomChestContent(Item.horseArmorDiamond.itemID, 0, 1, 1, 1)};
+    private static final WeightedRandomChestContent[] junglePyramidsChestContents = new WeightedRandomChestContent[] {new WeightedRandomChestContent(Item.diamond.itemID, 0, 1, 3, 3), new WeightedRandomChestContent(Item.ingotIron.itemID, 0, 1, 5, 10), new WeightedRandomChestContent(Item.ingotGold.itemID, 0, 2, 7, 15), new WeightedRandomChestContent(Item.emerald.itemID, 0, 1, 3, 2), new WeightedRandomChestContent(Item.bone.itemID, 0, 4, 6, 20), new WeightedRandomChestContent(Item.rottenFlesh.itemID, 0, 3, 7, 16), new WeightedRandomChestContent(Item.saddle.itemID, 0, 1, 1, 3), new WeightedRandomChestContent(Item.field_111215_ce.itemID, 0, 1, 1, 1), new WeightedRandomChestContent(Item.field_111216_cf.itemID, 0, 1, 1, 1), new WeightedRandomChestContent(Item.field_111213_cg.itemID, 0, 1, 1, 1)};
 
     /**
      * List of Dispenser contents to be generated in the Jungle Pyramid dispensers.
@@ -20,31 +20,11 @@
     /** List of random stones to be generated in the Jungle Pyramid. */
     private static StructureScatteredFeatureStones junglePyramidsRandomScatteredStones = new StructureScatteredFeatureStones((ComponentScatteredFeaturePieces2)null);
 
-    public ComponentScatteredFeatureJunglePyramid() {}
-
     public ComponentScatteredFeatureJunglePyramid(Random par1Random, int par2, int par3)
     {
         super(par1Random, par2, 64, par3, 12, 10, 15);
     }
 
-    protected void func_143012_a(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143012_a(par1NBTTagCompound);
-        par1NBTTagCompound.setBoolean("placedMainChest", this.field_74947_h);
-        par1NBTTagCompound.setBoolean("placedHiddenChest", this.field_74948_i);
-        par1NBTTagCompound.setBoolean("placedTrap1", this.field_74945_j);
-        par1NBTTagCompound.setBoolean("placedTrap2", this.field_74946_k);
-    }
-
-    protected void func_143011_b(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143011_b(par1NBTTagCompound);
-        this.field_74947_h = par1NBTTagCompound.getBoolean("placedMainChest");
-        this.field_74948_i = par1NBTTagCompound.getBoolean("placedHiddenChest");
-        this.field_74945_j = par1NBTTagCompound.getBoolean("placedTrap1");
-        this.field_74946_k = par1NBTTagCompound.getBoolean("placedTrap2");
-    }
-
     /**
      * second Part of Structure generating, this for example places Spiderwebs, Mob Spawners, it closes Mineshafts at
      * the end, it adds Fences...
