--- a/net/minecraft/src/ComponentVillageStartPiece.java
+++ b/net/minecraft/src/ComponentVillageStartPiece.java
@@ -6,13 +6,13 @@
 
 public class ComponentVillageStartPiece extends ComponentVillageWell
 {
-    public WorldChunkManager worldChunkMngr;
+    public final WorldChunkManager worldChunkMngr;
 
     /** Boolean that determines if the village is in a desert or not. */
-    public boolean inDesert;
+    public final boolean inDesert;
 
     /** World terrain type, 0 for normal, 1 for flap map */
-    public int terrainType;
+    public final int terrainType;
     public StructureVillagePieceWeight structVillagePieceWeight;
 
     /**
@@ -23,8 +23,6 @@
     public List field_74932_i = new ArrayList();
     public List field_74930_j = new ArrayList();
 
-    public ComponentVillageStartPiece() {}
-
     public ComponentVillageStartPiece(WorldChunkManager par1WorldChunkManager, int par2, Random par3Random, int par4, int par5, List par6List, int par7)
     {
         super((ComponentVillageStartPiece)null, 0, par3Random, par4, par5);
@@ -33,6 +31,7 @@
         this.terrainType = par7;
         BiomeGenBase var8 = par1WorldChunkManager.getBiomeGenAt(par4, par5);
         this.inDesert = var8 == BiomeGenBase.desert || var8 == BiomeGenBase.desertHills;
+        this.startPiece = this;
     }
 
     public WorldChunkManager getWorldChunkManager()
