--- a/net/minecraft/src/ComponentNetherBridgeEnd.java
+++ b/net/minecraft/src/ComponentNetherBridgeEnd.java
@@ -7,8 +7,6 @@
 {
     private int fillSeed;
 
-    public ComponentNetherBridgeEnd() {}
-
     public ComponentNetherBridgeEnd(int par1, Random par2Random, StructureBoundingBox par3StructureBoundingBox, int par4)
     {
         super(par1);
@@ -23,18 +21,6 @@
         return isAboveGround(var7) && StructureComponent.findIntersecting(par0List, var7) == null ? new ComponentNetherBridgeEnd(par6, par1Random, var7, par5) : null;
     }
 
-    protected void func_143011_b(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143011_b(par1NBTTagCompound);
-        this.fillSeed = par1NBTTagCompound.getInteger("Seed");
-    }
-
-    protected void func_143012_a(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143012_a(par1NBTTagCompound);
-        par1NBTTagCompound.setInteger("Seed", this.fillSeed);
-    }
-
     /**
      * second Part of Structure generating, this for example places Spiderwebs, Mob Spawners, it closes Mineshafts at
      * the end, it adds Fences...
