--- a/net/minecraft/src/ComponentNetherBridgeCrossing3.java
+++ b/net/minecraft/src/ComponentNetherBridgeCrossing3.java
@@ -23,7 +23,6 @@
             case 2:
                 this.boundingBox = new StructureBoundingBox(par2, 64, par3, par2 + 19 - 1, 73, par3 + 19 - 1);
                 break;
-
             default:
                 this.boundingBox = new StructureBoundingBox(par2, 64, par3, par2 + 19 - 1, 73, par3 + 19 - 1);
         }
