--- a/net/minecraft/src/ComponentMineshaftCross.java
+++ b/net/minecraft/src/ComponentMineshaftCross.java
@@ -5,22 +5,8 @@
 
 public class ComponentMineshaftCross extends StructureComponent
 {
-    private int corridorDirection;
-    private boolean isMultipleFloors;
-
-    public ComponentMineshaftCross() {}
-
-    protected void func_143012_a(NBTTagCompound par1NBTTagCompound)
-    {
-        par1NBTTagCompound.setBoolean("tf", this.isMultipleFloors);
-        par1NBTTagCompound.setInteger("D", this.corridorDirection);
-    }
-
-    protected void func_143011_b(NBTTagCompound par1NBTTagCompound)
-    {
-        this.isMultipleFloors = par1NBTTagCompound.getBoolean("tf");
-        this.corridorDirection = par1NBTTagCompound.getInteger("D");
-    }
+    private final int corridorDirection;
+    private final boolean isMultipleFloors;
 
     public ComponentMineshaftCross(int par1, Random par2Random, StructureBoundingBox par3StructureBoundingBox, int par4)
     {
@@ -46,19 +32,16 @@
                 var6.maxX = par2 + 3;
                 var6.maxZ = par4 + 4;
                 break;
-
             case 1:
                 var6.minX = par2 - 4;
                 var6.minZ = par4 - 1;
                 var6.maxZ = par4 + 3;
                 break;
-
             case 2:
                 var6.minX = par2 - 1;
                 var6.maxX = par2 + 3;
                 var6.minZ = par4 - 4;
                 break;
-
             case 3:
                 var6.maxX = par2 + 4;
                 var6.minZ = par4 - 1;
@@ -82,19 +65,16 @@
                 StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX - 1, this.boundingBox.minY, this.boundingBox.minZ + 1, 1, var4);
                 StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.maxX + 1, this.boundingBox.minY, this.boundingBox.minZ + 1, 3, var4);
                 break;
-
             case 1:
                 StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX + 1, this.boundingBox.minY, this.boundingBox.minZ - 1, 2, var4);
                 StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX + 1, this.boundingBox.minY, this.boundingBox.maxZ + 1, 0, var4);
                 StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX - 1, this.boundingBox.minY, this.boundingBox.minZ + 1, 1, var4);
                 break;
-
             case 2:
                 StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX + 1, this.boundingBox.minY, this.boundingBox.minZ - 1, 2, var4);
                 StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX - 1, this.boundingBox.minY, this.boundingBox.minZ + 1, 1, var4);
                 StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.maxX + 1, this.boundingBox.minY, this.boundingBox.minZ + 1, 3, var4);
                 break;
-
             case 3:
                 StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX + 1, this.boundingBox.minY, this.boundingBox.minZ - 1, 2, var4);
                 StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX + 1, this.boundingBox.minY, this.boundingBox.maxZ + 1, 0, var4);
