--- a/net/minecraft/src/ComponentStrongholdCrossing.java
+++ b/net/minecraft/src/ComponentStrongholdCrossing.java
@@ -5,18 +5,17 @@
 
 public class ComponentStrongholdCrossing extends ComponentStronghold
 {
+    protected final EnumDoor doorType;
     private boolean field_74996_b;
     private boolean field_74997_c;
     private boolean field_74995_d;
     private boolean field_74999_h;
 
-    public ComponentStrongholdCrossing() {}
-
     public ComponentStrongholdCrossing(int par1, Random par2Random, StructureBoundingBox par3StructureBoundingBox, int par4)
     {
         super(par1);
         this.coordBaseMode = par4;
-        this.field_143013_d = this.getRandomDoor(par2Random);
+        this.doorType = this.getRandomDoor(par2Random);
         this.boundingBox = par3StructureBoundingBox;
         this.field_74996_b = par2Random.nextBoolean();
         this.field_74997_c = par2Random.nextBoolean();
@@ -24,24 +23,6 @@
         this.field_74999_h = par2Random.nextInt(3) > 0;
     }
 
-    protected void func_143012_a(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143012_a(par1NBTTagCompound);
-        par1NBTTagCompound.setBoolean("leftLow", this.field_74996_b);
-        par1NBTTagCompound.setBoolean("leftHigh", this.field_74997_c);
-        par1NBTTagCompound.setBoolean("rightLow", this.field_74995_d);
-        par1NBTTagCompound.setBoolean("rightHigh", this.field_74999_h);
-    }
-
-    protected void func_143011_b(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143011_b(par1NBTTagCompound);
-        this.field_74996_b = par1NBTTagCompound.getBoolean("leftLow");
-        this.field_74997_c = par1NBTTagCompound.getBoolean("leftHigh");
-        this.field_74995_d = par1NBTTagCompound.getBoolean("rightLow");
-        this.field_74999_h = par1NBTTagCompound.getBoolean("rightHigh");
-    }
-
     /**
      * Initiates construction of the Structure Component picked, at the current Location of StructGen
      */
@@ -98,7 +79,7 @@
         else
         {
             this.fillWithRandomizedBlocks(par1World, par3StructureBoundingBox, 0, 0, 0, 9, 8, 10, true, par2Random, StructureStrongholdPieces.getStrongholdStones());
-            this.placeDoor(par1World, par2Random, par3StructureBoundingBox, this.field_143013_d, 4, 3, 0);
+            this.placeDoor(par1World, par2Random, par3StructureBoundingBox, this.doorType, 4, 3, 0);
 
             if (this.field_74996_b)
             {
