--- a/net/minecraft/src/ComponentNetherBridgeStairs.java
+++ b/net/minecraft/src/ComponentNetherBridgeStairs.java
@@ -5,8 +5,6 @@
 
 public class ComponentNetherBridgeStairs extends ComponentNetherBridgePiece
 {
-    public ComponentNetherBridgeStairs() {}
-
     public ComponentNetherBridgeStairs(int par1, Random par2Random, StructureBoundingBox par3StructureBoundingBox, int par4)
     {
         super(par1);
