--- a/net/minecraft/src/ComponentNetherBridgeCorridor.java
+++ b/net/minecraft/src/ComponentNetherBridgeCorridor.java
@@ -7,8 +7,6 @@
 {
     private boolean field_111021_b;
 
-    public ComponentNetherBridgeCorridor() {}
-
     public ComponentNetherBridgeCorridor(int par1, Random par2Random, StructureBoundingBox par3StructureBoundingBox, int par4)
     {
         super(par1);
@@ -17,18 +15,6 @@
         this.field_111021_b = par2Random.nextInt(3) == 0;
     }
 
-    protected void func_143011_b(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143011_b(par1NBTTagCompound);
-        this.field_111021_b = par1NBTTagCompound.getBoolean("Chest");
-    }
-
-    protected void func_143012_a(NBTTagCompound par1NBTTagCompound)
-    {
-        super.func_143012_a(par1NBTTagCompound);
-        par1NBTTagCompound.setBoolean("Chest", this.field_111021_b);
-    }
-
     /**
      * Initiates construction of the Structure Component picked, at the current Location of StructGen
      */
