--- a/net/minecraft/src/ComponentNetherBridgeStraight.java
+++ b/net/minecraft/src/ComponentNetherBridgeStraight.java
@@ -5,8 +5,6 @@
 
 public class ComponentNetherBridgeStraight extends ComponentNetherBridgePiece
 {
-    public ComponentNetherBridgeStraight() {}
-
     public ComponentNetherBridgeStraight(int par1, Random par2Random, StructureBoundingBox par3StructureBoundingBox, int par4)
     {
         super(par1);
