--- a/net/minecraft/src/ComponentScatteredFeature.java
+++ b/net/minecraft/src/ComponentScatteredFeature.java
@@ -28,7 +28,6 @@
             case 2:
                 this.boundingBox = new StructureBoundingBox(par2, par3, par4, par2 + par5 - 1, par3 + par6 - 1, par4 + par7 - 1);
                 break;
-
             default:
                 this.boundingBox = new StructureBoundingBox(par2, par3, par4, par2 + par7 - 1, par3 + par6 - 1, par4 + par5 - 1);
         }
