--- a/net/minecraft/src/ComponentVillageWell.java
+++ b/net/minecraft/src/ComponentVillageWell.java
@@ -19,7 +19,6 @@
             case 2:
                 this.boundingBox = new StructureBoundingBox(par4, 64, par5, par4 + 6 - 1, 78, par5 + 6 - 1);
                 break;
-
             default:
                 this.boundingBox = new StructureBoundingBox(par4, 64, par5, par4 + 6 - 1, 78, par5 + 6 - 1);
         }
