--- a/net/minecraft/src/ComponentNetherBridgeCorridor4.java
+++ b/net/minecraft/src/ComponentNetherBridgeCorridor4.java
@@ -5,8 +5,6 @@
 
 public class ComponentNetherBridgeCorridor4 extends ComponentNetherBridgePiece
 {
-    public ComponentNetherBridgeCorridor4() {}
-
     public ComponentNetherBridgeCorridor4(int par1, Random par2Random, StructureBoundingBox par3StructureBoundingBox, int par4)
     {
         super(par1);
