--- a/net/minecraft/src/ComponentMineshaftCorridor.java
+++ b/net/minecraft/src/ComponentMineshaftCorridor.java
@@ -5,8 +5,8 @@
 
 public class ComponentMineshaftCorridor extends StructureComponent
 {
-    private boolean hasRails;
-    private boolean hasSpiders;
+    private final boolean hasRails;
+    private final boolean hasSpiders;
     private boolean spawnerPlaced;
 
     /**
@@ -14,24 +14,6 @@
      */
     private int sectionCount;
 
-    public ComponentMineshaftCorridor() {}
-
-    protected void func_143012_a(NBTTagCompound par1NBTTagCompound)
-    {
-        par1NBTTagCompound.setBoolean("hr", this.hasRails);
-        par1NBTTagCompound.setBoolean("sc", this.hasSpiders);
-        par1NBTTagCompound.setBoolean("hps", this.spawnerPlaced);
-        par1NBTTagCompound.setInteger("Num", this.sectionCount);
-    }
-
-    protected void func_143011_b(NBTTagCompound par1NBTTagCompound)
-    {
-        this.hasRails = par1NBTTagCompound.getBoolean("hr");
-        this.hasSpiders = par1NBTTagCompound.getBoolean("sc");
-        this.spawnerPlaced = par1NBTTagCompound.getBoolean("hps");
-        this.sectionCount = par1NBTTagCompound.getInteger("Num");
-    }
-
     public ComponentMineshaftCorridor(int par1, Random par2Random, StructureBoundingBox par3StructureBoundingBox, int par4)
     {
         super(par1);
@@ -65,17 +47,14 @@
                     var6.maxX = par2 + 2;
                     var6.maxZ = par4 + (var8 - 1);
                     break;
-
                 case 1:
                     var6.minX = par2 - (var8 - 1);
                     var6.maxZ = par4 + 2;
                     break;
-
                 case 2:
                     var6.maxX = par2 + 2;
                     var6.minZ = par4 - (var8 - 1);
                     break;
-
                 case 3:
                     var6.maxX = par2 + (var8 - 1);
                     var6.maxZ = par4 + 2;
@@ -101,6 +80,7 @@
         switch (this.coordBaseMode)
         {
             case 0:
+
                 if (var5 <= 1)
                 {
                     StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX, this.boundingBox.minY - 1 + par3Random.nextInt(3), this.boundingBox.maxZ + 1, this.coordBaseMode, var4);
@@ -115,8 +95,8 @@
                 }
 
                 break;
-
             case 1:
+
                 if (var5 <= 1)
                 {
                     StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX - 1, this.boundingBox.minY - 1 + par3Random.nextInt(3), this.boundingBox.minZ, this.coordBaseMode, var4);
@@ -131,8 +111,8 @@
                 }
 
                 break;
-
             case 2:
+
                 if (var5 <= 1)
                 {
                     StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.minX, this.boundingBox.minY - 1 + par3Random.nextInt(3), this.boundingBox.minZ - 1, this.coordBaseMode, var4);
@@ -147,8 +127,8 @@
                 }
 
                 break;
-
             case 3:
+
                 if (var5 <= 1)
                 {
                     StructureMineshaftPieces.getNextComponent(par1StructureComponent, par2List, par3Random, this.boundingBox.maxX + 1, this.boundingBox.minY - 1 + par3Random.nextInt(3), this.boundingBox.minZ, this.coordBaseMode, var4);
