--- a/net/minecraft/src/Config.java
+++ b/net/minecraft/src/Config.java
@@ -36,8 +36,8 @@
     public static final String OF_NAME = "OptiFine";
     public static final String MC_VERSION = "1.6.2";
     public static final String OF_EDITION = "HD_U";
-    public static final String OF_RELEASE = "B2";
-    public static final String VERSION = "OptiFine_1.6.2_HD_U_B2";
+    public static final String OF_RELEASE = "B1";
+    public static final String VERSION = "OptiFine_1.6.2_HD_U_B1";
     private static String newRelease = null;
     private static GameSettings gameSettings = null;
     private static Minecraft minecraft = null;
@@ -67,7 +67,7 @@
 
     public static String getVersion()
     {
-        return "OptiFine_1.6.2_HD_U_B2";
+        return "OptiFine_1.6.2_HD_U_B1";
     }
 
     public static void initGameSettings(GameSettings settings)
