--- a/net/minecraft/src/Config.java
+++ b/net/minecraft/src/Config.java
@@ -222,26 +222,24 @@
             {
                 case 0:
                     return 9984;
-
                 case 1:
                     return 9986;
-
                 case 2:
+
                     if (isMultiTexture())
                     {
                         return 9985;
                     }
 
                     return 9986;
-
                 case 3:
+
                     if (isMultiTexture())
                     {
                         return 9987;
                     }
 
                     return 9986;
-
                 default:
                     return 9984;
             }
@@ -721,15 +719,12 @@
                     case 2:
                         --z;
                         break;
-
                     case 3:
                         ++z;
                         break;
-
                     case 4:
                         --x;
                         break;
-
                     case 5:
                         ++x;
                 }
@@ -761,15 +756,12 @@
                     case 2:
                         --z;
                         break;
-
                     case 3:
                         ++z;
                         break;
-
                     case 4:
                         --x;
                         break;
-
                     case 5:
                         ++x;
                 }
