--- a/net/minecraft/src/ConnectedProperties.java
+++ b/net/minecraft/src/ConnectedProperties.java
@@ -605,25 +605,18 @@
                         {
                             case 1:
                                 return this.isValidCtm(path);
-
                             case 2:
                                 return this.isValidHorizontal(path);
-
                             case 3:
                                 return this.isValidTop(path);
-
                             case 4:
                                 return this.isValidRandom(path);
-
                             case 5:
                                 return this.isValidRepeat(path);
-
                             case 6:
                                 return this.isValidVertical(path);
-
                             case 7:
                                 return this.isValidFixed(path);
-
                             default:
                                 Config.warn("Unknown method: " + path);
                                 return false;
