--- a/net/minecraft/src/ContainerRepair.java
+++ b/net/minecraft/src/ContainerRepair.java
@@ -210,10 +210,8 @@
                                 case 1:
                                     var23 = 8;
                                     break;
-
                                 case 2:
                                     var23 = 4;
-
                                 case 3:
                                 case 4:
                                 case 6:
@@ -222,11 +220,9 @@
                                 case 9:
                                 default:
                                     break;
-
                                 case 5:
                                     var23 = 2;
                                     break;
-
                                 case 10:
                                     var23 = 1;
                             }
@@ -279,10 +275,8 @@
                     case 1:
                         var14 = 8;
                         break;
-
                     case 2:
                         var14 = 4;
-
                     case 3:
                     case 4:
                     case 6:
@@ -291,11 +285,9 @@
                     case 9:
                     default:
                         break;
-
                     case 5:
                         var14 = 2;
                         break;
-
                     case 10:
                         var14 = 1;
                 }
