--- a/net/minecraft/src/ContainerEnchantment.java
+++ b/net/minecraft/src/ContainerEnchantment.java
@@ -192,7 +192,7 @@
                         {
                             if (var5)
                             {
-                                Item.enchantedBook.addEnchantment(var3, var8);
+                                Item.enchantedBook.getEnchantedItemStack_do(var3, var8);
                             }
                             else
                             {
