--- a/net/minecraft/src/ContainerHorseInventorySlotArmor.java
+++ b/net/minecraft/src/ContainerHorseInventorySlotArmor.java
@@ -2,7 +2,7 @@
 
 class ContainerHorseInventorySlotArmor extends Slot
 {
-    final EntityHorse theHorse;
+    final EntityHorse field_111241_a;
 
     final ContainerHorseInventory field_111240_b;
 
@@ -10,7 +10,7 @@
     {
         super(par2IInventory, par3, par4, par5);
         this.field_111240_b = par1ContainerHorseInventory;
-        this.theHorse = par6EntityHorse;
+        this.field_111241_a = par6EntityHorse;
     }
 
     /**
@@ -18,11 +18,11 @@
      */
     public boolean isItemValid(ItemStack par1ItemStack)
     {
-        return super.isItemValid(par1ItemStack) && this.theHorse.func_110259_cr() && EntityHorse.func_110211_v(par1ItemStack.itemID);
+        return super.isItemValid(par1ItemStack) && this.field_111241_a.func_110259_cr() && EntityHorse.func_110211_v(par1ItemStack.itemID);
     }
 
     public boolean func_111238_b()
     {
-        return this.theHorse.func_110259_cr();
+        return this.field_111241_a.func_110259_cr();
     }
 }
