--- a/net/minecraft/src/ContainerHorseInventory.java
+++ b/net/minecraft/src/ContainerHorseInventory.java
@@ -3,12 +3,12 @@
 public class ContainerHorseInventory extends Container
 {
     private IInventory field_111243_a;
-    private EntityHorse theHorse;
+    private EntityHorse field_111242_f;
 
     public ContainerHorseInventory(IInventory par1IInventory, IInventory par2IInventory, EntityHorse par3EntityHorse)
     {
         this.field_111243_a = par2IInventory;
-        this.theHorse = par3EntityHorse;
+        this.field_111242_f = par3EntityHorse;
         byte var4 = 3;
         par2IInventory.openChest();
         int var5 = (var4 - 4) * 18;
@@ -17,7 +17,7 @@
         int var6;
         int var7;
 
-        if (par3EntityHorse.isChested())
+        if (par3EntityHorse.func_110261_ca())
         {
             for (var6 = 0; var6 < var4; ++var6)
             {
@@ -44,7 +44,7 @@
 
     public boolean canInteractWith(EntityPlayer par1EntityPlayer)
     {
-        return this.field_111243_a.isUseableByPlayer(par1EntityPlayer) && this.theHorse.isEntityAlive() && this.theHorse.getDistanceToEntity(par1EntityPlayer) < 8.0F;
+        return this.field_111243_a.isUseableByPlayer(par1EntityPlayer) && this.field_111242_f.isEntityAlive() && this.field_111242_f.getDistanceToEntity(par1EntityPlayer) < 8.0F;
     }
 
     /**
