--- a/net/minecraft/src/ConvertingProgressUpdate.java
+++ b/net/minecraft/src/ConvertingProgressUpdate.java
@@ -12,7 +12,7 @@
     public ConvertingProgressUpdate(MinecraftServer par1MinecraftServer)
     {
         this.mcServer = par1MinecraftServer;
-        this.field_96245_b = MinecraftServer.getSystemTimeMillis();
+        this.field_96245_b = MinecraftServer.func_130071_aq();
     }
 
     /**
@@ -25,9 +25,9 @@
      */
     public void setLoadingProgress(int par1)
     {
-        if (MinecraftServer.getSystemTimeMillis() - this.field_96245_b >= 1000L)
+        if (MinecraftServer.func_130071_aq() - this.field_96245_b >= 1000L)
         {
-            this.field_96245_b = MinecraftServer.getSystemTimeMillis();
+            this.field_96245_b = MinecraftServer.func_130071_aq();
             this.mcServer.getLogAgent().logInfo("Converting... " + par1 + "%");
         }
     }
