--- a/net/minecraft/src/CustomColorizer.java
+++ b/net/minecraft/src/CustomColorizer.java
@@ -570,8 +570,8 @@
                         case 1:
                             var17 = blockAccess.getBiomeGenForCoords(r, g).getBiomeGrassColor();
                             break;
-
                         case 2:
+
                             if ((metadata & 3) == 1)
                             {
                                 var17 = ColorizerFoliage.getFoliageColorPine();
@@ -586,7 +586,6 @@
                             }
 
                             break;
-
                         default:
                             var17 = block.colorMultiplier(blockAccess, r, y, g);
                     }
