--- a/net/minecraft/src/CustomColorizer.java
+++ b/net/minecraft/src/CustomColorizer.java
@@ -908,7 +908,7 @@
         }
     }
 
-    public static boolean updateLightmap(World world, float torchFlickerX, int[] lmColors, boolean nightvision)
+    public static boolean updateLightmap(World world, EntityRenderer entityRenderer, int[] lmColors, boolean nightvision)
     {
         if (world == null)
         {
@@ -971,7 +971,7 @@
 
                             sun = Config.limitTo1(sun);
                             float sunX = sun * (float)(width - 1);
-                            float torchX = Config.limitTo1(torchFlickerX + 0.5F) * (float)(width - 1);
+                            float torchX = Config.limitTo1(entityRenderer.torchFlickerX + 0.5F) * (float)(width - 1);
                             float gamma = Config.limitTo1(Config.getGameSettings().gammaSetting);
                             boolean hasGamma = gamma > 1.0E-4F;
                             getLightMapColumn(lightMapRgb, sunX, startIndex, width, sunRgbs);
