--- a/net/minecraft/src/CustomSkyLayer.java
+++ b/net/minecraft/src/CustomSkyLayer.java
@@ -370,42 +370,36 @@
                 GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE);
                 GL11.glColor4f(1.0F, 1.0F, 1.0F, brightness);
                 break;
-
             case 1:
                 GL11.glDisable(GL11.GL_ALPHA_TEST);
                 GL11.glEnable(GL11.GL_BLEND);
                 GL11.glBlendFunc(GL11.GL_ONE_MINUS_DST_COLOR, GL11.GL_ZERO);
                 GL11.glColor4f(brightness, brightness, brightness, 1.0F);
                 break;
-
             case 2:
                 GL11.glDisable(GL11.GL_ALPHA_TEST);
                 GL11.glEnable(GL11.GL_BLEND);
                 GL11.glBlendFunc(GL11.GL_DST_COLOR, GL11.GL_ONE_MINUS_SRC_ALPHA);
                 GL11.glColor4f(brightness, brightness, brightness, brightness);
                 break;
-
             case 3:
                 GL11.glDisable(GL11.GL_ALPHA_TEST);
                 GL11.glEnable(GL11.GL_BLEND);
                 GL11.glBlendFunc(GL11.GL_ONE, GL11.GL_ONE);
                 GL11.glColor4f(brightness, brightness, brightness, 1.0F);
                 break;
-
             case 4:
                 GL11.glDisable(GL11.GL_ALPHA_TEST);
                 GL11.glEnable(GL11.GL_BLEND);
                 GL11.glBlendFunc(GL11.GL_ZERO, GL11.GL_ONE_MINUS_SRC_COLOR);
                 GL11.glColor4f(brightness, brightness, brightness, 1.0F);
                 break;
-
             case 5:
                 GL11.glDisable(GL11.GL_ALPHA_TEST);
                 GL11.glEnable(GL11.GL_BLEND);
                 GL11.glBlendFunc(GL11.GL_ONE, GL11.GL_ONE_MINUS_SRC_COLOR);
                 GL11.glColor4f(brightness, brightness, brightness, 1.0F);
                 break;
-
             case 6:
                 GL11.glEnable(GL11.GL_ALPHA_TEST);
                 GL11.glDisable(GL11.GL_BLEND);
