--- a/net/minecraft/src/CustomSky.java
+++ b/net/minecraft/src/CustomSky.java
@@ -74,7 +74,7 @@
                                 }
                                 else
                                 {
-                                    sl.textureId = tex.getGlTextureId();
+                                    sl.textureId = tex.func_110552_b();
                                     i.add(sl);
                                     in.close();
                                 }
