--- a/net/minecraft/src/DataWatcher.java
+++ b/net/minecraft/src/DataWatcher.java
@@ -85,7 +85,7 @@
         return ((Integer)this.getWatchedObject(par1).getObject()).intValue();
     }
 
-    public float getWatchableObjectFloat(int par1)
+    public float func_111145_d(int par1)
     {
         return ((Float)this.getWatchedObject(par1).getObject()).floatValue();
     }
@@ -253,28 +253,22 @@
             case 0:
                 par0DataOutput.writeByte(((Byte)par1WatchableObject.getObject()).byteValue());
                 break;
-
             case 1:
                 par0DataOutput.writeShort(((Short)par1WatchableObject.getObject()).shortValue());
                 break;
-
             case 2:
                 par0DataOutput.writeInt(((Integer)par1WatchableObject.getObject()).intValue());
                 break;
-
             case 3:
                 par0DataOutput.writeFloat(((Float)par1WatchableObject.getObject()).floatValue());
                 break;
-
             case 4:
                 Packet.writeString((String)par1WatchableObject.getObject(), par0DataOutput);
                 break;
-
             case 5:
                 ItemStack var4 = (ItemStack)par1WatchableObject.getObject();
                 Packet.writeItemStack(var4, par0DataOutput);
                 break;
-
             case 6:
                 ChunkCoordinates var3 = (ChunkCoordinates)par1WatchableObject.getObject();
                 par0DataOutput.writeInt(var3.posX);
@@ -303,27 +297,21 @@
                 case 0:
                     var5 = new WatchableObject(var3, var4, Byte.valueOf(par0DataInput.readByte()));
                     break;
-
                 case 1:
                     var5 = new WatchableObject(var3, var4, Short.valueOf(par0DataInput.readShort()));
                     break;
-
                 case 2:
                     var5 = new WatchableObject(var3, var4, Integer.valueOf(par0DataInput.readInt()));
                     break;
-
                 case 3:
                     var5 = new WatchableObject(var3, var4, Float.valueOf(par0DataInput.readFloat()));
                     break;
-
                 case 4:
                     var5 = new WatchableObject(var3, var4, Packet.readString(par0DataInput, 64));
                     break;
-
                 case 5:
                     var5 = new WatchableObject(var3, var4, Packet.readItemStack(par0DataInput));
                     break;
-
                 case 6:
                     int var6 = par0DataInput.readInt();
                     int var7 = par0DataInput.readInt();
