--- a/net/minecraft/src/DefaultResourcePack.java
+++ b/net/minecraft/src/DefaultResourcePack.java
@@ -14,19 +14,19 @@
 
 public class DefaultResourcePack implements ResourcePack
 {
-    public static final Set defaultResourceDomains = ImmutableSet.of("minecraft");
-    private final Map mapResourceFiles = Maps.newHashMap();
-    private final File fileAssets;
+    public static final Set field_110608_a = ImmutableSet.of("minecraft");
+    private final Map field_110606_b = Maps.newHashMap();
+    private final File field_110607_c;
 
     public DefaultResourcePack(File par1File)
     {
-        this.fileAssets = par1File;
-        this.readAssetsDir(this.fileAssets);
+        this.field_110607_c = par1File;
+        this.func_110603_a(this.field_110607_c);
     }
 
-    public InputStream getInputStream(ResourceLocation par1ResourceLocation) throws IOException
+    public InputStream func_110590_a(ResourceLocation par1ResourceLocation) throws IOException
     {
-        InputStream var2 = this.getResourceStream(par1ResourceLocation);
+        InputStream var2 = this.func_110605_c(par1ResourceLocation);
 
         if (var2 != null)
         {
@@ -34,7 +34,7 @@
         }
         else
         {
-            File var3 = (File)this.mapResourceFiles.get(par1ResourceLocation.toString());
+            File var3 = (File)this.field_110606_b.get(par1ResourceLocation.toString());
 
             if (var3 != null)
             {
@@ -42,32 +42,32 @@
             }
             else
             {
-                throw new FileNotFoundException(par1ResourceLocation.getResourcePath());
+                throw new FileNotFoundException(par1ResourceLocation.func_110623_a());
             }
         }
     }
 
-    private InputStream getResourceStream(ResourceLocation par1ResourceLocation)
+    private InputStream func_110605_c(ResourceLocation par1ResourceLocation)
     {
-        return DefaultResourcePack.class.getResourceAsStream("/assets/minecraft/" + par1ResourceLocation.getResourcePath());
+        return DefaultResourcePack.class.getResourceAsStream("/assets/minecraft/" + par1ResourceLocation.func_110623_a());
     }
 
-    public void addResourceFile(String par1Str, File par2File)
+    public void func_110604_a(String par1Str, File par2File)
     {
-        this.mapResourceFiles.put((new ResourceLocation(par1Str)).toString(), par2File);
+        this.field_110606_b.put((new ResourceLocation(par1Str)).toString(), par2File);
     }
 
-    public boolean resourceExists(ResourceLocation par1ResourceLocation)
+    public boolean func_110589_b(ResourceLocation par1ResourceLocation)
     {
-        return this.getResourceStream(par1ResourceLocation) != null || this.mapResourceFiles.containsKey(par1ResourceLocation.toString());
+        return this.func_110605_c(par1ResourceLocation) != null || this.field_110606_b.containsKey(par1ResourceLocation.toString());
     }
 
-    public Set getResourceDomains()
+    public Set func_110587_b()
     {
-        return defaultResourceDomains;
+        return field_110608_a;
     }
 
-    public void readAssetsDir(File par1File)
+    public void func_110603_a(File par1File)
     {
         if (par1File.isDirectory())
         {
@@ -77,26 +77,26 @@
             for (int var4 = 0; var4 < var3; ++var4)
             {
                 File var5 = var2[var4];
-                this.readAssetsDir(var5);
+                this.func_110603_a(var5);
             }
         }
         else
         {
-            this.addResourceFile(AbstractResourcePack.getRelativeName(this.fileAssets, par1File), par1File);
+            this.func_110604_a(AbstractResourcePack.func_110595_a(this.field_110607_c, par1File), par1File);
         }
     }
 
-    public MetadataSection getPackMetadata(MetadataSerializer par1MetadataSerializer, String par2Str) throws IOException
+    public MetadataSection func_135058_a(MetadataSerializer par1MetadataSerializer, String par2Str) throws IOException
     {
-        return AbstractResourcePack.readMetadata(par1MetadataSerializer, DefaultResourcePack.class.getResourceAsStream("/" + (new ResourceLocation("pack.mcmeta")).getResourcePath()), par2Str);
+        return AbstractResourcePack.func_110596_a(par1MetadataSerializer, DefaultResourcePack.class.getResourceAsStream("/" + (new ResourceLocation("pack.mcmeta")).func_110623_a()), par2Str);
     }
 
-    public BufferedImage getPackImage() throws IOException
+    public BufferedImage func_110586_a() throws IOException
     {
-        return ImageIO.read(DefaultResourcePack.class.getResourceAsStream("/" + (new ResourceLocation("pack.png")).getResourcePath()));
+        return ImageIO.read(DefaultResourcePack.class.getResourceAsStream("/" + (new ResourceLocation("pack.png")).func_110623_a()));
     }
 
-    public String getPackName()
+    public String func_130077_b()
     {
         return "Default";
     }
