--- a/net/minecraft/src/DispenserBehaviors.java
+++ b/net/minecraft/src/DispenserBehaviors.java
@@ -2,7 +2,7 @@
 
 public class DispenserBehaviors
 {
-    public static void registerDispenserBehaviours()
+    public static void func_96467_a()
     {
         BlockDispenser.dispenseBehaviorRegistry.putObject(Item.arrow, new DispenserBehaviorArrow());
         BlockDispenser.dispenseBehaviorRegistry.putObject(Item.egg, new DispenserBehaviorEgg());
