--- a/net/minecraft/src/EffectRenderer.java
+++ b/net/minecraft/src/EffectRenderer.java
@@ -85,11 +85,9 @@
                     default:
                         this.renderer.func_110577_a(field_110737_b);
                         break;
-
                     case 1:
                         this.renderer.func_110577_a(TextureMap.field_110575_b);
                         break;
-
                     case 2:
                         this.renderer.func_110577_a(TextureMap.field_110576_c);
                 }
