--- a/net/minecraft/src/EnchantmentProtection.java
+++ b/net/minecraft/src/EnchantmentProtection.java
@@ -102,10 +102,7 @@
         }
     }
 
-    /**
-     * Gets the amount of ticks an entity should be set fire, adjusted for fire protection.
-     */
-    public static int getFireTimeForEntity(Entity par0Entity, int par1)
+    public static int func_92093_a(Entity par0Entity, int par1)
     {
         int var2 = EnchantmentHelper.getMaxEnchantmentLevel(Enchantment.fireProtection.effectId, par0Entity.getLastActiveItems());
 
