--- a/net/minecraft/src/EntityMoveHelper.java
+++ b/net/minecraft/src/EntityMoveHelper.java
@@ -59,7 +59,7 @@
             {
                 float var10 = (float)(Math.atan2(var4, var2) * 180.0D / Math.PI) - 90.0F;
                 this.entity.rotationYaw = this.limitAngle(this.entity.rotationYaw, var10, 30.0F);
-                this.entity.setAIMoveSpeed((float)(this.speed * this.entity.getEntityAttribute(SharedMonsterAttributes.movementSpeed).getAttributeValue()));
+                this.entity.setAIMoveSpeed((float)(this.speed * this.entity.func_110148_a(SharedMonsterAttributes.field_111263_d).func_111126_e()));
 
                 if (var6 > 0.0D && var2 * var2 + var4 * var4 < 1.0D)
                 {
