--- a/net/minecraft/src/EntityAIOwnerHurtTarget.java
+++ b/net/minecraft/src/EntityAIOwnerHurtTarget.java
@@ -32,8 +32,8 @@
             }
             else
             {
-                this.theTarget = var1.getLastAttacker();
-                int var2 = var1.getLastAttackerTime();
+                this.theTarget = var1.func_110144_aD();
+                int var2 = var1.func_142013_aG();
                 return var2 != this.field_142050_e && this.isSuitableTarget(this.theTarget, false) && this.theEntityTameable.func_142018_a(this.theTarget, var1);
             }
         }
@@ -49,7 +49,7 @@
 
         if (var1 != null)
         {
-            this.field_142050_e = var1.getLastAttackerTime();
+            this.field_142050_e = var1.func_142013_aG();
         }
 
         super.startExecuting();
