--- a/net/minecraft/src/EntityAgeable.java
+++ b/net/minecraft/src/EntityAgeable.java
@@ -37,7 +37,7 @@
 
                         if (var2.hasDisplayName())
                         {
-                            var4.setCustomNameTag(var2.getDisplayName());
+                            var4.func_94058_c(var2.getDisplayName());
                         }
 
                         if (!par1EntityPlayer.capabilities.isCreativeMode)
@@ -77,11 +77,7 @@
         return this.dataWatcher.getWatchableObjectInt(12);
     }
 
-    /**
-     * "Adds the value of the parameter times 20 to the age of this entity. If the entity is an adult (if the entity's
-     * age is greater than 0), it will have no effect."
-     */
-    public void addGrowth(int par1)
+    public void func_110195_a(int par1)
     {
         int var2 = this.getGrowingAge();
         var2 += par1 * 20;
@@ -101,7 +97,7 @@
     public void setGrowingAge(int par1)
     {
         this.dataWatcher.updateObject(12, Integer.valueOf(par1));
-        this.setScaleForAge(this.isChild());
+        this.func_98054_a(this.isChild());
     }
 
     /**
@@ -132,7 +128,7 @@
 
         if (this.worldObj.isRemote)
         {
-            this.setScaleForAge(this.isChild());
+            this.func_98054_a(this.isChild());
         }
         else
         {
@@ -159,12 +155,9 @@
         return this.getGrowingAge() < 0;
     }
 
-    /**
-     * "Sets the scale for an ageable entity according to the boolean parameter, which says if it's a child."
-     */
-    public void setScaleForAge(boolean par1)
+    public void func_98054_a(boolean par1)
     {
-        this.setScale(par1 ? 0.5F : 1.0F);
+        this.func_98055_j(par1 ? 0.5F : 1.0F);
     }
 
     /**
@@ -178,11 +171,11 @@
 
         if (!var3)
         {
-            this.setScale(1.0F);
+            this.func_98055_j(1.0F);
         }
     }
 
-    protected final void setScale(float par1)
+    protected final void func_98055_j(float par1)
     {
         super.setSize(this.field_98056_d * par1, this.field_98057_e * par1);
     }
