--- a/net/minecraft/src/EntityPainting.java
+++ b/net/minecraft/src/EntityPainting.java
@@ -94,20 +94,17 @@
         super.readEntityFromNBT(par1NBTTagCompound);
     }
 
-    public int getWidthPixels()
+    public int func_82329_d()
     {
         return this.art.sizeX;
     }
 
-    public int getHeightPixels()
+    public int func_82330_g()
     {
         return this.art.sizeY;
     }
 
-    /**
-     * Called when this entity is broken. Entity parameter may be null.
-     */
-    public void onBroken(Entity par1Entity)
+    public void func_110128_b(Entity par1Entity)
     {
         if (par1Entity instanceof EntityPlayer)
         {
