--- a/net/minecraft/src/EntityAIMoveTowardsRestriction.java
+++ b/net/minecraft/src/EntityAIMoveTowardsRestriction.java
@@ -26,7 +26,7 @@
         }
         else
         {
-            ChunkCoordinates var1 = this.theEntity.getHomePosition();
+            ChunkCoordinates var1 = this.theEntity.func_110172_bL();
             Vec3 var2 = RandomPositionGenerator.findRandomTargetBlockTowards(this.theEntity, 16, 7, this.theEntity.worldObj.getWorldVec3Pool().getVecFromPool((double)var1.posX, (double)var1.posY, (double)var1.posZ));
 
             if (var2 == null)
