--- a/net/minecraft/src/EntitySheep.java
+++ b/net/minecraft/src/EntitySheep.java
@@ -66,11 +66,11 @@
         super.onLivingUpdate();
     }
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setAttribute(8.0D);
-        this.getEntityAttribute(SharedMonsterAttributes.movementSpeed).setAttribute(0.23000000417232513D);
+        super.func_110147_ax();
+        this.func_110148_a(SharedMonsterAttributes.field_111267_a).func_111128_a(8.0D);
+        this.func_110148_a(SharedMonsterAttributes.field_111263_d).func_111128_a(0.23000000417232513D);
     }
 
     protected void entityInit()
@@ -275,13 +275,13 @@
 
         if (this.isChild())
         {
-            this.addGrowth(60);
+            this.func_110195_a(60);
         }
     }
 
-    public EntityLivingData onSpawnWithEgg(EntityLivingData par1EntityLivingData)
+    public EntityLivingData func_110161_a(EntityLivingData par1EntityLivingData)
     {
-        par1EntityLivingData = super.onSpawnWithEgg(par1EntityLivingData);
+        par1EntityLivingData = super.func_110161_a(par1EntityLivingData);
         this.setFleeceColor(getRandomFleeceColor(this.worldObj.rand));
         return par1EntityLivingData;
     }
