--- a/net/minecraft/src/EntityBreakingFX.java
+++ b/net/minecraft/src/EntityBreakingFX.java
@@ -10,7 +10,7 @@
     public EntityBreakingFX(World par1World, double par2, double par4, double par6, Item par8Item, int par9)
     {
         super(par1World, par2, par4, par6, 0.0D, 0.0D, 0.0D);
-        this.setParticleIcon(par8Item.getIconFromDamage(par9));
+        this.func_110125_a(par8Item.getIconFromDamage(par9));
         this.particleRed = this.particleGreen = this.particleBlue = 1.0F;
         this.particleGravity = Block.blockSnow.blockParticleGravity;
         this.particleScale /= 2.0F;
