--- a/net/minecraft/src/EntityMob.java
+++ b/net/minecraft/src/EntityMob.java
@@ -83,7 +83,7 @@
 
     public boolean attackEntityAsMob(Entity par1Entity)
     {
-        float var2 = (float)this.getEntityAttribute(SharedMonsterAttributes.attackDamage).getAttributeValue();
+        float var2 = (float)this.func_110148_a(SharedMonsterAttributes.field_111264_e).func_111126_e();
         int var3 = 0;
 
         if (par1Entity instanceof EntityLivingBase)
@@ -177,9 +177,9 @@
         return this.worldObj.difficultySetting > 0 && this.isValidLightLevel() && super.getCanSpawnHere();
     }
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getAttributeMap().func_111150_b(SharedMonsterAttributes.attackDamage);
+        super.func_110147_ax();
+        this.func_110140_aT().func_111150_b(SharedMonsterAttributes.field_111264_e);
     }
 }
