--- a/net/minecraft/src/EntityZombie.java
+++ b/net/minecraft/src/EntityZombie.java
@@ -6,8 +6,8 @@
 public class EntityZombie extends EntityMob
 {
     protected static final Attribute field_110186_bp = (new RangedAttribute("zombie.spawnReinforcements", 0.0D, 0.0D, 1.0D)).func_111117_a("Spawn Reinforcements Chance");
-    private static final UUID babySpeedBoostUUID = UUID.fromString("B9766B59-9566-4402-BC1F-2EE2A276D836");
-    private static final AttributeModifier babySpeedBoostModifier = new AttributeModifier(babySpeedBoostUUID, "Baby speed boost", 0.5D, 1);
+    private static final UUID field_110187_bq = UUID.fromString("B9766B59-9566-4402-BC1F-2EE2A276D836");
+    private static final AttributeModifier field_110188_br = new AttributeModifier(field_110187_bq, "Baby speed boost", 0.5D, 1);
 
     /**
      * Ticker used to determine the time remaining for this zombie to convert into a villager when cured.
@@ -32,13 +32,13 @@
         this.targetTasks.addTask(2, new EntityAINearestAttackableTarget(this, EntityVillager.class, 0, false));
     }
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getEntityAttribute(SharedMonsterAttributes.followRange).setAttribute(40.0D);
-        this.getEntityAttribute(SharedMonsterAttributes.movementSpeed).setAttribute(0.23000000417232513D);
-        this.getEntityAttribute(SharedMonsterAttributes.attackDamage).setAttribute(3.0D);
-        this.getAttributeMap().func_111150_b(field_110186_bp).setAttribute(this.rand.nextDouble() * 0.10000000149011612D);
+        super.func_110147_ax();
+        this.func_110148_a(SharedMonsterAttributes.field_111265_b).func_111128_a(40.0D);
+        this.func_110148_a(SharedMonsterAttributes.field_111263_d).func_111128_a(0.23000000417232513D);
+        this.func_110148_a(SharedMonsterAttributes.field_111264_e).func_111128_a(3.0D);
+        this.func_110140_aT().func_111150_b(field_110186_bp).func_111128_a(this.rand.nextDouble() * 0.10000000149011612D);
     }
 
     protected void entityInit()
@@ -89,12 +89,12 @@
 
         if (this.worldObj != null && !this.worldObj.isRemote)
         {
-            AttributeInstance var2 = this.getEntityAttribute(SharedMonsterAttributes.movementSpeed);
-            var2.removeModifier(babySpeedBoostModifier);
+            AttributeInstance var2 = this.func_110148_a(SharedMonsterAttributes.field_111263_d);
+            var2.func_111124_b(field_110188_br);
 
             if (par1)
             {
-                var2.applyModifier(babySpeedBoostModifier);
+                var2.func_111121_a(field_110188_br);
             }
         }
     }
@@ -179,7 +179,7 @@
                 var3 = (EntityLivingBase)par1DamageSource.getEntity();
             }
 
-            if (var3 != null && this.worldObj.difficultySetting >= 3 && (double)this.rand.nextFloat() < this.getEntityAttribute(field_110186_bp).getAttributeValue())
+            if (var3 != null && this.worldObj.difficultySetting >= 3 && (double)this.rand.nextFloat() < this.func_110148_a(field_110186_bp).func_111126_e())
             {
                 int var4 = MathHelper.floor_double(this.posX);
                 int var5 = MathHelper.floor_double(this.posY);
@@ -200,9 +200,9 @@
                         {
                             this.worldObj.spawnEntityInWorld(var7);
                             var7.setAttackTarget(var3);
-                            var7.onSpawnWithEgg((EntityLivingData)null);
-                            this.getEntityAttribute(field_110186_bp).applyModifier(new AttributeModifier("Zombie reinforcement caller charge", -0.05000000074505806D, 0));
-                            var7.getEntityAttribute(field_110186_bp).applyModifier(new AttributeModifier("Zombie reinforcement callee charge", -0.05000000074505806D, 0));
+                            var7.func_110161_a((EntityLivingData)null);
+                            this.func_110148_a(field_110186_bp).func_111121_a(new AttributeModifier("Zombie reinforcement caller charge", -0.05000000074505806D, 0));
+                            var7.func_110148_a(field_110186_bp).func_111121_a(new AttributeModifier("Zombie reinforcement callee charge", -0.05000000074505806D, 0));
                             break;
                         }
                     }
@@ -299,11 +299,9 @@
             case 0:
                 this.dropItem(Item.ingotIron.itemID, 1);
                 break;
-
             case 1:
                 this.dropItem(Item.carrot.itemID, 1);
                 break;
-
             case 2:
                 this.dropItem(Item.potato.itemID, 1);
         }
@@ -391,7 +389,7 @@
             EntityZombie var2 = new EntityZombie(this.worldObj);
             var2.copyLocationAndAnglesFrom(par1EntityLivingBase);
             this.worldObj.removeEntity(par1EntityLivingBase);
-            var2.onSpawnWithEgg((EntityLivingData)null);
+            var2.func_110161_a((EntityLivingData)null);
             var2.setVillager(true);
 
             if (par1EntityLivingBase.isChild())
@@ -404,10 +402,10 @@
         }
     }
 
-    public EntityLivingData onSpawnWithEgg(EntityLivingData par1EntityLivingData)
+    public EntityLivingData func_110161_a(EntityLivingData par1EntityLivingData)
     {
-        Object par1EntityLivingData1 = super.onSpawnWithEgg(par1EntityLivingData);
-        float var2 = this.worldObj.getLocationTensionFactor(this.posX, this.posY, this.posZ);
+        Object par1EntityLivingData1 = super.func_110161_a(par1EntityLivingData);
+        float var2 = this.worldObj.func_110746_b(this.posX, this.posY, this.posZ);
         this.setCanPickUpLoot(this.rand.nextFloat() < 0.55F * var2);
 
         if (par1EntityLivingData1 == null)
@@ -444,13 +442,13 @@
             }
         }
 
-        this.getEntityAttribute(SharedMonsterAttributes.knockbackResistance).applyModifier(new AttributeModifier("Random spawn bonus", this.rand.nextDouble() * 0.05000000074505806D, 0));
-        this.getEntityAttribute(SharedMonsterAttributes.followRange).applyModifier(new AttributeModifier("Random zombie-spawn bonus", this.rand.nextDouble() * 1.5D, 2));
+        this.func_110148_a(SharedMonsterAttributes.field_111266_c).func_111121_a(new AttributeModifier("Random spawn bonus", this.rand.nextDouble() * 0.05000000074505806D, 0));
+        this.func_110148_a(SharedMonsterAttributes.field_111265_b).func_111121_a(new AttributeModifier("Random zombie-spawn bonus", this.rand.nextDouble() * 1.5D, 2));
 
         if (this.rand.nextFloat() < var2 * 0.05F)
         {
-            this.getEntityAttribute(field_110186_bp).applyModifier(new AttributeModifier("Leader zombie bonus", this.rand.nextDouble() * 0.25D + 0.5D, 0));
-            this.getEntityAttribute(SharedMonsterAttributes.maxHealth).applyModifier(new AttributeModifier("Leader zombie bonus", this.rand.nextDouble() * 3.0D + 1.0D, 2));
+            this.func_110148_a(field_110186_bp).func_111121_a(new AttributeModifier("Leader zombie bonus", this.rand.nextDouble() * 0.25D + 0.5D, 0));
+            this.func_110148_a(SharedMonsterAttributes.field_111267_a).func_111121_a(new AttributeModifier("Leader zombie bonus", this.rand.nextDouble() * 3.0D + 1.0D, 2));
         }
 
         return (EntityLivingData)par1EntityLivingData1;
@@ -536,7 +534,7 @@
     {
         EntityVillager var1 = new EntityVillager(this.worldObj);
         var1.copyLocationAndAnglesFrom(this);
-        var1.onSpawnWithEgg((EntityLivingData)null);
+        var1.func_110161_a((EntityLivingData)null);
         var1.func_82187_q();
 
         if (this.isChild())
