--- a/net/minecraft/src/EntityMinecartEmpty.java
+++ b/net/minecraft/src/EntityMinecartEmpty.java
@@ -12,10 +12,7 @@
         super(par1World, par2, par4, par6);
     }
 
-    /**
-     * First layer of player interaction
-     */
-    public boolean interactFirst(EntityPlayer par1EntityPlayer)
+    public boolean func_130002_c(EntityPlayer par1EntityPlayer)
     {
         if (this.riddenByEntity != null && this.riddenByEntity instanceof EntityPlayer && this.riddenByEntity != par1EntityPlayer)
         {
