--- a/net/minecraft/src/EntityRenderer.java
+++ b/net/minecraft/src/EntityRenderer.java
@@ -9,26 +9,28 @@
 import org.lwjgl.input.Keyboard;
 import org.lwjgl.input.Mouse;
 import org.lwjgl.opengl.Display;
+import org.lwjgl.opengl.DisplayMode;
 import org.lwjgl.opengl.GL11;
 import org.lwjgl.opengl.GLContext;
 import org.lwjgl.util.glu.Project;
 
 public class EntityRenderer
 {
-    private static final ResourceLocation field_110924_q = new ResourceLocation("textures/environment/rain.png");
-    private static final ResourceLocation field_110923_r = new ResourceLocation("textures/environment/snow.png");
+    protected static final ResourceLocation field_110924_q = new ResourceLocation("textures/environment/rain.png");
+    protected static final ResourceLocation field_110923_r = new ResourceLocation("textures/environment/snow.png");
     public static boolean anaglyphEnable;
 
     /** Anaglyph field (0=R, 1=GB) */
     public static int anaglyphField;
 
     /** A reference to the Minecraft object. */
-    private Minecraft mc;
-    private float farPlaneDistance;
+    protected Minecraft mc;
+    protected float farPlaneDistance = 0.0F;
+    protected float prevFarPlaneDistance = 0.0F;
     public ItemRenderer itemRenderer;
 
     /** Entity renderer update count */
-    private int rendererUpdateCount;
+    protected int rendererUpdateCount;
 
     /** Pointed entity */
     private Entity pointedEntity;
@@ -46,33 +48,31 @@
 
     /** Mouse filter dummy 4 */
     private MouseFilter mouseFilterDummy4 = new MouseFilter();
-    private float thirdPersonDistance = 4.0F;
+    protected float thirdPersonDistance = 4.0F;
 
     /** Third person distance temp */
-    private float thirdPersonDistanceTemp = 4.0F;
-    private float debugCamYaw;
-    private float prevDebugCamYaw;
-    private float debugCamPitch;
-    private float prevDebugCamPitch;
+    protected float thirdPersonDistanceTemp = 4.0F;
+    protected float debugCamYaw = 0.0F;
+    protected float prevDebugCamYaw = 0.0F;
+    protected float debugCamPitch = 0.0F;
+    protected float prevDebugCamPitch = 0.0F;
 
     /** Smooth cam yaw */
-    private float smoothCamYaw;
+    protected float smoothCamYaw;
 
     /** Smooth cam pitch */
     private float smoothCamPitch;
 
     /** Smooth cam filter X */
-    private float smoothCamFilterX;
+    protected float smoothCamFilterX;
 
     /** Smooth cam filter Y */
-    private float smoothCamFilterY;
+    protected float smoothCamFilterY;
 
     /** Smooth cam partial ticks */
-    private float smoothCamPartialTicks;
+    protected float smoothCamPartialTicks;
     private float debugCamFOV;
     private float prevDebugCamFOV;
-    private float camRoll;
-    private float prevCamRoll;
 
     /**
      * The texture id of the blocklight/skylight texture used for lighting effects
@@ -97,37 +97,44 @@
     private float field_82832_V;
 
     /** Cloud fog mode */
-    private boolean cloudFog;
-    private double cameraZoom = 1.0D;
-    private double cameraYaw;
-    private double cameraPitch;
+    protected boolean cloudFog = false;
+    protected double cameraZoom = 1.0D;
+    
+    protected float cameraYaw = 0.0F; //head + body
+    protected float cameraPitch = 0.0F; //head + additional input(moues if enabled)
+    protected float cameraRoll = 0.0F;
+    protected float prevCamYaw = 0.0F;
+    protected float prevCamPitch = 0.0F;
+    protected float prevCamRoll = 0.0F;
+
+    protected boolean checkAspectRatio = true;
 
     /** Previous frame time in milliseconds */
     private long prevFrameTime = Minecraft.getSystemTime();
 
     /** End time of last render (ns) */
-    private long renderEndNanoTime;
+    private long renderEndNanoTime = 0L;
 
     /**
      * Is set, updateCameraAndRender() calls updateLightmap(); set by updateTorchFlicker()
      */
-    private boolean lightmapUpdateNeeded;
+    protected boolean lightmapUpdateNeeded = false;
 
     /** Torch flicker X */
-    float torchFlickerX;
+    float torchFlickerX = 0.0F;
 
     /** Torch flicker DX */
-    float torchFlickerDX;
+    float torchFlickerDX = 0.0F;
 
     /** Torch flicker Y */
-    float torchFlickerY;
+    float torchFlickerY = 0.0F;
 
     /** Torch flicker DY */
-    float torchFlickerDY;
+    float torchFlickerDY = 0.0F;
     private Random random = new Random();
 
     /** Rain sound counter */
-    private int rainSoundCounter;
+    private int rainSoundCounter = 0;
 
     /** Rain X coords */
     float[] rainXCoords;
@@ -139,13 +146,13 @@
     FloatBuffer fogColorBuffer = GLAllocation.createDirectFloatBuffer(16);
 
     /** red component of the fog color */
-    float fogColorRed;
+    protected float fogColorRed;
 
     /** green component of the fog color */
-    float fogColorGreen;
+    protected float fogColorGreen;
 
     /** blue component of the fog color */
-    float fogColorBlue;
+    protected float fogColorBlue;
 
     /** Fog color 2 */
     private float fogColor2;
@@ -158,11 +165,11 @@
      */
     public int debugViewDirection;
     private boolean initialized = false;
-    private World updatedWorld = null;
+    protected World updatedWorld = null;
     private boolean showDebugInfo = false;
     public boolean fogStandard = false;
-    private long lastServerTime = 0L;
-    private int lastServerTicks = 0;
+    protected long lastServerTime = 0L;
+    protected int lastServerTicks = 0;
     private int serverWaitTime = 0;
     private int serverWaitTimeCurrent = 0;
     private float avgServerTimeDiff = 0.0F;
@@ -197,7 +204,7 @@
         this.prevDebugCamYaw = this.debugCamYaw;
         this.prevDebugCamPitch = this.debugCamPitch;
         this.prevDebugCamFOV = this.debugCamFOV;
-        this.prevCamRoll = this.camRoll;
+        this.prevCamRoll = this.cameraRoll;
         float var1;
         float var2;
 
@@ -277,36 +284,43 @@
             }
 
             Vec3 var7 = this.mc.renderViewEntity.getLook(par1);
-            Vec3 var8 = var6.addVector(var7.xCoord * var2, var7.yCoord * var2, var7.zCoord * var2);
-            this.pointedEntity = null;
-            float var9 = 1.0F;
-            List var10 = this.mc.theWorld.getEntitiesWithinAABBExcludingEntity(this.mc.renderViewEntity, this.mc.renderViewEntity.boundingBox.addCoord(var7.xCoord * var2, var7.yCoord * var2, var7.zCoord * var2).expand((double)var9, (double)var9, (double)var9));
-            double var11 = var4;
+            getPointedEntity(var6, var7, var2, var4);
+        }
+    }
+
+    protected void getPointedEntity( Vec3 var6, Vec3 var7, double var2, double var4 )
+    {
+        Vec3 var8 = var6.addVector(var7.xCoord * var2, var7.yCoord * var2, var7.zCoord * var2);
+
+        this.pointedEntity = null;
+        float var9 = 1.0F;
+        List var10 = this.mc.theWorld.getEntitiesWithinAABBExcludingEntity(this.mc.renderViewEntity, this.mc.renderViewEntity.boundingBox.addCoord(var7.xCoord * var2, var7.yCoord * var2, var7.zCoord * var2).expand((double)var9, (double)var9, (double)var9));
+        double var11 = var4;
 
-            for (int var13 = 0; var13 < var10.size(); ++var13)
+        for (int var13 = 0; var13 < var10.size(); ++var13)
+        {
+            Entity var14 = (Entity)var10.get(var13);
+
+            if (var14.canBeCollidedWith())
             {
-                Entity var14 = (Entity)var10.get(var13);
+                float var15 = var14.getCollisionBorderSize();
+                AxisAlignedBB var16 = var14.boundingBox.expand((double)var15, (double)var15, (double)var15);
+                MovingObjectPosition var17 = var16.calculateIntercept(var6, var8);
 
-                if (var14.canBeCollidedWith())
+                if (var16.isVecInside(var6))
                 {
-                    float var15 = var14.getCollisionBorderSize();
-                    AxisAlignedBB var16 = var14.boundingBox.expand((double)var15, (double)var15, (double)var15);
-                    MovingObjectPosition var17 = var16.calculateIntercept(var6, var8);
-
-                    if (var16.isVecInside(var6))
+                    if (0.0D < var11 || var11 == 0.0D)
                     {
-                        if (0.0D < var11 || var11 == 0.0D)
-                        {
-                            this.pointedEntity = var14;
-                            var11 = 0.0D;
-                        }
+                        this.pointedEntity = var14;
+                        var11 = 0.0D;
                     }
-                    else if (var17 != null)
-                    {
-                        double var18 = var6.distanceTo(var17.hitVec);
+                }
+                else if (var17 != null)
+                {
+                    double var18 = var6.distanceTo(var17.hitVec);
 
-                        if (var18 < var11 || var11 == 0.0D)
-                        {
+                    if (var18 < var11 || var11 == 0.0D)
+                    {
                             if (var14 == this.mc.renderViewEntity.ridingEntity)
                             {
                                 if (var11 == 0.0D)
@@ -333,7 +347,6 @@
                     this.mc.pointedEntityLiving = (EntityLivingBase)this.pointedEntity;
                 }
             }
-        }
     }
 
     /**
@@ -437,7 +450,7 @@
         }
     }
 
-    private void hurtCameraEffect(float par1)
+    protected void hurtCameraEffect(float par1)
     {
         EntityLivingBase var2 = this.mc.renderViewEntity;
         float var3 = (float)var2.hurtTime - par1;
@@ -463,7 +476,7 @@
     /**
      * Setups all the GL settings for view bobbing. Args: partialTickTime
      */
-    private void setupViewBobbing(float par1)
+    protected void setupViewBobbing(float par1)
     {
         if (this.mc.renderViewEntity instanceof EntityPlayer)
         {
@@ -489,7 +502,7 @@
         double var4 = var2.prevPosX + (var2.posX - var2.prevPosX) * (double)par1;
         double var6 = var2.prevPosY + (var2.posY - var2.prevPosY) * (double)par1 - (double)var3;
         double var8 = var2.prevPosZ + (var2.posZ - var2.prevPosZ) * (double)par1;
-        GL11.glRotatef(this.prevCamRoll + (this.camRoll - this.prevCamRoll) * par1, 0.0F, 0.0F, 1.0F);
+        GL11.glRotatef(this.prevCamRoll + (this.cameraRoll - this.prevCamRoll) * par1, 0.0F, 0.0F, 1.0F);
 
         if (var2.isPlayerSleeping())
         {
@@ -823,7 +836,7 @@
         this.lightmapUpdateNeeded = true;
     }
 
-    private void updateLightmap(float par1)
+    protected void updateLightmap(float par1)
     {
         WorldClient var2 = this.mc.theWorld;
 
@@ -975,7 +988,45 @@
         int var3 = par1EntityPlayer.getActivePotionEffect(Potion.nightVision).getDuration();
         return var3 > 200 ? 1.0F : 0.7F + MathHelper.sin(((float)var3 - par2) * (float)Math.PI * 0.2F) * 0.3F;
     }
+    
+    protected void updateCamera( float par1, boolean var14 )
+    {
+        this.mc.mcProfiler.startSection("mouse");
+
+        if (this.mc.inGameHasFocus && var14)
+        {
+            this.mc.mouseHelper.mouseXYChange();
+            float var4 = this.mc.gameSettings.mouseSensitivity * 0.6F + 0.2F;
+            float var5 = var4 * var4 * var4 * 8.0F;
+            float var6 = (float)this.mc.mouseHelper.deltaX * var5;
+            float var7 = (float)this.mc.mouseHelper.deltaY * var5;
+            byte var8 = 1;
+
+            if (this.mc.gameSettings.invertMouse)
+            {
+                var8 = -1;
+            }
+
+            if (this.mc.gameSettings.smoothCamera)
+            {
+                this.smoothCamYaw += var6;
+                this.smoothCamPitch += var7;
+                float var9 = par1 - this.smoothCamPartialTicks;
+                this.smoothCamPartialTicks = par1;
+                var6 = this.smoothCamFilterX * var9;
+                var7 = this.smoothCamFilterY * var9;
+                this.mc.thePlayer.setAngles(var6, var7 * (float)var8);
+            }
+            else
+            {
+                this.mc.thePlayer.setAngles(var6, var7 * (float)var8);
+            }
+        }
 
+        this.mc.mcProfiler.endSection();
+
+    }
+    
     /**
      * Will update any inputs that effect the camera angle (mouse) and then render the world and GUI
      */
@@ -999,6 +1050,29 @@
             Config.setNewRelease((String)null);
         }
 
+        if (var2 != null)
+        {
+            if (checkAspectRatio)
+            {
+                if (this.mc.hmdInfo != null && !this.mc.hmdInfo.getHMDInfo().isFakeData)
+                {
+                    int hmdScreenWidthPixels = this.mc.hmdInfo.getHMDInfo().HResolution;
+                    int hmdScreenHeightPixels = this.mc.hmdInfo.getHMDInfo().VResolution;
+
+                    DisplayMode e = Display.getDisplayMode();
+                    if (hmdScreenWidthPixels != e.getWidth() || hmdScreenHeightPixels != e.getHeight())
+                    {
+                        this.mc.printChatMessage("Warning: Non-native display resolution, perspective may be skewed! Ideally set your display to full-screen " + hmdScreenWidthPixels + "x" + hmdScreenHeightPixels + ".");
+                    }
+                }
+            }
+            checkAspectRatio = false;
+        }
+        else
+        {
+            checkAspectRatio = true;
+        }
+
         if (this.mc.currentScreen instanceof GuiMainMenu)
         {
             this.updateMainMenu((GuiMainMenu)this.mc.currentScreen);
@@ -1028,7 +1102,7 @@
         {
             if (Minecraft.getSystemTime() - this.prevFrameTime > 500L)
             {
-                this.mc.displayInGameMenu();
+            	this.mc.displayInGameMenu();
             }
         }
         else
@@ -1036,124 +1110,98 @@
             this.prevFrameTime = Minecraft.getSystemTime();
         }
 
-        this.mc.mcProfiler.startSection("mouse");
+        updateCamera( par1, var14 );
 
-        if (this.mc.inGameHasFocus && var14)
+        if (!this.mc.skipRenderWorld)
         {
-            this.mc.mouseHelper.mouseXYChange();
-            float var4 = this.mc.gameSettings.mouseSensitivity * 0.6F + 0.2F;
-            float var5 = var4 * var4 * var4 * 8.0F;
-            float var6 = (float)this.mc.mouseHelper.deltaX * var5;
-            float var7 = (float)this.mc.mouseHelper.deltaY * var5;
-            byte var8 = 1;
-
-            if (this.mc.gameSettings.invertMouse)
-            {
-                var8 = -1;
-            }
-
-            if (this.mc.gameSettings.smoothCamera)
-            {
-                this.smoothCamYaw += var6;
-                this.smoothCamPitch += var7;
-                float var9 = par1 - this.smoothCamPartialTicks;
-                this.smoothCamPartialTicks = par1;
-                var6 = this.smoothCamFilterX * var9;
-                var7 = this.smoothCamFilterY * var9;
-                this.mc.thePlayer.setAngles(var6, var7 * (float)var8);
-            }
-            else
-            {
-                this.mc.thePlayer.setAngles(var6, var7 * (float)var8);
-            }
+        	renderGUIandWorld(par1);
         }
 
-        this.mc.mcProfiler.endSection();
+        this.waitForServerThread();
 
-        if (!this.mc.skipRenderWorld)
+        if (this.mc.gameSettings.showDebugInfo != this.lastShowDebugInfo)
         {
-            anaglyphEnable = this.mc.gameSettings.anaglyph;
-            ScaledResolution var15 = new ScaledResolution(this.mc.gameSettings, this.mc.displayWidth, this.mc.displayHeight);
-            int var16 = var15.getScaledWidth();
-            int var17 = var15.getScaledHeight();
-            int var18 = Mouse.getX() * var16 / this.mc.displayWidth;
-            int var20 = var17 - Mouse.getY() * var17 / this.mc.displayHeight - 1;
-            int var19 = performanceToFps(this.mc.gameSettings.limitFramerate);
+            this.showExtendedDebugInfo = this.mc.gameSettings.showDebugProfilerChart;
+            this.lastShowDebugInfo = this.mc.gameSettings.showDebugInfo;
+        }
 
-            if (this.mc.theWorld != null)
-            {
-                this.mc.mcProfiler.startSection("level");
+        if (this.mc.gameSettings.showDebugInfo)
+        {
+            this.showLagometer(this.mc.mcProfiler.timeTickNano, this.mc.mcProfiler.timeUpdateChunksNano);
+        }
 
-                if (this.mc.gameSettings.limitFramerate == 0)
-                {
-                    this.renderWorld(par1, 0L);
-                }
-                else
-                {
-                    this.renderWorld(par1, this.renderEndNanoTime + (long)(1000000000 / var19));
-                }
+        if (this.mc.gameSettings.ofProfiler)
+        {
+            this.mc.gameSettings.showDebugProfilerChart = true;
+        }
+    }
 
-                this.renderEndNanoTime = System.nanoTime();
-                this.mc.mcProfiler.endStartSection("gui");
+    protected void renderGUIandWorld(float par1)
+    {
+        anaglyphEnable = this.mc.gameSettings.anaglyph;
+        ScaledResolution var15 = new ScaledResolution(this.mc.gameSettings, this.mc.displayWidth, this.mc.displayHeight);
+        int var16 = var15.getScaledWidth();
+        int var17 = var15.getScaledHeight();
+        int var18 = Mouse.getX() * var16 / this.mc.displayWidth;
+        int var20 = var17 - Mouse.getY() * var17 / this.mc.displayHeight - 1;
+        int var19 = performanceToFps(this.mc.gameSettings.limitFramerate);
 
-                if (!this.mc.gameSettings.hideGUI || this.mc.currentScreen != null)
-                {
-                    this.mc.ingameGUI.renderGameOverlay(par1, this.mc.currentScreen != null, var18, var20);
-                }
+        if (this.mc.theWorld != null)
+        {
+            this.mc.mcProfiler.startSection("level");
 
-                this.mc.mcProfiler.endSection();
+            if (this.mc.gameSettings.limitFramerate == 0)
+            {
+                this.renderWorld(par1, 0L);
             }
             else
             {
-                GL11.glViewport(0, 0, this.mc.displayWidth, this.mc.displayHeight);
-                GL11.glMatrixMode(GL11.GL_PROJECTION);
-                GL11.glLoadIdentity();
-                GL11.glMatrixMode(GL11.GL_MODELVIEW);
-                GL11.glLoadIdentity();
-                this.setupOverlayRendering();
-                this.renderEndNanoTime = System.nanoTime();
+                this.renderWorld(par1, this.renderEndNanoTime + (long)(1000000000 / var19));
             }
 
-            if (this.mc.currentScreen != null)
-            {
-                GL11.glClear(GL11.GL_DEPTH_BUFFER_BIT);
+            this.renderEndNanoTime = System.nanoTime();
+            this.mc.mcProfiler.endStartSection("gui");
 
-                try
-                {
-                    this.mc.currentScreen.drawScreen(var18, var20, par1);
-                }
-                catch (Throwable var13)
-                {
-                    CrashReport var11 = CrashReport.makeCrashReport(var13, "Rendering screen");
-                    CrashReportCategory var12 = var11.makeCategory("Screen render details");
-                    var12.addCrashSectionCallable("Screen name", new CallableScreenName(this));
-                    var12.addCrashSectionCallable("Mouse location", new CallableMouseLocation(this, var18, var20));
-                    var12.addCrashSectionCallable("Screen size", new CallableScreenSize(this, var15));
-                    throw new ReportedException(var11);
-                }
+            if (!this.mc.gameSettings.hideGUI || this.mc.currentScreen != null)
+            {
+                this.mc.ingameGUI.renderGameOverlay(par1, this.mc.currentScreen != null, var18, var20);
             }
-        }
-
-        this.waitForServerThread();
 
-        if (this.mc.gameSettings.showDebugInfo != this.lastShowDebugInfo)
-        {
-            this.showExtendedDebugInfo = this.mc.gameSettings.showDebugProfilerChart;
-            this.lastShowDebugInfo = this.mc.gameSettings.showDebugInfo;
+            this.mc.mcProfiler.endSection();
         }
-
-        if (this.mc.gameSettings.showDebugInfo)
+        else
         {
-            this.showLagometer(this.mc.mcProfiler.timeTickNano, this.mc.mcProfiler.timeUpdateChunksNano);
+            GL11.glViewport(0, 0, this.mc.displayWidth, this.mc.displayHeight);
+            GL11.glMatrixMode(GL11.GL_PROJECTION);
+            GL11.glLoadIdentity();
+            GL11.glMatrixMode(GL11.GL_MODELVIEW);
+            GL11.glLoadIdentity();
+            this.setupOverlayRendering();
+            this.renderEndNanoTime = System.nanoTime();
         }
 
-        if (this.mc.gameSettings.ofProfiler)
+        if (this.mc.currentScreen != null)
         {
-            this.mc.gameSettings.showDebugProfilerChart = true;
+            GL11.glClear(GL11.GL_DEPTH_BUFFER_BIT);
+
+            try
+            {
+                this.mc.currentScreen.drawScreen(var18, var20, par1);
+                GL11.glDisable(GL11.GL_LIGHTING);
+            }
+            catch (Throwable var13)
+            {
+                CrashReport var11 = CrashReport.makeCrashReport(var13, "Rendering screen");
+                CrashReportCategory var12 = var11.makeCategory("Screen render details");
+                var12.addCrashSectionCallable("Screen name", new CallableScreenName(this));
+                var12.addCrashSectionCallable("Mouse location", new CallableMouseLocation(this, var18, var20));
+                var12.addCrashSectionCallable("Screen size", new CallableScreenSize(this, var15));
+                throw new ReportedException(var11);
+            }
         }
     }
 
-    private void waitForServerThread()
+    protected void waitForServerThread()
     {
         this.serverWaitTimeCurrent = 0;
 
@@ -1600,7 +1648,7 @@
     /**
      * Render clouds if enabled
      */
-    private void renderCloudsCheck(RenderGlobal par1RenderGlobal, float par2)
+    protected void renderCloudsCheck(RenderGlobal par1RenderGlobal, float par2)
     {
         if (this.mc.gameSettings.shouldRenderClouds())
         {
@@ -1904,7 +1952,7 @@
     /**
      * calculates fog and calls glClearColor
      */
-    private void updateFogColor(float par1)
+    protected void updateFogColor(float par1)
     {
         WorldClient var2 = this.mc.theWorld;
         EntityLivingBase var3 = this.mc.renderViewEntity;
@@ -2115,7 +2163,7 @@
      * Sets up the fog to be rendered. If the arg passed in is -1 the fog starts at 0 and goes to 80% of far plane
      * distance and is used for sky rendering.
      */
-    private void setupFog(int par1, float par2)
+    protected void setupFog(int par1, float par2)
     {
         EntityLivingBase var3 = this.mc.renderViewEntity;
         boolean var4 = false;
@@ -2293,25 +2341,14 @@
     {
         Minecraft var1 = Config.getMinecraft();
 
-        if (var1.currentScreen != null && var1.currentScreen instanceof GuiMainMenu)
-        {
-            return 35;
-        }
-        else if (var1.theWorld == null)
+        int var2 = Config.getGameSettings().ofLimitFramerateFine;
+
+        if (var2 <= 0)
         {
-            return 35;
+            var2 = 10000;
         }
-        else
-        {
-            int var2 = Config.getGameSettings().ofLimitFramerateFine;
 
-            if (var2 <= 0)
-            {
-                var2 = 10000;
-            }
-
-            return var2;
-        }
+        return var2;
     }
 
     /**
