--- a/net/minecraft/src/EntityMagmaCube.java
+++ b/net/minecraft/src/EntityMagmaCube.java
@@ -8,10 +8,10 @@
         this.isImmuneToFire = true;
     }
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getEntityAttribute(SharedMonsterAttributes.movementSpeed).setAttribute(0.20000000298023224D);
+        super.func_110147_ax();
+        this.func_110148_a(SharedMonsterAttributes.field_111263_d).func_111128_a(0.20000000298023224D);
     }
 
     /**
@@ -104,9 +104,9 @@
         return super.getJumpDelay() * 4;
     }
 
-    protected void alterSquishAmount()
+    protected void func_70808_l()
     {
-        this.squishAmount *= 0.9F;
+        this.field_70813_a *= 0.9F;
     }
 
     /**
