--- a/net/minecraft/src/EntitySpider.java
+++ b/net/minecraft/src/EntitySpider.java
@@ -27,11 +27,11 @@
         }
     }
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setAttribute(16.0D);
-        this.getEntityAttribute(SharedMonsterAttributes.movementSpeed).setAttribute(0.800000011920929D);
+        super.func_110147_ax();
+        this.func_110148_a(SharedMonsterAttributes.field_111267_a).func_111128_a(16.0D);
+        this.func_110148_a(SharedMonsterAttributes.field_111263_d).func_111128_a(0.800000011920929D);
     }
 
     /**
@@ -194,15 +194,15 @@
         this.dataWatcher.updateObject(16, Byte.valueOf(var2));
     }
 
-    public EntityLivingData onSpawnWithEgg(EntityLivingData par1EntityLivingData)
+    public EntityLivingData func_110161_a(EntityLivingData par1EntityLivingData)
     {
-        Object par1EntityLivingData1 = super.onSpawnWithEgg(par1EntityLivingData);
+        Object par1EntityLivingData1 = super.func_110161_a(par1EntityLivingData);
 
         if (this.worldObj.rand.nextInt(100) == 0)
         {
             EntitySkeleton var2 = new EntitySkeleton(this.worldObj);
             var2.setLocationAndAngles(this.posX, this.posY, this.posZ, this.rotationYaw, 0.0F);
-            var2.onSpawnWithEgg((EntityLivingData)null);
+            var2.func_110161_a((EntityLivingData)null);
             this.worldObj.spawnEntityInWorld(var2);
             var2.mountEntity(this);
         }
@@ -211,7 +211,7 @@
         {
             par1EntityLivingData1 = new SpiderEffectsGroupData();
 
-            if (this.worldObj.difficultySetting > 2 && this.worldObj.rand.nextFloat() < 0.1F * this.worldObj.getLocationTensionFactor(this.posX, this.posY, this.posZ))
+            if (this.worldObj.difficultySetting > 2 && this.worldObj.rand.nextFloat() < 0.1F * this.worldObj.func_110746_b(this.posX, this.posY, this.posZ))
             {
                 ((SpiderEffectsGroupData)par1EntityLivingData1).func_111104_a(this.worldObj.rand);
             }
