--- a/net/minecraft/src/EntityMinecartContainer.java
+++ b/net/minecraft/src/EntityMinecartContainer.java
@@ -269,10 +269,7 @@
         }
     }
 
-    /**
-     * First layer of player interaction
-     */
-    public boolean interactFirst(EntityPlayer par1EntityPlayer)
+    public boolean func_130002_c(EntityPlayer par1EntityPlayer)
     {
         if (!this.worldObj.isRemote)
         {
