--- a/net/minecraft/src/EntityHanging.java
+++ b/net/minecraft/src/EntityHanging.java
@@ -280,15 +280,12 @@
             case 0:
                 par1NBTTagCompound.setByte("Dir", (byte)2);
                 break;
-
             case 1:
                 par1NBTTagCompound.setByte("Dir", (byte)1);
                 break;
-
             case 2:
                 par1NBTTagCompound.setByte("Dir", (byte)0);
                 break;
-
             case 3:
                 par1NBTTagCompound.setByte("Dir", (byte)3);
         }
@@ -310,15 +307,12 @@
                 case 0:
                     this.hangingDirection = 2;
                     break;
-
                 case 1:
                     this.hangingDirection = 1;
                     break;
-
                 case 2:
                     this.hangingDirection = 0;
                     break;
-
                 case 3:
                     this.hangingDirection = 3;
             }
