--- a/net/minecraft/src/EntityCreeper.java
+++ b/net/minecraft/src/EntityCreeper.java
@@ -31,10 +31,10 @@
         this.targetTasks.addTask(2, new EntityAIHurtByTarget(this, false));
     }
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getEntityAttribute(SharedMonsterAttributes.movementSpeed).setAttribute(0.25D);
+        super.func_110147_ax();
+        this.func_110148_a(SharedMonsterAttributes.field_111263_d).func_111128_a(0.25D);
     }
 
     /**
@@ -45,12 +45,9 @@
         return true;
     }
 
-    /**
-     * The number of iterations PathFinder.getSafePoint will execute before giving up.
-     */
-    public int getMaxSafePointTries()
+    public int func_82143_as()
     {
-        return this.getAttackTarget() == null ? 3 : 3 + (int)(this.getHealth() - 1.0F);
+        return this.getAttackTarget() == null ? 3 : 3 + (int)(this.func_110143_aJ() - 1.0F);
     }
 
     /**
