--- a/net/minecraft/src/EntityBoat.java
+++ b/net/minecraft/src/EntityBoat.java
@@ -482,10 +482,7 @@
         return 0.0F;
     }
 
-    /**
-     * First layer of player interaction
-     */
-    public boolean interactFirst(EntityPlayer par1EntityPlayer)
+    public boolean func_130002_c(EntityPlayer par1EntityPlayer)
     {
         if (this.riddenByEntity != null && this.riddenByEntity instanceof EntityPlayer && this.riddenByEntity != par1EntityPlayer)
         {
@@ -515,7 +512,7 @@
      */
     public float getDamageTaken()
     {
-        return this.dataWatcher.getWatchableObjectFloat(19);
+        return this.dataWatcher.func_111145_d(19);
     }
 
     /**
