--- a/net/minecraft/src/EntityMinecart.java
+++ b/net/minecraft/src/EntityMinecart.java
@@ -44,19 +44,14 @@
         {
             case 1:
                 return new EntityMinecartChest(par0World, par1, par3, par5);
-
             case 2:
                 return new EntityMinecartFurnace(par0World, par1, par3, par5);
-
             case 3:
                 return new EntityMinecartTNT(par0World, par1, par3, par5);
-
             case 4:
                 return new EntityMinecartMobSpawner(par0World, par1, par3, par5);
-
             case 5:
                 return new EntityMinecartHopper(par0World, par1, par3, par5);
-
             default:
                 return new EntityMinecartEmpty(par0World, par1, par3, par5);
         }
