--- a/net/minecraft/src/EntityFireworkStarterFX.java
+++ b/net/minecraft/src/EntityFireworkStarterFX.java
@@ -2,10 +2,10 @@
 
 public class EntityFireworkStarterFX extends EntityFX
 {
-    private int fireworkAge;
-    private final EffectRenderer theEffectRenderer;
+    private int field_92042_ax;
+    private final EffectRenderer field_92040_ay;
     private NBTTagList fireworkExplosions;
-    boolean twinkle;
+    boolean field_92041_a;
 
     public EntityFireworkStarterFX(World par1World, double par2, double par4, double par6, double par8, double par10, double par12, EffectRenderer par14EffectRenderer, NBTTagCompound par15NBTTagCompound)
     {
@@ -13,7 +13,7 @@
         this.motionX = par8;
         this.motionY = par10;
         this.motionZ = par12;
-        this.theEffectRenderer = par14EffectRenderer;
+        this.field_92040_ay = par14EffectRenderer;
         this.particleMaxAge = 8;
 
         if (par15NBTTagCompound != null)
@@ -34,7 +34,7 @@
 
                     if (var17.getBoolean("Flicker"))
                     {
-                        this.twinkle = true;
+                        this.field_92041_a = true;
                         this.particleMaxAge += 15;
                         break;
                     }
@@ -52,7 +52,7 @@
     {
         boolean var1;
 
-        if (this.fireworkAge == 0 && this.fireworkExplosions != null)
+        if (this.field_92042_ax == 0 && this.fireworkExplosions != null)
         {
             var1 = this.func_92037_i();
             boolean var2 = false;
@@ -79,9 +79,9 @@
             this.worldObj.playSound(this.posX, this.posY, this.posZ, var15, 20.0F, 0.95F + this.rand.nextFloat() * 0.1F, true);
         }
 
-        if (this.fireworkAge % 2 == 0 && this.fireworkExplosions != null && this.fireworkAge / 2 < this.fireworkExplosions.tagCount())
+        if (this.field_92042_ax % 2 == 0 && this.fireworkExplosions != null && this.field_92042_ax / 2 < this.fireworkExplosions.tagCount())
         {
-            int var13 = this.fireworkAge / 2;
+            int var13 = this.field_92042_ax / 2;
             NBTTagCompound var14 = (NBTTagCompound)this.fireworkExplosions.tagAt(var13);
             byte var17 = var14.getByte("Type");
             boolean var18 = var14.getBoolean("Trail");
@@ -91,23 +91,23 @@
 
             if (var17 == 1)
             {
-                this.createBall(0.5D, 4, var6, var7, var18, var5);
+                this.func_92035_a(0.5D, 4, var6, var7, var18, var5);
             }
             else if (var17 == 2)
             {
-                this.createShaped(0.5D, new double[][] {{0.0D, 1.0D}, {0.3455D, 0.309D}, {0.9511D, 0.309D}, {0.3795918367346939D, -0.12653061224489795D}, {0.6122448979591837D, -0.8040816326530612D}, {0.0D, -0.35918367346938773D}}, var6, var7, var18, var5, false);
+                this.func_92038_a(0.5D, new double[][] {{0.0D, 1.0D}, {0.3455D, 0.309D}, {0.9511D, 0.309D}, {0.3795918367346939D, -0.12653061224489795D}, {0.6122448979591837D, -0.8040816326530612D}, {0.0D, -0.35918367346938773D}}, var6, var7, var18, var5, false);
             }
             else if (var17 == 3)
             {
-                this.createShaped(0.5D, new double[][] {{0.0D, 0.2D}, {0.2D, 0.2D}, {0.2D, 0.6D}, {0.6D, 0.6D}, {0.6D, 0.2D}, {0.2D, 0.2D}, {0.2D, 0.0D}, {0.4D, 0.0D}, {0.4D, -0.6D}, {0.2D, -0.6D}, {0.2D, -0.4D}, {0.0D, -0.4D}}, var6, var7, var18, var5, true);
+                this.func_92038_a(0.5D, new double[][] {{0.0D, 0.2D}, {0.2D, 0.2D}, {0.2D, 0.6D}, {0.6D, 0.6D}, {0.6D, 0.2D}, {0.2D, 0.2D}, {0.2D, 0.0D}, {0.4D, 0.0D}, {0.4D, -0.6D}, {0.2D, -0.6D}, {0.2D, -0.4D}, {0.0D, -0.4D}}, var6, var7, var18, var5, true);
             }
             else if (var17 == 4)
             {
-                this.createBurst(var6, var7, var18, var5);
+                this.func_92036_a(var6, var7, var18, var5);
             }
             else
             {
-                this.createBall(0.25D, 2, var6, var7, var18, var5);
+                this.func_92035_a(0.25D, 2, var6, var7, var18, var5);
             }
 
             int var8 = var6[0];
@@ -116,14 +116,14 @@
             float var11 = (float)((var8 & 255) >> 0) / 255.0F;
             EntityFireworkOverlayFX var12 = new EntityFireworkOverlayFX(this.worldObj, this.posX, this.posY, this.posZ);
             var12.setRBGColorF(var9, var10, var11);
-            this.theEffectRenderer.addEffect(var12);
+            this.field_92040_ay.addEffect(var12);
         }
 
-        ++this.fireworkAge;
+        ++this.field_92042_ax;
 
-        if (this.fireworkAge > this.particleMaxAge)
+        if (this.field_92042_ax > this.particleMaxAge)
         {
-            if (this.twinkle)
+            if (this.field_92041_a)
             {
                 var1 = this.func_92037_i();
                 String var16 = "fireworks." + (var1 ? "twinkle_far" : "twinkle");
@@ -140,31 +140,23 @@
         return var1 == null || var1.renderViewEntity == null || var1.renderViewEntity.getDistanceSq(this.posX, this.posY, this.posZ) >= 256.0D;
     }
 
-    /**
-     * Creates a single particle. Args: x, y, z, x velocity, y velocity, z velocity, colours, fade colours, whether to
-     * trail, whether to twinkle
-     */
-    private void createParticle(double par1, double par3, double par5, double par7, double par9, double par11, int[] par13ArrayOfInteger, int[] par14ArrayOfInteger, boolean par15, boolean par16)
+    private void func_92034_a(double par1, double par3, double par5, double par7, double par9, double par11, int[] par13ArrayOfInteger, int[] par14ArrayOfInteger, boolean par15, boolean par16)
     {
-        EntityFireworkSparkFX var17 = new EntityFireworkSparkFX(this.worldObj, par1, par3, par5, par7, par9, par11, this.theEffectRenderer);
-        var17.setTrail(par15);
-        var17.setTwinkle(par16);
+        EntityFireworkSparkFX var17 = new EntityFireworkSparkFX(this.worldObj, par1, par3, par5, par7, par9, par11, this.field_92040_ay);
+        var17.func_92045_e(par15);
+        var17.func_92043_f(par16);
         int var18 = this.rand.nextInt(par13ArrayOfInteger.length);
-        var17.setColour(par13ArrayOfInteger[var18]);
+        var17.func_92044_a(par13ArrayOfInteger[var18]);
 
         if (par14ArrayOfInteger != null && par14ArrayOfInteger.length > 0)
         {
-            var17.setFadeColour(par14ArrayOfInteger[this.rand.nextInt(par14ArrayOfInteger.length)]);
+            var17.func_92046_g(par14ArrayOfInteger[this.rand.nextInt(par14ArrayOfInteger.length)]);
         }
 
-        this.theEffectRenderer.addEffect(var17);
+        this.field_92040_ay.addEffect(var17);
     }
 
-    /**
-     * Creates a small ball or large ball type explosion. Args: particle speed, size, colours, fade colours, whether to
-     * trail, whether to flicker
-     */
-    private void createBall(double par1, int par3, int[] par4ArrayOfInteger, int[] par5ArrayOfInteger, boolean par6, boolean par7)
+    private void func_92035_a(double par1, int par3, int[] par4ArrayOfInteger, int[] par5ArrayOfInteger, boolean par6, boolean par7)
     {
         double var8 = this.posX;
         double var10 = this.posY;
@@ -180,7 +172,7 @@
                     double var19 = (double)var14 + (this.rand.nextDouble() - this.rand.nextDouble()) * 0.5D;
                     double var21 = (double)var16 + (this.rand.nextDouble() - this.rand.nextDouble()) * 0.5D;
                     double var23 = (double)MathHelper.sqrt_double(var17 * var17 + var19 * var19 + var21 * var21) / par1 + this.rand.nextGaussian() * 0.05D;
-                    this.createParticle(var8, var10, var12, var17 / var23, var19 / var23, var21 / var23, par4ArrayOfInteger, par5ArrayOfInteger, par6, par7);
+                    this.func_92034_a(var8, var10, var12, var17 / var23, var19 / var23, var21 / var23, par4ArrayOfInteger, par5ArrayOfInteger, par6, par7);
 
                     if (var14 != -par3 && var14 != par3 && var15 != -par3 && var15 != par3)
                     {
@@ -191,15 +183,11 @@
         }
     }
 
-    /**
-     * Creates a creeper-shaped or star-shaped explosion. Args: particle speed, shape, colours, fade colours, whether to
-     * trail, whether to flicker, unknown
-     */
-    private void createShaped(double par1, double[][] par3ArrayOfDouble, int[] par4ArrayOfInteger, int[] par5ArrayOfInteger, boolean par6, boolean par7, boolean par8)
+    private void func_92038_a(double par1, double[][] par3ArrayOfDouble, int[] par4ArrayOfInteger, int[] par5ArrayOfInteger, boolean par6, boolean par7, boolean par8)
     {
         double var9 = par3ArrayOfDouble[0][0];
         double var11 = par3ArrayOfDouble[0][1];
-        this.createParticle(this.posX, this.posY, this.posZ, var9 * par1, var11 * par1, 0.0D, par4ArrayOfInteger, par5ArrayOfInteger, par6, par7);
+        this.func_92034_a(this.posX, this.posY, this.posZ, var9 * par1, var11 * par1, 0.0D, par4ArrayOfInteger, par5ArrayOfInteger, par6, par7);
         float var13 = this.rand.nextFloat() * (float)Math.PI;
         double var14 = par8 ? 0.034D : 0.34D;
 
@@ -223,7 +211,7 @@
 
                     for (double var36 = -1.0D; var36 <= 1.0D; var36 += 2.0D)
                     {
-                        this.createParticle(this.posX, this.posY, this.posZ, var30 * var36, var32, var34 * var36, par4ArrayOfInteger, par5ArrayOfInteger, par6, par7);
+                        this.func_92034_a(this.posX, this.posY, this.posZ, var30 * var36, var32, var34 * var36, par4ArrayOfInteger, par5ArrayOfInteger, par6, par7);
                     }
                 }
 
@@ -233,10 +221,7 @@
         }
     }
 
-    /**
-     * Creates a burst type explosion. Args: colours, fade colours, whether to trail, whether to flicker
-     */
-    private void createBurst(int[] par1ArrayOfInteger, int[] par2ArrayOfInteger, boolean par3, boolean par4)
+    private void func_92036_a(int[] par1ArrayOfInteger, int[] par2ArrayOfInteger, boolean par3, boolean par4)
     {
         double var5 = this.rand.nextGaussian() * 0.05D;
         double var7 = this.rand.nextGaussian() * 0.05D;
@@ -246,7 +231,7 @@
             double var10 = this.motionX * 0.5D + this.rand.nextGaussian() * 0.15D + var5;
             double var12 = this.motionZ * 0.5D + this.rand.nextGaussian() * 0.15D + var7;
             double var14 = this.motionY * 0.5D + this.rand.nextDouble() * 0.5D;
-            this.createParticle(this.posX, this.posY, this.posZ, var10, var14, var12, par1ArrayOfInteger, par2ArrayOfInteger, par3, par4);
+            this.func_92034_a(this.posX, this.posY, this.posZ, var10, var14, var12, par1ArrayOfInteger, par2ArrayOfInteger, par3, par4);
         }
     }
 
