--- a/net/minecraft/src/EntityMinecartFurnace.java
+++ b/net/minecraft/src/EntityMinecartFurnace.java
@@ -112,10 +112,7 @@
         super.applyDrag();
     }
 
-    /**
-     * First layer of player interaction
-     */
-    public boolean interactFirst(EntityPlayer par1EntityPlayer)
+    public boolean func_130002_c(EntityPlayer par1EntityPlayer)
     {
         ItemStack var2 = par1EntityPlayer.inventory.getCurrentItem();
 
