--- a/net/minecraft/src/EntityWither.java
+++ b/net/minecraft/src/EntityWither.java
@@ -18,7 +18,7 @@
     public EntityWither(World par1World)
     {
         super(par1World);
-        this.setHealth(this.getMaxHealth());
+        this.setEntityHealth(this.func_110138_aP());
         this.setSize(0.9F, 4.0F);
         this.isImmuneToFire = true;
         this.getNavigator().setCanSwim(true);
@@ -352,7 +352,7 @@
     public void func_82206_m()
     {
         this.func_82215_s(220);
-        this.setHealth(this.getMaxHealth() / 3.0F);
+        this.setEntityHealth(this.func_110138_aP() / 3.0F);
     }
 
     /**
@@ -555,12 +555,12 @@
         return true;
     }
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setAttribute(300.0D);
-        this.getEntityAttribute(SharedMonsterAttributes.movementSpeed).setAttribute(0.6000000238418579D);
-        this.getEntityAttribute(SharedMonsterAttributes.followRange).setAttribute(40.0D);
+        super.func_110147_ax();
+        this.func_110148_a(SharedMonsterAttributes.field_111267_a).func_111128_a(300.0D);
+        this.func_110148_a(SharedMonsterAttributes.field_111263_d).func_111128_a(0.6000000238418579D);
+        this.func_110148_a(SharedMonsterAttributes.field_111265_b).func_111128_a(40.0D);
     }
 
     public float func_82207_a(int par1)
@@ -602,7 +602,7 @@
      */
     public boolean isArmored()
     {
-        return this.getHealth() <= this.getMaxHealth() / 2.0F;
+        return this.func_110143_aJ() <= this.func_110138_aP() / 2.0F;
     }
 
     /**
