--- a/net/minecraft/src/EntityItemFrame.java
+++ b/net/minecraft/src/EntityItemFrame.java
@@ -22,12 +22,12 @@
         this.getDataWatcher().addObject(3, Byte.valueOf((byte)0));
     }
 
-    public int getWidthPixels()
+    public int func_82329_d()
     {
         return 9;
     }
 
-    public int getHeightPixels()
+    public int func_82330_g()
     {
         return 9;
     }
@@ -43,10 +43,7 @@
         return par1 < var3 * var3;
     }
 
-    /**
-     * Called when this entity is broken. Entity parameter may be null.
-     */
-    public void onBroken(Entity par1Entity)
+    public void func_110128_b(Entity par1Entity)
     {
         ItemStack var2 = this.getDisplayedItem();
 
@@ -56,7 +53,7 @@
 
             if (var3.capabilities.isCreativeMode)
             {
-                this.removeFrameFromMap(var2);
+                this.func_110131_b(var2);
                 return;
             }
         }
@@ -66,15 +63,12 @@
         if (var2 != null && this.rand.nextFloat() < this.itemDropChance)
         {
             var2 = var2.copy();
-            this.removeFrameFromMap(var2);
+            this.func_110131_b(var2);
             this.entityDropItem(var2, 0.0F);
         }
     }
 
-    /**
-     * Removes the dot representing this frame's position from the map when the item frame is broken.
-     */
-    private void removeFrameFromMap(ItemStack par1ItemStack)
+    private void func_110131_b(ItemStack par1ItemStack)
     {
         if (par1ItemStack != null)
         {
@@ -151,10 +145,7 @@
         super.readEntityFromNBT(par1NBTTagCompound);
     }
 
-    /**
-     * First layer of player interaction
-     */
-    public boolean interactFirst(EntityPlayer par1EntityPlayer)
+    public boolean func_130002_c(EntityPlayer par1EntityPlayer)
     {
         if (this.getDisplayedItem() == null)
         {
