--- a/net/minecraft/src/EntityGiantZombie.java
+++ b/net/minecraft/src/EntityGiantZombie.java
@@ -9,12 +9,12 @@
         this.setSize(this.width * 6.0F, this.height * 6.0F);
     }
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setAttribute(100.0D);
-        this.getEntityAttribute(SharedMonsterAttributes.movementSpeed).setAttribute(0.5D);
-        this.getEntityAttribute(SharedMonsterAttributes.attackDamage).setAttribute(50.0D);
+        super.func_110147_ax();
+        this.func_110148_a(SharedMonsterAttributes.field_111267_a).func_111128_a(100.0D);
+        this.func_110148_a(SharedMonsterAttributes.field_111263_d).func_111128_a(0.5D);
+        this.func_110148_a(SharedMonsterAttributes.field_111264_e).func_111128_a(50.0D);
     }
 
     /**
