--- a/net/minecraft/src/EntityGhast.java
+++ b/net/minecraft/src/EntityGhast.java
@@ -56,10 +56,10 @@
         this.dataWatcher.addObject(16, Byte.valueOf((byte)0));
     }
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setAttribute(10.0D);
+        super.func_110147_ax();
+        this.func_110148_a(SharedMonsterAttributes.field_111267_a).func_111128_a(10.0D);
     }
 
     protected void updateEntityActionState()
