--- a/net/minecraft/src/EntityBlaze.java
+++ b/net/minecraft/src/EntityBlaze.java
@@ -16,10 +16,10 @@
         this.experienceValue = 10;
     }
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getEntityAttribute(SharedMonsterAttributes.attackDamage).setAttribute(6.0D);
+        super.func_110147_ax();
+        this.func_110148_a(SharedMonsterAttributes.field_111264_e).func_111128_a(6.0D);
     }
 
     protected void entityInit()
