--- a/net/minecraft/src/EntityFallingSand.java
+++ b/net/minecraft/src/EntityFallingSand.java
@@ -299,9 +299,9 @@
         return false;
     }
 
-    public void addEntityCrashInfo(CrashReportCategory par1CrashReportCategory)
+    public void func_85029_a(CrashReportCategory par1CrashReportCategory)
     {
-        super.addEntityCrashInfo(par1CrashReportCategory);
+        super.func_85029_a(par1CrashReportCategory);
         par1CrashReportCategory.addCrashSection("Immitating block ID", Integer.valueOf(this.blockID));
         par1CrashReportCategory.addCrashSection("Immitating block data", Integer.valueOf(this.metadata));
     }
