--- a/net/minecraft/src/EntityHorse.java
+++ b/net/minecraft/src/EntityHorse.java
@@ -103,16 +103,12 @@
                 case 0:
                 default:
                     return StatCollector.translateToLocal("entity.horse.name");
-
                 case 1:
                     return StatCollector.translateToLocal("entity.donkey.name");
-
                 case 2:
                     return StatCollector.translateToLocal("entity.mule.name");
-
                 case 3:
                     return StatCollector.translateToLocal("entity.zombiehorse.name");
-
                 case 4:
                     return StatCollector.translateToLocal("entity.skeletonhorse.name");
             }
