--- a/net/minecraft/src/EntityLeashKnot.java
+++ b/net/minecraft/src/EntityLeashKnot.java
@@ -23,12 +23,12 @@
 
     public void setDirection(int par1) {}
 
-    public int getWidthPixels()
+    public int func_82329_d()
     {
         return 9;
     }
 
-    public int getHeightPixels()
+    public int func_82330_g()
     {
         return 9;
     }
@@ -42,17 +42,12 @@
         return par1 < 1024.0D;
     }
 
-    /**
-     * Called when this entity is broken. Entity parameter may be null.
-     */
-    public void onBroken(Entity par1Entity) {}
+    public void func_110128_b(Entity par1Entity) {}
 
     /**
-     * Either write this entity to the NBT tag given and return true, or return false without doing anything. If this
-     * returns false the entity is not saved on disk. Ridden entities return false here as they are saved with their
-     * rider.
+     * adds the ID of this entity to the NBT given
      */
-    public boolean writeToNBTOptional(NBTTagCompound par1NBTTagCompound)
+    public boolean addEntityID(NBTTagCompound par1NBTTagCompound)
     {
         return false;
     }
@@ -67,10 +62,7 @@
      */
     public void readEntityFromNBT(NBTTagCompound par1NBTTagCompound) {}
 
-    /**
-     * First layer of player interaction
-     */
-    public boolean interactFirst(EntityPlayer par1EntityPlayer)
+    public boolean func_130002_c(EntityPlayer par1EntityPlayer)
     {
         ItemStack var2 = par1EntityPlayer.getHeldItem();
         boolean var3 = false;
@@ -79,7 +71,7 @@
         Iterator var7;
         EntityLiving var8;
 
-        if (var2 != null && var2.itemID == Item.leash.itemID && !this.worldObj.isRemote)
+        if (var2 != null && var2.itemID == Item.field_111214_ch.itemID && !this.worldObj.isRemote)
         {
             var4 = 7.0D;
             var6 = this.worldObj.getEntitiesWithinAABB(EntityLiving.class, AxisAlignedBB.getAABBPool().getAABB(this.posX - var4, this.posY - var4, this.posZ - var4, this.posX + var4, this.posY + var4, this.posZ + var4));
@@ -92,9 +84,9 @@
                 {
                     var8 = (EntityLiving)var7.next();
 
-                    if (var8.getLeashed() && var8.getLeashedToEntity() == par1EntityPlayer)
+                    if (var8.func_110167_bD() && var8.func_110166_bE() == par1EntityPlayer)
                     {
-                        var8.setLeashedToEntity(this, true);
+                        var8.func_110162_b(this, true);
                         var3 = true;
                     }
                 }
@@ -118,9 +110,9 @@
                     {
                         var8 = (EntityLiving)var7.next();
 
-                        if (var8.getLeashed() && var8.getLeashedToEntity() == this)
+                        if (var8.func_110167_bD() && var8.func_110166_bE() == this)
                         {
-                            var8.clearLeashed(true, false);
+                            var8.func_110160_i(true, false);
                         }
                     }
                 }
@@ -142,12 +134,12 @@
     public static EntityLeashKnot func_110129_a(World par0World, int par1, int par2, int par3)
     {
         EntityLeashKnot var4 = new EntityLeashKnot(par0World, par1, par2, par3);
-        var4.forceSpawn = true;
+        var4.field_98038_p = true;
         par0World.spawnEntityInWorld(var4);
         return var4;
     }
 
-    public static EntityLeashKnot getKnotForBlock(World par0World, int par1, int par2, int par3)
+    public static EntityLeashKnot func_110130_b(World par0World, int par1, int par2, int par3)
     {
         List var4 = par0World.getEntitiesWithinAABB(EntityLeashKnot.class, AxisAlignedBB.getAABBPool().getAABB((double)par1 - 1.0D, (double)par2 - 1.0D, (double)par3 - 1.0D, (double)par1 + 1.0D, (double)par2 + 1.0D, (double)par3 + 1.0D));
         Object var5 = null;
