--- a/net/minecraft/src/EntityPlayer.java
+++ b/net/minecraft/src/EntityPlayer.java
@@ -1346,15 +1346,12 @@
                 case 0:
                     var7 = 0.9F;
                     break;
-
                 case 1:
                     var10 = 0.1F;
                     break;
-
                 case 2:
                     var7 = 0.1F;
                     break;
-
                 case 3:
                     var10 = 0.9F;
             }
@@ -1390,15 +1387,12 @@
             case 0:
                 this.field_71089_bV = -1.8F;
                 break;
-
             case 1:
                 this.field_71079_bU = 1.8F;
                 break;
-
             case 2:
                 this.field_71089_bV = 1.8F;
                 break;
-
             case 3:
                 this.field_71079_bU = -1.8F;
         }
@@ -1498,13 +1492,10 @@
             {
                 case 0:
                     return 90.0F;
-
                 case 1:
                     return 0.0F;
-
                 case 2:
                     return 270.0F;
-
                 case 3:
                     return 180.0F;
             }
