--- a/net/minecraft/src/EntitySmallFireball.java
+++ b/net/minecraft/src/EntitySmallFireball.java
@@ -45,23 +45,18 @@
                     case 0:
                         --var3;
                         break;
-
                     case 1:
                         ++var3;
                         break;
-
                     case 2:
                         --var4;
                         break;
-
                     case 3:
                         ++var4;
                         break;
-
                     case 4:
                         --var2;
                         break;
-
                     case 5:
                         ++var2;
                 }
