--- a/net/minecraft/src/EntityMinecartHopper.java
+++ b/net/minecraft/src/EntityMinecartHopper.java
@@ -41,10 +41,7 @@
         return 5;
     }
 
-    /**
-     * First layer of player interaction
-     */
-    public boolean interactFirst(EntityPlayer par1EntityPlayer)
+    public boolean func_130002_c(EntityPlayer par1EntityPlayer)
     {
         if (!this.worldObj.isRemote)
         {
@@ -151,7 +148,7 @@
 
             if (var1.size() > 0)
             {
-                TileEntityHopper.insertStackFromEntity(this, (EntityItem)var1.get(0));
+                TileEntityHopper.func_96114_a(this, (EntityItem)var1.get(0));
             }
 
             return false;
