--- a/net/minecraft/src/EntityBat.java
+++ b/net/minecraft/src/EntityBat.java
@@ -75,10 +75,10 @@
 
     protected void collideWithNearbyEntities() {}
 
-    protected void applyEntityAttributes()
+    protected void func_110147_ax()
     {
-        super.applyEntityAttributes();
-        this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setAttribute(6.0D);
+        super.func_110147_ax();
+        this.func_110148_a(SharedMonsterAttributes.field_111267_a).func_111128_a(6.0D);
     }
 
     public boolean getIsBatHanging()
