--- a/net/minecraft/src/EnumChatFormatting.java
+++ b/net/minecraft/src/EnumChatFormatting.java
@@ -59,10 +59,7 @@
         return this.field_96303_A;
     }
 
-    /**
-     * Checks if typo is a color.
-     */
-    public boolean isColor()
+    public boolean func_96302_c()
     {
         return !this.field_96303_A && this != RESET;
     }
@@ -97,7 +94,7 @@
         {
             EnumChatFormatting var6 = var3[var5];
 
-            if ((!var6.isColor() || par0) && (!var6.func_96301_b() || par1))
+            if ((!var6.func_96302_c() || par0) && (!var6.func_96301_b() || par1))
             {
                 var2.add(var6.func_96297_d());
             }
