--- a/net/minecraft/src/EnumOptions.java
+++ b/net/minecraft/src/EnumOptions.java
@@ -97,7 +97,77 @@
     HELD_ITEM_TOOLTIPS("Held Item Tooltips", false, false),
     DROPPED_ITEMS("Dropped Items", false, false),
     LAZY_CHUNK_LOADING("Lazy Chunk Loading", false, false),
-    CUSTOM_SKY("Custom Sky", false, false);
+    CUSTOM_SKY("Custom Sky", false, false),
+    // Minecrift below here
+    
+    //General
+    USE_VR("VR mode", false, true),
+    HUD_SCALE("HUD Scale", true, false),
+    HUD_DISTANCE("HUD Distance", true, false),
+    HUD_PITCH("HUD Pitch Offset", true, false),
+    HUD_OPACITY("HUD Opacity", false, true),
+    HUD_HIDE("Hide HUD (F1)", false, true),
+    HUD_OCCLUSION("HUD Occlusion", false, true),
+    HEAD_TRACKING("Head Tracking", false, true),
+    DUMMY("Dummy", false, true),
+    VR_HEAD_ORIENTATION("Head Orientation", false, true),
+    VR_HEAD_POSITION("Head Position", false, true),
+    VR_CONTROLLER("Controller", false, true),
+    CROSSHAIR_SCALE("Crosshair Size", true, false),
+    CROSSHAIR_ALWAYS_SHOW("Show Crosshair", false, true),
+    CROSSHAIR_ROLL("Roll Crosshair", false, true),
+    BLOCK_OUTLINE_ALWAYS_SHOW("Show Block Outline", false, true),
+
+    // Player
+    EYE_HEIGHT("Eye Height", true, false),
+    EYE_PROTRUSION("Eye Protrusion", true, false),
+    NECK_LENGTH("Neck Length", true, false),
+    RENDER_OWN_HEADWEAR("Render Own Headwear", false, true),
+    RENDER_FULL_FIRST_PERSON_MODEL("First Person Model", false, true),
+    RENDER_PLAYER_OFFSET("View Body Offset", true, false),
+    IPD("IPD", true, false),
+    OCULUS_PROFILE("Use Oculus Profile", false, false),
+    OCULUS_PROFILE_NAME("Name", false, true),
+    OCULUS_PROFILE_GENDER("Gender", false, true),
+
+    //HMD/render
+    USE_DISTORTION("Distortion", false, true),
+    CHROM_AB_CORRECTION("Chrom. Ab. Correction", false, true),
+    FOV_SCALE_FACTOR("FOV Scale", true, false),
+    DISTORTION_FIT_POINT("Distortion Border", true, false),
+    SUPERSAMPLING("FSAA", false, true),
+    SUPERSAMPLE_SCALEFACTOR("FSAA Render Scale", true, false),
+
+    //Head orientation tracking
+    HEAD_TRACK_PREDICTION("Head Track Prediction", false, true),
+    HEAD_TRACK_SENSITIVITY("Head Track Sensitivity", true, false),
+    HEAD_TRACK_PREDICTION_TIME("Prediction time", true, false),
+
+    //eye center position tracking
+    POS_TRACK_HYDRALOC("Position", false, false),
+    POS_TRACK_HYDRA_OFFSET_X("Hydra X Offset", true, false),
+    POS_TRACK_HYDRA_OFFSET_Y("Hydra Y Offset", true, false),
+    POS_TRACK_HYDRA_OFFSET_Z("Hydra Z Offset", true, false),
+    POS_TRACK_OFFSET_SET_DEFAULT("Default Offsets", false, true),
+    POS_TRACK_HYDRA_DISTANCE_SCALE("Dist. Scale", true, false),
+    POS_TRACK_HYDRA_USE_CONTROLLER_ONE("Controller", false, true),
+    POS_TRACK_HYDRA_AT_BACKOFHEAD_IS_POINTING_LEFT("Hydra Direction", false, true),
+    HYDRA_USE_FILTER("Filter", false, true),
+    POS_TRACK_Y_AXIS_DISTANCE_SKEW("Distance Skew Angle", true, false),
+
+    //Movement/aiming controls
+    DECOUPLE_LOOK_MOVE("Decouple Look/Move", false, true),
+    MOVEMENT_MULTIPLIER("Move. Speed Multiplier", true, false),
+    PITCH_AFFECTS_CAMERA("Pitch Affects Camera", false, true),
+    JOYSTICK_SENSITIVITY("Joystick Sensitivity",true,false),
+    KEYHOLE_WIDTH("Keyhole Width",true,false),
+    KEYHOLE_HEIGHT("Keyhole Height",true,false),
+    KEYHOLE_HEAD_RELATIVE("Keyhole Moves With Head",false,true),
+    MOVEAIM_HYDRA_USE_CONTROLLER_ONE("Controller", false, true),
+
+    // Calibration
+    CALIBRATION_STRATEGY("Initial Calibration", false, false);
+
     private final boolean enumFloat;
     private final boolean enumBoolean;
     private final String enumString;
