--- a/net/minecraft/src/FontRenderer.java
+++ b/net/minecraft/src/FontRenderer.java
@@ -280,9 +280,9 @@
         if (field_111274_c[par1] == null)
         {
             field_111274_c[par1] = new ResourceLocation(String.format("textures/font/unicode_page_%02x.png", new Object[] {Integer.valueOf(par1)}));
-            field_111274_c[par1] = getHdFontLocation(field_111274_c[par1]);
         }
 
+        field_111274_c[par1] = getHdFontLocation(field_111274_c[par1]);
         return field_111274_c[par1];
     }
 
