--- a/net/minecraft/src/FontMetadataSection.java
+++ b/net/minecraft/src/FontMetadataSection.java
@@ -2,14 +2,14 @@
 
 public class FontMetadataSection implements MetadataSection
 {
-    private final float[] charWidths;
-    private final float[] charLefts;
-    private final float[] charSpacings;
+    private final float[] field_110467_a;
+    private final float[] field_110465_b;
+    private final float[] field_110466_c;
 
     public FontMetadataSection(float[] par1ArrayOfFloat, float[] par2ArrayOfFloat, float[] par3ArrayOfFloat)
     {
-        this.charWidths = par1ArrayOfFloat;
-        this.charLefts = par2ArrayOfFloat;
-        this.charSpacings = par3ArrayOfFloat;
+        this.field_110467_a = par1ArrayOfFloat;
+        this.field_110465_b = par2ArrayOfFloat;
+        this.field_110466_c = par3ArrayOfFloat;
     }
 }
