--- a/net/minecraft/src/FontMetadataSectionSerializer.java
+++ b/net/minecraft/src/FontMetadataSectionSerializer.java
@@ -69,10 +69,7 @@
         return new FontMetadataSection(var5, var7, var6);
     }
 
-    /**
-     * The name of this section type as it appears in JSON.
-     */
-    public String getSectionName()
+    public String func_110483_a()
     {
         return "font";
     }
