--- a/net/minecraft/src/FoodStats.java
+++ b/net/minecraft/src/FoodStats.java
@@ -71,7 +71,7 @@
 
             if (this.foodTimer >= 80)
             {
-                if (par1EntityPlayer.getHealth() > 10.0F || var2 >= 3 || par1EntityPlayer.getHealth() > 1.0F && var2 >= 2)
+                if (par1EntityPlayer.func_110143_aJ() > 10.0F || var2 >= 3 || par1EntityPlayer.func_110143_aJ() > 1.0F && var2 >= 2)
                 {
                     par1EntityPlayer.attackEntityFrom(DamageSource.starve, 1.0F);
                 }
