--- a/net/minecraft/src/GuiAchievements.java
+++ b/net/minecraft/src/GuiAchievements.java
@@ -415,8 +415,8 @@
         {
             String var33 = I18n.getString(var30.getName());
             String var35 = var30.getDescription();
-            var41 = par1 + 12;
-            var25 = par2 - 4;
+            var41 = par1 - 40; //+ 12;
+            var25 = par2 + 12; //- 4;
 
             if (this.statFileWriter.canUnlockAchievement(var30))
             {
