--- a/net/minecraft/src/GuiBeaconButton.java
+++ b/net/minecraft/src/GuiBeaconButton.java
@@ -25,7 +25,7 @@
     {
         if (this.drawButton)
         {
-            par1Minecraft.getTextureManager().bindTexture(GuiBeacon.getBeaconGuiTextures());
+            par1Minecraft.func_110434_K().func_110577_a(GuiBeacon.func_110427_g());
             GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
             this.field_82253_i = par2 >= this.xPosition && par3 >= this.yPosition && par2 < this.xPosition + this.width && par3 < this.yPosition + this.height;
             short var4 = 219;
@@ -46,9 +46,9 @@
 
             this.drawTexturedModalRect(this.xPosition, this.yPosition, var5, var4, this.width, this.height);
 
-            if (!GuiBeacon.getBeaconGuiTextures().equals(this.buttonTexture))
+            if (!GuiBeacon.func_110427_g().equals(this.buttonTexture))
             {
-                par1Minecraft.getTextureManager().bindTexture(this.buttonTexture);
+                par1Minecraft.func_110434_K().func_110577_a(this.buttonTexture);
             }
 
             this.drawTexturedModalRect(this.xPosition + 2, this.yPosition + 2, this.field_82257_l, this.field_82258_m, 18, 18);
