--- a/net/minecraft/src/GuiButtonMerchant.java
+++ b/net/minecraft/src/GuiButtonMerchant.java
@@ -22,7 +22,7 @@
     {
         if (this.drawButton)
         {
-            par1Minecraft.getTextureManager().bindTexture(GuiMerchant.func_110417_h());
+            par1Minecraft.func_110434_K().func_110577_a(GuiMerchant.func_110417_h());
             GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
             boolean var4 = par2 >= this.xPosition && par3 >= this.yPosition && par2 < this.xPosition + this.width && par3 < this.yPosition + this.height;
             int var5 = 0;
