--- a/net/minecraft/src/GuiBrewingStand.java
+++ b/net/minecraft/src/GuiBrewingStand.java
@@ -51,27 +51,21 @@
                 case 0:
                     var7 = 29;
                     break;
-
                 case 1:
                     var7 = 24;
                     break;
-
                 case 2:
                     var7 = 20;
                     break;
-
                 case 3:
                     var7 = 16;
                     break;
-
                 case 4:
                     var7 = 11;
                     break;
-
                 case 5:
                     var7 = 6;
                     break;
-
                 case 6:
                     var7 = 0;
             }
