--- a/net/minecraft/src/GuiCommandBlock.java
+++ b/net/minecraft/src/GuiCommandBlock.java
@@ -34,8 +34,8 @@
     {
         Keyboard.enableRepeatEvents(true);
         this.buttonList.clear();
-        this.buttonList.add(this.doneBtn = new GuiButton(0, this.width / 2 - 100, this.height / 4 + 96 + 12, I18n.getString("gui.done")));
-        this.buttonList.add(this.cancelBtn = new GuiButton(1, this.width / 2 - 100, this.height / 4 + 120 + 12, I18n.getString("gui.cancel")));
+        this.buttonList.add(this.doneBtn = new GuiButton(0, this.width / 2 - 100, this.height / 4 + 96 + 12, I18n.func_135053_a("gui.done")));
+        this.buttonList.add(this.cancelBtn = new GuiButton(1, this.width / 2 - 100, this.height / 4 + 120 + 12, I18n.func_135053_a("gui.cancel")));
         this.commandTextField = new GuiTextField(this.fontRenderer, this.width / 2 - 150, 60, 300, 20);
         this.commandTextField.setMaxStringLength(32767);
         this.commandTextField.setFocused(true);
@@ -122,11 +122,11 @@
     public void drawScreen(int par1, int par2, float par3)
     {
         this.drawDefaultBackground();
-        this.drawCenteredString(this.fontRenderer, I18n.getString("advMode.setCommand"), this.width / 2, 20, 16777215);
-        this.drawString(this.fontRenderer, I18n.getString("advMode.command"), this.width / 2 - 150, 47, 10526880);
-        this.drawString(this.fontRenderer, I18n.getString("advMode.nearestPlayer"), this.width / 2 - 150, 97, 10526880);
-        this.drawString(this.fontRenderer, I18n.getString("advMode.randomPlayer"), this.width / 2 - 150, 108, 10526880);
-        this.drawString(this.fontRenderer, I18n.getString("advMode.allPlayers"), this.width / 2 - 150, 119, 10526880);
+        this.drawCenteredString(this.fontRenderer, I18n.func_135053_a("advMode.setCommand"), this.width / 2, 20, 16777215);
+        this.drawString(this.fontRenderer, I18n.func_135053_a("advMode.command"), this.width / 2 - 150, 47, 10526880);
+        this.drawString(this.fontRenderer, I18n.func_135053_a("advMode.nearestPlayer"), this.width / 2 - 150, 97, 10526880);
+        this.drawString(this.fontRenderer, I18n.func_135053_a("advMode.randomPlayer"), this.width / 2 - 150, 108, 10526880);
+        this.drawString(this.fontRenderer, I18n.func_135053_a("advMode.allPlayers"), this.width / 2 - 150, 119, 10526880);
         this.commandTextField.drawTextBox();
         super.drawScreen(par1, par2, par3);
     }
