--- a/net/minecraft/src/GuiCreateFlatWorldListSlot.java
+++ b/net/minecraft/src/GuiCreateFlatWorldListSlot.java
@@ -24,7 +24,7 @@
         if (par3ItemStack != null)
         {
             RenderHelper.enableGUIStandardItemLighting();
-            GuiCreateFlatWorld.getRenderItem().renderItemIntoGUI(this.createFlatWorldGui.fontRenderer, this.createFlatWorldGui.mc.getTextureManager(), par3ItemStack, par1 + 2, par2 + 2);
+            GuiCreateFlatWorld.getRenderItem().renderItemIntoGUI(this.createFlatWorldGui.fontRenderer, this.createFlatWorldGui.mc.func_110434_K(), par3ItemStack, par1 + 2, par2 + 2);
             RenderHelper.disableStandardItemLighting();
         }
 
@@ -39,7 +39,7 @@
     private void func_82450_b(int par1, int par2, int par3, int par4)
     {
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-        this.createFlatWorldGui.mc.getTextureManager().bindTexture(Gui.statIcons);
+        this.createFlatWorldGui.mc.func_110434_K().func_110577_a(Gui.field_110323_l);
         float var5 = 0.0078125F;
         float var6 = 0.0078125F;
         boolean var7 = true;
@@ -91,15 +91,15 @@
 
         if (par1 == 0)
         {
-            var9 = I18n.getStringParams("createWorld.customize.flat.layer.top", new Object[] {Integer.valueOf(var6.getLayerCount())});
+            var9 = I18n.func_135052_a("createWorld.customize.flat.layer.top", new Object[] {Integer.valueOf(var6.getLayerCount())});
         }
         else if (par1 == GuiCreateFlatWorld.func_82271_a(this.createFlatWorldGui).getFlatLayers().size() - 1)
         {
-            var9 = I18n.getStringParams("createWorld.customize.flat.layer.bottom", new Object[] {Integer.valueOf(var6.getLayerCount())});
+            var9 = I18n.func_135052_a("createWorld.customize.flat.layer.bottom", new Object[] {Integer.valueOf(var6.getLayerCount())});
         }
         else
         {
-            var9 = I18n.getStringParams("createWorld.customize.flat.layer", new Object[] {Integer.valueOf(var6.getLayerCount())});
+            var9 = I18n.func_135052_a("createWorld.customize.flat.layer", new Object[] {Integer.valueOf(var6.getLayerCount())});
         }
 
         this.createFlatWorldGui.fontRenderer.drawString(var9, par2 + 2 + 213 - this.createFlatWorldGui.fontRenderer.getStringWidth(var9), par3 + 3, 16777215);
