--- a/net/minecraft/src/GuiErrorScreen.java
+++ b/net/minecraft/src/GuiErrorScreen.java
@@ -24,7 +24,7 @@
     public void initGui()
     {
         super.initGui();
-        this.buttonList.add(new GuiButton(0, this.width / 2 - 100, 140, I18n.getString("gui.cancel")));
+        this.buttonList.add(new GuiButton(0, this.width / 2 - 100, 140, I18n.func_135053_a("gui.cancel")));
     }
 
     /**
