--- a/net/minecraft/src/GuiFlatPresetsListSlot.java
+++ b/net/minecraft/src/GuiFlatPresetsListSlot.java
@@ -21,7 +21,7 @@
         this.func_82456_d(par1 + 1, par2 + 1);
         GL11.glEnable(GL12.GL_RESCALE_NORMAL);
         RenderHelper.enableGUIStandardItemLighting();
-        GuiFlatPresets.getPresetIconRenderer().renderItemIntoGUI(this.flatPresetsGui.fontRenderer, this.flatPresetsGui.mc.getTextureManager(), new ItemStack(par3, 1, 0), par1 + 2, par2 + 2);
+        GuiFlatPresets.getPresetIconRenderer().renderItemIntoGUI(this.flatPresetsGui.fontRenderer, this.flatPresetsGui.mc.func_110434_K(), new ItemStack(par3, 1, 0), par1 + 2, par2 + 2);
         RenderHelper.disableStandardItemLighting();
         GL11.glDisable(GL12.GL_RESCALE_NORMAL);
     }
@@ -34,7 +34,7 @@
     private void func_82455_b(int par1, int par2, int par3, int par4)
     {
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-        this.flatPresetsGui.mc.getTextureManager().bindTexture(Gui.statIcons);
+        this.flatPresetsGui.mc.func_110434_K().func_110577_a(Gui.field_110323_l);
         float var5 = 0.0078125F;
         float var6 = 0.0078125F;
         boolean var7 = true;
