--- a/net/minecraft/src/GuiIngameMenu.java
+++ b/net/minecraft/src/GuiIngameMenu.java
@@ -43,33 +43,27 @@
             case 0:
                 this.mc.displayGuiScreen(new GuiOptions(this, this.mc.gameSettings));
                 break;
-
             case 1:
                 par1GuiButton.enabled = false;
                 this.mc.statFileWriter.readStat(StatList.leaveGameStat, 1);
                 this.mc.theWorld.sendQuittingDisconnectingPacket();
                 this.mc.loadWorld((WorldClient)null);
                 this.mc.displayGuiScreen(new GuiMainMenu());
-
             case 2:
             case 3:
             default:
                 break;
-
             case 4:
                 this.mc.displayGuiScreen((GuiScreen)null);
                 this.mc.setIngameFocus();
                 this.mc.sndManager.resumeAllSounds();
                 break;
-
             case 5:
                 this.mc.displayGuiScreen(new GuiAchievements(this.mc.statFileWriter));
                 break;
-
             case 6:
                 this.mc.displayGuiScreen(new GuiStats(this, this.mc.statFileWriter));
                 break;
-
             case 7:
                 this.mc.displayGuiScreen(new GuiShareToLan(this));
         }
