--- a/net/minecraft/src/GuiIngame.java
+++ b/net/minecraft/src/GuiIngame.java
@@ -10,9 +10,9 @@
 
 public class GuiIngame extends Gui
 {
-    private static final ResourceLocation vignetteTexPath = new ResourceLocation("textures/misc/vignette.png");
-    private static final ResourceLocation widgetsTexPath = new ResourceLocation("textures/gui/widgets.png");
-    private static final ResourceLocation pumpkinBlurTexPath = new ResourceLocation("textures/misc/pumpkinblur.png");
+    private static final ResourceLocation field_110329_b = new ResourceLocation("textures/misc/vignette.png");
+    private static final ResourceLocation field_110330_c = new ResourceLocation("textures/gui/widgets.png");
+    private static final ResourceLocation field_110328_d = new ResourceLocation("textures/misc/pumpkinblur.png");
     private static final RenderItem itemRenderer = new RenderItem();
     private final Random rand = new Random();
     private final Minecraft mc;
@@ -64,13 +64,6 @@
             GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
         }
 
-        ItemStack var9 = this.mc.thePlayer.inventory.armorItemInSlot(3);
-
-        if (this.mc.gameSettings.thirdPersonView == 0 && var9 != null && var9.itemID == Block.pumpkin.blockID)
-        {
-            this.renderPumpkinBlur(var6, var7);
-        }
-
         if (!this.mc.thePlayer.isPotionActive(Potion.confusion))
         {
             float var10 = this.mc.thePlayer.prevTimeInPortal + (this.mc.thePlayer.timeInPortal - this.mc.thePlayer.prevTimeInPortal) * par1;
@@ -88,16 +81,19 @@
         if (!this.mc.playerController.enableEverythingIsScrewedUpMode())
         {
             GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-            this.mc.getTextureManager().bindTexture(widgetsTexPath);
+            this.mc.func_110434_K().func_110577_a(field_110330_c);
             InventoryPlayer var31 = this.mc.thePlayer.inventory;
             this.zLevel = -90.0F;
             this.drawTexturedModalRect(var6 / 2 - 91, var7 - 22, 0, 0, 182, 22);
             this.drawTexturedModalRect(var6 / 2 - 91 - 1 + var31.currentItem * 20, var7 - 22 - 1, 0, 22, 24, 22);
-            this.mc.getTextureManager().bindTexture(icons);
-            GL11.glEnable(GL11.GL_BLEND);
-            GL11.glBlendFunc(GL11.GL_ONE_MINUS_DST_COLOR, GL11.GL_ONE_MINUS_SRC_COLOR);
-            this.drawTexturedModalRect(var6 / 2 - 7, var7 / 2 - 7, 0, 0, 16, 16);
-            GL11.glDisable(GL11.GL_BLEND);
+            this.mc.func_110434_K().func_110577_a(field_110324_m);
+            if( !this.mc.vrSettings.useVRRenderer )
+            {
+	            GL11.glEnable(GL11.GL_BLEND);
+	            GL11.glBlendFunc(GL11.GL_ONE_MINUS_DST_COLOR, GL11.GL_ONE_MINUS_SRC_COLOR);
+	            this.drawTexturedModalRect(var6 / 2 - 7, var7 / 2 - 7, 0, 0, 16, 16);
+	            GL11.glDisable(GL11.GL_BLEND);
+            }
             this.mc.mcProfiler.startSection("bossHealth");
             this.renderBossHealth();
             this.mc.mcProfiler.endSection();
@@ -156,11 +152,11 @@
         float var33;
         short var37;
 
-        if (this.mc.thePlayer.isRidingHorse())
+        if (this.mc.thePlayer.func_110317_t())
         {
             this.mc.mcProfiler.startSection("jumpBar");
-            this.mc.getTextureManager().bindTexture(Gui.icons);
-            var33 = this.mc.thePlayer.getHorseJumpPower();
+            this.mc.func_110434_K().func_110577_a(Gui.field_110324_m);
+            var33 = this.mc.thePlayer.func_110319_bJ();
             var37 = 182;
             var14 = (int)(var33 * (float)(var37 + 1));
             var15 = var7 - 32 + 3;
@@ -176,7 +172,7 @@
         else if (this.mc.playerController.func_78763_f())
         {
             this.mc.mcProfiler.startSection("expBar");
-            this.mc.getTextureManager().bindTexture(Gui.icons);
+            this.mc.func_110434_K().func_110577_a(Gui.field_110324_m);
             var12 = this.mc.thePlayer.xpBarCap();
 
             if (var12 > 0)
@@ -257,11 +253,11 @@
 
             if (this.mc.theWorld.getTotalWorldTime() >= 120500L)
             {
-                var36 = I18n.getString("demo.demoExpired");
+                var36 = I18n.func_135053_a("demo.demoExpired");
             }
             else
             {
-                var36 = I18n.getStringParams("demo.remainingTime", new Object[] {StringUtils.ticksToElapsedTime((int)(120500L - this.mc.theWorld.getTotalWorldTime()))});
+                var36 = I18n.func_135052_a("demo.remainingTime", new Object[] {StringUtils.ticksToElapsedTime((int)(120500L - this.mc.theWorld.getTotalWorldTime()))});
             }
 
             var13 = var8.getStringWidth(var36);
@@ -277,7 +273,7 @@
         {
             this.mc.mcProfiler.startSection("debug");
             GL11.glPushMatrix();
-            var8.drawStringWithShadow("Minecraft 1.6.4 (" + this.mc.debug + ")", 2, 2, 16777215);
+            var8.drawStringWithShadow("Minecraft 1.6.2 (" + this.mc.debug + ")", 2, 2, 16777215);
             var8.drawStringWithShadow(this.mc.debugInfoRenders(), 2, 12, 16777215);
             var8.drawStringWithShadow(this.mc.getEntityDebug(), 2, 22, 16777215);
             var8.drawStringWithShadow(this.mc.debugInfoEntities(), 2, 32, 16777215);
@@ -414,7 +410,7 @@
                     }
 
                     GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-                    this.mc.getTextureManager().bindTexture(icons);
+                    this.mc.func_110434_K().func_110577_a(field_110324_m);
                     byte var53 = 0;
                     boolean var51 = false;
                     byte var50;
@@ -513,19 +509,19 @@
             var3 = false;
         }
 
-        int var4 = MathHelper.ceiling_float_int(this.mc.thePlayer.getHealth());
+        int var4 = MathHelper.ceiling_float_int(this.mc.thePlayer.func_110143_aJ());
         int var5 = MathHelper.ceiling_float_int(this.mc.thePlayer.prevHealth);
         this.rand.setSeed((long)(this.updateCounter * 312871));
         boolean var6 = false;
         FoodStats var7 = this.mc.thePlayer.getFoodStats();
         int var8 = var7.getFoodLevel();
         int var9 = var7.getPrevFoodLevel();
-        AttributeInstance var10 = this.mc.thePlayer.getEntityAttribute(SharedMonsterAttributes.maxHealth);
+        AttributeInstance var10 = this.mc.thePlayer.func_110148_a(SharedMonsterAttributes.field_111267_a);
         int var11 = par1 / 2 - 91;
         int var12 = par1 / 2 + 91;
         int var13 = par2 - 39;
-        float var14 = (float)var10.getAttributeValue();
-        float var15 = this.mc.thePlayer.getAbsorptionAmount();
+        float var14 = (float)var10.func_111126_e();
+        float var15 = this.mc.thePlayer.func_110139_bj();
         int var16 = MathHelper.ceiling_float_int((var14 + var15) / 2.0F / 10.0F);
         int var17 = Math.max(10 - (var16 - 2), 3);
         int var18 = var13 - (var16 - 1) * var17 - 10;
@@ -713,8 +709,8 @@
         {
             this.mc.mcProfiler.endStartSection("mountHealth");
             EntityLivingBase var38 = (EntityLivingBase)var34;
-            var35 = (int)Math.ceil((double)var38.getHealth());
-            float var37 = var38.getMaxHealth();
+            var35 = (int)Math.ceil((double)var38.func_110143_aJ());
+            float var37 = var38.func_110138_aP();
             var26 = (int)(var37 + 0.5F) / 2;
 
             if (var26 > 30)
@@ -807,7 +803,7 @@
             String var8 = BossStatus.bossName;
             var1.drawStringWithShadow(var8, var3 / 2 - var1.getStringWidth(var8) / 2, var7 - 10, 16777215);
             GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-            this.mc.getTextureManager().bindTexture(icons);
+            this.mc.func_110434_K().func_110577_a(field_110324_m);
         }
     }
 
@@ -818,7 +814,7 @@
         GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
         GL11.glDisable(GL11.GL_ALPHA_TEST);
-        this.mc.getTextureManager().bindTexture(pumpkinBlurTexPath);
+        this.mc.func_110434_K().func_110577_a(field_110328_d);
         Tessellator var3 = Tessellator.instance;
         var3.startDrawingQuads();
         var3.addVertexWithUV(0.0D, (double)par2, -90.0D, 0.0D, 1.0D);
@@ -854,7 +850,7 @@
         GL11.glDepthMask(false);
         GL11.glBlendFunc(GL11.GL_ZERO, GL11.GL_ONE_MINUS_SRC_COLOR);
         GL11.glColor4f(this.prevVignetteBrightness, this.prevVignetteBrightness, this.prevVignetteBrightness, 1.0F);
-        this.mc.getTextureManager().bindTexture(vignetteTexPath);
+        this.mc.func_110434_K().func_110577_a(field_110329_b);
         Tessellator var4 = Tessellator.instance;
         var4.startDrawingQuads();
         var4.addVertexWithUV(0.0D, (double)par3, -90.0D, 0.0D, 1.0D);
@@ -883,7 +879,7 @@
         GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
         GL11.glColor4f(1.0F, 1.0F, 1.0F, par1);
         Icon var4 = Block.portal.getBlockTextureFromSide(1);
-        this.mc.getTextureManager().bindTexture(TextureMap.locationBlocksTexture);
+        this.mc.func_110434_K().func_110577_a(TextureMap.field_110575_b);
         float var5 = var4.getMinU();
         float var6 = var4.getMinV();
         float var7 = var4.getMaxU();
@@ -921,14 +917,14 @@
                 GL11.glTranslatef((float)(-(par2 + 8)), (float)(-(par3 + 12)), 0.0F);
             }
 
-            itemRenderer.renderItemAndEffectIntoGUI(this.mc.fontRenderer, this.mc.getTextureManager(), var5, par2, par3);
+            itemRenderer.renderItemAndEffectIntoGUI(this.mc.fontRenderer, this.mc.func_110434_K(), var5, par2, par3);
 
             if (var6 > 0.0F)
             {
                 GL11.glPopMatrix();
             }
 
-            itemRenderer.renderItemOverlayIntoGUI(this.mc.fontRenderer, this.mc.getTextureManager(), var5, par2, par3);
+            itemRenderer.renderItemOverlayIntoGUI(this.mc.fontRenderer, this.mc.func_110434_K(), var5, par2, par3);
         }
     }
 
