--- a/net/minecraft/src/GuiLanguage.java
+++ b/net/minecraft/src/GuiLanguage.java
@@ -43,11 +43,9 @@
             {
                 case 5:
                     break;
-
                 case 6:
                     this.mc.displayGuiScreen(this.parentGui);
                     break;
-
                 default:
                     this.languageList.actionPerformed(par1GuiButton);
             }
