--- a/net/minecraft/src/GuiMultiplayer.java
+++ b/net/minecraft/src/GuiMultiplayer.java
@@ -106,13 +106,13 @@
      */
     public void initGuiControls()
     {
-        this.buttonList.add(this.field_96289_p = new GuiButton(7, this.width / 2 - 154, this.height - 28, 70, 20, I18n.getString("selectServer.edit")));
-        this.buttonList.add(this.buttonDelete = new GuiButton(2, this.width / 2 - 74, this.height - 28, 70, 20, I18n.getString("selectServer.delete")));
-        this.buttonList.add(this.buttonSelect = new GuiButton(1, this.width / 2 - 154, this.height - 52, 100, 20, I18n.getString("selectServer.select")));
-        this.buttonList.add(new GuiButton(4, this.width / 2 - 50, this.height - 52, 100, 20, I18n.getString("selectServer.direct")));
-        this.buttonList.add(new GuiButton(3, this.width / 2 + 4 + 50, this.height - 52, 100, 20, I18n.getString("selectServer.add")));
-        this.buttonList.add(new GuiButton(8, this.width / 2 + 4, this.height - 28, 70, 20, I18n.getString("selectServer.refresh")));
-        this.buttonList.add(new GuiButton(0, this.width / 2 + 4 + 76, this.height - 28, 75, 20, I18n.getString("gui.cancel")));
+        this.buttonList.add(this.field_96289_p = new GuiButton(7, this.width / 2 - 154, this.height - 28, 70, 20, I18n.func_135053_a("selectServer.edit")));
+        this.buttonList.add(this.buttonDelete = new GuiButton(2, this.width / 2 - 74, this.height - 28, 70, 20, I18n.func_135053_a("selectServer.delete")));
+        this.buttonList.add(this.buttonSelect = new GuiButton(1, this.width / 2 - 154, this.height - 52, 100, 20, I18n.func_135053_a("selectServer.select")));
+        this.buttonList.add(new GuiButton(4, this.width / 2 - 50, this.height - 52, 100, 20, I18n.func_135053_a("selectServer.direct")));
+        this.buttonList.add(new GuiButton(3, this.width / 2 + 4 + 50, this.height - 52, 100, 20, I18n.func_135053_a("selectServer.add")));
+        this.buttonList.add(new GuiButton(8, this.width / 2 + 4, this.height - 28, 70, 20, I18n.func_135053_a("selectServer.refresh")));
+        this.buttonList.add(new GuiButton(0, this.width / 2 + 4 + 76, this.height - 28, 75, 20, I18n.func_135053_a("gui.cancel")));
         boolean var1 = this.selectedServer >= 0 && this.selectedServer < this.serverSlotContainer.getSize();
         this.buttonSelect.enabled = var1;
         this.field_96289_p.enabled = var1;
@@ -162,10 +162,10 @@
                 if (var2 != null)
                 {
                     this.deleteClicked = true;
-                    String var3 = I18n.getString("selectServer.deleteQuestion");
-                    String var4 = "\'" + var2 + "\' " + I18n.getString("selectServer.deleteWarning");
-                    String var5 = I18n.getString("selectServer.deleteButton");
-                    String var6 = I18n.getString("gui.cancel");
+                    String var3 = I18n.func_135053_a("selectServer.deleteQuestion");
+                    String var4 = "\'" + var2 + "\' " + I18n.func_135053_a("selectServer.deleteWarning");
+                    String var5 = I18n.func_135053_a("selectServer.deleteButton");
+                    String var6 = I18n.func_135053_a("gui.cancel");
                     GuiYesNo var7 = new GuiYesNo(this, var3, var4, var5, var6, this.selectedServer);
                     this.mc.displayGuiScreen(var7);
                 }
@@ -177,12 +177,12 @@
             else if (par1GuiButton.id == 4)
             {
                 this.directClicked = true;
-                this.mc.displayGuiScreen(new GuiScreenServerList(this, this.theServerData = new ServerData(I18n.getString("selectServer.defaultName"), "")));
+                this.mc.displayGuiScreen(new GuiScreenServerList(this, this.theServerData = new ServerData(I18n.func_135053_a("selectServer.defaultName"), "")));
             }
             else if (par1GuiButton.id == 3)
             {
                 this.addClicked = true;
-                this.mc.displayGuiScreen(new GuiScreenAddServer(this, this.theServerData = new ServerData(I18n.getString("selectServer.defaultName"), "")));
+                this.mc.displayGuiScreen(new GuiScreenAddServer(this, this.theServerData = new ServerData(I18n.func_135053_a("selectServer.defaultName"), "")));
             }
             else if (par1GuiButton.id == 7)
             {
@@ -324,7 +324,7 @@
         this.lagTooltip = null;
         this.drawDefaultBackground();
         this.serverSlotContainer.drawScreen(par1, par2, par3);
-        this.drawCenteredString(this.fontRenderer, I18n.getString("multiplayer.title"), this.width / 2, 20, 16777215);
+        this.drawCenteredString(this.fontRenderer, I18n.func_135053_a("multiplayer.title"), this.width / 2, 20, 16777215);
         super.drawScreen(par1, par2, par3);
 
         if (this.lagTooltip != null)
@@ -375,7 +375,7 @@
             var2.connect(new InetSocketAddress(var1.getIP(), var1.getPort()), 3000);
             var3 = new DataInputStream(var2.getInputStream());
             var4 = new DataOutputStream(var2.getOutputStream());
-            Packet254ServerPing var5 = new Packet254ServerPing(78, var1.getIP(), var1.getPort());
+            Packet254ServerPing var5 = new Packet254ServerPing(74, var1.getIP(), var1.getPort());
             var4.writeByte(var5.getPacketId());
             var5.writePacketData(var4);
 
@@ -425,7 +425,7 @@
                 {
                     par0ServerData.gameVersion = "???";
                     par0ServerData.serverMOTD = "" + EnumChatFormatting.DARK_GRAY + "???";
-                    par0ServerData.field_82821_f = 79;
+                    par0ServerData.field_82821_f = 75;
                     par0ServerData.populationInfo = "" + EnumChatFormatting.DARK_GRAY + "???";
                 }
             }
@@ -458,7 +458,7 @@
                 }
 
                 par0ServerData.gameVersion = "1.3";
-                par0ServerData.field_82821_f = 77;
+                par0ServerData.field_82821_f = 73;
             }
         }
         finally
