--- a/net/minecraft/src/GuiOptions.java
+++ b/net/minecraft/src/GuiOptions.java
@@ -1,5 +1,7 @@
 package net.minecraft.src;
 
+import com.mtbs3d.minecrift.GuiMinecriftSettings;
+
 public class GuiOptions extends GuiScreen
 {
     /**
@@ -122,6 +124,12 @@
                 this.mc.gameSettings.saveOptions();
                 this.mc.displayGuiScreen(new GuiTexturePacks(this, this.options));
             }
+
+            if (par1GuiButton.id == 666)
+            {
+                this.mc.gameSettings.saveOptions();
+                this.mc.displayGuiScreen(new GuiMinecriftSettings(this, this, this.options));
+            }
         }
     }
 
@@ -134,4 +142,5 @@
         this.drawCenteredString(this.fontRenderer, this.screenTitle, this.width / 2, 15, 16777215);
         super.drawScreen(par1, par2, par3);
     }
+
 }
