--- a/net/minecraft/src/GuiOptions.java
+++ b/net/minecraft/src/GuiOptions.java
@@ -1,11 +1,13 @@
 package net.minecraft.src;
 
+import com.mtbs3d.minecrift.gui.GuiMinecriftSettings;
+
 public class GuiOptions extends GuiScreen
 {
     /**
      * An array of options that can be changed directly from the options GUI.
      */
-    private static final EnumOptions[] relevantOptions = new EnumOptions[] {EnumOptions.MUSIC, EnumOptions.SOUND, EnumOptions.INVERT_MOUSE, EnumOptions.SENSITIVITY, EnumOptions.FOV, EnumOptions.DIFFICULTY, EnumOptions.TOUCHSCREEN};
+    private static final EnumOptions[] relevantOptions = new EnumOptions[] {EnumOptions.MUSIC, EnumOptions.SOUND, EnumOptions.INVERT_MOUSE, EnumOptions.SENSITIVITY, EnumOptions.TOUCHSCREEN, EnumOptions.DIFFICULTY, EnumOptions.FOV};
 
     /**
      * A reference to the screen object that created this. Used for navigating between screens.
@@ -38,8 +40,13 @@
         for (int var5 = 0; var5 < var4; ++var5)
         {
             EnumOptions var6 = var3[var5];
-
-            if (var6.getEnumFloat())
+            if (var6.getEnumString().equalsIgnoreCase("options.fov"))    // Replace FOV slider with 'Minecrift settings' button
+            {
+                // Minecrift settings
+                GuiSmallButton var7 = new GuiSmallButton(666, this.width / 2 - 155 + var2 % 2 * 160, this.height / 6 - 12 + 24 * (var2 >> 1), var6, "VR Settings...");
+                this.buttonList.add(var7);
+            }
+            else if (var6.getEnumFloat())
             {
                 this.buttonList.add(new GuiSlider(var6.returnEnumOrdinal(), this.width / 2 - 155 + var2 % 2 * 160, this.height / 6 - 12 + 24 * (var2 >> 1), var6, this.options.getKeyBinding(var6), this.options.getOptionFloatValue(var6)));
             }
@@ -122,6 +129,12 @@
                 this.mc.gameSettings.saveOptions();
                 this.mc.displayGuiScreen(new GuiTexturePacks(this, this.options));
             }
+
+            if (par1GuiButton.id == 666)
+            {
+                this.mc.gameSettings.saveOptions();
+                this.mc.displayGuiScreen(new GuiMinecriftSettings(this, this.options));
+            }
         }
     }
 
@@ -134,4 +147,5 @@
         this.drawCenteredString(this.fontRenderer, this.screenTitle, this.width / 2, 15, 16777215);
         super.drawScreen(par1, par2, par3);
     }
+
 }
