--- a/net/minecraft/src/GuiOptions.java
+++ b/net/minecraft/src/GuiOptions.java
@@ -1,11 +1,13 @@
 package net.minecraft.src;
 
+import com.mtbs3d.minecrift.gui.GuiMinecriftSettings;
+
 public class GuiOptions extends GuiScreen
 {
     /**
      * An array of options that can be changed directly from the options GUI.
      */
-    private static final EnumOptions[] relevantOptions = new EnumOptions[] {EnumOptions.MUSIC, EnumOptions.SOUND, EnumOptions.INVERT_MOUSE, EnumOptions.SENSITIVITY, EnumOptions.FOV, EnumOptions.DIFFICULTY, EnumOptions.TOUCHSCREEN};
+    private static final EnumOptions[] relevantOptions = new EnumOptions[] {EnumOptions.MUSIC, EnumOptions.SOUND, EnumOptions.INVERT_MOUSE, EnumOptions.SENSITIVITY, EnumOptions.TOUCHSCREEN, EnumOptions.DIFFICULTY, EnumOptions.FOV};
 
     /**
      * A reference to the screen object that created this. Used for navigating between screens.
@@ -37,8 +39,13 @@
         for (int var4 = 0; var4 < var3; ++var4)
         {
             EnumOptions var5 = var2[var4];
-
-            if (var5.getEnumFloat())
+            if (var5.getEnumString().equalsIgnoreCase("options.fov"))    // Replace FOV slider with 'Minecrift settings' button
+            {
+                // Minecrift settings
+                GuiSmallButton var7 = new GuiSmallButton(666, this.width / 2 - 155 + var1 % 2 * 160, this.height / 6 - 12 + 24 * (var1 >> 1), var5, "VR Settings...");
+                this.buttonList.add(var7);
+            }
+            else if (var5.getEnumFloat())
             {
                 this.buttonList.add(new GuiSlider(var5.returnEnumOrdinal(), this.width / 2 - 155 + var1 % 2 * 160, this.height / 6 - 12 + 24 * (var1 >> 1), var5, this.options.getKeyBinding(var5), this.options.getOptionFloatValue(var5)));
             }
@@ -58,7 +65,7 @@
             ++var1;
         }
 
-        this.buttonList.add(new GuiButton(101, this.width / 2 - 152, this.height / 6 + 96 - 6, 150, 20, I18n.getString("options.video")));
+        this.buttonList.add(new GuiButton(101, this.width / 2 - 152, this.height / 6 + 96 - 6, 150, 20, "OptiFine Settings..."));
         this.buttonList.add(new GuiButton(100, this.width / 2 + 2, this.height / 6 + 96 - 6, 150, 20, I18n.getString("options.controls")));
         this.buttonList.add(new GuiButton(102, this.width / 2 - 152, this.height / 6 + 120 - 6, 150, 20, I18n.getString("options.language")));
         this.buttonList.add(new GuiButton(103, this.width / 2 + 2, this.height / 6 + 120 - 6, 150, 20, I18n.getString("options.multiplayer.title")));
@@ -121,6 +128,12 @@
                 this.mc.gameSettings.saveOptions();
                 this.mc.displayGuiScreen(new GuiScreenTemporaryResourcePackSelect(this, this.options));
             }
+
+            if (par1GuiButton.id == 666)
+            {
+                this.mc.gameSettings.saveOptions();
+                this.mc.displayGuiScreen(new GuiMinecriftSettings(this, this.mc.vrSettings));
+            }
         }
     }
 
