--- a/net/minecraft/src/GuiRepair.java
+++ b/net/minecraft/src/GuiRepair.java
@@ -6,7 +6,7 @@
 
 public class GuiRepair extends GuiContainer implements ICrafting
 {
-    private static final ResourceLocation anvilGuiTextures = new ResourceLocation("textures/gui/container/anvil.png");
+    private static final ResourceLocation field_110429_t = new ResourceLocation("textures/gui/container/anvil.png");
     private ContainerRepair repairContainer;
     private GuiTextField itemNameField;
     private InventoryPlayer field_82325_q;
@@ -52,17 +52,17 @@
     protected void drawGuiContainerForegroundLayer(int par1, int par2)
     {
         GL11.glDisable(GL11.GL_LIGHTING);
-        this.fontRenderer.drawString(I18n.getString("container.repair"), 60, 6, 4210752);
+        this.fontRenderer.drawString(I18n.func_135053_a("container.repair"), 60, 6, 4210752);
 
         if (this.repairContainer.maximumCost > 0)
         {
             int var3 = 8453920;
             boolean var4 = true;
-            String var5 = I18n.getStringParams("container.repair.cost", new Object[] {Integer.valueOf(this.repairContainer.maximumCost)});
+            String var5 = I18n.func_135052_a("container.repair.cost", new Object[] {Integer.valueOf(this.repairContainer.maximumCost)});
 
             if (this.repairContainer.maximumCost >= 40 && !this.mc.thePlayer.capabilities.isCreativeMode)
             {
-                var5 = I18n.getString("container.repair.expensive");
+                var5 = I18n.func_135053_a("container.repair.expensive");
                 var3 = 16736352;
             }
             else if (!this.repairContainer.getSlot(2).getHasStack())
@@ -153,7 +153,7 @@
     protected void drawGuiContainerBackgroundLayer(float par1, int par2, int par3)
     {
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-        this.mc.getTextureManager().bindTexture(anvilGuiTextures);
+        this.mc.func_110434_K().func_110577_a(field_110429_t);
         int var4 = (this.width - this.xSize) / 2;
         int var5 = (this.height - this.ySize) / 2;
         this.drawTexturedModalRect(var4, var5, 0, 0, this.xSize, this.ySize);
