--- a/net/minecraft/src/GuiScreenConfigureWorld.java
+++ b/net/minecraft/src/GuiScreenConfigureWorld.java
@@ -47,22 +47,22 @@
 
         if (this.field_96280_b.field_96404_d.equals("CLOSED"))
         {
-            this.buttonList.add(this.field_96281_r = new GuiButton(0, this.field_96277_d, this.func_96264_a(12), this.field_96286_n / 2 - 2, 20, I18n.getString("mco.configure.world.buttons.open")));
+            this.buttonList.add(this.field_96281_r = new GuiButton(0, this.field_96277_d, this.func_96264_a(12), this.field_96286_n / 2 - 2, 20, I18n.func_135053_a("mco.configure.world.buttons.open")));
             this.field_96281_r.enabled = !this.field_96280_b.field_98166_h;
         }
         else
         {
-            this.buttonList.add(this.field_96279_s = new GuiButton(1, this.field_96277_d, this.func_96264_a(12), this.field_96286_n / 2 - 2, 20, I18n.getString("mco.configure.world.buttons.close")));
+            this.buttonList.add(this.field_96279_s = new GuiButton(1, this.field_96277_d, this.func_96264_a(12), this.field_96286_n / 2 - 2, 20, I18n.func_135053_a("mco.configure.world.buttons.close")));
             this.field_96279_s.enabled = !this.field_96280_b.field_98166_h;
         }
 
-        this.buttonList.add(this.field_98129_x = new GuiButton(7, this.field_96277_d + this.field_96286_n / 2 + 2, this.func_96264_a(12), this.field_96286_n / 2 - 2, 20, I18n.getString("mco.configure.world.buttons.subscription")));
-        this.buttonList.add(this.field_96278_t = new GuiButton(5, this.field_96277_d, this.func_96264_a(10), this.field_96286_n / 2 - 2, 20, I18n.getString("mco.configure.world.buttons.edit")));
-        this.buttonList.add(this.field_96276_u = new GuiButton(6, this.field_96277_d + this.field_96286_n / 2 + 2, this.func_96264_a(10), this.field_96286_n / 2 - 2, 20, I18n.getString("mco.configure.world.buttons.reset")));
-        this.buttonList.add(this.field_98128_v = new GuiButton(4, this.field_96287_o, this.func_96264_a(10), this.field_96286_n / 2 - 2, 20, I18n.getString("mco.configure.world.buttons.invite")));
-        this.buttonList.add(this.field_98127_w = new GuiButton(3, this.field_96287_o + this.field_96286_n / 2 + 2, this.func_96264_a(10), this.field_96286_n / 2 - 2, 20, I18n.getString("mco.configure.world.buttons.uninvite")));
-        this.buttonList.add(this.field_110381_z = new GuiButton(8, this.field_96287_o, this.func_96264_a(12), this.field_96286_n / 2 - 2, 20, I18n.getString("mco.configure.world.buttons.backup")));
-        this.buttonList.add(new GuiButton(10, this.field_96287_o + this.field_96286_n / 2 + 2, this.func_96264_a(12), this.field_96286_n / 2 - 2, 20, I18n.getString("gui.back")));
+        this.buttonList.add(this.field_98129_x = new GuiButton(7, this.field_96277_d + this.field_96286_n / 2 + 2, this.func_96264_a(12), this.field_96286_n / 2 - 2, 20, I18n.func_135053_a("mco.configure.world.buttons.subscription")));
+        this.buttonList.add(this.field_96278_t = new GuiButton(5, this.field_96277_d, this.func_96264_a(10), this.field_96286_n / 2 - 2, 20, I18n.func_135053_a("mco.configure.world.buttons.edit")));
+        this.buttonList.add(this.field_96276_u = new GuiButton(6, this.field_96277_d + this.field_96286_n / 2 + 2, this.func_96264_a(10), this.field_96286_n / 2 - 2, 20, I18n.func_135053_a("mco.configure.world.buttons.reset")));
+        this.buttonList.add(this.field_98128_v = new GuiButton(4, this.field_96287_o, this.func_96264_a(10), this.field_96286_n / 2 - 2, 20, I18n.func_135053_a("mco.configure.world.buttons.invite")));
+        this.buttonList.add(this.field_98127_w = new GuiButton(3, this.field_96287_o + this.field_96286_n / 2 + 2, this.func_96264_a(10), this.field_96286_n / 2 - 2, 20, I18n.func_135053_a("mco.configure.world.buttons.uninvite")));
+        this.buttonList.add(this.field_110381_z = new GuiButton(8, this.field_96287_o, this.func_96264_a(12), this.field_96286_n / 2 - 2, 20, I18n.func_135053_a("mco.configure.world.buttons.backup")));
+        this.buttonList.add(new GuiButton(10, this.field_96287_o + this.field_96286_n / 2 + 2, this.func_96264_a(12), this.field_96286_n / 2 - 2, 20, I18n.func_135053_a("gui.back")));
         this.field_96282_c = new SelectionListInvited(this);
         this.field_96278_t.enabled = !this.field_96280_b.field_98166_h;
         this.field_96276_u.enabled = !this.field_96280_b.field_98166_h;
@@ -106,8 +106,8 @@
             }
             else if (par1GuiButton.id == 1)
             {
-                String var2 = I18n.getString("mco.configure.world.close.question.line1");
-                String var3 = I18n.getString("mco.configure.world.close.question.line2");
+                String var2 = I18n.func_135053_a("mco.configure.world.close.question.line1");
+                String var3 = I18n.func_135053_a("mco.configure.world.close.question.line2");
                 this.mc.displayGuiScreen(new GuiScreenConfirmation(this, GuiScreenConfirmationType.Info, var2, var3, 1));
             }
             else if (par1GuiButton.id == 0)
@@ -139,7 +139,7 @@
 
     private void func_96268_g()
     {
-        McoClient var1 = new McoClient(this.mc.getSession());
+        McoClient var1 = new McoClient(this.mc.func_110432_I());
 
         try
         {
@@ -164,7 +164,7 @@
 
     private void func_96275_h()
     {
-        McoClient var1 = new McoClient(this.mc.getSession());
+        McoClient var1 = new McoClient(this.mc.func_110432_I());
 
         try
         {
@@ -192,7 +192,7 @@
         if (this.field_96284_p >= 0 && this.field_96284_p < this.field_96280_b.field_96402_f.size())
         {
             this.field_96283_q = (String)this.field_96280_b.field_96402_f.get(this.field_96284_p);
-            GuiYesNo var1 = new GuiYesNo(this, "Warning!", I18n.getString("mco.configure.world.uninvite.question") + " \'" + this.field_96283_q + "\'", 3);
+            GuiYesNo var1 = new GuiYesNo(this, "Warning!", I18n.func_135053_a("mco.configure.world.uninvite.question") + " \'" + this.field_96283_q + "\'", 3);
             this.mc.displayGuiScreen(var1);
         }
     }
@@ -203,7 +203,7 @@
         {
             if (par1)
             {
-                McoClient var3 = new McoClient(this.mc.getSession());
+                McoClient var3 = new McoClient(this.mc.func_110432_I());
 
                 try
                 {
@@ -256,14 +256,14 @@
     {
         this.drawDefaultBackground();
         this.field_96282_c.func_96612_a(par1, par2, par3);
-        this.drawCenteredString(this.fontRenderer, I18n.getString("mco.configure.world.title"), this.width / 2, 17, 16777215);
-        this.drawString(this.fontRenderer, I18n.getString("mco.configure.world.name"), this.field_96277_d, this.func_96264_a(1), 10526880);
+        this.drawCenteredString(this.fontRenderer, I18n.func_135053_a("mco.configure.world.title"), this.width / 2, 17, 16777215);
+        this.drawString(this.fontRenderer, I18n.func_135053_a("mco.configure.world.name"), this.field_96277_d, this.func_96264_a(1), 10526880);
         this.drawString(this.fontRenderer, this.field_96280_b.func_96398_b(), this.field_96277_d, this.func_96264_a(2), 16777215);
-        this.drawString(this.fontRenderer, I18n.getString("mco.configure.world.description"), this.field_96277_d, this.func_96264_a(4), 10526880);
+        this.drawString(this.fontRenderer, I18n.func_135053_a("mco.configure.world.description"), this.field_96277_d, this.func_96264_a(4), 10526880);
         this.drawString(this.fontRenderer, this.field_96280_b.func_96397_a(), this.field_96277_d, this.func_96264_a(5), 16777215);
-        this.drawString(this.fontRenderer, I18n.getString("mco.configure.world.status"), this.field_96277_d, this.func_96264_a(7), 10526880);
+        this.drawString(this.fontRenderer, I18n.func_135053_a("mco.configure.world.status"), this.field_96277_d, this.func_96264_a(7), 10526880);
         this.drawString(this.fontRenderer, this.func_104045_j(), this.field_96277_d, this.func_96264_a(8), 16777215);
-        this.drawString(this.fontRenderer, I18n.getString("mco.configure.world.invited"), this.field_96287_o, this.func_96264_a(1), 10526880);
+        this.drawString(this.fontRenderer, I18n.func_135053_a("mco.configure.world.invited"), this.field_96287_o, this.func_96264_a(1), 10526880);
         super.drawScreen(par1, par2, par3);
     }
 
@@ -280,7 +280,7 @@
         }
     }
 
-    static Minecraft getMinecraft(GuiScreenConfigureWorld par0GuiScreenConfigureWorld)
+    static Minecraft func_96265_a(GuiScreenConfigureWorld par0GuiScreenConfigureWorld)
     {
         return par0GuiScreenConfigureWorld.mc;
     }
