--- a/net/minecraft/src/GuiScreenBook.java
+++ b/net/minecraft/src/GuiScreenBook.java
@@ -7,7 +7,7 @@
 
 public class GuiScreenBook extends GuiScreen
 {
-    private static final ResourceLocation bookGuiTextures = new ResourceLocation("textures/gui/book.png");
+    private static final ResourceLocation field_110405_a = new ResourceLocation("textures/gui/book.png");
 
     /** The player editing the book */
     private final EntityPlayer editingPlayer;
@@ -85,14 +85,14 @@
 
         if (this.bookIsUnsigned)
         {
-            this.buttonList.add(this.buttonSign = new GuiButton(3, this.width / 2 - 100, 4 + this.bookImageHeight, 98, 20, I18n.getString("book.signButton")));
-            this.buttonList.add(this.buttonDone = new GuiButton(0, this.width / 2 + 2, 4 + this.bookImageHeight, 98, 20, I18n.getString("gui.done")));
-            this.buttonList.add(this.buttonFinalize = new GuiButton(5, this.width / 2 - 100, 4 + this.bookImageHeight, 98, 20, I18n.getString("book.finalizeButton")));
-            this.buttonList.add(this.buttonCancel = new GuiButton(4, this.width / 2 + 2, 4 + this.bookImageHeight, 98, 20, I18n.getString("gui.cancel")));
+            this.buttonList.add(this.buttonSign = new GuiButton(3, this.width / 2 - 100, 4 + this.bookImageHeight, 98, 20, I18n.func_135053_a("book.signButton")));
+            this.buttonList.add(this.buttonDone = new GuiButton(0, this.width / 2 + 2, 4 + this.bookImageHeight, 98, 20, I18n.func_135053_a("gui.done")));
+            this.buttonList.add(this.buttonFinalize = new GuiButton(5, this.width / 2 - 100, 4 + this.bookImageHeight, 98, 20, I18n.func_135053_a("book.finalizeButton")));
+            this.buttonList.add(this.buttonCancel = new GuiButton(4, this.width / 2 + 2, 4 + this.bookImageHeight, 98, 20, I18n.func_135053_a("gui.cancel")));
         }
         else
         {
-            this.buttonList.add(this.buttonDone = new GuiButton(0, this.width / 2 - 100, 4 + this.bookImageHeight, 200, 20, I18n.getString("gui.done")));
+            this.buttonList.add(this.buttonDone = new GuiButton(0, this.width / 2 - 100, 4 + this.bookImageHeight, 200, 20, I18n.func_135053_a("gui.done")));
         }
 
         int var1 = (this.width - this.bookImageWidth) / 2;
@@ -272,8 +272,8 @@
             case 22:
                 this.func_74160_b(GuiScreen.getClipboardString());
                 return;
-
             default:
+
                 switch (par2)
                 {
                     case 14:
@@ -285,13 +285,12 @@
                         }
 
                         return;
-
                     case 28:
                     case 156:
                         this.func_74160_b("\n");
                         return;
-
                     default:
+
                         if (ChatAllowedCharacters.isAllowedCharacter(par1))
                         {
                             this.func_74160_b(Character.toString(par1));
@@ -305,6 +304,7 @@
         switch (par2)
         {
             case 14:
+
                 if (!this.bookTitle.isEmpty())
                 {
                     this.bookTitle = this.bookTitle.substring(0, this.bookTitle.length() - 1);
@@ -312,9 +312,9 @@
                 }
 
                 return;
-
             case 28:
             case 156:
+
                 if (!this.bookTitle.isEmpty())
                 {
                     this.sendBookToServer(true);
@@ -322,8 +322,8 @@
                 }
 
                 return;
-
             default:
+
                 if (this.bookTitle.length() < 16 && ChatAllowedCharacters.isAllowedCharacter(par1))
                 {
                     this.bookTitle = this.bookTitle + Character.toString(par1);
@@ -374,7 +374,7 @@
     public void drawScreen(int par1, int par2, float par3)
     {
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-        this.mc.getTextureManager().bindTexture(bookGuiTextures);
+        this.mc.func_110434_K().func_110577_a(field_110405_a);
         int var4 = (this.width - this.bookImageWidth) / 2;
         byte var5 = 2;
         this.drawTexturedModalRect(var4, var5, 0, 0, this.bookImageWidth, this.bookImageHeight);
@@ -398,20 +398,20 @@
                 }
             }
 
-            var7 = I18n.getString("book.editTitle");
+            var7 = I18n.func_135053_a("book.editTitle");
             var8 = this.fontRenderer.getStringWidth(var7);
             this.fontRenderer.drawString(var7, var4 + 36 + (116 - var8) / 2, var5 + 16 + 16, 0);
             int var9 = this.fontRenderer.getStringWidth(var6);
             this.fontRenderer.drawString(var6, var4 + 36 + (116 - var9) / 2, var5 + 48, 0);
-            String var10 = String.format(I18n.getString("book.byAuthor"), new Object[] {this.editingPlayer.getCommandSenderName()});
+            String var10 = String.format(I18n.func_135053_a("book.byAuthor"), new Object[] {this.editingPlayer.getCommandSenderName()});
             int var11 = this.fontRenderer.getStringWidth(var10);
             this.fontRenderer.drawString(EnumChatFormatting.DARK_GRAY + var10, var4 + 36 + (116 - var11) / 2, var5 + 48 + 10, 0);
-            String var12 = I18n.getString("book.finalizeWarning");
+            String var12 = I18n.func_135053_a("book.finalizeWarning");
             this.fontRenderer.drawSplitString(var12, var4 + 36, var5 + 80, 116, 0);
         }
         else
         {
-            var6 = String.format(I18n.getString("book.pageIndicator"), new Object[] {Integer.valueOf(this.currPage + 1), Integer.valueOf(this.bookTotalPages)});
+            var6 = String.format(I18n.func_135053_a("book.pageIndicator"), new Object[] {Integer.valueOf(this.currPage + 1), Integer.valueOf(this.bookTotalPages)});
             var7 = "";
 
             if (this.bookPages != null && this.currPage >= 0 && this.currPage < this.bookPages.tagCount())
@@ -446,6 +446,6 @@
 
     static ResourceLocation func_110404_g()
     {
-        return bookGuiTextures;
+        return field_110405_a;
     }
 }
