--- a/net/minecraft/src/GuiScreenBackupDownloadThread.java
+++ b/net/minecraft/src/GuiScreenBackupDownloadThread.java
@@ -11,7 +11,7 @@
 
     public void run()
     {
-        McoClient var1 = new McoClient(GuiScreenBackup.func_110366_a(this.field_111250_a).getSession());
+        McoClient var1 = new McoClient(GuiScreenBackup.func_110366_a(this.field_111250_a).func_110432_I());
 
         try
         {
