--- a/net/minecraft/src/GuiSleepMP.java
+++ b/net/minecraft/src/GuiSleepMP.java
@@ -8,7 +8,7 @@
     public void initGui()
     {
         super.initGui();
-        this.buttonList.add(new GuiButton(1, this.width / 2 - 100, this.height - 40, I18n.getString("multiplayer.stopSleeping")));
+        this.buttonList.add(new GuiButton(1, this.width / 2 - 100, this.height - 40, I18n.func_135053_a("multiplayer.stopSleeping")));
     }
 
     /**
