--- a/net/minecraft/src/GuiScreenOnlineServers.java
+++ b/net/minecraft/src/GuiScreenOnlineServers.java
@@ -43,7 +43,7 @@
     {
         Keyboard.enableRepeatEvents(true);
         this.buttonList.clear();
-        field_96194_t.func_130129_a(this.mc.getSession());
+        field_96194_t.func_130129_a(this.mc.func_110432_I());
 
         if (!this.field_96193_u)
         {
@@ -60,18 +60,18 @@
 
     public void func_96178_g()
     {
-        this.buttonList.add(this.field_96196_r = new GuiButton(1, this.width / 2 - 154, this.height - 52, 100, 20, I18n.getString("mco.selectServer.play")));
-        this.buttonList.add(this.field_96198_p = new GuiButton(2, this.width / 2 - 48, this.height - 52, 100, 20, I18n.getString("mco.selectServer.create")));
-        this.buttonList.add(this.field_96190_o = new GuiButton(3, this.width / 2 + 58, this.height - 52, 100, 20, I18n.getString("mco.selectServer.configure")));
-        this.buttonList.add(this.field_96197_q = new GuiButtonLink(4, this.width / 2 - 154, this.height - 28, 154, 20, I18n.getString("mco.selectServer.moreinfo")));
-        this.buttonList.add(new GuiButton(0, this.width / 2 + 6, this.height - 28, 153, 20, I18n.getString("gui.cancel")));
+        this.buttonList.add(this.field_96196_r = new GuiButton(1, this.width / 2 - 154, this.height - 52, 100, 20, I18n.func_135053_a("mco.selectServer.play")));
+        this.buttonList.add(this.field_96198_p = new GuiButton(2, this.width / 2 - 48, this.height - 52, 100, 20, I18n.func_135053_a("mco.selectServer.create")));
+        this.buttonList.add(this.field_96190_o = new GuiButton(3, this.width / 2 + 58, this.height - 52, 100, 20, I18n.func_135053_a("mco.selectServer.configure")));
+        this.buttonList.add(this.field_96197_q = new GuiButtonLink(4, this.width / 2 - 154, this.height - 28, 154, 20, I18n.func_135053_a("mco.selectServer.moreinfo")));
+        this.buttonList.add(new GuiButton(0, this.width / 2 + 6, this.height - 28, 153, 20, I18n.func_135053_a("gui.cancel")));
         McoServer var1 = this.func_140030_b(this.field_96189_n);
         this.field_96196_r.enabled = var1 != null && var1.field_96404_d.equals("OPEN") && !var1.field_98166_h;
         this.field_96198_p.enabled = this.field_96199_x > 0;
 
-        if (var1 != null && !var1.field_96405_e.equals(this.mc.getSession().getUsername()))
+        if (var1 != null && !var1.field_96405_e.equals(this.mc.func_110432_I().func_111285_a()))
         {
-            this.field_96190_o.displayString = I18n.getString("mco.selectServer.leave");
+            this.field_96190_o.displayString = I18n.func_135053_a("mco.selectServer.leave");
         }
     }
 
@@ -170,7 +170,7 @@
 
         if (var1 != null)
         {
-            if (this.mc.getSession().getUsername().equals(var1.field_96405_e))
+            if (this.mc.func_110432_I().func_111285_a().equals(var1.field_96405_e))
             {
                 McoServer var2 = this.func_98086_a(var1.field_96408_a);
 
@@ -182,8 +182,8 @@
             }
             else
             {
-                String var4 = I18n.getString("mco.configure.world.leave.question.line1");
-                String var3 = I18n.getString("mco.configure.world.leave.question.line2");
+                String var4 = I18n.func_135053_a("mco.configure.world.leave.question.line1");
+                String var3 = I18n.func_135053_a("mco.configure.world.leave.question.line2");
                 this.mc.displayGuiScreen(new GuiScreenConfirmation(this, GuiScreenConfirmationType.Info, var4, var3, 3));
             }
         }
@@ -259,7 +259,7 @@
 
     private McoServer func_98086_a(long par1)
     {
-        McoClient var3 = new McoClient(this.mc.getSession());
+        McoClient var3 = new McoClient(this.mc.func_110432_I());
 
         try
         {
@@ -308,7 +308,7 @@
         this.field_96195_s = null;
         this.drawDefaultBackground();
         this.field_96186_b.drawScreen(par1, par2, par3);
-        this.drawCenteredString(this.fontRenderer, I18n.getString("mco.title"), this.width / 2, 20, 16777215);
+        this.drawCenteredString(this.fontRenderer, I18n.func_135053_a("mco.title"), this.width / 2, 20, 16777215);
         super.drawScreen(par1, par2, par3);
 
         if (this.field_96195_s != null)
@@ -337,7 +337,7 @@
     {
         int var3 = field_96194_t.func_130124_d();
         boolean var4 = this.func_130037_c(par1, par2);
-        this.mc.getTextureManager().bindTexture(field_130039_a);
+        this.mc.func_110434_K().func_110577_a(field_130039_a);
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
         GL11.glPushMatrix();
         this.drawTexturedModalRect(this.width / 2 + 58, 15, var4 ? 166 : 182, 22, 16, 16);
@@ -349,7 +349,7 @@
         {
             var5 = 198 + (Math.min(var3, 6) - 1) * 8;
             var6 = (int)(Math.max(0.0F, Math.max(MathHelper.sin((float)(10 + this.field_104044_y) * 0.57F), MathHelper.cos((float)this.field_104044_y * 0.35F))) * -6.0F);
-            this.mc.getTextureManager().bindTexture(field_130039_a);
+            this.mc.func_110434_K().func_110577_a(field_130039_a);
             GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
             GL11.glPushMatrix();
             this.drawTexturedModalRect(this.width / 2 + 58 + 4, 19 + var6, var5, 22, 8, 8);
@@ -360,7 +360,7 @@
         {
             var5 = par1 + 12;
             var6 = par2 - 12;
-            String var7 = I18n.getString("mco.invites.pending");
+            String var7 = I18n.func_135053_a("mco.invites.pending");
             int var8 = this.fontRenderer.getStringWidth(var7);
             this.drawGradientRect(var5 - 3, var6 - 3, var5 + var8 + 3, var6 + 8 + 3, -1073741824, -1073741824);
             this.fontRenderer.drawStringWithShadow(var7, var5, var6, -1);
@@ -391,7 +391,7 @@
 
     private void func_101008_c(int par1, int par2, int par3, int par4)
     {
-        this.mc.getTextureManager().bindTexture(field_130039_a);
+        this.mc.func_110434_K().func_110577_a(field_130039_a);
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
         GL11.glPushMatrix();
         GL11.glScalef(0.5F, 0.5F, 0.5F);
@@ -400,7 +400,7 @@
 
         if (par3 >= par1 && par3 <= par1 + 9 && par4 >= par2 && par4 <= par2 + 9)
         {
-            this.field_96195_s = I18n.getString("mco.selectServer.expired");
+            this.field_96195_s = I18n.func_135053_a("mco.selectServer.expired");
         }
     }
 
@@ -408,7 +408,7 @@
     {
         if (this.field_104044_y % 20 < 10)
         {
-            this.mc.getTextureManager().bindTexture(field_130039_a);
+            this.mc.func_110434_K().func_110577_a(field_130039_a);
             GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
             GL11.glPushMatrix();
             GL11.glScalef(0.5F, 0.5F, 0.5F);
@@ -420,22 +420,22 @@
         {
             if (par5 == 0)
             {
-                this.field_96195_s = I18n.getString("mco.selectServer.expires.soon");
+                this.field_96195_s = I18n.func_135053_a("mco.selectServer.expires.soon");
             }
             else if (par5 == 1)
             {
-                this.field_96195_s = I18n.getString("mco.selectServer.expires.day");
+                this.field_96195_s = I18n.func_135053_a("mco.selectServer.expires.day");
             }
             else
             {
-                this.field_96195_s = I18n.getStringParams("mco.selectServer.expires.days", new Object[] {Integer.valueOf(par5)});
+                this.field_96195_s = I18n.func_135052_a("mco.selectServer.expires.days", new Object[] {Integer.valueOf(par5)});
             }
         }
     }
 
     private void func_101006_d(int par1, int par2, int par3, int par4)
     {
-        this.mc.getTextureManager().bindTexture(field_130039_a);
+        this.mc.func_110434_K().func_110577_a(field_130039_a);
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
         GL11.glPushMatrix();
         GL11.glScalef(0.5F, 0.5F, 0.5F);
@@ -444,13 +444,13 @@
 
         if (par3 >= par1 && par3 <= par1 + 9 && par4 >= par2 && par4 <= par2 + 9)
         {
-            this.field_96195_s = I18n.getString("mco.selectServer.open");
+            this.field_96195_s = I18n.func_135053_a("mco.selectServer.open");
         }
     }
 
     private void func_101001_e(int par1, int par2, int par3, int par4)
     {
-        this.mc.getTextureManager().bindTexture(field_130039_a);
+        this.mc.func_110434_K().func_110577_a(field_130039_a);
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
         GL11.glPushMatrix();
         GL11.glScalef(0.5F, 0.5F, 0.5F);
@@ -459,7 +459,7 @@
 
         if (par3 >= par1 && par3 <= par1 + 9 && par4 >= par2 && par4 <= par2 + 9)
         {
-            this.field_96195_s = I18n.getString("mco.selectServer.closed");
+            this.field_96195_s = I18n.func_135053_a("mco.selectServer.closed");
         }
     }
 
@@ -482,7 +482,7 @@
             par1McoServer.field_96414_k = EnumChatFormatting.GRAY + "" + 0;
         }
 
-        par1McoServer.field_96415_h = 78;
+        par1McoServer.field_96415_h = 74;
         ServerAddress var2 = ServerAddress.func_78860_a(par1McoServer.field_96403_g);
         Socket var3 = null;
         DataInputStream var4 = null;
@@ -542,7 +542,7 @@
                 }
                 else
                 {
-                    par1McoServer.field_96415_h = 79;
+                    par1McoServer.field_96415_h = 75;
                     par1McoServer.field_96414_k = "" + EnumChatFormatting.DARK_GRAY + "???";
                 }
             }
@@ -574,7 +574,7 @@
                     par1McoServer.field_96414_k = "" + EnumChatFormatting.DARK_GRAY + "???";
                 }
 
-                par1McoServer.field_96415_h = 77;
+                par1McoServer.field_96415_h = 73;
             }
         }
         finally
