--- a/net/minecraft/src/GuiScreenConfirmation.java
+++ b/net/minecraft/src/GuiScreenConfirmation.java
@@ -17,8 +17,8 @@
         this.field_140045_e = par2GuiScreenConfirmationType;
         this.field_140049_p = par3Str;
         this.field_96288_n = par4Str;
-        this.field_140046_b = I18n.getString("gui.yes");
-        this.field_140047_c = I18n.getString("gui.no");
+        this.field_140046_b = I18n.func_135053_a("gui.yes");
+        this.field_140047_c = I18n.func_135053_a("gui.no");
     }
 
     /**
