--- a/net/minecraft/src/GuiScreen.java
+++ b/net/minecraft/src/GuiScreen.java
@@ -39,6 +39,9 @@
     private long field_85043_c = 0L;
     private int field_92018_d = 0;
 
+	public int mouseOffsetX = 0;
+	public int mouseOffsetY = 0;
+
     /**
      * Draws the screen and all the components in it.
      */
@@ -177,14 +180,38 @@
             this.handleKeyboardInput();
         }
     }
+    
+    public void mouseDown( int rawX, int rawY, int button )
+    {
+        int var1 = rawX * this.width / this.mc.displayFBWidth;
+        int var2 = this.height - rawY * this.height / this.mc.displayFBHeight - 1;
+        this.eventButton = button;
+        this.field_85043_c = Minecraft.getSystemTime();
+        mouseClicked(var1, var2, button);
+    }
+    
+    public void mouseUp( int rawX, int rawY, int button )
+    {
+        int var1 = rawX * this.width / this.mc.displayFBWidth;
+        int var2 = this.height - rawY * this.height / this.mc.displayFBHeight - 1;
+        mouseMovedOrUp(var1, var2, button);
+    }
+    
+    public void mouseDrag( int rawX, int rawY )
+    {
+        int var1 = rawX * this.width / this.mc.displayFBWidth;
+        int var2 = this.height - rawY * this.height / this.mc.displayFBHeight - 1;
+        long var3 = Minecraft.getSystemTime() - this.field_85043_c;
+        this.func_85041_a(var1, var2, this.eventButton, var3);
+    }
 
     /**
      * Handles mouse input.
      */
     public void handleMouseInput()
     {
-        int var1 = Mouse.getEventX() * this.width / this.mc.displayWidth;
-        int var2 = this.height - Mouse.getEventY() * this.height / this.mc.displayHeight - 1;
+        int var1 = ( Mouse.getEventX() + mouseOffsetX ) * this.width  / this.mc.displayFBWidth;
+        int var2 = this.height - ( Mouse.getEventY()  + mouseOffsetY) * this.height/ this.mc.displayFBHeight - 1;
 
         if (Mouse.getEventButtonState())
         {
@@ -261,7 +288,7 @@
     {
         if (this.mc.theWorld != null)
         {
-            this.drawGradientRect(0, 0, this.width, this.height, -1072689136, -804253680);
+            this.drawGradientRect(0, 0, this.width, this.height, 0, 0);
         }
         else
         {
