--- a/net/minecraft/src/GuiScreenCreateOnlineWorld.java
+++ b/net/minecraft/src/GuiScreenCreateOnlineWorld.java
@@ -42,8 +42,8 @@
     {
         Keyboard.enableRepeatEvents(true);
         this.buttonList.clear();
-        this.buttonList.add(new GuiButton(field_96253_d, this.width / 2 - 100, this.height / 4 + 120 + 17, 97, 20, I18n.getString("mco.create.world")));
-        this.buttonList.add(new GuiButton(field_96261_n, this.width / 2 + 5, this.height / 4 + 120 + 17, 95, 20, I18n.getString("gui.cancel")));
+        this.buttonList.add(new GuiButton(field_96253_d, this.width / 2 - 100, this.height / 4 + 120 + 17, 97, 20, I18n.func_135053_a("mco.create.world")));
+        this.buttonList.add(new GuiButton(field_96261_n, this.width / 2 + 5, this.height / 4 + 120 + 17, 95, 20, I18n.func_135053_a("gui.cancel")));
         this.field_96257_c = new GuiTextField(this.fontRenderer, this.width / 2 - 100, 65, 200, 20);
         this.field_96257_c.setFocused(true);
 
@@ -61,14 +61,14 @@
 
         if (this.field_110356_u == null)
         {
-            this.buttonList.add(new GuiButton(field_110357_r, this.width / 2 - 100, 147, 200, 20, I18n.getString("mco.template.default.name")));
+            this.buttonList.add(new GuiButton(field_110357_r, this.width / 2 - 100, 147, 200, 20, I18n.func_135053_a("mco.template.default.name")));
         }
         else
         {
             this.field_96255_b.setText("");
             this.field_96255_b.setEnabled(false);
             this.field_96255_b.setFocused(false);
-            this.buttonList.add(new GuiButton(field_110357_r, this.width / 2 - 100, 147, 200, 20, I18n.getString("mco.template.name") + ": " + this.field_110356_u.field_110732_b));
+            this.buttonList.add(new GuiButton(field_110357_r, this.width / 2 - 100, 147, 200, 20, I18n.func_135053_a("mco.template.name") + ": " + this.field_110356_u.field_110732_b));
         }
     }
 
@@ -155,9 +155,9 @@
     public void drawScreen(int par1, int par2, float par3)
     {
         this.drawDefaultBackground();
-        this.drawCenteredString(this.fontRenderer, I18n.getString("mco.selectServer.create"), this.width / 2, 11, 16777215);
-        this.drawString(this.fontRenderer, I18n.getString("mco.configure.world.name"), this.width / 2 - 100, 52, 10526880);
-        this.drawString(this.fontRenderer, I18n.getString("mco.create.world.seed"), this.width / 2 - 100, 98, 10526880);
+        this.drawCenteredString(this.fontRenderer, I18n.func_135053_a("mco.selectServer.create"), this.width / 2, 11, 16777215);
+        this.drawString(this.fontRenderer, I18n.func_135053_a("mco.configure.world.name"), this.width / 2 - 100, 52, 10526880);
+        this.drawString(this.fontRenderer, I18n.func_135053_a("mco.create.world.seed"), this.width / 2 - 100, 98, 10526880);
 
         if (this.field_96256_r)
         {
