--- a/net/minecraft/src/GuiScreenMcoWorldTemplateDownloadThread.java
+++ b/net/minecraft/src/GuiScreenMcoWorldTemplateDownloadThread.java
@@ -11,7 +11,7 @@
 
     public void run()
     {
-        McoClient var1 = new McoClient(GuiScreenMcoWorldTemplate.func_110382_a(this.field_111256_a).getSession());
+        McoClient var1 = new McoClient(GuiScreenMcoWorldTemplate.func_110382_a(this.field_111256_a).func_110432_I());
 
         try
         {
