--- a/net/minecraft/src/GuiStats.java
+++ b/net/minecraft/src/GuiStats.java
@@ -35,7 +35,7 @@
      */
     public void initGui()
     {
-        this.statsTitle = I18n.getString("gui.stats");
+        this.statsTitle = I18n.func_135053_a("gui.stats");
         this.slotGeneral = new GuiSlotStatsGeneral(this);
         this.slotGeneral.registerScrollButtons(1, 1);
         this.slotItem = new GuiSlotStatsItem(this);
@@ -51,12 +51,12 @@
      */
     public void addHeaderButtons()
     {
-        this.buttonList.add(new GuiButton(0, this.width / 2 + 4, this.height - 28, 150, 20, I18n.getString("gui.done")));
-        this.buttonList.add(new GuiButton(1, this.width / 2 - 154, this.height - 52, 100, 20, I18n.getString("stat.generalButton")));
+        this.buttonList.add(new GuiButton(0, this.width / 2 + 4, this.height - 28, 150, 20, I18n.func_135053_a("gui.done")));
+        this.buttonList.add(new GuiButton(1, this.width / 2 - 154, this.height - 52, 100, 20, I18n.func_135053_a("stat.generalButton")));
         GuiButton var1;
-        this.buttonList.add(var1 = new GuiButton(2, this.width / 2 - 46, this.height - 52, 100, 20, I18n.getString("stat.blocksButton")));
+        this.buttonList.add(var1 = new GuiButton(2, this.width / 2 - 46, this.height - 52, 100, 20, I18n.func_135053_a("stat.blocksButton")));
         GuiButton var2;
-        this.buttonList.add(var2 = new GuiButton(3, this.width / 2 + 62, this.height - 52, 100, 20, I18n.getString("stat.itemsButton")));
+        this.buttonList.add(var2 = new GuiButton(3, this.width / 2 + 62, this.height - 52, 100, 20, I18n.func_135053_a("stat.itemsButton")));
 
         if (this.slotBlock.getSize() == 0)
         {
@@ -117,7 +117,7 @@
         this.drawButtonBackground(par1 + 1, par2 + 1);
         GL11.glEnable(GL12.GL_RESCALE_NORMAL);
         RenderHelper.enableGUIStandardItemLighting();
-        renderItem.renderItemIntoGUI(this.fontRenderer, this.mc.getTextureManager(), new ItemStack(par3, 1, 0), par1 + 2, par2 + 2);
+        renderItem.renderItemIntoGUI(this.fontRenderer, this.mc.func_110434_K(), new ItemStack(par3, 1, 0), par1 + 2, par2 + 2);
         RenderHelper.disableStandardItemLighting();
         GL11.glDisable(GL12.GL_RESCALE_NORMAL);
     }
@@ -131,12 +131,12 @@
     }
 
     /**
-     * "Draws a sprite from assets/textures/gui/container/stats_icons.png"
+     * Draws a sprite from /gui/slot.png.
      */
     private void drawSprite(int par1, int par2, int par3, int par4)
     {
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-        this.mc.getTextureManager().bindTexture(statIcons);
+        this.mc.func_110434_K().func_110577_a(field_110323_l);
         float var5 = 0.0078125F;
         float var6 = 0.0078125F;
         boolean var7 = true;
