--- a/net/minecraft/src/GuiSlotServer.java
+++ b/net/minecraft/src/GuiSlotServer.java
@@ -31,7 +31,7 @@
             int var3 = GuiMultiplayer.getSelectedServer(this.parentGui);
             GuiMultiplayer.getAndSetSelectedServer(this.parentGui, par1);
             ServerData var4 = GuiMultiplayer.getInternetServerList(this.parentGui).countServers() > par1 ? GuiMultiplayer.getInternetServerList(this.parentGui).getServerData(par1) : null;
-            boolean var5 = GuiMultiplayer.getSelectedServer(this.parentGui) >= 0 && GuiMultiplayer.getSelectedServer(this.parentGui) < this.getSize() && (var4 == null || var4.field_82821_f == 78);
+            boolean var5 = GuiMultiplayer.getSelectedServer(this.parentGui) >= 0 && GuiMultiplayer.getSelectedServer(this.parentGui) < this.getSize() && (var4 == null || var4.field_82821_f == 74);
             boolean var6 = GuiMultiplayer.getSelectedServer(this.parentGui) < GuiMultiplayer.getInternetServerList(this.parentGui).countServers();
             GuiMultiplayer.getButtonSelect(this.parentGui).enabled = var5;
             GuiMultiplayer.getButtonEdit(this.parentGui).enabled = var6;
@@ -88,12 +88,12 @@
     private void func_77248_b(int par1, int par2, int par3, int par4, Tessellator par5Tessellator)
     {
         LanServer var6 = (LanServer)GuiMultiplayer.getListOfLanServers(this.parentGui).get(par1 - GuiMultiplayer.getInternetServerList(this.parentGui).countServers());
-        this.parentGui.drawString(this.parentGui.fontRenderer, I18n.getString("lanServer.title"), par2 + 2, par3 + 1, 16777215);
+        this.parentGui.drawString(this.parentGui.fontRenderer, I18n.func_135053_a("lanServer.title"), par2 + 2, par3 + 1, 16777215);
         this.parentGui.drawString(this.parentGui.fontRenderer, var6.getServerMotd(), par2 + 2, par3 + 12, 8421504);
 
         if (this.parentGui.mc.gameSettings.hideServerAddress)
         {
-            this.parentGui.drawString(this.parentGui.fontRenderer, I18n.getString("selectServer.hiddenAddress"), par2 + 2, par3 + 12 + 11, 3158064);
+            this.parentGui.drawString(this.parentGui.fontRenderer, I18n.func_135053_a("selectServer.hiddenAddress"), par2 + 2, par3 + 12 + 11, 3158064);
         }
         else
         {
@@ -103,7 +103,7 @@
 
     private void func_77249_c(int par1, int par2, int par3, int par4, Tessellator par5Tessellator)
     {
-        this.parentGui.drawCenteredString(this.parentGui.fontRenderer, I18n.getString("lanServer.scanning"), this.parentGui.width / 2, par3 + 1, 16777215);
+        this.parentGui.drawCenteredString(this.parentGui.fontRenderer, I18n.func_135053_a("lanServer.scanning"), this.parentGui.width / 2, par3 + 1, 16777215);
         String var6;
 
         switch (GuiMultiplayer.getTicksOpened(this.parentGui) / 3 % 4)
@@ -112,12 +112,10 @@
             default:
                 var6 = "O o o";
                 break;
-
             case 1:
             case 3:
                 var6 = "o O o";
                 break;
-
             case 2:
                 var6 = "o o O";
         }
@@ -142,8 +140,8 @@
             }
         }
 
-        boolean var7 = var6.field_82821_f > 78;
-        boolean var8 = var6.field_82821_f < 78;
+        boolean var7 = var6.field_82821_f > 74;
+        boolean var8 = var6.field_82821_f < 74;
         boolean var9 = var7 || var8;
         this.parentGui.drawString(this.parentGui.fontRenderer, var6.serverName, par2 + 2, par3 + 1, 16777215);
         this.parentGui.drawString(this.parentGui.fontRenderer, var6.serverMOTD, par2 + 2, par3 + 12, 8421504);
@@ -161,11 +159,11 @@
         }
         else
         {
-            this.parentGui.drawString(this.parentGui.fontRenderer, I18n.getString("selectServer.hiddenAddress"), par2 + 2, par3 + 12 + 11, 3158064);
+            this.parentGui.drawString(this.parentGui.fontRenderer, I18n.func_135053_a("selectServer.hiddenAddress"), par2 + 2, par3 + 12 + 11, 3158064);
         }
 
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-        this.parentGui.mc.getTextureManager().bindTexture(Gui.icons);
+        this.parentGui.mc.func_110434_K().func_110577_a(Gui.field_110324_m);
         byte var16 = 0;
         boolean var11 = false;
         String var12 = "";
