--- a/net/minecraft/src/GuiScreenTemporaryResourcePackSelect.java
+++ b/net/minecraft/src/GuiScreenTemporaryResourcePackSelect.java
@@ -23,9 +23,9 @@
      */
     public void initGui()
     {
-        this.buttonList.add(new GuiSmallButton(5, this.width / 2 - 154, this.height - 48, I18n.getString("resourcePack.openFolder")));
-        this.buttonList.add(new GuiSmallButton(6, this.width / 2 + 4, this.height - 48, I18n.getString("gui.done")));
-        this.field_110346_c = new GuiScreenTemporaryResourcePackSelectSelectionList(this, this.mc.getResourcePackRepository());
+        this.buttonList.add(new GuiSmallButton(5, this.width / 2 - 154, this.height - 48, I18n.func_135053_a("resourcePack.openFolder")));
+        this.buttonList.add(new GuiSmallButton(6, this.width / 2 + 4, this.height - 48, I18n.func_135053_a("gui.done")));
+        this.field_110346_c = new GuiScreenTemporaryResourcePackSelectSelectionList(this, this.mc.func_110438_M());
         this.field_110346_c.registerScrollButtons(7, 8);
     }
 
@@ -38,10 +38,10 @@
         {
             if (par1GuiButton.id == 5)
             {
-                File var2 = GuiScreenTemporaryResourcePackSelectSelectionList.func_110510_a(this.field_110346_c).getDirResourcepacks();
+                File var2 = GuiScreenTemporaryResourcePackSelectSelectionList.func_110510_a(this.field_110346_c).func_110612_e();
                 String var3 = var2.getAbsolutePath();
 
-                if (Util.getOSType() == EnumOS.MACOS)
+                if (Util.func_110647_a() == EnumOS.MACOS)
                 {
                     try
                     {
@@ -54,7 +54,7 @@
                         var9.printStackTrace();
                     }
                 }
-                else if (Util.getOSType() == EnumOS.WINDOWS)
+                else if (Util.func_110647_a() == EnumOS.WINDOWS)
                 {
                     String var4 = String.format("cmd.exe /C start \"Open file\" \"%s\"", new Object[] {var3});
 
@@ -126,12 +126,12 @@
 
         if (this.refreshTimer <= 0)
         {
-            GuiScreenTemporaryResourcePackSelectSelectionList.func_110510_a(this.field_110346_c).updateRepositoryEntriesAll();
+            GuiScreenTemporaryResourcePackSelectSelectionList.func_110510_a(this.field_110346_c).func_110611_a();
             this.refreshTimer = 20;
         }
 
-        this.drawCenteredString(this.fontRenderer, I18n.getString("resourcePack.title"), this.width / 2, 16, 16777215);
-        this.drawCenteredString(this.fontRenderer, I18n.getString("resourcePack.folderInfo"), this.width / 2 - 77, this.height - 26, 8421504);
+        this.drawCenteredString(this.fontRenderer, I18n.func_135053_a("resourcePack.title"), this.width / 2, 16, 16777215);
+        this.drawCenteredString(this.fontRenderer, I18n.func_135053_a("resourcePack.folderInfo"), this.width / 2 - 77, this.height - 26, 8421504);
         super.drawScreen(par1, par2, par3);
     }
 
