--- a/net/minecraft/src/GuiScreenMcoWorldTemplate.java
+++ b/net/minecraft/src/GuiScreenMcoWorldTemplate.java
@@ -33,8 +33,8 @@
 
     private void func_110385_g()
     {
-        this.buttonList.add(new GuiButton(0, this.width / 2 + 6, this.height - 52, 153, 20, I18n.getString("gui.cancel")));
-        this.buttonList.add(this.field_110400_p = new GuiButton(1, this.width / 2 - 154, this.height - 52, 153, 20, I18n.getString("mco.template.button.select")));
+        this.buttonList.add(new GuiButton(0, this.width / 2 + 6, this.height - 52, 153, 20, I18n.func_135053_a("gui.cancel")));
+        this.buttonList.add(this.field_110400_p = new GuiButton(1, this.width / 2 - 154, this.height - 52, 153, 20, I18n.func_135053_a("mco.template.button.select")));
     }
 
     /**
@@ -84,7 +84,7 @@
     {
         this.drawDefaultBackground();
         this.field_110396_d.drawScreen(par1, par2, par3);
-        this.drawCenteredString(this.fontRenderer, I18n.getString("mco.template.title"), this.width / 2, 20, 16777215);
+        this.drawCenteredString(this.fontRenderer, I18n.func_135053_a("mco.template.title"), this.width / 2, 20, 16777215);
         super.drawScreen(par1, par2, par3);
     }
 
