--- a/net/minecraft/src/GuiSlotLanguage.java
+++ b/net/minecraft/src/GuiSlotLanguage.java
@@ -19,13 +19,13 @@
         this.languageGui = par1GuiLanguage;
         this.field_77251_g = Lists.newArrayList();
         this.field_77253_h = Maps.newHashMap();
-        Iterator var2 = GuiLanguage.func_135011_a(par1GuiLanguage).getLanguages().iterator();
+        Iterator var2 = GuiLanguage.func_135011_a(par1GuiLanguage).func_135040_d().iterator();
 
         while (var2.hasNext())
         {
             Language var3 = (Language)var2.next();
-            this.field_77253_h.put(var3.getLanguageCode(), var3);
-            this.field_77251_g.add(var3.getLanguageCode());
+            this.field_77253_h.put(var3.func_135034_a(), var3);
+            this.field_77251_g.add(var3.func_135034_a());
         }
     }
 
@@ -43,12 +43,12 @@
     protected void elementClicked(int par1, boolean par2)
     {
         Language var3 = (Language)this.field_77253_h.get(this.field_77251_g.get(par1));
-        GuiLanguage.func_135011_a(this.languageGui).setCurrentLanguage(var3);
-        GuiLanguage.getGameSettings(this.languageGui).language = var3.getLanguageCode();
-        this.languageGui.mc.refreshResources();
-        this.languageGui.fontRenderer.setUnicodeFlag(GuiLanguage.func_135011_a(this.languageGui).isCurrentLocaleUnicode());
-        this.languageGui.fontRenderer.setBidiFlag(GuiLanguage.func_135011_a(this.languageGui).isCurrentLanguageBidirectional());
-        GuiLanguage.getDoneButton(this.languageGui).displayString = I18n.getString("gui.done");
+        GuiLanguage.func_135011_a(this.languageGui).func_135045_a(var3);
+        GuiLanguage.getGameSettings(this.languageGui).language = var3.func_135034_a();
+        this.languageGui.mc.func_110436_a();
+        this.languageGui.fontRenderer.setUnicodeFlag(GuiLanguage.func_135011_a(this.languageGui).func_135042_a());
+        this.languageGui.fontRenderer.setBidiFlag(GuiLanguage.func_135011_a(this.languageGui).func_135044_b());
+        GuiLanguage.getDoneButton(this.languageGui).displayString = I18n.func_135053_a("gui.done");
         GuiLanguage.getGameSettings(this.languageGui).saveOptions();
     }
 
@@ -57,7 +57,7 @@
      */
     protected boolean isSelected(int par1)
     {
-        return ((String)this.field_77251_g.get(par1)).equals(GuiLanguage.func_135011_a(this.languageGui).getCurrentLanguage().getLanguageCode());
+        return ((String)this.field_77251_g.get(par1)).equals(GuiLanguage.func_135011_a(this.languageGui).func_135041_c().func_135034_a());
     }
 
     /**
@@ -77,6 +77,6 @@
     {
         this.languageGui.fontRenderer.setBidiFlag(true);
         this.languageGui.drawCenteredString(this.languageGui.fontRenderer, ((Language)this.field_77253_h.get(this.field_77251_g.get(par1))).toString(), this.languageGui.width / 2, par3 + 1, 16777215);
-        this.languageGui.fontRenderer.setBidiFlag(GuiLanguage.func_135011_a(this.languageGui).getCurrentLanguage().isBidirectional());
+        this.languageGui.fontRenderer.setBidiFlag(GuiLanguage.func_135011_a(this.languageGui).func_135041_c().func_135035_b());
     }
 }
