--- a/net/minecraft/src/GuiScreen.java
+++ b/net/minecraft/src/GuiScreen.java
@@ -23,7 +23,7 @@
     public int height;
 
     /** A list of all the buttons in this container. */
-    protected List buttonList = new ArrayList();
+    public List<GuiButton> buttonList = new ArrayList<GuiButton>();
     public boolean allowUserInput;
 
     /** The FontRenderer used by GuiScreen */
@@ -35,6 +35,12 @@
     private long lastMouseEvent;
     private int field_92018_d;
 
+	public int mouseOffsetX = 0;
+	public int mouseOffsetY = 0;
+	
+	public int joystickX = 0;
+	public int joystickY = 0;
+
     /**
      * Draws the screen and all the components in it.
      */
@@ -42,7 +48,7 @@
     {
         for (int var4 = 0; var4 < this.buttonList.size(); ++var4)
         {
-            GuiButton var5 = (GuiButton)this.buttonList.get(var4);
+            GuiButton var5 = this.buttonList.get(var4);
             var5.drawButton(this.mc, par1, par2);
         }
     }
@@ -106,7 +112,7 @@
         {
             for (int var4 = 0; var4 < this.buttonList.size(); ++var4)
             {
-                GuiButton var5 = (GuiButton)this.buttonList.get(var4);
+                GuiButton var5 = this.buttonList.get(var4);
 
                 if (var5.mousePressed(this.mc, par1, par2))
                 {
@@ -171,22 +177,73 @@
             this.handleMouseInput();
         }
 
-        while (Keyboard.next())
+        while (Keyboard.isCreated() && Keyboard.next())
         {
             this.handleKeyboardInput();
         }
     }
+    
+    public void mouseDown( int rawX, int rawY, int button )
+    {
+        int var1 = rawX * this.width / this.mc.displayFBWidth;
+        int var2 = this.height - rawY * this.height / this.mc.displayFBHeight - 1;
+        this.eventButton = button;
+        this.lastMouseEvent = Minecraft.getSystemTime();
+        mouseClicked(var1, var2, button);
+    }
+    
+    public void mouseUp( int rawX, int rawY, int button )
+    {
+        int var1 = rawX * this.width / this.mc.displayFBWidth;
+        int var2 = this.height - rawY * this.height / this.mc.displayFBHeight - 1;
+        mouseMovedOrUp(var1, var2, button);
+    }
+    
+    public void mouseDrag( int rawX, int rawY )
+    {
+        int var1 = rawX * this.width / this.mc.displayFBWidth;
+        int var2 = this.height - rawY * this.height / this.mc.displayFBHeight - 1;
+        long var3 = Minecraft.getSystemTime() - this.lastMouseEvent;
+        this.mouseClickMove(var1, var2, this.eventButton, var3);
+    }
+
+    public void mouseGuiDown( int guiX, int guiY, int button )
+    {
+        this.eventButton = button;
+        this.lastMouseEvent = Minecraft.getSystemTime();
+        mouseClicked(guiX, guiY, button);
+    }
+    
+    public void mouseGuiUp( int guiX, int guiY, int button )
+    {
+        mouseMovedOrUp(guiX, guiY, button);
+    }
+    
+    public void mouseGuiDrag( int guiX, int guiY )
+    {
+        long var3 = Minecraft.getSystemTime() - this.lastMouseEvent;
+        this.mouseClickMove(guiX, guiY, this.eventButton, var3);
+    }
 
     /**
      * Handles mouse input.
      */
     public void handleMouseInput()
     {
-        int var1 = Mouse.getEventX() * this.width / this.mc.displayWidth;
-        int var2 = this.height - Mouse.getEventY() * this.height / this.mc.displayHeight - 1;
+        int var1 = ( Mouse.getEventX() + mouseOffsetX ) * this.width  / this.mc.displayFBWidth;
+        int var2 = this.height - ( Mouse.getEventY()  + mouseOffsetY) * this.height/ this.mc.displayFBHeight - 1;
         int var3 = Mouse.getEventButton();
 
-        if (Minecraft.isRunningOnMac && var3 == 0 && (Keyboard.isKeyDown(29) || Keyboard.isKeyDown(157)))
+        mouseOffsetX = 0;
+        mouseOffsetY = 0;
+        if( joystickX > 0 )
+        {
+        	var1 = joystickX;
+        	var2 = joystickY;
+        	joystickX = 0;
+        	joystickY = 0;
+        }
+        if (Minecraft.field_142025_a && var3 == 0 && (Keyboard.isKeyDown(29) || Keyboard.isKeyDown(157)))
         {
             var3 = 1;
         }
@@ -261,7 +318,7 @@
     {
         if (this.mc.theWorld != null)
         {
-            this.drawGradientRect(0, 0, this.width, this.height, -1072689136, -804253680);
+            this.drawGradientRect(0, 0, this.width, this.height, 0, 0);
         }
         else
         {
@@ -277,7 +334,7 @@
         GL11.glDisable(GL11.GL_LIGHTING);
         GL11.glDisable(GL11.GL_FOG);
         Tessellator var2 = Tessellator.instance;
-        this.mc.getTextureManager().bindTexture(optionsBackground);
+        this.mc.func_110434_K().func_110577_a(field_110325_k);
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
         float var3 = 32.0F;
         var2.startDrawingQuads();
@@ -301,7 +358,7 @@
 
     public static boolean isCtrlKeyDown()
     {
-        return Minecraft.isRunningOnMac ? Keyboard.isKeyDown(219) || Keyboard.isKeyDown(220) : Keyboard.isKeyDown(29) || Keyboard.isKeyDown(157);
+        return Minecraft.field_142025_a ? Keyboard.isKeyDown(219) || Keyboard.isKeyDown(220) : Keyboard.isKeyDown(29) || Keyboard.isKeyDown(157);
     }
 
     public static boolean isShiftKeyDown()
