--- a/net/minecraft/src/GuiScreenBackupSelectionList.java
+++ b/net/minecraft/src/GuiScreenBackupSelectionList.java
@@ -67,7 +67,7 @@
     private void func_111246_b(int par1, int par2, int par3, int par4, Tessellator par5Tessellator)
     {
         Backup var6 = (Backup)GuiScreenBackup.func_110370_e(this.field_111249_a).get(par1);
-        this.field_111249_a.drawString(GuiScreenBackup.func_130032_i(this.field_111249_a), "Backup (" + this.func_111248_a(Long.valueOf(MinecraftServer.getSystemTimeMillis() - var6.field_110725_b.getTime())) + ")", par2 + 2, par3 + 1, 16777215);
+        this.field_111249_a.drawString(GuiScreenBackup.func_130032_i(this.field_111249_a), "Backup (" + this.func_111248_a(Long.valueOf(MinecraftServer.func_130071_aq() - var6.field_110725_b.getTime())) + ")", par2 + 2, par3 + 1, 16777215);
         this.field_111249_a.drawString(GuiScreenBackup.func_130033_j(this.field_111249_a), this.func_111247_a(var6.field_110725_b), par2 + 2, par3 + 12, 7105644);
     }
 
