--- a/net/minecraft/src/GuiTextField.java
+++ b/net/minecraft/src/GuiTextField.java
@@ -314,24 +314,22 @@
                     this.setCursorPositionEnd();
                     this.setSelectionPos(0);
                     return true;
-
                 case 3:
                     GuiScreen.setClipboardString(this.getSelectedtext());
                     return true;
-
                 case 22:
                     this.writeText(GuiScreen.getClipboardString());
                     return true;
-
                 case 24:
                     GuiScreen.setClipboardString(this.getSelectedtext());
                     this.writeText("");
                     return true;
-
                 default:
+
                     switch (par2)
                     {
                         case 14:
+
                             if (GuiScreen.isCtrlKeyDown())
                             {
                                 this.deleteWords(-1);
@@ -342,8 +340,8 @@
                             }
 
                             return true;
-
                         case 199:
+
                             if (GuiScreen.isShiftKeyDown())
                             {
                                 this.setSelectionPos(0);
@@ -354,8 +352,8 @@
                             }
 
                             return true;
-
                         case 203:
+
                             if (GuiScreen.isShiftKeyDown())
                             {
                                 if (GuiScreen.isCtrlKeyDown())
@@ -377,8 +375,8 @@
                             }
 
                             return true;
-
                         case 205:
+
                             if (GuiScreen.isShiftKeyDown())
                             {
                                 if (GuiScreen.isCtrlKeyDown())
@@ -400,8 +398,8 @@
                             }
 
                             return true;
-
                         case 207:
+
                             if (GuiScreen.isShiftKeyDown())
                             {
                                 this.setSelectionPos(this.text.length());
@@ -412,8 +410,8 @@
                             }
 
                             return true;
-
                         case 211:
+
                             if (GuiScreen.isCtrlKeyDown())
                             {
                                 this.deleteWords(1);
@@ -424,8 +422,8 @@
                             }
 
                             return true;
-
                         default:
+
                             if (ChatAllowedCharacters.isAllowedCharacter(par1))
                             {
                                 this.writeText(Character.toString(par1));
