--- a/net/minecraft/src/GuiVideoSettings.java
+++ b/net/minecraft/src/GuiVideoSettings.java
@@ -32,7 +32,7 @@
      */
     public void initGui()
     {
-        this.screenTitle = I18n.getString("options.videoTitle");
+        this.screenTitle = I18n.func_135053_a("options.videoTitle");
         this.buttonList.clear();
         this.is64bit = false;
         String[] var1 = new String[] {"sun.arch.data.model", "com.ibm.vm.bitmode", "os.arch"};
@@ -89,7 +89,7 @@
         this.buttonList.add(new GuiSmallButton(111, var9, var16, "Animations..."));
         var9 = this.width / 2 - 155 + 160;
         this.buttonList.add(new GuiSmallButton(122, var9, var16, "Other..."));
-        this.buttonList.add(new GuiButton(200, this.width / 2 - 100, this.height / 6 + 168 + 11, I18n.getString("gui.done")));
+        this.buttonList.add(new GuiButton(200, this.width / 2 - 100, this.height / 6 + 168 + 11, I18n.func_135053_a("gui.done")));
     }
 
     /**
