--- a/net/minecraft/src/GuiWinGame.java
+++ b/net/minecraft/src/GuiWinGame.java
@@ -10,7 +10,7 @@
 
 public class GuiWinGame extends GuiScreen
 {
-    private static final ResourceLocation minecraftLogoTexture = new ResourceLocation("textures/gui/title/minecraft.png");
+    private static final ResourceLocation field_110362_a = new ResourceLocation("textures/gui/title/minecraft.png");
     private static final ResourceLocation field_110361_b = new ResourceLocation("textures/misc/vignette.png");
 
     /** Counts the number of screen updates. */
@@ -77,7 +77,7 @@
                 String var1 = "";
                 String var2 = "" + EnumChatFormatting.WHITE + EnumChatFormatting.OBFUSCATED + EnumChatFormatting.GREEN + EnumChatFormatting.AQUA;
                 short var3 = 274;
-                BufferedReader var4 = new BufferedReader(new InputStreamReader(this.mc.getResourceManager().getResource(new ResourceLocation("texts/end.txt")).getInputStream(), Charsets.UTF_8));
+                BufferedReader var4 = new BufferedReader(new InputStreamReader(this.mc.func_110442_L().func_110536_a(new ResourceLocation("texts/end.txt")).func_110527_b(), Charsets.UTF_8));
                 Random var5 = new Random(8124371L);
                 int var6;
 
@@ -86,7 +86,7 @@
                     String var7;
                     String var8;
 
-                    for (var1 = var1.replaceAll("PLAYERNAME", this.mc.getSession().getUsername()); var1.contains(var2); var1 = var7 + EnumChatFormatting.WHITE + EnumChatFormatting.OBFUSCATED + "XXXXXXXX".substring(0, var5.nextInt(4) + 3) + var8)
+                    for (var1 = var1.replaceAll("PLAYERNAME", this.mc.func_110432_I().func_111285_a()); var1.contains(var2); var1 = var7 + EnumChatFormatting.WHITE + EnumChatFormatting.OBFUSCATED + "XXXXXXXX".substring(0, var5.nextInt(4) + 3) + var8)
                     {
                         var6 = var1.indexOf(var2);
                         var7 = var1.substring(0, var6);
@@ -102,11 +102,11 @@
                     this.lines.add("");
                 }
 
-                var4 = new BufferedReader(new InputStreamReader(this.mc.getResourceManager().getResource(new ResourceLocation("texts/credits.txt")).getInputStream(), Charsets.UTF_8));
+                var4 = new BufferedReader(new InputStreamReader(this.mc.func_110442_L().func_110536_a(new ResourceLocation("texts/credits.txt")).func_110527_b(), Charsets.UTF_8));
 
                 while ((var1 = var4.readLine()) != null)
                 {
-                    var1 = var1.replaceAll("PLAYERNAME", this.mc.getSession().getUsername());
+                    var1 = var1.replaceAll("PLAYERNAME", this.mc.func_110432_I().func_111285_a());
                     var1 = var1.replaceAll("\t", "    ");
                     this.lines.addAll(this.mc.fontRenderer.listFormattedStringToWidth(var1, var3));
                     this.lines.add("");
@@ -124,7 +124,7 @@
     private void func_73986_b(int par1, int par2, float par3)
     {
         Tessellator var4 = Tessellator.instance;
-        this.mc.getTextureManager().bindTexture(Gui.optionsBackground);
+        this.mc.func_110434_K().func_110577_a(Gui.field_110325_k);
         var4.startDrawingQuads();
         var4.setColorRGBA_F(1.0F, 1.0F, 1.0F, 1.0F);
         int var5 = this.width;
@@ -168,7 +168,7 @@
         float var8 = -((float)this.updateCounter + par3) * this.field_73987_d;
         GL11.glPushMatrix();
         GL11.glTranslatef(0.0F, var8, 0.0F);
-        this.mc.getTextureManager().bindTexture(minecraftLogoTexture);
+        this.mc.func_110434_K().func_110577_a(field_110362_a);
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
         this.drawTexturedModalRect(var6, var7, 0, 0, 155, 44);
         this.drawTexturedModalRect(var6 + 155, var7, 0, 45, 155, 44);
@@ -207,7 +207,7 @@
         }
 
         GL11.glPopMatrix();
-        this.mc.getTextureManager().bindTexture(field_110361_b);
+        this.mc.func_110434_K().func_110577_a(field_110361_b);
         GL11.glEnable(GL11.GL_BLEND);
         GL11.glBlendFunc(GL11.GL_ZERO, GL11.GL_ONE_MINUS_SRC_COLOR);
         var4.startDrawingQuads();
