--- a/net/minecraft/src/GuiWorldSlot.java
+++ b/net/minecraft/src/GuiWorldSlot.java
@@ -84,12 +84,12 @@
 
             if (var6.isHardcoreModeEnabled())
             {
-                var9 = EnumChatFormatting.DARK_RED + I18n.getString("gameMode.hardcore") + EnumChatFormatting.RESET;
+                var9 = EnumChatFormatting.DARK_RED + I18n.func_135053_a("gameMode.hardcore") + EnumChatFormatting.RESET;
             }
 
             if (var6.getCheatsEnabled())
             {
-                var9 = var9 + ", " + I18n.getString("selectWorld.cheats");
+                var9 = var9 + ", " + I18n.func_135053_a("selectWorld.cheats");
             }
         }
 
