--- a/net/minecraft/src/GuiYesNo.java
+++ b/net/minecraft/src/GuiYesNo.java
@@ -28,8 +28,8 @@
         this.message1 = par2Str;
         this.message2 = par3Str;
         this.worldNumber = par4;
-        this.buttonText1 = I18n.getString("gui.yes");
-        this.buttonText2 = I18n.getString("gui.no");
+        this.buttonText1 = I18n.func_135053_a("gui.yes");
+        this.buttonText2 = I18n.func_135053_a("gui.no");
     }
 
     public GuiYesNo(GuiScreen par1GuiScreen, String par2Str, String par3Str, String par4Str, String par5Str, int par6)
