--- a/net/minecraft/src/IconFlipped.java
+++ b/net/minecraft/src/IconFlipped.java
@@ -14,19 +14,19 @@
     }
 
     /**
-     * Returns the width of the icon, in pixels.
+     * Returns the X position of this icon on its texture sheet, in pixels.
      */
-    public int getIconWidth()
+    public int getOriginX()
     {
-        return this.baseIcon.getIconWidth();
+        return this.baseIcon.getOriginX();
     }
 
     /**
-     * Returns the height of the icon, in pixels.
+     * Returns the Y position of this icon on its texture sheet, in pixels.
      */
-    public int getIconHeight()
+    public int getOriginY()
     {
-        return this.baseIcon.getIconHeight();
+        return this.baseIcon.getOriginY();
     }
 
     /**
