--- a/net/minecraft/src/ImageBufferDownload.java
+++ b/net/minecraft/src/ImageBufferDownload.java
@@ -21,23 +21,29 @@
         {
             this.imageWidth = 64;
             this.imageHeight = 32;
+            int var3 = par1BufferedImage.getWidth();
+            int var4 = par1BufferedImage.getHeight();
 
-            for (BufferedImage var2 = par1BufferedImage; this.imageWidth < var2.getWidth() || this.imageHeight < var2.getHeight(); this.imageHeight *= 2)
+            if (var3 != 64 || var4 != 32 && var4 != 64)
             {
-                this.imageWidth *= 2;
+                while (this.imageWidth < var3 || this.imageHeight < var4)
+                {
+                    this.imageWidth *= 2;
+                    this.imageHeight *= 2;
+                }
             }
 
-            BufferedImage var3 = new BufferedImage(this.imageWidth, this.imageHeight, 2);
-            Graphics var4 = var3.getGraphics();
-            var4.drawImage(par1BufferedImage, 0, 0, (ImageObserver)null);
-            var4.dispose();
-            this.imageData = ((DataBufferInt)var3.getRaster().getDataBuffer()).getData();
-            int var5 = this.imageWidth;
-            int var6 = this.imageHeight;
-            this.setAreaOpaque(0, 0, var5 / 2, var6 / 2);
-            this.setAreaTransparent(var5 / 2, 0, var5, var6);
-            this.setAreaOpaque(0, var6 / 2, var5, var6);
-            return var3;
+            BufferedImage var5 = new BufferedImage(this.imageWidth, this.imageHeight, 2);
+            Graphics var6 = var5.getGraphics();
+            var6.drawImage(par1BufferedImage, 0, 0, (ImageObserver)null);
+            var6.dispose();
+            this.imageData = ((DataBufferInt)var5.getRaster().getDataBuffer()).getData();
+            int var7 = this.imageWidth;
+            int var8 = this.imageHeight;
+            this.setAreaOpaque(0, 0, var7 / 2, var8 / 2);
+            this.setAreaTransparent(var7 / 2, 0, var7, var8);
+            this.setAreaOpaque(0, var8 / 2, var7, var8);
+            return var5;
         }
     }
 
