--- a/net/minecraft/src/IntegratedServer.java
+++ b/net/minecraft/src/IntegratedServer.java
@@ -21,7 +21,7 @@
     {
         super(new File(par1Minecraft.mcDataDir, "saves"));
         this.serverLogAgent = new LogAgent("Minecraft-Server", " [SERVER]", (new File(par1Minecraft.mcDataDir, "output-server.log")).getAbsolutePath());
-        this.setServerOwner(par1Minecraft.getSession().getUsername());
+        this.setServerOwner(par1Minecraft.func_110432_I().func_111285_a());
         this.setFolderName(par2Str);
         this.setWorldName(par3Str);
         this.setDemo(par1Minecraft.isDemo());
@@ -29,9 +29,8 @@
         this.setBuildLimit(256);
         this.setConfigurationManager(new IntegratedPlayerList(this));
         this.mc = par1Minecraft;
-        this.serverProxy = par1Minecraft.getProxy();
+        this.field_110456_c = par1Minecraft.func_110437_J();
         this.theWorldSettings = par4WorldSettings;
-        Reflector.callVoid(Reflector.ModLoader_registerServer, new Object[] {this});
 
         try
         {
@@ -123,7 +122,7 @@
      */
     protected boolean startServer() throws IOException
     {
-        this.serverLogAgent.logInfo("Starting integrated minecraft server version 1.6.4");
+        this.serverLogAgent.logInfo("Starting integrated minecraft server version 1.6.2");
         this.setOnlineMode(false);
         this.setCanSpawnAnimals(true);
         this.setCanSpawnNPCs(true);
@@ -240,8 +239,8 @@
     public CrashReport addServerInfoToCrashReport(CrashReport par1CrashReport)
     {
         par1CrashReport = super.addServerInfoToCrashReport(par1CrashReport);
-        par1CrashReport.getCategory().addCrashSectionCallable("Type", new CallableType3(this));
-        par1CrashReport.getCategory().addCrashSectionCallable("Is Modded", new CallableIsModded(this));
+        par1CrashReport.func_85056_g().addCrashSectionCallable("Type", new CallableType3(this));
+        par1CrashReport.func_85056_g().addCrashSectionCallable("Is Modded", new CallableIsModded(this));
         return par1CrashReport;
     }
 
