--- a/net/minecraft/src/InventoryEffectRenderer.java
+++ b/net/minecraft/src/InventoryEffectRenderer.java
@@ -66,7 +66,7 @@
                 PotionEffect var7 = (PotionEffect)var6.next();
                 Potion var8 = Potion.potionTypes[var7.getPotionID()];
                 GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-                this.mc.getTextureManager().bindTexture(field_110408_a);
+                this.mc.func_110434_K().func_110577_a(field_110408_a);
                 this.drawTexturedModalRect(var1, var2, 0, 166, 140, 32);
 
                 if (var8.hasStatusIcon())
@@ -75,7 +75,7 @@
                     this.drawTexturedModalRect(var1 + 6, var2 + 7, 0 + var9 % 8 * 18, 198 + var9 / 8 * 18, 18, 18);
                 }
 
-                String var11 = I18n.getString(var8.getName());
+                String var11 = I18n.func_135053_a(var8.getName());
 
                 if (var7.getAmplifier() == 1)
                 {
