--- a/net/minecraft/src/ItemFishingRod.java
+++ b/net/minecraft/src/ItemFishingRod.java
@@ -57,8 +57,8 @@
 
     public void registerIcons(IconRegister par1IconRegister)
     {
-        this.itemIcon = par1IconRegister.registerIcon(this.getIconString() + "_uncast");
-        this.theIcon = par1IconRegister.registerIcon(this.getIconString() + "_cast");
+        this.itemIcon = par1IconRegister.registerIcon(this.func_111208_A() + "_uncast");
+        this.theIcon = par1IconRegister.registerIcon(this.func_111208_A() + "_cast");
     }
 
     public Icon func_94597_g()
