--- a/net/minecraft/src/ItemStack.java
+++ b/net/minecraft/src/ItemStack.java
@@ -377,7 +377,7 @@
 
     public boolean func_111282_a(EntityPlayer par1EntityPlayer, EntityLivingBase par2EntityLivingBase)
     {
-        return Item.itemsList[this.itemID].itemInteractionForEntity(this, par1EntityPlayer, par2EntityLivingBase);
+        return Item.itemsList[this.itemID].func_111207_a(this, par1EntityPlayer, par2EntityLivingBase);
     }
 
     /**
@@ -425,7 +425,7 @@
         return this.itemID == par1ItemStack.itemID && this.itemDamage == par1ItemStack.itemDamage;
     }
 
-    public String getUnlocalizedName()
+    public String getItemName()
     {
         return Item.itemsList[this.itemID].getUnlocalizedName(this);
     }
@@ -639,13 +639,13 @@
 
             if (this.stackTagCompound.hasKey("display"))
             {
-                NBTTagCompound var17 = this.stackTagCompound.getCompoundTag("display");
+                NBTTagCompound var16 = this.stackTagCompound.getCompoundTag("display");
 
-                if (var17.hasKey("color"))
+                if (var16.hasKey("color"))
                 {
                     if (par2)
                     {
-                        var3.add("Color: #" + Integer.toHexString(var17.getInteger("color")).toUpperCase());
+                        var3.add("Color: #" + Integer.toHexString(var16.getInteger("color")).toUpperCase());
                     }
                     else
                     {
@@ -653,9 +653,9 @@
                     }
                 }
 
-                if (var17.hasKey("Lore"))
+                if (var16.hasKey("Lore"))
                 {
-                    NBTTagList var19 = var17.getTagList("Lore");
+                    NBTTagList var19 = var16.getTagList("Lore");
 
                     if (var19.tagCount() > 0)
                     {
@@ -668,37 +668,37 @@
             }
         }
 
-        Multimap var16 = this.getAttributeModifiers();
+        Multimap var17 = this.func_111283_C();
 
-        if (!var16.isEmpty())
+        if (!var17.isEmpty())
         {
             var3.add("");
-            Iterator var15 = var16.entries().iterator();
+            Iterator var15 = var17.entries().iterator();
 
             while (var15.hasNext())
             {
                 Entry var18 = (Entry)var15.next();
                 AttributeModifier var21 = (AttributeModifier)var18.getValue();
-                double var10 = var21.getAmount();
+                double var10 = var21.func_111164_d();
                 double var12;
 
-                if (var21.getOperation() != 1 && var21.getOperation() != 2)
+                if (var21.func_111169_c() != 1 && var21.func_111169_c() != 2)
                 {
-                    var12 = var21.getAmount();
+                    var12 = var21.func_111164_d();
                 }
                 else
                 {
-                    var12 = var21.getAmount() * 100.0D;
+                    var12 = var21.func_111164_d() * 100.0D;
                 }
 
                 if (var10 > 0.0D)
                 {
-                    var3.add(EnumChatFormatting.BLUE + StatCollector.translateToLocalFormatted("attribute.modifier.plus." + var21.getOperation(), new Object[] {field_111284_a.format(var12), StatCollector.translateToLocal("attribute.name." + (String)var18.getKey())}));
+                    var3.add(EnumChatFormatting.BLUE + StatCollector.translateToLocalFormatted("attribute.modifier.plus." + var21.func_111169_c(), new Object[] {field_111284_a.format(var12), StatCollector.translateToLocal("attribute.name." + (String)var18.getKey())}));
                 }
                 else if (var10 < 0.0D)
                 {
                     var12 *= -1.0D;
-                    var3.add(EnumChatFormatting.RED + StatCollector.translateToLocalFormatted("attribute.modifier.take." + var21.getOperation(), new Object[] {field_111284_a.format(var12), StatCollector.translateToLocal("attribute.name." + (String)var18.getKey())}));
+                    var3.add(EnumChatFormatting.RED + StatCollector.translateToLocalFormatted("attribute.modifier.take." + var21.func_111169_c(), new Object[] {field_111284_a.format(var12), StatCollector.translateToLocal("attribute.name." + (String)var18.getKey())}));
                 }
             }
         }
@@ -819,11 +819,7 @@
         this.stackTagCompound.setInteger("RepairCost", par1);
     }
 
-    /**
-     * Gets the attribute modifiers for this ItemStack.\nWill check for an NBT tag list containing modifiers for the
-     * stack.
-     */
-    public Multimap getAttributeModifiers()
+    public Multimap func_111283_C()
     {
         Object var1;
 
@@ -837,7 +833,7 @@
                 NBTTagCompound var4 = (NBTTagCompound)var2.tagAt(var3);
                 AttributeModifier var5 = SharedMonsterAttributes.func_111259_a(var4);
 
-                if (var5.getID().getLeastSignificantBits() != 0L && var5.getID().getMostSignificantBits() != 0L)
+                if (var5.func_111167_a().getLeastSignificantBits() != 0L && var5.func_111167_a().getMostSignificantBits() != 0L)
                 {
                     ((Multimap)var1).put(var4.getString("AttributeName"), var5);
                 }
@@ -845,7 +841,7 @@
         }
         else
         {
-            var1 = this.getItem().getItemAttributeModifiers();
+            var1 = this.getItem().func_111205_h();
         }
 
         return (Multimap)var1;
