--- a/net/minecraft/src/ItemFireworkCharge.java
+++ b/net/minecraft/src/ItemFireworkCharge.java
@@ -216,6 +216,6 @@
     public void registerIcons(IconRegister par1IconRegister)
     {
         super.registerIcons(par1IconRegister);
-        this.theIcon = par1IconRegister.registerIcon(this.getIconString() + "_overlay");
+        this.theIcon = par1IconRegister.registerIcon(this.func_111208_A() + "_overlay");
     }
 }
