--- a/net/minecraft/src/Material.java
+++ b/net/minecraft/src/Material.java
@@ -21,7 +21,7 @@
     public static final Material fire = (new MaterialTransparent(MapColor.airColor)).setNoPushMobility();
     public static final Material sand = new Material(MapColor.sandColor);
     public static final Material circuits = (new MaterialLogic(MapColor.airColor)).setNoPushMobility();
-    public static final Material materialCarpet = (new MaterialLogic(MapColor.clothColor)).setBurning();
+    public static final Material field_111018_r = (new MaterialLogic(MapColor.clothColor)).setBurning();
     public static final Material glass = (new Material(MapColor.airColor)).setTranslucent().setAdventureModeExempt();
     public static final Material redstoneLight = (new Material(MapColor.airColor)).setAdventureModeExempt();
     public static final Material tnt = (new Material(MapColor.tntColor)).setBurning().setTranslucent();
