--- a/net/minecraft/src/McoClient.java
+++ b/net/minecraft/src/McoClient.java
@@ -15,8 +15,8 @@
 
     public McoClient(Session par1Session)
     {
-        this.field_96390_a = par1Session.getSessionID();
-        this.field_100007_c = par1Session.getUsername();
+        this.field_96390_a = par1Session.func_111286_b();
+        this.field_100007_c = par1Session.func_111285_a();
     }
 
     public ValueObjectList func_96382_a() throws ExceptionMcoService, IOException
@@ -136,10 +136,6 @@
         {
             var7.append("?motd=").append(this.func_96380_a(par4Str));
         }
-        else
-        {
-            var7.append("?motd=");
-        }
 
         var7.append("&difficulty=").append(par5).append("&gameMode=").append(par6);
         this.func_96377_a(Request.func_96363_c(var7.toString(), ""));
@@ -237,7 +233,7 @@
     {
         par1Request.func_100006_a("sid", this.field_96390_a);
         par1Request.func_100006_a("user", this.field_100007_c);
-        par1Request.func_100006_a("version", "1.6.4");
+        par1Request.func_100006_a("version", "1.6.2");
 
         try
         {
