--- a/net/minecraft/src/MobSpawnerBaseLogic.java
+++ b/net/minecraft/src/MobSpawnerBaseLogic.java
@@ -149,7 +149,7 @@
         if (this.getRandomMinecart() != null)
         {
             NBTTagCompound var2 = new NBTTagCompound();
-            par1Entity.writeToNBTOptional(var2);
+            par1Entity.addEntityID(var2);
             Iterator var3 = this.getRandomMinecart().field_98222_b.getTags().iterator();
 
             while (var3.hasNext())
@@ -175,7 +175,7 @@
                 if (var5 != null)
                 {
                     NBTTagCompound var6 = new NBTTagCompound();
-                    var5.writeToNBTOptional(var6);
+                    var5.addEntityID(var6);
                     Iterator var7 = var10.getTags().iterator();
 
                     while (var7.hasNext())
@@ -200,7 +200,7 @@
         }
         else if (par1Entity instanceof EntityLivingBase && par1Entity.worldObj != null)
         {
-            ((EntityLiving)par1Entity).onSpawnWithEgg((EntityLivingData)null);
+            ((EntityLiving)par1Entity).func_110161_a((EntityLivingData)null);
             this.getSpawnerWorld().spawnEntityInWorld(par1Entity);
         }
 
