--- a/net/minecraft/src/ModelMagmaCube.java
+++ b/net/minecraft/src/ModelMagmaCube.java
@@ -38,7 +38,7 @@
     public void setLivingAnimations(EntityLivingBase par1EntityLivingBase, float par2, float par3, float par4)
     {
         EntityMagmaCube var5 = (EntityMagmaCube)par1EntityLivingBase;
-        float var6 = var5.prevSquishFactor + (var5.squishFactor - var5.prevSquishFactor) * par4;
+        float var6 = var5.field_70812_c + (var5.field_70811_b - var5.field_70812_c) * par4;
 
         if (var6 < 0.0F)
         {
