--- a/net/minecraft/src/MovementInputFromOptions.java
+++ b/net/minecraft/src/MovementInputFromOptions.java
@@ -11,8 +11,8 @@
 
     public void updatePlayerMoveState()
     {
-        this.moveStrafe = 0.0F;
-        this.moveForward = 0.0F;
+        this.moveStrafe = baseMoveStrafe;
+        this.moveForward = baseMoveForward;
 
         if (this.gameSettings.keyBindForward.pressed)
         {
