--- a/net/minecraft/src/NetClientHandler.java
+++ b/net/minecraft/src/NetClientHandler.java
@@ -75,7 +75,7 @@
     {
         this.mc = par1Minecraft;
         this.netManager = new MemoryConnection(par1Minecraft.getLogAgent(), this);
-        par2IntegratedServer.getServerListeningThread().func_71754_a((MemoryConnection)this.netManager, par1Minecraft.getSession().getUsername());
+        par2IntegratedServer.getServerListeningThread().func_71754_a((MemoryConnection)this.netManager, par1Minecraft.func_110432_I().func_111285_a());
     }
 
     /**
@@ -117,7 +117,7 @@
         if (!"-".equals(var2))
         {
             String var5 = (new BigInteger(CryptManager.getServerIdHash(var2, var3, var4))).toString(16);
-            String var6 = this.sendSessionRequest(this.mc.getSession().getUsername(), this.mc.getSession().getSessionID(), var5);
+            String var6 = this.sendSessionRequest(this.mc.func_110432_I().func_111285_a(), this.mc.func_110432_I().func_111286_b(), var5);
 
             if (!"ok".equalsIgnoreCase(var6))
             {
@@ -137,7 +137,7 @@
         try
         {
             URL var4 = new URL("http://session.minecraft.net/game/joinserver.jsp?user=" + urlEncode(par1Str) + "&sessionId=" + urlEncode(par2Str) + "&serverId=" + urlEncode(par3Str));
-            InputStream var5 = var4.openConnection(this.mc.getProxy()).getInputStream();
+            InputStream var5 = var4.openConnection(this.mc.func_110437_J()).getInputStream();
             BufferedReader var6 = new BufferedReader(new InputStreamReader(var5));
             String var7 = var6.readLine();
             var6.close();
@@ -688,7 +688,7 @@
 
     public void handleChat(Packet3Chat par1Packet3Chat)
     {
-        this.mc.ingameGUI.getChatGUI().printChatMessage(ChatMessageComponent.createFromJson(par1Packet3Chat.message).toStringWithFormatting(true));
+        this.mc.ingameGUI.getChatGUI().printChatMessage(ChatMessageComponent.func_111078_c(par1Packet3Chat.message).func_111068_a(true));
     }
 
     public void handleAnimation(Packet18Animation par1Packet18Animation)
@@ -809,14 +809,14 @@
      */
     public void handleAttachEntity(Packet39AttachEntity par1Packet39AttachEntity)
     {
-        Object var2 = this.getEntityByID(par1Packet39AttachEntity.ridingEntityId);
+        Object var2 = this.getEntityByID(par1Packet39AttachEntity.field_111006_b);
         Entity var3 = this.getEntityByID(par1Packet39AttachEntity.vehicleEntityId);
 
-        if (par1Packet39AttachEntity.attachState == 0)
+        if (par1Packet39AttachEntity.field_111007_a == 0)
         {
             boolean var4 = false;
 
-            if (par1Packet39AttachEntity.ridingEntityId == this.mc.thePlayer.entityId)
+            if (par1Packet39AttachEntity.field_111006_b == this.mc.thePlayer.entityId)
             {
                 var2 = this.mc.thePlayer;
 
@@ -842,18 +842,18 @@
             if (var4)
             {
                 GameSettings var5 = this.mc.gameSettings;
-                this.mc.ingameGUI.func_110326_a(I18n.getStringParams("mount.onboard", new Object[] {GameSettings.getKeyDisplayString(var5.keyBindSneak.keyCode)}), false);
+                this.mc.ingameGUI.func_110326_a(I18n.func_135052_a("mount.onboard", new Object[] {GameSettings.getKeyDisplayString(var5.keyBindSneak.keyCode)}), false);
             }
         }
-        else if (par1Packet39AttachEntity.attachState == 1 && var2 != null && var2 instanceof EntityLiving)
+        else if (par1Packet39AttachEntity.field_111007_a == 1 && var2 != null && var2 instanceof EntityLiving)
         {
             if (var3 != null)
             {
-                ((EntityLiving)var2).setLeashedToEntity(var3, false);
+                ((EntityLiving)var2).func_110162_b(var3, false);
             }
             else
             {
-                ((EntityLiving)var2).clearLeashed(false, false);
+                ((EntityLiving)var2).func_110160_i(false, false);
             }
         }
     }
@@ -881,7 +881,7 @@
      */
     public void handleUpdateHealth(Packet8UpdateHealth par1Packet8UpdateHealth)
     {
-        this.mc.thePlayer.setPlayerSPHealth(par1Packet8UpdateHealth.healthMP);
+        this.mc.thePlayer.setHealth(par1Packet8UpdateHealth.healthMP);
         this.mc.thePlayer.getFoodStats().setFoodLevel(par1Packet8UpdateHealth.food);
         this.mc.thePlayer.getFoodStats().setFoodSaturationLevel(par1Packet8UpdateHealth.foodSaturation);
     }
@@ -935,12 +935,10 @@
                 var2.displayGUIChest(new InventoryBasic(par1Packet100OpenWindow.windowTitle, par1Packet100OpenWindow.useProvidedWindowTitle, par1Packet100OpenWindow.slotsCount));
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 1:
                 var2.displayGUIWorkbench(MathHelper.floor_double(var2.posX), MathHelper.floor_double(var2.posY), MathHelper.floor_double(var2.posZ));
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 2:
                 TileEntityFurnace var4 = new TileEntityFurnace();
 
@@ -952,7 +950,6 @@
                 var2.displayGUIFurnace(var4);
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 3:
                 TileEntityDispenser var7 = new TileEntityDispenser();
 
@@ -964,12 +961,10 @@
                 var2.displayGUIDispenser(var7);
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 4:
                 var2.displayGUIEnchantment(MathHelper.floor_double(var2.posX), MathHelper.floor_double(var2.posY), MathHelper.floor_double(var2.posZ), par1Packet100OpenWindow.useProvidedWindowTitle ? par1Packet100OpenWindow.windowTitle : null);
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 5:
                 TileEntityBrewingStand var5 = new TileEntityBrewingStand();
 
@@ -981,12 +976,10 @@
                 var2.displayGUIBrewingStand(var5);
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 6:
                 var2.displayGUIMerchant(new NpcMerchant(var2), par1Packet100OpenWindow.useProvidedWindowTitle ? par1Packet100OpenWindow.windowTitle : null);
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 7:
                 TileEntityBeacon var8 = new TileEntityBeacon();
                 var2.displayGUIBeacon(var8);
@@ -998,12 +991,10 @@
 
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 8:
                 var2.displayGUIAnvil(MathHelper.floor_double(var2.posX), MathHelper.floor_double(var2.posY), MathHelper.floor_double(var2.posZ));
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 9:
                 TileEntityHopper var3 = new TileEntityHopper();
 
@@ -1015,7 +1006,6 @@
                 var2.displayGUIHopper(var3);
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 10:
                 TileEntityDropper var6 = new TileEntityDropper();
 
@@ -1027,13 +1017,12 @@
                 var2.displayGUIDispenser(var6);
                 var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 break;
-
             case 11:
                 Entity var9 = this.getEntityByID(par1Packet100OpenWindow.field_111008_f);
 
                 if (var9 != null && var9 instanceof EntityHorse)
                 {
-                    var2.displayGUIHorse((EntityHorse)var9, new AnimalChest(par1Packet100OpenWindow.windowTitle, par1Packet100OpenWindow.useProvidedWindowTitle, par1Packet100OpenWindow.slotsCount));
+                    var2.func_110298_a((EntityHorse)var9, new AnimalChest(par1Packet100OpenWindow.windowTitle, par1Packet100OpenWindow.useProvidedWindowTitle, par1Packet100OpenWindow.slotsCount));
                     var2.openContainer.windowId = par1Packet100OpenWindow.windowId;
                 }
         }
@@ -1159,7 +1148,7 @@
 
         if (!var2 && this.mc.thePlayer != null)
         {
-            this.mc.thePlayer.sendChatToPlayer(ChatMessageComponent.createFromText("Unable to locate sign at " + par1Packet130UpdateSign.xPosition + ", " + par1Packet130UpdateSign.yPosition + ", " + par1Packet130UpdateSign.zPosition));
+            this.mc.thePlayer.sendChatToPlayer(ChatMessageComponent.func_111066_d("Unable to locate sign at " + par1Packet130UpdateSign.xPosition + ", " + par1Packet130UpdateSign.yPosition + ", " + par1Packet130UpdateSign.zPosition));
         }
     }
 
@@ -1173,19 +1162,19 @@
             {
                 if (par1Packet132TileEntityData.actionType == 1 && var2 instanceof TileEntityMobSpawner)
                 {
-                    var2.readFromNBT(par1Packet132TileEntityData.data);
+                    var2.readFromNBT(par1Packet132TileEntityData.customParam1);
                 }
                 else if (par1Packet132TileEntityData.actionType == 2 && var2 instanceof TileEntityCommandBlock)
                 {
-                    var2.readFromNBT(par1Packet132TileEntityData.data);
+                    var2.readFromNBT(par1Packet132TileEntityData.customParam1);
                 }
                 else if (par1Packet132TileEntityData.actionType == 3 && var2 instanceof TileEntityBeacon)
                 {
-                    var2.readFromNBT(par1Packet132TileEntityData.data);
+                    var2.readFromNBT(par1Packet132TileEntityData.customParam1);
                 }
                 else if (par1Packet132TileEntityData.actionType == 4 && var2 instanceof TileEntitySkull)
                 {
-                    var2.readFromNBT(par1Packet132TileEntityData.data);
+                    var2.readFromNBT(par1Packet132TileEntityData.customParam1);
                 }
             }
         }
@@ -1636,27 +1625,27 @@
             }
             else
             {
-                BaseAttributeMap var3 = ((EntityLivingBase)var2).getAttributeMap();
+                BaseAttributeMap var3 = ((EntityLivingBase)var2).func_110140_aT();
                 Iterator var4 = par1Packet44UpdateAttributes.func_111003_f().iterator();
 
                 while (var4.hasNext())
                 {
                     Packet44UpdateAttributesSnapshot var5 = (Packet44UpdateAttributesSnapshot)var4.next();
-                    AttributeInstance var6 = var3.getAttributeInstanceByName(var5.func_142040_a());
+                    AttributeInstance var6 = var3.func_111152_a(var5.func_142040_a());
 
                     if (var6 == null)
                     {
                         var6 = var3.func_111150_b(new RangedAttribute(var5.func_142040_a(), 0.0D, 2.2250738585072014E-308D, Double.MAX_VALUE));
                     }
 
-                    var6.setAttribute(var5.func_142041_b());
+                    var6.func_111128_a(var5.func_142041_b());
                     var6.func_142049_d();
                     Iterator var7 = var5.func_142039_c().iterator();
 
                     while (var7.hasNext())
                     {
                         AttributeModifier var8 = (AttributeModifier)var7.next();
-                        var6.applyModifier(var8);
+                        var6.func_111121_a(var8);
                     }
                 }
             }
