--- a/net/minecraft/src/NetServerHandler.java
+++ b/net/minecraft/src/NetServerHandler.java
@@ -99,7 +99,7 @@
             this.playerEntity.mountEntityAndWakeUp();
             this.sendPacketToPlayer(new Packet255KickDisconnect(par1Str));
             this.netManager.serverShutdown();
-            this.mcServer.getConfigurationManager().sendChatMsg(ChatMessageComponent.createFromTranslationWithSubstitutions("multiplayer.player.left", new Object[] {this.playerEntity.getTranslatedEntityName()}).setColor(EnumChatFormatting.YELLOW));
+            this.mcServer.getConfigurationManager().sendChatMsg(ChatMessageComponent.func_111082_b("multiplayer.player.left", new Object[] {this.playerEntity.getTranslatedEntityName()}).func_111059_a(EnumChatFormatting.YELLOW));
             this.mcServer.getConfigurationManager().playerLoggedOut(this.playerEntity);
             this.connectionClosed = true;
         }
@@ -107,7 +107,7 @@
 
     public void func_110774_a(Packet27PlayerInput par1Packet27PlayerInput)
     {
-        this.playerEntity.setEntityActionState(par1Packet27PlayerInput.func_111010_d(), par1Packet27PlayerInput.func_111012_f(), par1Packet27PlayerInput.func_111013_g(), par1Packet27PlayerInput.func_111011_h());
+        this.playerEntity.func_110430_a(par1Packet27PlayerInput.func_111010_d(), par1Packet27PlayerInput.func_111012_f(), par1Packet27PlayerInput.func_111013_g(), par1Packet27PlayerInput.func_111011_h());
     }
 
     public void handleFlying(Packet10Flying par1Packet10Flying)
@@ -336,7 +336,6 @@
     public void handleBlockDig(Packet14BlockDig par1Packet14BlockDig)
     {
         WorldServer var2 = this.mcServer.worldServerForDimension(this.playerEntity.dimension);
-        this.playerEntity.func_143004_u();
 
         if (par1Packet14BlockDig.status == 4)
         {
@@ -393,7 +392,7 @@
 
             if (par1Packet14BlockDig.status == 0)
             {
-                if (!this.mcServer.isBlockProtected(var2, var4, var5, var6, this.playerEntity))
+                if (!this.mcServer.func_96290_a(var2, var4, var5, var6, this.playerEntity))
                 {
                     this.playerEntity.theItemInWorldManager.onBlockClicked(var4, var5, var6, par1Packet14BlockDig.face);
                 }
@@ -432,7 +431,6 @@
         int var6 = par1Packet15Place.getYPosition();
         int var7 = par1Packet15Place.getZPosition();
         int var8 = par1Packet15Place.getDirection();
-        this.playerEntity.func_143004_u();
 
         if (par1Packet15Place.getDirection() == 255)
         {
@@ -445,12 +443,12 @@
         }
         else if (par1Packet15Place.getYPosition() >= this.mcServer.getBuildLimit() - 1 && (par1Packet15Place.getDirection() == 1 || par1Packet15Place.getYPosition() >= this.mcServer.getBuildLimit()))
         {
-            this.playerEntity.playerNetServerHandler.sendPacketToPlayer(new Packet3Chat(ChatMessageComponent.createFromTranslationWithSubstitutions("build.tooHigh", new Object[] {Integer.valueOf(this.mcServer.getBuildLimit())}).setColor(EnumChatFormatting.RED)));
+            this.playerEntity.playerNetServerHandler.sendPacketToPlayer(new Packet3Chat(ChatMessageComponent.func_111082_b("build.tooHigh", new Object[] {Integer.valueOf(this.mcServer.getBuildLimit())}).func_111059_a(EnumChatFormatting.RED)));
             var4 = true;
         }
         else
         {
-            if (this.hasMoved && this.playerEntity.getDistanceSq((double)var5 + 0.5D, (double)var6 + 0.5D, (double)var7 + 0.5D) < 64.0D && !this.mcServer.isBlockProtected(var2, var5, var6, var7, this.playerEntity))
+            if (this.hasMoved && this.playerEntity.getDistanceSq((double)var5 + 0.5D, (double)var6 + 0.5D, (double)var7 + 0.5D) < 64.0D && !this.mcServer.func_96290_a(var2, var5, var6, var7, this.playerEntity))
             {
                 this.playerEntity.theItemInWorldManager.activateBlockOrUseItem(this.playerEntity, var2, var3, var5, var6, var7, var8, par1Packet15Place.getXOffset(), par1Packet15Place.getYOffset(), par1Packet15Place.getZOffset());
             }
@@ -521,7 +519,7 @@
     public void handleErrorMessage(String par1Str, Object[] par2ArrayOfObj)
     {
         this.mcServer.getLogAgent().logInfo(this.playerEntity.getCommandSenderName() + " lost connection: " + par1Str);
-        this.mcServer.getConfigurationManager().sendChatMsg(ChatMessageComponent.createFromTranslationWithSubstitutions("multiplayer.player.left", new Object[] {this.playerEntity.getTranslatedEntityName()}).setColor(EnumChatFormatting.YELLOW));
+        this.mcServer.getConfigurationManager().sendChatMsg(ChatMessageComponent.func_111082_b("multiplayer.player.left", new Object[] {this.playerEntity.getTranslatedEntityName()}).func_111059_a(EnumChatFormatting.YELLOW));
         this.mcServer.getConfigurationManager().playerLoggedOut(this.playerEntity);
         this.connectionClosed = true;
 
@@ -582,7 +580,6 @@
         if (par1Packet16BlockItemSwitch.id >= 0 && par1Packet16BlockItemSwitch.id < InventoryPlayer.getHotbarSize())
         {
             this.playerEntity.inventory.currentItem = par1Packet16BlockItemSwitch.id;
-            this.playerEntity.func_143004_u();
         }
         else
         {
@@ -594,11 +591,10 @@
     {
         if (this.playerEntity.getChatVisibility() == 2)
         {
-            this.sendPacketToPlayer(new Packet3Chat(ChatMessageComponent.createFromTranslationKey("chat.cannotSend").setColor(EnumChatFormatting.RED)));
+            this.sendPacketToPlayer(new Packet3Chat(ChatMessageComponent.func_111077_e("chat.cannotSend").func_111059_a(EnumChatFormatting.RED)));
         }
         else
         {
-            this.playerEntity.func_143004_u();
             String var2 = par1Packet3Chat.message;
 
             if (var2.length() > 100)
@@ -626,17 +622,17 @@
                 {
                     if (this.playerEntity.getChatVisibility() == 1)
                     {
-                        this.sendPacketToPlayer(new Packet3Chat(ChatMessageComponent.createFromTranslationKey("chat.cannotSend").setColor(EnumChatFormatting.RED)));
+                        this.sendPacketToPlayer(new Packet3Chat(ChatMessageComponent.func_111077_e("chat.cannotSend").func_111059_a(EnumChatFormatting.RED)));
                         return;
                     }
 
-                    ChatMessageComponent var4 = ChatMessageComponent.createFromTranslationWithSubstitutions("chat.type.text", new Object[] {this.playerEntity.getTranslatedEntityName(), var2});
+                    ChatMessageComponent var4 = ChatMessageComponent.func_111082_b("chat.type.text", new Object[] {this.playerEntity.getTranslatedEntityName(), var2});
                     this.mcServer.getConfigurationManager().func_110459_a(var4, false);
                 }
 
                 this.chatSpamThresholdCount += 20;
 
-                if (this.chatSpamThresholdCount > 200 && !this.mcServer.getConfigurationManager().isPlayerOpped(this.playerEntity.getCommandSenderName()))
+                if (this.chatSpamThresholdCount > 200 && !this.mcServer.getConfigurationManager().areCommandsAllowed(this.playerEntity.getCommandSenderName()))
                 {
                     this.kickPlayerFromServer("disconnect.spam");
                 }
@@ -654,8 +650,6 @@
 
     public void handleAnimation(Packet18Animation par1Packet18Animation)
     {
-        this.playerEntity.func_143004_u();
-
         if (par1Packet18Animation.animate == 1)
         {
             this.playerEntity.swingItem();
@@ -667,39 +661,37 @@
      */
     public void handleEntityAction(Packet19EntityAction par1Packet19EntityAction)
     {
-        this.playerEntity.func_143004_u();
-
-        if (par1Packet19EntityAction.action == 1)
+        if (par1Packet19EntityAction.state == 1)
         {
             this.playerEntity.setSneaking(true);
         }
-        else if (par1Packet19EntityAction.action == 2)
+        else if (par1Packet19EntityAction.state == 2)
         {
             this.playerEntity.setSneaking(false);
         }
-        else if (par1Packet19EntityAction.action == 4)
+        else if (par1Packet19EntityAction.state == 4)
         {
             this.playerEntity.setSprinting(true);
         }
-        else if (par1Packet19EntityAction.action == 5)
+        else if (par1Packet19EntityAction.state == 5)
         {
             this.playerEntity.setSprinting(false);
         }
-        else if (par1Packet19EntityAction.action == 3)
+        else if (par1Packet19EntityAction.state == 3)
         {
             this.playerEntity.wakeUpPlayer(false, true, true);
             this.hasMoved = false;
         }
-        else if (par1Packet19EntityAction.action == 6)
+        else if (par1Packet19EntityAction.state == 6)
         {
             if (this.playerEntity.ridingEntity != null && this.playerEntity.ridingEntity instanceof EntityHorse)
             {
-                ((EntityHorse)this.playerEntity.ridingEntity).setJumpPower(par1Packet19EntityAction.auxData);
+                ((EntityHorse)this.playerEntity.ridingEntity).func_110206_u(par1Packet19EntityAction.field_111009_c);
             }
         }
-        else if (par1Packet19EntityAction.action == 7 && this.playerEntity.ridingEntity != null && this.playerEntity.ridingEntity instanceof EntityHorse)
+        else if (par1Packet19EntityAction.state == 7 && this.playerEntity.ridingEntity != null && this.playerEntity.ridingEntity instanceof EntityHorse)
         {
-            ((EntityHorse)this.playerEntity.ridingEntity).openGUI(this.playerEntity);
+            ((EntityHorse)this.playerEntity.ridingEntity).func_110199_f(this.playerEntity);
         }
     }
 
@@ -720,7 +712,6 @@
     {
         WorldServer var2 = this.mcServer.worldServerForDimension(this.playerEntity.dimension);
         Entity var3 = var2.getEntityByID(par1Packet7UseEntity.targetEntity);
-        this.playerEntity.func_143004_u();
 
         if (var3 != null)
         {
@@ -755,8 +746,6 @@
 
     public void handleClientCommand(Packet205ClientCommand par1Packet205ClientCommand)
     {
-        this.playerEntity.func_143004_u();
-
         if (par1Packet205ClientCommand.forceRespawn == 1)
         {
             if (this.playerEntity.playerConqueredTheEnd)
@@ -780,7 +769,7 @@
             }
             else
             {
-                if (this.playerEntity.getHealth() > 0.0F)
+                if (this.playerEntity.func_110143_aJ() > 0.0F)
                 {
                     return;
                 }
@@ -812,8 +801,6 @@
 
     public void handleWindowClick(Packet102WindowClick par1Packet102WindowClick)
     {
-        this.playerEntity.func_143004_u();
-
         if (this.playerEntity.openContainer.windowId == par1Packet102WindowClick.window_Id && this.playerEntity.openContainer.isPlayerNotUsingContainer(this.playerEntity))
         {
             ItemStack var2 = this.playerEntity.openContainer.slotClick(par1Packet102WindowClick.inventorySlot, par1Packet102WindowClick.mouseClick, par1Packet102WindowClick.holdingShift, this.playerEntity);
@@ -845,8 +832,6 @@
 
     public void handleEnchantItem(Packet108EnchantItem par1Packet108EnchantItem)
     {
-        this.playerEntity.func_143004_u();
-
         if (this.playerEntity.openContainer.windowId == par1Packet108EnchantItem.windowId && this.playerEntity.openContainer.isPlayerNotUsingContainer(this.playerEntity))
         {
             this.playerEntity.openContainer.enchantItem(this.playerEntity, par1Packet108EnchantItem.enchantment);
@@ -908,7 +893,6 @@
      */
     public void handleUpdateSign(Packet130UpdateSign par1Packet130UpdateSign)
     {
-        this.playerEntity.func_143004_u();
         WorldServer var2 = this.mcServer.worldServerForDimension(this.playerEntity.dimension);
 
         if (var2.blockExists(par1Packet130UpdateSign.xPosition, par1Packet130UpdateSign.yPosition, par1Packet130UpdateSign.zPosition))
@@ -1105,7 +1089,7 @@
                 {
                     if (!this.mcServer.isCommandBlockEnabled())
                     {
-                        this.playerEntity.sendChatToPlayer(ChatMessageComponent.createFromTranslationKey("advMode.notEnabled"));
+                        this.playerEntity.sendChatToPlayer(ChatMessageComponent.func_111077_e("advMode.notEnabled"));
                     }
                     else if (this.playerEntity.canCommandSenderUseCommand(2, "") && this.playerEntity.capabilities.isCreativeMode)
                     {
@@ -1122,7 +1106,7 @@
                             {
                                 ((TileEntityCommandBlock)var7).setCommand(var6);
                                 this.playerEntity.worldObj.markBlockForUpdate(var14, var18, var5);
-                                this.playerEntity.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("advMode.setCommand.success", new Object[] {var6}));
+                                this.playerEntity.sendChatToPlayer(ChatMessageComponent.func_111082_b("advMode.setCommand.success", new Object[] {var6}));
                             }
                         }
                         catch (Exception var9)
@@ -1132,7 +1116,7 @@
                     }
                     else
                     {
-                        this.playerEntity.sendChatToPlayer(ChatMessageComponent.createFromTranslationKey("advMode.notAllowed"));
+                        this.playerEntity.sendChatToPlayer(ChatMessageComponent.func_111077_e("advMode.notAllowed"));
                     }
                 }
                 else if ("MC|Beacon".equals(par1Packet250CustomPayload.channel))
@@ -1184,7 +1168,7 @@
         }
     }
 
-    public boolean isConnectionClosed()
+    public boolean func_142032_c()
     {
         return this.connectionClosed;
     }
