--- a/net/minecraft/src/Packet39AttachEntity.java
+++ b/net/minecraft/src/Packet39AttachEntity.java
@@ -6,17 +6,16 @@
 
 public class Packet39AttachEntity extends Packet
 {
-    /** 0 for riding, 1 for leashed. */
-    public int attachState;
-    public int ridingEntityId;
+    public int field_111007_a;
+    public int field_111006_b;
     public int vehicleEntityId;
 
     public Packet39AttachEntity() {}
 
     public Packet39AttachEntity(int par1, Entity par2Entity, Entity par3Entity)
     {
-        this.attachState = par1;
-        this.ridingEntityId = par2Entity.entityId;
+        this.field_111007_a = par1;
+        this.field_111006_b = par2Entity.entityId;
         this.vehicleEntityId = par3Entity != null ? par3Entity.entityId : -1;
     }
 
@@ -33,9 +32,9 @@
      */
     public void readPacketData(DataInput par1DataInput) throws IOException
     {
-        this.ridingEntityId = par1DataInput.readInt();
+        this.field_111006_b = par1DataInput.readInt();
         this.vehicleEntityId = par1DataInput.readInt();
-        this.attachState = par1DataInput.readUnsignedByte();
+        this.field_111007_a = par1DataInput.readUnsignedByte();
     }
 
     /**
@@ -43,9 +42,9 @@
      */
     public void writePacketData(DataOutput par1DataOutput) throws IOException
     {
-        par1DataOutput.writeInt(this.ridingEntityId);
+        par1DataOutput.writeInt(this.field_111006_b);
         par1DataOutput.writeInt(this.vehicleEntityId);
-        par1DataOutput.writeByte(this.attachState);
+        par1DataOutput.writeByte(this.field_111007_a);
     }
 
     /**
@@ -71,6 +70,6 @@
     public boolean containsSameEntityIDAs(Packet par1Packet)
     {
         Packet39AttachEntity var2 = (Packet39AttachEntity)par1Packet;
-        return var2.ridingEntityId == this.ridingEntityId;
+        return var2.field_111006_b == this.field_111006_b;
     }
 }
