--- a/net/minecraft/src/PackMetadataSection.java
+++ b/net/minecraft/src/PackMetadataSection.java
@@ -2,22 +2,22 @@
 
 public class PackMetadataSection implements MetadataSection
 {
-    private final String packDescription;
-    private final int packFormat;
+    private final String field_110464_a;
+    private final int field_110463_b;
 
     public PackMetadataSection(String par1Str, int par2)
     {
-        this.packDescription = par1Str;
-        this.packFormat = par2;
+        this.field_110464_a = par1Str;
+        this.field_110463_b = par2;
     }
 
-    public String getPackDescription()
+    public String func_110461_a()
     {
-        return this.packDescription;
+        return this.field_110464_a;
     }
 
-    public int getPackFormat()
+    public int func_110462_b()
     {
-        return this.packFormat;
+        return this.field_110463_b;
     }
 }
