--- a/net/minecraft/src/Packet.java
+++ b/net/minecraft/src/Packet.java
@@ -27,7 +27,7 @@
     protected ILogAgent field_98193_m;
 
     /** the system time in milliseconds when this packet was created. */
-    public final long creationTimeMillis = MinecraftServer.getSystemTimeMillis();
+    public final long creationTimeMillis = MinecraftServer.func_130071_aq();
     public static long receivedID;
     public static long receivedSize;
 
