--- a/net/minecraft/src/PathNavigate.java
+++ b/net/minecraft/src/PathNavigate.java
@@ -49,7 +49,7 @@
     {
         this.theEntity = par1EntityLiving;
         this.worldObj = par2World;
-        this.pathSearchRange = par1EntityLiving.getEntityAttribute(SharedMonsterAttributes.followRange);
+        this.pathSearchRange = par1EntityLiving.func_110148_a(SharedMonsterAttributes.field_111265_b);
     }
 
     public void setAvoidsWater(boolean par1)
@@ -109,7 +109,7 @@
 
     public float func_111269_d()
     {
-        return (float)this.pathSearchRange.getAttributeValue();
+        return (float)this.pathSearchRange.func_111126_e();
     }
 
     /**
