--- a/net/minecraft/src/PlayerManager.java
+++ b/net/minecraft/src/PlayerManager.java
@@ -167,12 +167,12 @@
         }
     }
 
-    public PlayerInstance getOrCreateChunkWatcher(int par1, int par2, boolean par3)
+    private PlayerInstance getOrCreateChunkWatcher(int par1, int par2, boolean par3)
     {
         return this.getOrCreateChunkWatcher(par1, par2, par3, false);
     }
 
-    public PlayerInstance getOrCreateChunkWatcher(int par1, int par2, boolean par3, boolean lazy)
+    private PlayerInstance getOrCreateChunkWatcher(int par1, int par2, boolean par3, boolean lazy)
     {
         long var4 = (long)par1 + 2147483647L | (long)par2 + 2147483647L << 32;
         PlayerInstance var6 = (PlayerInstance)this.playerInstances.getValueByKey(var4);
