--- a/net/minecraft/src/PlayerControllerMP.java
+++ b/net/minecraft/src/PlayerControllerMP.java
@@ -399,7 +399,7 @@
 
     public EntityClientPlayerMP func_78754_a(World par1World)
     {
-        return new EntityClientPlayerMP(this.mc, par1World, this.mc.getSession(), this.netClientHandler);
+        return new EntityClientPlayerMP(this.mc, par1World, this.mc.func_110432_I(), this.netClientHandler);
     }
 
     /**
