--- a/net/minecraft/src/PlayerSelector.java
+++ b/net/minecraft/src/PlayerSelector.java
@@ -308,15 +308,12 @@
                     case 0:
                         var5 = "x";
                         break;
-
                     case 1:
                         var5 = "y";
                         break;
-
                     case 2:
                         var5 = "z";
                         break;
-
                     case 3:
                         var5 = "r";
                 }
