--- a/net/minecraft/src/PotionAbsoption.java
+++ b/net/minecraft/src/PotionAbsoption.java
@@ -7,15 +7,15 @@
         super(par1, par2, par3);
     }
 
-    public void removeAttributesModifiersFromEntity(EntityLivingBase par1EntityLivingBase, BaseAttributeMap par2BaseAttributeMap, int par3)
+    public void func_111187_a(EntityLivingBase par1EntityLivingBase, BaseAttributeMap par2BaseAttributeMap, int par3)
     {
-        par1EntityLivingBase.setAbsorptionAmount(par1EntityLivingBase.getAbsorptionAmount() - (float)(4 * (par3 + 1)));
-        super.removeAttributesModifiersFromEntity(par1EntityLivingBase, par2BaseAttributeMap, par3);
+        par1EntityLivingBase.func_110149_m(par1EntityLivingBase.func_110139_bj() - (float)(4 * (par3 + 1)));
+        super.func_111187_a(par1EntityLivingBase, par2BaseAttributeMap, par3);
     }
 
-    public void applyAttributesModifiersToEntity(EntityLivingBase par1EntityLivingBase, BaseAttributeMap par2BaseAttributeMap, int par3)
+    public void func_111185_a(EntityLivingBase par1EntityLivingBase, BaseAttributeMap par2BaseAttributeMap, int par3)
     {
-        par1EntityLivingBase.setAbsorptionAmount(par1EntityLivingBase.getAbsorptionAmount() + (float)(4 * (par3 + 1)));
-        super.applyAttributesModifiersToEntity(par1EntityLivingBase, par2BaseAttributeMap, par3);
+        par1EntityLivingBase.func_110149_m(par1EntityLivingBase.func_110139_bj() + (float)(4 * (par3 + 1)));
+        super.func_111185_a(par1EntityLivingBase, par2BaseAttributeMap, par3);
     }
 }
