--- a/net/minecraft/src/Potion.java
+++ b/net/minecraft/src/Potion.java
@@ -11,11 +11,11 @@
     /** The array of potion types. */
     public static final Potion[] potionTypes = new Potion[32];
     public static final Potion field_76423_b = null;
-    public static final Potion moveSpeed = (new Potion(1, false, 8171462)).setPotionName("potion.moveSpeed").setIconIndex(0, 0).func_111184_a(SharedMonsterAttributes.movementSpeed, "91AEAA56-376B-4498-935B-2F7F68070635", 0.20000000298023224D, 2);
-    public static final Potion moveSlowdown = (new Potion(2, true, 5926017)).setPotionName("potion.moveSlowdown").setIconIndex(1, 0).func_111184_a(SharedMonsterAttributes.movementSpeed, "7107DE5E-7CE8-4030-940E-514C1F160890", -0.15000000596046448D, 2);
+    public static final Potion moveSpeed = (new Potion(1, false, 8171462)).setPotionName("potion.moveSpeed").setIconIndex(0, 0).func_111184_a(SharedMonsterAttributes.field_111263_d, "91AEAA56-376B-4498-935B-2F7F68070635", 0.20000000298023224D, 2);
+    public static final Potion moveSlowdown = (new Potion(2, true, 5926017)).setPotionName("potion.moveSlowdown").setIconIndex(1, 0).func_111184_a(SharedMonsterAttributes.field_111263_d, "7107DE5E-7CE8-4030-940E-514C1F160890", -0.15000000596046448D, 2);
     public static final Potion digSpeed = (new Potion(3, false, 14270531)).setPotionName("potion.digSpeed").setIconIndex(2, 0).setEffectiveness(1.5D);
     public static final Potion digSlowdown = (new Potion(4, true, 4866583)).setPotionName("potion.digSlowDown").setIconIndex(3, 0);
-    public static final Potion damageBoost = (new PotionAttackDamage(5, false, 9643043)).setPotionName("potion.damageBoost").setIconIndex(4, 0).func_111184_a(SharedMonsterAttributes.attackDamage, "648D7064-6A60-4F59-8ABE-C2C23A6DD7A9", 3.0D, 2);
+    public static final Potion damageBoost = (new PotionAttackDamage(5, false, 9643043)).setPotionName("potion.damageBoost").setIconIndex(4, 0).func_111184_a(SharedMonsterAttributes.field_111264_e, "648D7064-6A60-4F59-8ABE-C2C23A6DD7A9", 3.0D, 2);
     public static final Potion heal = (new PotionHealth(6, false, 16262179)).setPotionName("potion.heal");
     public static final Potion harm = (new PotionHealth(7, true, 4393481)).setPotionName("potion.harm");
     public static final Potion jump = (new Potion(8, false, 7889559)).setPotionName("potion.jump").setIconIndex(2, 1);
@@ -44,14 +44,14 @@
     public static final Potion hunger = (new Potion(17, true, 5797459)).setPotionName("potion.hunger").setIconIndex(1, 1);
 
     /** The weakness Potion object. */
-    public static final Potion weakness = (new PotionAttackDamage(18, true, 4738376)).setPotionName("potion.weakness").setIconIndex(5, 0).func_111184_a(SharedMonsterAttributes.attackDamage, "22653B89-116E-49DC-9B6B-9971489B5BE5", 2.0D, 0);
+    public static final Potion weakness = (new PotionAttackDamage(18, true, 4738376)).setPotionName("potion.weakness").setIconIndex(5, 0).func_111184_a(SharedMonsterAttributes.field_111264_e, "22653B89-116E-49DC-9B6B-9971489B5BE5", 2.0D, 0);
 
     /** The poison Potion object. */
     public static final Potion poison = (new Potion(19, true, 5149489)).setPotionName("potion.poison").setIconIndex(6, 0).setEffectiveness(0.25D);
 
     /** The wither Potion object. */
     public static final Potion wither = (new Potion(20, true, 3484199)).setPotionName("potion.wither").setIconIndex(1, 2).setEffectiveness(0.25D);
-    public static final Potion field_76434_w = (new PotionHealthBoost(21, false, 16284963)).setPotionName("potion.healthBoost").setIconIndex(2, 2).func_111184_a(SharedMonsterAttributes.maxHealth, "5D6F0BA2-1186-46AC-B896-C61C5CEE99CC", 4.0D, 0);
+    public static final Potion field_76434_w = (new PotionHealthBoost(21, false, 16284963)).setPotionName("potion.healthBoost").setIconIndex(2, 2).func_111184_a(SharedMonsterAttributes.field_111267_a, "5D6F0BA2-1186-46AC-B896-C61C5CEE99CC", 4.0D, 0);
     public static final Potion field_76444_x = (new PotionAbsoption(22, false, 2445989)).setPotionName("potion.absorption").setIconIndex(2, 2);
     public static final Potion field_76443_y = (new PotionHealth(23, false, 16262179)).setPotionName("potion.saturation");
     public static final Potion field_76442_z = null;
@@ -122,14 +122,14 @@
     {
         if (this.id == regeneration.id)
         {
-            if (par1EntityLivingBase.getHealth() < par1EntityLivingBase.getMaxHealth())
+            if (par1EntityLivingBase.func_110143_aJ() < par1EntityLivingBase.func_110138_aP())
             {
                 par1EntityLivingBase.heal(1.0F);
             }
         }
         else if (this.id == poison.id)
         {
-            if (par1EntityLivingBase.getHealth() > 1.0F)
+            if (par1EntityLivingBase.func_110143_aJ() > 1.0F)
             {
                 par1EntityLivingBase.attackEntityFrom(DamageSource.magic, 1.0F);
             }
@@ -318,42 +318,42 @@
         return this.field_111188_I;
     }
 
-    public void removeAttributesModifiersFromEntity(EntityLivingBase par1EntityLivingBase, BaseAttributeMap par2BaseAttributeMap, int par3)
+    public void func_111187_a(EntityLivingBase par1EntityLivingBase, BaseAttributeMap par2BaseAttributeMap, int par3)
     {
         Iterator var4 = this.field_111188_I.entrySet().iterator();
 
         while (var4.hasNext())
         {
             Entry var5 = (Entry)var4.next();
-            AttributeInstance var6 = par2BaseAttributeMap.getAttributeInstance((Attribute)var5.getKey());
+            AttributeInstance var6 = par2BaseAttributeMap.func_111151_a((Attribute)var5.getKey());
 
             if (var6 != null)
             {
-                var6.removeModifier((AttributeModifier)var5.getValue());
+                var6.func_111124_b((AttributeModifier)var5.getValue());
             }
         }
     }
 
-    public void applyAttributesModifiersToEntity(EntityLivingBase par1EntityLivingBase, BaseAttributeMap par2BaseAttributeMap, int par3)
+    public void func_111185_a(EntityLivingBase par1EntityLivingBase, BaseAttributeMap par2BaseAttributeMap, int par3)
     {
         Iterator var4 = this.field_111188_I.entrySet().iterator();
 
         while (var4.hasNext())
         {
             Entry var5 = (Entry)var4.next();
-            AttributeInstance var6 = par2BaseAttributeMap.getAttributeInstance((Attribute)var5.getKey());
+            AttributeInstance var6 = par2BaseAttributeMap.func_111151_a((Attribute)var5.getKey());
 
             if (var6 != null)
             {
                 AttributeModifier var7 = (AttributeModifier)var5.getValue();
-                var6.removeModifier(var7);
-                var6.applyModifier(new AttributeModifier(var7.getID(), this.getName() + " " + par3, this.func_111183_a(par3, var7), var7.getOperation()));
+                var6.func_111124_b(var7);
+                var6.func_111121_a(new AttributeModifier(var7.func_111167_a(), this.getName() + " " + par3, this.func_111183_a(par3, var7), var7.func_111169_c()));
             }
         }
     }
 
     public double func_111183_a(int par1, AttributeModifier par2AttributeModifier)
     {
-        return par2AttributeModifier.getAmount() * (double)(par1 + 1);
+        return par2AttributeModifier.func_111164_d() * (double)(par1 + 1);
     }
 }
