--- a/net/minecraft/src/RandomMobs.java
+++ b/net/minecraft/src/RandomMobs.java
@@ -99,7 +99,7 @@
         }
         else
         {
-            ResourceLocation entity;
+            ResourceLocation name;
 
             try
             {
@@ -110,45 +110,44 @@
                     initialize();
                 }
 
-                if (renderGlobal != null)
+                if (renderGlobal == null)
                 {
-                    Entity entity1 = renderGlobal.renderedEntity;
-                    ResourceLocation name1;
+                    ResourceLocation entity1 = loc;
+                    return entity1;
+                }
 
-                    if (entity1 == null)
-                    {
-                        name1 = loc;
-                        return name1;
-                    }
+                Entity entity = renderGlobal.renderedEntity;
 
-                    if (!(entity1 instanceof EntityLiving))
-                    {
-                        name1 = loc;
-                        return name1;
-                    }
+                if (entity == null)
+                {
+                    name = loc;
+                    return name;
+                }
 
-                    String name = loc.getResourcePath();
+                if (entity instanceof EntityLiving)
+                {
+                    String name1 = loc.getResourcePath();
 
-                    if (!name.startsWith("textures/entity/"))
+                    if (!name1.startsWith("textures/entity/"))
                     {
                         ResourceLocation uuidLow1 = loc;
                         return uuidLow1;
                     }
 
-                    long uuidLow = entity1.getUniqueID().getLeastSignificantBits();
+                    long uuidLow = entity.getUniqueID().getLeastSignificantBits();
                     int id = (int)(uuidLow & 2147483647L);
                     ResourceLocation var6 = getTextureLocation(loc, id);
                     return var6;
                 }
 
-                entity = loc;
+                name = loc;
             }
             finally
             {
                 working = false;
             }
 
-            return entity;
+            return name;
         }
     }
 
