--- a/net/minecraft/src/RandomMobs.java
+++ b/net/minecraft/src/RandomMobs.java
@@ -118,14 +118,14 @@
 
                 Entity entity = renderGlobal.renderedEntity;
 
-                if (entity == null)
+                if (entity != null)
                 {
-                    entityLiving = loc;
-                    return entityLiving;
-                }
+                    if (!(entity instanceof EntityLiving))
+                    {
+                        entityLiving = loc;
+                        return entityLiving;
+                    }
 
-                if (entity instanceof EntityLiving)
-                {
                     EntityLiving entityLiving1 = (EntityLiving)entity;
                     String name = loc.func_110623_a();
 
