--- a/net/minecraft/src/RandomPositionGenerator.java
+++ b/net/minecraft/src/RandomPositionGenerator.java
@@ -54,9 +54,9 @@
         float var9 = -99999.0F;
         boolean var10;
 
-        if (par0EntityCreature.hasHome())
+        if (par0EntityCreature.func_110175_bO())
         {
-            double var11 = (double)(par0EntityCreature.getHomePosition().getDistanceSquared(MathHelper.floor_double(par0EntityCreature.posX), MathHelper.floor_double(par0EntityCreature.posY), MathHelper.floor_double(par0EntityCreature.posZ)) + 4.0F);
+            double var11 = (double)(par0EntityCreature.func_110172_bL().getDistanceSquared(MathHelper.floor_double(par0EntityCreature.posX), MathHelper.floor_double(par0EntityCreature.posY), MathHelper.floor_double(par0EntityCreature.posZ)) + 4.0F);
             double var13 = (double)(par0EntityCreature.func_110174_bM() + (float)par1);
             var10 = var11 < var13 * var13;
         }
