--- a/net/minecraft/src/RandomMobs.java
+++ b/net/minecraft/src/RandomMobs.java
@@ -99,7 +99,7 @@
         }
         else
         {
-            ResourceLocation entityLiving;
+            ResourceLocation entity;
 
             try
             {
@@ -110,23 +110,24 @@
                     initialize();
                 }
 
-                if (renderGlobal == null)
+                if (renderGlobal != null)
                 {
-                    ResourceLocation entity1 = loc;
-                    return entity1;
-                }
+                    Entity entity1 = renderGlobal.renderedEntity;
+                    ResourceLocation entityLiving1;
 
-                Entity entity = renderGlobal.renderedEntity;
+                    if (entity1 == null)
+                    {
+                        entityLiving1 = loc;
+                        return entityLiving1;
+                    }
 
-                if (entity == null)
-                {
-                    entityLiving = loc;
-                    return entityLiving;
-                }
+                    if (!(entity1 instanceof EntityLiving))
+                    {
+                        entityLiving1 = loc;
+                        return entityLiving1;
+                    }
 
-                if (entity instanceof EntityLiving)
-                {
-                    EntityLiving entityLiving1 = (EntityLiving)entity;
+                    EntityLiving entityLiving = (EntityLiving)entity1;
                     String name = loc.func_110623_a();
 
                     if (!name.startsWith("textures/entity/"))
@@ -135,19 +136,19 @@
                         return id1;
                     }
 
-                    int id = (int)entity.func_110124_au().getLeastSignificantBits();
+                    int id = (int)entity1.func_110124_au().getLeastSignificantBits();
                     ResourceLocation var5 = getTextureLocation(loc, id);
                     return var5;
                 }
 
-                entityLiving = loc;
+                entity = loc;
             }
             finally
             {
                 working = false;
             }
 
-            return entityLiving;
+            return entity;
         }
     }
 
