--- a/net/minecraft/src/RandomMobs.java
+++ b/net/minecraft/src/RandomMobs.java
@@ -99,7 +99,7 @@
         }
         else
         {
-            ResourceLocation var6;
+            ResourceLocation name;
 
             try
             {
@@ -117,38 +117,37 @@
                 }
 
                 Entity entity = renderGlobal.renderedEntity;
-                ResourceLocation name1;
 
                 if (entity == null)
                 {
-                    name1 = loc;
-                    return name1;
+                    name = loc;
+                    return name;
                 }
 
-                if (!(entity instanceof EntityLiving))
+                if (entity instanceof EntityLiving)
                 {
-                    name1 = loc;
-                    return name1;
-                }
+                    String name1 = loc.getResourcePath();
 
-                String name = loc.getResourcePath();
+                    if (!name1.startsWith("textures/entity/"))
+                    {
+                        ResourceLocation uuidLow1 = loc;
+                        return uuidLow1;
+                    }
 
-                if (!name.startsWith("textures/entity/"))
-                {
-                    ResourceLocation uuidLow1 = loc;
-                    return uuidLow1;
+                    long uuidLow = entity.getUniqueID().getLeastSignificantBits();
+                    int id = (int)(uuidLow & 2147483647L);
+                    ResourceLocation var6 = getTextureLocation(loc, id);
+                    return var6;
                 }
 
-                long uuidLow = entity.getUniqueID().getLeastSignificantBits();
-                int id = (int)(uuidLow & 2147483647L);
-                var6 = getTextureLocation(loc, id);
+                name = loc;
             }
             finally
             {
                 working = false;
             }
 
-            return var6;
+            return name;
         }
     }
 
