--- a/net/minecraft/src/RangedAttribute.java
+++ b/net/minecraft/src/RangedAttribute.java
@@ -2,15 +2,15 @@
 
 public class RangedAttribute extends BaseAttribute
 {
-    private final double minimumValue;
-    private final double maximumValue;
+    private final double field_111120_a;
+    private final double field_111118_b;
     private String field_111119_c;
 
     public RangedAttribute(String par1Str, double par2, double par4, double par6)
     {
         super(par1Str, par2);
-        this.minimumValue = par4;
-        this.maximumValue = par6;
+        this.field_111120_a = par4;
+        this.field_111118_b = par6;
 
         if (par4 > par6)
         {
@@ -37,16 +37,16 @@
         return this.field_111119_c;
     }
 
-    public double clampValue(double par1)
+    public double func_111109_a(double par1)
     {
-        if (par1 < this.minimumValue)
+        if (par1 < this.field_111120_a)
         {
-            par1 = this.minimumValue;
+            par1 = this.field_111120_a;
         }
 
-        if (par1 > this.maximumValue)
+        if (par1 > this.field_111118_b)
         {
-            par1 = this.maximumValue;
+            par1 = this.field_111118_b;
         }
 
         return par1;
