--- a/net/minecraft/src/RConThreadQuery.java
+++ b/net/minecraft/src/RConThreadQuery.java
@@ -137,6 +137,7 @@
             switch (var2[2])
             {
                 case 0:
+
                     if (!this.verifyClientAuth(par1DatagramPacket).booleanValue())
                     {
                         this.logDebug("Invalid challenge [" + var4 + "]");
@@ -167,7 +168,6 @@
                     this.sendAuthChallenge(par1DatagramPacket);
                     this.logDebug("Challenge [" + var4 + "]");
                     return true;
-
                 default:
                     return true;
             }
@@ -184,7 +184,7 @@
      */
     private byte[] createQueryResponse(DatagramPacket par1DatagramPacket) throws IOException
     {
-        long var2 = MinecraftServer.getSystemTimeMillis();
+        long var2 = MinecraftServer.func_130071_aq();
 
         if (var2 < this.lastQueryResponseTime + 5000L)
         {
@@ -285,7 +285,7 @@
     {
         if (this.running)
         {
-            long var1 = MinecraftServer.getSystemTimeMillis();
+            long var1 = MinecraftServer.func_130071_aq();
 
             if (var1 >= this.lastAuthCheckTime + 30000L)
             {
@@ -308,7 +308,7 @@
     public void run()
     {
         this.logInfo("Query running on " + this.serverHostname + ":" + this.queryPort);
-        this.lastAuthCheckTime = MinecraftServer.getSystemTimeMillis();
+        this.lastAuthCheckTime = MinecraftServer.func_130071_aq();
         this.incomingPacket = new DatagramPacket(this.buffer, this.buffer.length);
 
         try
