--- a/net/minecraft/src/RConThreadQuery.java
+++ b/net/minecraft/src/RConThreadQuery.java
@@ -137,6 +137,7 @@
             switch (var2[2])
             {
                 case 0:
+
                     if (!this.verifyClientAuth(par1DatagramPacket).booleanValue())
                     {
                         this.logDebug("Invalid challenge [" + var4 + "]");
@@ -167,7 +168,6 @@
                     this.sendAuthChallenge(par1DatagramPacket);
                     this.logDebug("Challenge [" + var4 + "]");
                     return true;
-
                 default:
                     return true;
             }
