--- a/net/minecraft/src/RConThreadClient.java
+++ b/net/minecraft/src/RConThreadClient.java
@@ -55,16 +55,16 @@
                 BufferedInputStream var1 = new BufferedInputStream(this.clientSocket.getInputStream());
                 int var2 = var1.read(this.buffer, 0, 1460);
 
-                if (10 <= var2)
+                if (10 > var2)
                 {
-                    byte var3 = 0;
-                    int var4 = RConUtils.getBytesAsLEInt(this.buffer, 0, var2);
+                    return;
+                }
 
-                    if (var4 != var2 - 4)
-                    {
-                        return;
-                    }
+                byte var3 = 0;
+                int var4 = RConUtils.getBytesAsLEInt(this.buffer, 0, var2);
 
+                if (var4 == var2 - 4)
+                {
                     int var21 = var3 + 4;
                     int var5 = RConUtils.getBytesAsLEInt(this.buffer, var21, var2);
                     var21 += 4;
@@ -74,6 +74,7 @@
                     switch (var6)
                     {
                         case 2:
+
                             if (this.loggedIn)
                             {
                                 String var8 = RConUtils.getBytesAsString(this.buffer, var21, var2);
@@ -92,7 +93,6 @@
 
                             this.sendLoginFailedResponse();
                             continue;
-
                         case 3:
                             String var7 = RConUtils.getBytesAsString(this.buffer, var21, var2);
                             int var10000 = var21 + var7.length();
@@ -107,7 +107,6 @@
                             this.loggedIn = false;
                             this.sendLoginFailedResponse();
                             continue;
-
                         default:
                             this.sendMultipacketResponse(var5, String.format("Unknown request %s", new Object[] {Integer.toHexString(var6)}));
                             continue;
