--- a/net/minecraft/src/RecipesDyes.java
+++ b/net/minecraft/src/RecipesDyes.java
@@ -12,7 +12,7 @@
         for (var2 = 0; var2 < 16; ++var2)
         {
             par1CraftingManager.addShapelessRecipe(new ItemStack(Block.cloth, 1, BlockColored.getDyeFromBlock(var2)), new Object[] {new ItemStack(Item.dyePowder, 1, var2), new ItemStack(Item.itemsList[Block.cloth.blockID], 1, 0)});
-            par1CraftingManager.addRecipe(new ItemStack(Block.stainedClay, 8, BlockColored.getDyeFromBlock(var2)), new Object[] {"###", "#X#", "###", '#', new ItemStack(Block.hardenedClay), 'X', new ItemStack(Item.dyePowder, 1, var2)});
+            par1CraftingManager.addRecipe(new ItemStack(Block.field_111039_cA, 8, BlockColored.getDyeFromBlock(var2)), new Object[] {"###", "#X#", "###", '#', new ItemStack(Block.field_111032_cD), 'X', new ItemStack(Item.dyePowder, 1, var2)});
         }
 
         par1CraftingManager.addShapelessRecipe(new ItemStack(Item.dyePowder, 2, 11), new Object[] {Block.plantYellow});
@@ -33,7 +33,7 @@
 
         for (var2 = 0; var2 < 16; ++var2)
         {
-            par1CraftingManager.addRecipe(new ItemStack(Block.carpet, 3, var2), new Object[] {"##", '#', new ItemStack(Block.cloth, 1, var2)});
+            par1CraftingManager.addRecipe(new ItemStack(Block.field_111031_cC, 3, var2), new Object[] {"##", '#', new ItemStack(Block.cloth, 1, var2)});
         }
     }
 }
