--- a/net/minecraft/src/Reflector.java
+++ b/net/minecraft/src/Reflector.java
@@ -4,11 +4,17 @@
 import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.List;
+
 import net.minecraft.block.Block;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiButton;
 import net.minecraft.client.gui.GuiScreen;
 import net.minecraft.client.renderer.EntityRenderer;
+import net.minecraft.client.renderer.entity.RenderPlayer;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
@@ -117,6 +123,86 @@
     public static ReflectorClass ForgeItemStack = new ReflectorClass(ItemStack.class);
     public static ReflectorMethod ForgeItemStack_hasEffect = new ReflectorMethod(ForgeItemStack, "hasEffect", new Class[] {Integer.TYPE});
 
+    /** Minecrift **/
+    public static ReflectorClass ForgeIntegratedServer = new ReflectorClass("net.minecraftforge.client.integrated.IntegratedServer");
+    public static ReflectorMethod ForgeIntegratedServer_isServerStopped = new ReflectorMethod(ForgeIntegratedServer, "isServerStopped");
+    public static ReflectorClass ForgeGuiIngame = new ReflectorClass("net.minecraftforge.client.GuiIngameForge");
+    public static ReflectorConstructor ForgeGuiIngame_Constructor = new ReflectorConstructor(ForgeGuiIngame, new Class[]{Minecraft.class});
+    public static ReflectorField ForgeGuiIngame_renderCrosshairs = new ReflectorField(ForgeGuiIngame, "renderCrosshairs");
+    public static ReflectorField ForgeGuiIngame_renderHelmet = new ReflectorField(ForgeGuiIngame, "renderHelmet");
+    public static ReflectorMethod FMLClientHandler_beginMinecraftLoading = new ReflectorMethod(FMLClientHandler, "beginMinecraftLoading");
+    public static ReflectorMethod FMLClientHandler_finishMinecraftLoading = new ReflectorMethod(FMLClientHandler, "finishMinecraftLoading");
+    public static ReflectorMethod FMLClientHandler_connectToServerAtStartup = new ReflectorMethod(FMLClientHandler, "connectToServerAtStartup");
+    public static ReflectorMethod FMLClientHandler_onInitializationComplete = new ReflectorMethod(FMLClientHandler, "onInitializationComplete");
+    public static ReflectorMethod FMLClientHandler_startIntegratedServer = new ReflectorMethod(FMLClientHandler, "startIntegratedServer");
+    public static ReflectorMethod FMLClientHandler_handleClientWorldClosing = new ReflectorMethod(FMLClientHandler, "handleClientWorldClosing");
+    public static ReflectorMethod FMLClientHandler_handleLoadingScreen = new ReflectorMethod(FMLClientHandler, "handleLoadingScreen");
+    public static ReflectorMethod FMLCommonHandler_handleExit = new ReflectorMethod(FMLCommonHandler, "handleExit");
+    public static ReflectorMethod FMLCommonHandler_onRenderTickStart = new ReflectorMethod(FMLCommonHandler, "onRenderTickStart");
+    public static ReflectorMethod FMLCommonHandler_onRenderTickEnd = new ReflectorMethod(FMLCommonHandler, "onRenderTickEnd");
+    public static ReflectorMethod FMLCommonHandler_onPreClientTick = new ReflectorMethod(FMLCommonHandler, "onPreClientTick");
+    public static ReflectorMethod FMLCommonHandler_onPostClientTick = new ReflectorMethod(FMLCommonHandler, "onPostClientTick");
+    public static ReflectorMethod FMLCommonHandler_fireMouseInput = new ReflectorMethod(FMLCommonHandler, "fireMouseInput");
+    public static ReflectorMethod FMLCommonHandler_fireKeyInput = new ReflectorMethod(FMLCommonHandler, "fireKeyInput");
+    public static ReflectorMethod FMLCommonHandler_getBrandings = new ReflectorMethod(FMLCommonHandler, "getBrandings");
+    public static ReflectorClass FMLGuiModList = new ReflectorClass("cpw.mods.fml.client.GuiModList");
+    public static ReflectorConstructor FMLGuiModList_Constructor = new ReflectorConstructor(FMLGuiModList, new Class[]{GuiScreen.class});
+    public static ReflectorClass FMLStartupQuery = new ReflectorClass("cpw.mods.fml.common.StartupQuery");
+    public static ReflectorMethod FMLStartupQuery_check = new ReflectorMethod(FMLStartupQuery, "check");
+    public static ReflectorClass ForgeActionPerformedEvent_Pre = new ReflectorClass("net.minecraftforge.client.event.GuiScreenEvent$ActionPerformedEvent$Pre");
+    public static ReflectorConstructor ForgeActionPerformedEvent_Pre_Constructor = new ReflectorConstructor(ForgeActionPerformedEvent_Pre, new Class[] {GuiScreen.class, GuiButton.class, List.class});
+    public static ReflectorField ForgeActionPerformedEvent_Pre_button = new ReflectorField(ForgeActionPerformedEvent_Pre, "button");
+    public static ReflectorClass ForgeActionPerformedEvent_Post = new ReflectorClass("net.minecraftforge.client.event.GuiScreenEvent$ActionPerformedEvent$Post");
+    public static ReflectorConstructor ForgeActionPerformedEvent_Post_Constructor = new ReflectorConstructor(ForgeActionPerformedEvent_Post, new Class[] {GuiScreen.class, GuiButton.class, List.class});
+    public static ReflectorMethod ForgeHooks_onPickBlock = new ReflectorMethod(ForgeHooks, "onPickBlock");
+    public static ReflectorMethod ForgeHooksClient_postMouseEvent = new ReflectorMethod(ForgeHooksClient, "postMouseEvent");
+    public static ReflectorMethod ForgeHooksClient_createDisplay = new ReflectorMethod(ForgeHooksClient, "createDisplay");
+    public static ReflectorMethod ForgeHooksClient_getArmorModel = new ReflectorMethod(ForgeHooksClient, "getArmorModel");
+    public static ReflectorMethod ForgeHooksClient_renderMainMenu = new ReflectorMethod(ForgeHooksClient, "renderMainMenu");
+    public static ReflectorClass ForgeEventFactory = new ReflectorClass("net.minecraftforge.event.ForgeEventFactory");
+    public static ReflectorMethod ForgeEventFactory_onPlayerInteract = new ReflectorMethod(ForgeEventFactory, "onPlayerInteract");
+    public static ReflectorClass ForgeInitGuiEvent_Pre = new ReflectorClass("net.minecraftforge.client.event.GuiScreenEvent$InitGuiEvent$Pre");
+    public static ReflectorConstructor ForgeInitGuiEvent_Pre_Constructor = new ReflectorConstructor(ForgeInitGuiEvent_Pre, new Class[] {GuiScreen.class, List.class});
+    public static ReflectorClass ForgeInitGuiEvent_Post = new ReflectorClass("net.minecraftforge.client.event.GuiScreenEvent$InitGuiEvent$Post");
+    public static ReflectorConstructor ForgeInitGuiEvent_Post_Constructor = new ReflectorConstructor(ForgeInitGuiEvent_Post, new Class[] {GuiScreen.class, List.class});
+    public static ReflectorClass ForgeGuiOpenEvent = new ReflectorClass("net.minecraftforge.client.event.GuiOpenEvent");
+    public static ReflectorConstructor ForgeGuiOpenEvent_Constructor = new ReflectorConstructor(ForgeGuiOpenEvent, new Class[] {GuiScreen.class});
+    public static ReflectorField ForgeGuiOpenEvent_gui = new ReflectorField(ForgeGuiOpenEvent, "gui");
+    public static ReflectorClass ForgeRenderPlayerEvent_Pre = new ReflectorClass("net.minecraftforge.client.event.RenderPlayerEvent$Pre");
+    public static ReflectorConstructor ForgeRenderPlayerEvent_Pre_Constructor = new ReflectorConstructor(ForgeRenderPlayerEvent_Pre, new Class[] {EntityPlayer.class, RenderPlayer.class, Float.TYPE});
+    public static ReflectorClass ForgeRenderPlayerEvent_Post = new ReflectorClass("net.minecraftforge.client.event.RenderPlayerEvent$Post");
+    public static ReflectorConstructor ForgeRenderPlayerEvent_Post_Constructor = new ReflectorConstructor(ForgeRenderPlayerEvent_Post, new Class[] {EntityPlayer.class, RenderPlayer.class, Float.TYPE});
+    public static ReflectorClass ForgeRenderPlayerEvent_Specials_Pre = new ReflectorClass("net.minecraftforge.client.event.RenderPlayerEvent$Specials$Pre");
+    public static ReflectorConstructor ForgeRenderPlayerEvent_Specials_Pre_Constructor = new ReflectorConstructor(ForgeRenderPlayerEvent_Specials_Pre, new Class[] {EntityPlayer.class, RenderPlayer.class, Float.TYPE});
+    public static ReflectorField ForgeRenderPlayerEvent_Specials_Pre_renderCape = new ReflectorField(ForgeRenderPlayerEvent_Specials_Pre, "renderCape");
+    public static ReflectorField ForgeRenderPlayerEvent_Specials_Pre_renderItem = new ReflectorField(ForgeRenderPlayerEvent_Specials_Pre, "renderItem");
+    public static ReflectorClass ForgeRenderPlayerEvent_Specials_Post = new ReflectorClass("net.minecraftforge.client.event.RenderPlayerEvent$Specials$Post");
+    public static ReflectorConstructor ForgeRenderPlayerEvent_Specials_Post_Constructor = new ReflectorConstructor(ForgeRenderPlayerEvent_Specials_Post, new Class[] {EntityPlayer.class, RenderPlayer.class, Float.TYPE});
+    public static ReflectorClass ForgeRenderPlayerEvent_SetArmorModel = new ReflectorClass("net.minecraftforge.client.event.RenderPlayerEvent$SetArmorModel");
+    public static ReflectorConstructor ForgeRenderPlayerEvent_SetArmorModel_Constructor = new ReflectorConstructor(ForgeRenderPlayerEvent_SetArmorModel, new Class[] {EntityPlayer.class, RenderPlayer.class, Integer.TYPE, Float.TYPE, ItemStack.class});
+    public static ReflectorField ForgeRenderPlayerEvent_SetArmorModel_result = new ReflectorField(ForgeRenderPlayerEvent_SetArmorModel, "result");
+    public static ReflectorClass ForgeIItemRenderer = new ReflectorClass("net.minecraftforge.client.IItemRenderer");
+    public static ReflectorMethod ForgeIItemRenderer_shouldUseRenderHelper = new ReflectorMethod(ForgeIItemRenderer, "shouldUseRenderHelper");
+    public static ReflectorClass ForgeItemRendererHelper = new ReflectorClass("net.minecraftforge.client.IItemRenderer$ItemRendererHelper");
+    public static ReflectorField ForgeItemRendererHelper_BLOCK_3D = new ReflectorField(ForgeItemRendererHelper, "BLOCK_3D");
+    public static ReflectorClass ForgePlayerInteractEvent = new ReflectorClass("net.minecraftforge.event.entity.player.PlayerInteractEvent");
+    public static ReflectorMethod ForgePlayerInteractEvent_isCanceled = new ReflectorMethod(ForgePlayerInteractEvent, "isCanceled");
+    public static ReflectorClass ForgePlayerInteractEvent_Action = new ReflectorClass("net.minecraftforge.event.entity.player.PlayerInteractEvent$Action");
+    public static ReflectorField ForgePlayerInteractEvent_Action_RIGHT_CLICK_BLOCK = new ReflectorField(ForgePlayerInteractEvent_Action, "RIGHT_CLICK_BLOCK");
+    public static ReflectorField ForgePlayerInteractEvent_Action_RIGHT_CLICK_AIR = new ReflectorField(ForgePlayerInteractEvent_Action, "RIGHT_CLICK_AIR");
+    public static ReflectorClass ForgeWorldEvent_Unload = new ReflectorClass("net.minecraftforge.event.world.WorldEvent$Unload");
+    public static ReflectorConstructor ForgeWorldEvent_Unload_Constructor = new ReflectorConstructor(ForgeWorldEvent_Unload, new Class[]{World.class});
+    public static ReflectorMethod MinecraftForgeClient_getStencilBits = new ReflectorMethod(MinecraftForgeClient, "getStencilBits");
+    public static ReflectorClass MinecraftItem = new ReflectorClass(Item.class);
+    public static ReflectorMethod MinecraftItem_Forge_getFontRenderer = new ReflectorMethod(MinecraftItem, "getFontRenderer");
+    public static ReflectorMethod MinecraftItem_Forge_getRenderPasses = new ReflectorMethod(MinecraftItem, "getRenderPasses");
+
+    public static boolean forgeExists()
+    {
+        return EventBus.exists();
+    }
+    /** end Minecrift **/
+
     public static void callVoid(ReflectorMethod refMethod, Object ... params)
     {
         try
