--- a/net/minecraft/src/ReflectorMethod.java
+++ b/net/minecraft/src/ReflectorMethod.java
@@ -53,7 +53,7 @@
                 {
                     if (i >= ms.length)
                     {
-                        Config.log("(Reflector) Method not pesent: " + cls.getName() + "." + this.targetMethodName);
+                        Config.log("(Reflector) Method not present: " + cls.getName() + "." + this.targetMethodName);
                         return null;
                     }
 
