--- a/net/minecraft/src/RegistryDefaulted.java
+++ b/net/minecraft/src/RegistryDefaulted.java
@@ -12,9 +12,9 @@
         this.defaultObject = par1Obj;
     }
 
-    public Object getObject(Object par1Obj)
+    public Object func_82594_a(Object par1Obj)
     {
-        Object var2 = super.getObject(par1Obj);
+        Object var2 = super.func_82594_a(par1Obj);
         return var2 == null ? this.defaultObject : var2;
     }
 }
