--- a/net/minecraft/src/RenderVillager.java
+++ b/net/minecraft/src/RenderVillager.java
@@ -39,19 +39,14 @@
         {
             case 0:
                 return field_110904_g;
-
             case 1:
                 return field_110908_h;
-
             case 2:
                 return field_110907_k;
-
             case 3:
                 return field_110905_l;
-
             case 4:
                 return field_110906_m;
-
             default:
                 return field_110903_f;
         }
