--- a/net/minecraft/src/RenderVillager.java
+++ b/net/minecraft/src/RenderVillager.java
@@ -4,12 +4,12 @@
 
 public class RenderVillager extends RenderLiving
 {
-    private static final ResourceLocation villagerTextures = new ResourceLocation("textures/entity/villager/villager.png");
-    private static final ResourceLocation farmerVillagerTextures = new ResourceLocation("textures/entity/villager/farmer.png");
-    private static final ResourceLocation librarianVillagerTextures = new ResourceLocation("textures/entity/villager/librarian.png");
-    private static final ResourceLocation priestVillagerTextures = new ResourceLocation("textures/entity/villager/priest.png");
-    private static final ResourceLocation smithVillagerTextures = new ResourceLocation("textures/entity/villager/smith.png");
-    private static final ResourceLocation butcherVillagerTextures = new ResourceLocation("textures/entity/villager/butcher.png");
+    private static final ResourceLocation field_110903_f = new ResourceLocation("textures/entity/villager/villager.png");
+    private static final ResourceLocation field_110904_g = new ResourceLocation("textures/entity/villager/farmer.png");
+    private static final ResourceLocation field_110908_h = new ResourceLocation("textures/entity/villager/librarian.png");
+    private static final ResourceLocation field_110907_k = new ResourceLocation("textures/entity/villager/priest.png");
+    private static final ResourceLocation field_110905_l = new ResourceLocation("textures/entity/villager/smith.png");
+    private static final ResourceLocation field_110906_m = new ResourceLocation("textures/entity/villager/butcher.png");
 
     /** Model of the villager. */
     protected ModelVillager villagerModel;
@@ -38,22 +38,17 @@
         switch (par1EntityVillager.getProfession())
         {
             case 0:
-                return farmerVillagerTextures;
-
+                return field_110904_g;
             case 1:
-                return librarianVillagerTextures;
-
+                return field_110908_h;
             case 2:
-                return priestVillagerTextures;
-
+                return field_110907_k;
             case 3:
-                return smithVillagerTextures;
-
+                return field_110905_l;
             case 4:
-                return butcherVillagerTextures;
-
+                return field_110906_m;
             default:
-                return villagerTextures;
+                return field_110903_f;
         }
     }
 
@@ -111,10 +106,7 @@
         this.renderVillager((EntityVillager)par1EntityLivingBase, par2, par4, par6, par8, par9);
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
         return this.func_110902_a((EntityVillager)par1Entity);
     }
