--- a/net/minecraft/src/RenderEntity.java
+++ b/net/minecraft/src/RenderEntity.java
@@ -17,10 +17,7 @@
         GL11.glPopMatrix();
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
         return null;
     }
