--- a/net/minecraft/src/RenderItemFrame.java
+++ b/net/minecraft/src/RenderItemFrame.java
@@ -93,11 +93,9 @@
                 case 1:
                     GL11.glTranslatef(-0.16F, -0.16F, 0.0F);
                     break;
-
                 case 2:
                     GL11.glTranslatef(0.0F, -0.32F, 0.0F);
                     break;
-
                 case 3:
                     GL11.glTranslatef(0.16F, -0.16F, 0.0F);
             }
