--- a/net/minecraft/src/RenderGlobal.java
+++ b/net/minecraft/src/RenderGlobal.java
@@ -339,12 +339,14 @@
                 var1 = var3;
             }
 
+            // TODO: Set chunk renderer attribs for greater draw distance
             this.prevReposX = -9999.0D;
             this.prevReposY = -9999.0D;
             this.prevReposZ = -9999.0D;
             this.renderChunksWide = var1 / 16 + 1;
             this.renderChunksTall = 16;
             this.renderChunksDeep = var1 / 16 + 1;
+
             this.worldRenderers = new WorldRenderer[this.renderChunksWide * this.renderChunksTall * this.renderChunksDeep];
             this.sortedWorldRenderers = new WorldRenderer[this.renderChunksWide * this.renderChunksTall * this.renderChunksDeep];
             int var4 = 0;
@@ -470,20 +472,30 @@
             for (var6 = 0; var6 < var5.size(); ++var6)
             {
                 var7 = (Entity)var5.get(var6);
-                boolean var9 = var7.isInRangeToRenderVec3D(par1Vec3) && (var7.ignoreFrustumCheck || par2ICamera.isBoundingBoxInFrustum(var7.boundingBox) || var7.riddenByEntity == this.mc.thePlayer);
+                boolean render = var7.isInRangeToRenderVec3D(par1Vec3) && (var7.ignoreFrustumCheck || par2ICamera.isBoundingBoxInFrustum(var7.boundingBox) || var7.riddenByEntity == this.mc.thePlayer);
 
-                if (!var9 && var7 instanceof EntityLiving)
+                if (!render && var7 instanceof EntityLiving)
                 {
                     EntityLiving var10 = (EntityLiving)var7;
 
                     if (var10.func_110167_bD() && var10.func_110166_bE() != null)
                     {
                         Entity var11 = var10.func_110166_bE();
-                        var9 = par2ICamera.isBoundingBoxInFrustum(var11.boundingBox);
+                        render = par2ICamera.isBoundingBoxInFrustum(var11.boundingBox);
                     }
                 }
 
-                if (var9 && (var7 != this.mc.renderViewEntity || this.mc.gameSettings.thirdPersonView != 0 || this.mc.renderViewEntity.isPlayerSleeping()) && this.theWorld.blockExists(MathHelper.floor_double(var7.posX), 0, MathHelper.floor_double(var7.posZ)))
+
+                RenderPlayer renderPlayer = (RenderPlayer)RenderManager.instance.getEntityRenderObject(this.mc.thePlayer);
+
+                if (var7 == this.mc.renderViewEntity && this.mc.gameSettings.thirdPersonView == 0 && !mc.renderViewEntity.isPlayerSleeping() )
+                {
+                	renderPlayer.setVR(true,this.mc.gameSettings.renderHeadWear, this.mc.gameSettings.renderPlayerOffset);
+                    if (this.mc.gameSettings.renderFullFirstPersonModel == false)
+                        render = false;
+                }
+
+                if (render)
                 {
                     ++this.countEntitiesRendered;
 
@@ -496,6 +508,11 @@
                     RenderManager.instance.renderEntity(var7, par3);
                     this.renderedEntity = null;
                 }
+
+                if (var7 == this.mc.renderViewEntity && this.mc.gameSettings.thirdPersonView == 0 && !mc.renderViewEntity.isPlayerSleeping() )
+                {
+                	renderPlayer.setVR(false,true,0.0F);
+                }
             }
 
             this.mc.gameSettings.fancyGraphics = var8;
@@ -2289,6 +2306,7 @@
         {
             case 1013:
             case 1018:
+
                 if (this.mc.renderViewEntity != null)
                 {
                     double var7 = (double)par2 - this.mc.renderViewEntity.posX;
@@ -2343,16 +2361,14 @@
             case 1000:
                 this.theWorld.playSound((double)par3, (double)par4, (double)par5, "random.click", 1.0F, 1.0F, false);
                 break;
-
             case 1001:
                 this.theWorld.playSound((double)par3, (double)par4, (double)par5, "random.click", 1.0F, 1.2F, false);
                 break;
-
             case 1002:
                 this.theWorld.playSound((double)par3, (double)par4, (double)par5, "random.bow", 1.0F, 1.2F, false);
                 break;
-
             case 1003:
+
                 if (Math.random() < 0.5D)
                 {
                     this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "random.door_open", 1.0F, this.theWorld.rand.nextFloat() * 0.1F + 0.9F, false);
@@ -2363,12 +2379,11 @@
                 }
 
                 break;
-
             case 1004:
                 this.theWorld.playSound((double)((float)par3 + 0.5F), (double)((float)par4 + 0.5F), (double)((float)par5 + 0.5F), "random.fizz", 0.5F, 2.6F + (var7.nextFloat() - var7.nextFloat()) * 0.8F, false);
                 break;
-
             case 1005:
+
                 if (Item.itemsList[par6] instanceof ItemRecord)
                 {
                     this.theWorld.playRecord(((ItemRecord)Item.itemsList[par6]).recordName, par3, par4, par5);
@@ -2379,59 +2394,45 @@
                 }
 
                 break;
-
             case 1007:
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "mob.ghast.charge", 10.0F, (var7.nextFloat() - var7.nextFloat()) * 0.2F + 1.0F, false);
                 break;
-
             case 1008:
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "mob.ghast.fireball", 10.0F, (var7.nextFloat() - var7.nextFloat()) * 0.2F + 1.0F, false);
                 break;
-
             case 1009:
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "mob.ghast.fireball", 2.0F, (var7.nextFloat() - var7.nextFloat()) * 0.2F + 1.0F, false);
                 break;
-
             case 1010:
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "mob.zombie.wood", 2.0F, (var7.nextFloat() - var7.nextFloat()) * 0.2F + 1.0F, false);
                 break;
-
             case 1011:
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "mob.zombie.metal", 2.0F, (var7.nextFloat() - var7.nextFloat()) * 0.2F + 1.0F, false);
                 break;
-
             case 1012:
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "mob.zombie.woodbreak", 2.0F, (var7.nextFloat() - var7.nextFloat()) * 0.2F + 1.0F, false);
                 break;
-
             case 1014:
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "mob.wither.shoot", 2.0F, (var7.nextFloat() - var7.nextFloat()) * 0.2F + 1.0F, false);
                 break;
-
             case 1015:
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "mob.bat.takeoff", 0.05F, (var7.nextFloat() - var7.nextFloat()) * 0.2F + 1.0F, false);
                 break;
-
             case 1016:
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "mob.zombie.infect", 2.0F, (var7.nextFloat() - var7.nextFloat()) * 0.2F + 1.0F, false);
                 break;
-
             case 1017:
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "mob.zombie.unfect", 2.0F, (var7.nextFloat() - var7.nextFloat()) * 0.2F + 1.0F, false);
                 break;
-
             case 1020:
                 this.theWorld.playSound((double)((float)par3 + 0.5F), (double)((float)par4 + 0.5F), (double)((float)par5 + 0.5F), "random.anvil_break", 1.0F, this.theWorld.rand.nextFloat() * 0.1F + 0.9F, false);
                 break;
-
             case 1021:
                 this.theWorld.playSound((double)((float)par3 + 0.5F), (double)((float)par4 + 0.5F), (double)((float)par5 + 0.5F), "random.anvil_use", 1.0F, this.theWorld.rand.nextFloat() * 0.1F + 0.9F, false);
                 break;
-
             case 1022:
                 this.theWorld.playSound((double)((float)par3 + 0.5F), (double)((float)par4 + 0.5F), (double)((float)par5 + 0.5F), "random.anvil_land", 0.3F, this.theWorld.rand.nextFloat() * 0.1F + 0.9F, false);
                 break;
-
             case 2000:
                 int var27 = par6 % 3 - 1;
                 int var28 = par6 / 3 % 3 - 1;
@@ -2452,7 +2453,6 @@
                 }
 
                 return;
-
             case 2001:
                 var16 = par6 & 4095;
 
@@ -2464,7 +2464,6 @@
 
                 this.mc.effectRenderer.addBlockDestroyEffects(par3, par4, par5, par6 & 4095, par6 >> 12 & 255);
                 break;
-
             case 2002:
                 var8 = (double)par3;
                 var10 = (double)par4;
@@ -2506,7 +2505,6 @@
 
                 this.theWorld.playSound((double)par3 + 0.5D, (double)par4 + 0.5D, (double)par5 + 0.5D, "random.glass", 1.0F, this.theWorld.rand.nextFloat() * 0.1F + 0.9F, false);
                 break;
-
             case 2003:
                 var8 = (double)par3 + 0.5D;
                 var10 = (double)par4;
@@ -2525,8 +2523,8 @@
                 }
 
                 return;
-
             case 2004:
+
                 for (int var35 = 0; var35 < 20; ++var35)
                 {
                     double var36 = (double)par3 + 0.5D + ((double)this.theWorld.rand.nextFloat() - 0.5D) * 2.0D;
@@ -2537,7 +2535,6 @@
                 }
 
                 return;
-
             case 2005:
                 ItemDye.func_96603_a(this.theWorld, par3, par4, par5, par6);
         }
