--- a/net/minecraft/src/RendererLivingEntity.java
+++ b/net/minecraft/src/RendererLivingEntity.java
@@ -6,7 +6,7 @@
 
 public abstract class RendererLivingEntity extends Render
 {
-    private static final ResourceLocation RES_ITEM_GLINT = new ResourceLocation("textures/misc/enchanted_item_glint.png");
+    private static final ResourceLocation field_110814_a = new ResourceLocation("textures/misc/enchanted_item_glint.png");
     protected ModelBase mainModel;
 
     /** The model to be used during the render passes. */
@@ -49,7 +49,7 @@
         return par1 + par3 * var4;
     }
 
-    public void doRenderLiving(EntityLivingBase par1EntityLivingBase, double par2, double par4, double par6, float par8, float par9)
+    public void func_130000_a(EntityLivingBase par1EntityLivingBase, double par2, double par4, double par6, float par8, float par9)
     {
         GL11.glPushMatrix();
         GL11.glDisable(GL11.GL_CULL_FACE);
@@ -113,8 +113,8 @@
             GL11.glScalef(-1.0F, -1.0F, 1.0F);
             this.preRenderCallback(par1EntityLivingBase, par9);
             GL11.glTranslatef(0.0F, -24.0F * var14 - 0.0078125F, 0.0F);
-            float var15 = par1EntityLivingBase.prevLimbSwingAmount + (par1EntityLivingBase.limbSwingAmount - par1EntityLivingBase.prevLimbSwingAmount) * par9;
-            float var16 = par1EntityLivingBase.limbSwing - par1EntityLivingBase.limbSwingAmount * (1.0F - par9);
+            float var15 = par1EntityLivingBase.prevLimbYaw + (par1EntityLivingBase.limbYaw - par1EntityLivingBase.prevLimbYaw) * par9;
+            float var16 = par1EntityLivingBase.limbSwing - par1EntityLivingBase.limbYaw * (1.0F - par9);
 
             if (par1EntityLivingBase.isChild())
             {
@@ -152,7 +152,7 @@
                     if ((var18 & 15) == 15)
                     {
                         var19 = (float)par1EntityLivingBase.ticksExisted + par9;
-                        this.bindTexture(RES_ITEM_GLINT);
+                        this.func_110776_a(field_110814_a);
                         GL11.glEnable(GL11.GL_BLEND);
                         var20 = 0.5F;
                         GL11.glColor4f(var20, var20, var20, 1.0F);
@@ -267,13 +267,13 @@
      */
     protected void renderModel(EntityLivingBase par1EntityLivingBase, float par2, float par3, float par4, float par5, float par6, float par7)
     {
-        this.bindEntityTexture(par1EntityLivingBase);
+        this.func_110777_b(par1EntityLivingBase);
 
         if (!par1EntityLivingBase.isInvisible())
         {
             this.mainModel.render(par1EntityLivingBase, par2, par3, par4, par5, par6, par7);
         }
-        else if (!par1EntityLivingBase.isInvisibleToPlayer(Minecraft.getMinecraft().thePlayer))
+        else if (!par1EntityLivingBase.func_98034_c(Minecraft.getMinecraft().thePlayer))
         {
             GL11.glPushMatrix();
             GL11.glColor4f(1.0F, 1.0F, 1.0F, 0.15F);
@@ -317,15 +317,10 @@
 
             GL11.glRotatef(var5 * this.getDeathMaxRotation(par1EntityLivingBase), 0.0F, 0.0F, 1.0F);
         }
-        else
+        else if ((par1EntityLivingBase.getEntityName().equals("Dinnerbone") || par1EntityLivingBase.getEntityName().equals("Grumm")) && (!(par1EntityLivingBase instanceof EntityPlayer) || !((EntityPlayer)par1EntityLivingBase).getHideCape()))
         {
-            String var6 = EnumChatFormatting.func_110646_a(par1EntityLivingBase.getEntityName());
-
-            if ((var6.equals("Dinnerbone") || var6.equals("Grumm")) && (!(par1EntityLivingBase instanceof EntityPlayer) || !((EntityPlayer)par1EntityLivingBase).getHideCape()))
-            {
-                GL11.glTranslatef(0.0F, par1EntityLivingBase.height + 0.1F, 0.0F);
-                GL11.glRotatef(180.0F, 0.0F, 0.0F, 1.0F);
-            }
+            GL11.glTranslatef(0.0F, par1EntityLivingBase.height + 0.1F, 0.0F);
+            GL11.glRotatef(180.0F, 0.0F, 0.0F, 1.0F);
         }
     }
 
@@ -483,7 +478,7 @@
 
     protected boolean func_110813_b(EntityLivingBase par1EntityLivingBase)
     {
-        return Minecraft.isGuiEnabled() && par1EntityLivingBase != this.renderManager.livingPlayer && !par1EntityLivingBase.isInvisibleToPlayer(Minecraft.getMinecraft().thePlayer) && par1EntityLivingBase.riddenByEntity == null;
+        return Minecraft.isGuiEnabled() && par1EntityLivingBase != this.renderManager.livingPlayer && !par1EntityLivingBase.func_98034_c(Minecraft.getMinecraft().thePlayer) && par1EntityLivingBase.riddenByEntity == null;
     }
 
     protected void func_96449_a(EntityLivingBase par1EntityLivingBase, double par2, double par4, double par6, String par8Str, float par9, double par10)
@@ -558,6 +553,6 @@
      */
     public void doRender(Entity par1Entity, double par2, double par4, double par6, float par8, float par9)
     {
-        this.doRenderLiving((EntityLivingBase)par1Entity, par2, par4, par6, par8, par9);
+        this.func_130000_a((EntityLivingBase)par1Entity, par2, par4, par6, par8, par9);
     }
 }
