--- a/net/minecraft/src/RenderItemFrame.java
+++ b/net/minecraft/src/RenderItemFrame.java
@@ -4,7 +4,7 @@
 
 public class RenderItemFrame extends Render
 {
-    private static final ResourceLocation mapBackgroundTextures = new ResourceLocation("textures/map/map_background.png");
+    private static final ResourceLocation field_110789_a = new ResourceLocation("textures/map/map_background.png");
     private final RenderBlocks renderBlocksInstance = new RenderBlocks();
     private Icon field_94147_f;
 
@@ -40,7 +40,7 @@
     {
         GL11.glPushMatrix();
         GL11.glRotatef(par1EntityItemFrame.rotationYaw, 0.0F, 1.0F, 0.0F);
-        this.renderManager.renderEngine.bindTexture(TextureMap.locationBlocksTexture);
+        this.renderManager.renderEngine.func_110577_a(TextureMap.field_110575_b);
         Block var2 = Block.planks;
         float var3 = 0.0625F;
         float var4 = 0.75F;
@@ -93,18 +93,16 @@
                 case 1:
                     GL11.glTranslatef(-0.16F, -0.16F, 0.0F);
                     break;
-
                 case 2:
                     GL11.glTranslatef(0.0F, -0.32F, 0.0F);
                     break;
-
                 case 3:
                     GL11.glTranslatef(0.16F, -0.16F, 0.0F);
             }
 
             if (var3.getEntityItem().getItem() == Item.map)
             {
-                this.renderManager.renderEngine.bindTexture(mapBackgroundTextures);
+                this.renderManager.renderEngine.func_110577_a(field_110789_a);
                 Tessellator var4 = Tessellator.instance;
                 GL11.glRotatef(180.0F, 0.0F, 1.0F, 0.0F);
                 GL11.glRotatef(180.0F, 0.0F, 0.0F, 1.0F);
@@ -130,9 +128,9 @@
             {
                 if (var3.getEntityItem().getItem() == Item.compass)
                 {
-                    TextureManager var11 = Minecraft.getMinecraft().getTextureManager();
-                    var11.bindTexture(TextureMap.locationItemsTexture);
-                    TextureAtlasSprite var13 = ((TextureMap)var11.getTexture(TextureMap.locationItemsTexture)).getAtlasSprite(Item.compass.getIconIndex(var3.getEntityItem()).getIconName());
+                    TextureManager var11 = Minecraft.getMinecraft().func_110434_K();
+                    var11.func_110577_a(TextureMap.field_110576_c);
+                    TextureAtlasSprite var13 = ((TextureMap)var11.func_110581_b(TextureMap.field_110576_c)).func_110572_b(Item.compass.getIconIndex(var3.getEntityItem()).getIconName());
 
                     if (var13 instanceof TextureCompass)
                     {
@@ -153,9 +151,9 @@
 
                 if (var3.getEntityItem().getItem() == Item.compass)
                 {
-                    TextureAtlasSprite var12 = ((TextureMap)Minecraft.getMinecraft().getTextureManager().getTexture(TextureMap.locationItemsTexture)).getAtlasSprite(Item.compass.getIconIndex(var3.getEntityItem()).getIconName());
+                    TextureAtlasSprite var12 = ((TextureMap)Minecraft.getMinecraft().func_110434_K().func_110581_b(TextureMap.field_110576_c)).func_110572_b(Item.compass.getIconIndex(var3.getEntityItem()).getIconName());
 
-                    if (var12.getFrameCount() > 0)
+                    if (var12.func_110970_k() > 0)
                     {
                         var12.updateAnimation();
                     }
@@ -166,10 +164,7 @@
         }
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
         return this.func_110788_a((EntityItemFrame)par1Entity);
     }
