--- a/net/minecraft/src/RenderTNTPrimed.java
+++ b/net/minecraft/src/RenderTNTPrimed.java
@@ -38,7 +38,7 @@
         }
 
         var10 = (1.0F - ((float)par1EntityTNTPrimed.fuse - par9 + 1.0F) / 100.0F) * 0.8F;
-        this.bindEntityTexture(par1EntityTNTPrimed);
+        this.func_110777_b(par1EntityTNTPrimed);
         this.blockRenderer.renderBlockAsItem(Block.tnt, 0, par1EntityTNTPrimed.getBrightness(par9));
 
         if (par1EntityTNTPrimed.fuse / 5 % 2 == 0)
@@ -60,13 +60,10 @@
 
     protected ResourceLocation func_110808_a(EntityTNTPrimed par1EntityTNTPrimed)
     {
-        return TextureMap.locationBlocksTexture;
+        return TextureMap.field_110575_b;
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
         return this.func_110808_a((EntityTNTPrimed)par1Entity);
     }
