--- a/net/minecraft/src/RenderSilverfish.java
+++ b/net/minecraft/src/RenderSilverfish.java
@@ -2,7 +2,7 @@
 
 public class RenderSilverfish extends RenderLiving
 {
-    private static final ResourceLocation silverfishTextures = new ResourceLocation("textures/entity/silverfish.png");
+    private static final ResourceLocation field_110882_a = new ResourceLocation("textures/entity/silverfish.png");
 
     public RenderSilverfish()
     {
@@ -25,9 +25,9 @@
         super.doRenderLiving(par1EntitySilverfish, par2, par4, par6, par8, par9);
     }
 
-    protected ResourceLocation getSilverfishTextures(EntitySilverfish par1EntitySilverfish)
+    protected ResourceLocation func_110881_b(EntitySilverfish par1EntitySilverfish)
     {
-        return silverfishTextures;
+        return field_110882_a;
     }
 
     /**
@@ -61,12 +61,9 @@
         this.renderSilverfish((EntitySilverfish)par1EntityLivingBase, par2, par4, par6, par8, par9);
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
-        return this.getSilverfishTextures((EntitySilverfish)par1Entity);
+        return this.func_110881_b((EntitySilverfish)par1Entity);
     }
 
     /**
