--- a/net/minecraft/src/RenderSpider.java
+++ b/net/minecraft/src/RenderSpider.java
@@ -4,8 +4,8 @@
 
 public class RenderSpider extends RenderLiving
 {
-    private static final ResourceLocation spiderEyesTextures = new ResourceLocation("textures/entity/spider_eyes.png");
-    private static final ResourceLocation spiderTextures = new ResourceLocation("textures/entity/spider/spider.png");
+    private static final ResourceLocation field_110891_a = new ResourceLocation("textures/entity/spider_eyes.png");
+    private static final ResourceLocation field_110890_f = new ResourceLocation("textures/entity/spider/spider.png");
 
     public RenderSpider()
     {
@@ -29,7 +29,7 @@
         }
         else
         {
-            this.bindTexture(spiderEyesTextures);
+            this.func_110776_a(field_110891_a);
             float var4 = 1.0F;
             GL11.glEnable(GL11.GL_BLEND);
             GL11.glDisable(GL11.GL_ALPHA_TEST);
@@ -54,9 +54,9 @@
         }
     }
 
-    protected ResourceLocation getSpiderTextures(EntitySpider par1EntitySpider)
+    protected ResourceLocation func_110889_a(EntitySpider par1EntitySpider)
     {
-        return spiderTextures;
+        return field_110890_f;
     }
 
     protected float getDeathMaxRotation(EntityLivingBase par1EntityLivingBase)
@@ -72,11 +72,8 @@
         return this.setSpiderEyeBrightness((EntitySpider)par1EntityLivingBase, par2, par3);
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
-        return this.getSpiderTextures((EntitySpider)par1Entity);
+        return this.func_110889_a((EntitySpider)par1Entity);
     }
 }
