--- a/net/minecraft/src/RenderChicken.java
+++ b/net/minecraft/src/RenderChicken.java
@@ -2,7 +2,7 @@
 
 public class RenderChicken extends RenderLiving
 {
-    private static final ResourceLocation chickenTextures = new ResourceLocation("textures/entity/chicken.png");
+    private static final ResourceLocation field_110920_a = new ResourceLocation("textures/entity/chicken.png");
 
     public RenderChicken(ModelBase par1ModelBase, float par2)
     {
@@ -14,9 +14,9 @@
         super.doRenderLiving(par1EntityChicken, par2, par4, par6, par8, par9);
     }
 
-    protected ResourceLocation getChickenTextures(EntityChicken par1EntityChicken)
+    protected ResourceLocation func_110919_a(EntityChicken par1EntityChicken)
     {
-        return chickenTextures;
+        return field_110920_a;
     }
 
     protected float getWingRotation(EntityChicken par1EntityChicken, float par2)
@@ -44,12 +44,9 @@
         this.renderChicken((EntityChicken)par1EntityLivingBase, par2, par4, par6, par8, par9);
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
-        return this.getChickenTextures((EntityChicken)par1Entity);
+        return this.func_110919_a((EntityChicken)par1Entity);
     }
 
     /**
