--- a/net/minecraft/src/RenderZombie.java
+++ b/net/minecraft/src/RenderZombie.java
@@ -2,11 +2,11 @@
 
 public class RenderZombie extends RenderBiped
 {
-    private static final ResourceLocation zombiePigmanTextures = new ResourceLocation("textures/entity/zombie_pigman.png");
-    private static final ResourceLocation zombieTextures = new ResourceLocation("textures/entity/zombie/zombie.png");
-    private static final ResourceLocation zombieVillagerTextures = new ResourceLocation("textures/entity/zombie/zombie_villager.png");
+    private static final ResourceLocation field_110866_o = new ResourceLocation("textures/entity/zombie_pigman.png");
+    private static final ResourceLocation field_110865_p = new ResourceLocation("textures/entity/zombie/zombie.png");
+    private static final ResourceLocation field_110864_q = new ResourceLocation("textures/entity/zombie/zombie_villager.png");
     private ModelBiped field_82434_o;
-    private ModelZombieVillager zombieVillagerModel;
+    private ModelZombieVillager field_82432_p;
     protected ModelBiped field_82437_k;
     protected ModelBiped field_82435_l;
     protected ModelBiped field_82436_m;
@@ -17,7 +17,7 @@
     {
         super(new ModelZombie(), 0.5F, 1.0F);
         this.field_82434_o = this.modelBipedMain;
-        this.zombieVillagerModel = new ModelZombieVillager();
+        this.field_82432_p = new ModelZombieVillager();
     }
 
     protected void func_82421_b()
@@ -44,7 +44,7 @@
 
     protected ResourceLocation func_110863_a(EntityZombie par1EntityZombie)
     {
-        return par1EntityZombie instanceof EntityPigZombie ? zombiePigmanTextures : (par1EntityZombie.isVillager() ? zombieVillagerTextures : zombieTextures);
+        return par1EntityZombie instanceof EntityPigZombie ? field_110866_o : (par1EntityZombie.isVillager() ? field_110864_q : field_110865_p);
     }
 
     protected void func_82428_a(EntityZombie par1EntityZombie, float par2)
@@ -57,15 +57,15 @@
     {
         if (par1EntityZombie.isVillager())
         {
-            if (this.field_82431_q != this.zombieVillagerModel.func_82897_a())
+            if (this.field_82431_q != this.field_82432_p.func_82897_a())
             {
-                this.zombieVillagerModel = new ModelZombieVillager();
-                this.field_82431_q = this.zombieVillagerModel.func_82897_a();
+                this.field_82432_p = new ModelZombieVillager();
+                this.field_82431_q = this.field_82432_p.func_82897_a();
                 this.field_82436_m = new ModelZombieVillager(1.0F, 0.0F, true);
                 this.field_82433_n = new ModelZombieVillager(0.5F, 0.0F, true);
             }
 
-            this.mainModel = this.zombieVillagerModel;
+            this.mainModel = this.field_82432_p;
             this.field_82423_g = this.field_82436_m;
             this.field_82425_h = this.field_82433_n;
         }
@@ -132,10 +132,7 @@
         this.func_82426_a((EntityZombie)par1EntityLivingBase, par2, par4, par6, par8, par9);
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
         return this.func_110863_a((EntityZombie)par1Entity);
     }
