--- a/net/minecraft/src/RenderLiving.java
+++ b/net/minecraft/src/RenderLiving.java
@@ -16,7 +16,7 @@
 
     public void doRenderLiving(EntityLiving par1EntityLiving, double par2, double par4, double par6, float par8, float par9)
     {
-        super.doRenderLiving(par1EntityLiving, par2, par4, par6, par8, par9);
+        super.func_130000_a(par1EntityLiving, par2, par4, par6, par8, par9);
         this.func_110827_b(par1EntityLiving, par2, par4, par6, par8, par9);
     }
 
@@ -27,7 +27,7 @@
 
     protected void func_110827_b(EntityLiving par1EntityLiving, double par2, double par4, double par6, float par8, float par9)
     {
-        Entity var10 = par1EntityLiving.getLeashedToEntity();
+        Entity var10 = par1EntityLiving.func_110166_bE();
 
         if (var10 != null)
         {
