--- a/net/minecraft/src/RenderBlocks.java
+++ b/net/minecraft/src/RenderBlocks.java
@@ -409,72 +409,51 @@
             {
                 case 0:
                     return this.renderStandardBlock(par1Block, par2, par3, par4);
-
                 case 1:
                     return this.renderCrossedSquares(par1Block, par2, par3, par4);
-
                 case 2:
                     return this.renderBlockTorch(par1Block, par2, par3, par4);
-
                 case 3:
                     return this.renderBlockFire((BlockFire)par1Block, par2, par3, par4);
-
                 case 4:
                     return this.renderBlockFluids(par1Block, par2, par3, par4);
-
                 case 5:
                     return this.renderBlockRedstoneWire(par1Block, par2, par3, par4);
-
                 case 6:
                     return this.renderBlockCrops(par1Block, par2, par3, par4);
-
                 case 7:
                     return this.renderBlockDoor(par1Block, par2, par3, par4);
-
                 case 8:
                     return this.renderBlockLadder(par1Block, par2, par3, par4);
-
                 case 9:
                     return this.renderBlockMinecartTrack((BlockRailBase)par1Block, par2, par3, par4);
-
                 case 10:
                     return this.renderBlockStairs((BlockStairs)par1Block, par2, par3, par4);
-
                 case 11:
                     return this.renderBlockFence((BlockFence)par1Block, par2, par3, par4);
-
                 case 12:
                     return this.renderBlockLever(par1Block, par2, par3, par4);
-
                 case 13:
                     return this.renderBlockCactus(par1Block, par2, par3, par4);
-
                 case 14:
                     return this.renderBlockBed(par1Block, par2, par3, par4);
-
                 case 15:
                     return this.renderBlockRepeater((BlockRedstoneRepeater)par1Block, par2, par3, par4);
-
                 case 16:
                     return this.renderPistonBase(par1Block, par2, par3, par4, false);
-
                 case 17:
                     return this.renderPistonExtension(par1Block, par2, par3, par4, true);
-
                 case 18:
                     return this.renderBlockPane((BlockPane)par1Block, par2, par3, par4);
-
                 case 19:
                     return this.renderBlockStem(par1Block, par2, par3, par4);
-
                 case 20:
                     return this.renderBlockVine(par1Block, par2, par3, par4);
-
                 case 21:
                     return this.renderBlockFenceGate((BlockFenceGate)par1Block, par2, par3, par4);
-
                 case 22:
                 default:
+
                     if (Reflector.ModLoader.exists())
                     {
                         return Reflector.callBoolean(Reflector.ModLoader_renderWorldBlock, new Object[] {this, this.blockAccess, Integer.valueOf(par2), Integer.valueOf(par3), Integer.valueOf(par4), par1Block, Integer.valueOf(var5)});
@@ -491,52 +470,36 @@
 
                 case 23:
                     return this.renderBlockLilyPad(par1Block, par2, par3, par4);
-
                 case 24:
                     return this.renderBlockCauldron((BlockCauldron)par1Block, par2, par3, par4);
-
                 case 25:
                     return this.renderBlockBrewingStand((BlockBrewingStand)par1Block, par2, par3, par4);
-
                 case 26:
                     return this.renderBlockEndPortalFrame((BlockEndPortalFrame)par1Block, par2, par3, par4);
-
                 case 27:
                     return this.renderBlockDragonEgg((BlockDragonEgg)par1Block, par2, par3, par4);
-
                 case 28:
                     return this.renderBlockCocoa((BlockCocoa)par1Block, par2, par3, par4);
-
                 case 29:
                     return this.renderBlockTripWireSource(par1Block, par2, par3, par4);
-
                 case 30:
                     return this.renderBlockTripWire(par1Block, par2, par3, par4);
-
                 case 31:
                     return this.renderBlockLog(par1Block, par2, par3, par4);
-
                 case 32:
                     return this.renderBlockWall((BlockWall)par1Block, par2, par3, par4);
-
                 case 33:
                     return this.renderBlockFlowerpot((BlockFlowerPot)par1Block, par2, par3, par4);
-
                 case 34:
                     return this.renderBlockBeacon((BlockBeacon)par1Block, par2, par3, par4);
-
                 case 35:
                     return this.renderBlockAnvil((BlockAnvil)par1Block, par2, par3, par4);
-
                 case 36:
                     return this.renderBlockRedstoneLogic((BlockRedstoneLogic)par1Block, par2, par3, par4);
-
                 case 37:
                     return this.renderBlockComparator((BlockComparator)par1Block, par2, par3, par4);
-
                 case 38:
                     return this.renderBlockHopper((BlockHopper)par1Block, par2, par3, par4);
-
                 case 39:
                     return this.renderBlockQuartz(par1Block, par2, par3, par4);
             }
@@ -704,14 +667,11 @@
             case 0:
                 var50 = 5;
                 break;
-
             case 1:
                 var50 = 3;
-
             case 2:
             default:
                 break;
-
             case 3:
                 var50 = 2;
         }
@@ -933,21 +893,17 @@
                 case 1:
                     var17 = Block.plantRed;
                     break;
-
                 case 2:
                     var17 = Block.plantYellow;
-
                 case 3:
                 case 4:
                 case 5:
                 case 6:
                 default:
                     break;
-
                 case 7:
                     var17 = Block.mushroomRed;
                     break;
-
                 case 8:
                     var17 = Block.mushroomBrown;
             }
@@ -1059,7 +1015,6 @@
                 this.uvRotateTop = 3;
                 this.uvRotateBottom = 3;
                 break;
-
             case 1:
                 this.uvRotateEast = 1;
                 this.uvRotateWest = 2;
@@ -1067,12 +1022,10 @@
                 this.uvRotateBottom = 1;
                 var8 = true;
                 break;
-
             case 2:
                 this.uvRotateSouth = 1;
                 this.uvRotateNorth = 2;
                 break;
-
             case 3:
                 this.uvRotateEast = 2;
                 this.uvRotateWest = 1;
@@ -1214,17 +1167,14 @@
                 var18 = -0.3125D;
                 var14 = BlockRedstoneRepeater.repeaterTorchOffset[var7];
                 break;
-
             case 1:
                 var16 = 0.3125D;
                 var12 = -BlockRedstoneRepeater.repeaterTorchOffset[var7];
                 break;
-
             case 2:
                 var18 = 0.3125D;
                 var14 = -BlockRedstoneRepeater.repeaterTorchOffset[var7];
                 break;
-
             case 3:
                 var16 = -0.3125D;
                 var12 = BlockRedstoneRepeater.repeaterTorchOffset[var7];
@@ -1251,7 +1201,6 @@
                     var22 = 9.0F;
                     var23 = 2.0F;
                     var24 = 14.0F;
-
                 case 0:
                 case 2:
                 default:
@@ -1307,17 +1256,14 @@
                 var12 = -0.3125D;
                 var16 = 1.0D;
                 break;
-
             case 1:
                 var8 = 0.3125D;
                 var14 = -1.0D;
                 break;
-
             case 2:
                 var12 = 0.3125D;
                 var16 = -1.0D;
                 break;
-
             case 3:
                 var8 = -0.3125D;
                 var14 = 1.0D;
@@ -1421,17 +1367,14 @@
                     this.uvRotateNorth = 3;
                     this.setRenderBounds(0.0D, 0.25D, 0.0D, 1.0D, 1.0D, 1.0D);
                     break;
-
                 case 1:
                     this.setRenderBounds(0.0D, 0.0D, 0.0D, 1.0D, 0.75D, 1.0D);
                     break;
-
                 case 2:
                     this.uvRotateSouth = 1;
                     this.uvRotateNorth = 2;
                     this.setRenderBounds(0.0D, 0.0D, 0.25D, 1.0D, 1.0D, 1.0D);
                     break;
-
                 case 3:
                     this.uvRotateSouth = 2;
                     this.uvRotateNorth = 1;
@@ -1439,7 +1382,6 @@
                     this.uvRotateBottom = 3;
                     this.setRenderBounds(0.0D, 0.0D, 0.0D, 1.0D, 1.0D, 0.75D);
                     break;
-
                 case 4:
                     this.uvRotateEast = 1;
                     this.uvRotateWest = 2;
@@ -1447,7 +1389,6 @@
                     this.uvRotateBottom = 1;
                     this.setRenderBounds(0.25D, 0.0D, 0.0D, 1.0D, 1.0D, 1.0D);
                     break;
-
                 case 5:
                     this.uvRotateEast = 2;
                     this.uvRotateWest = 1;
@@ -1476,30 +1417,25 @@
                     this.uvRotateWest = 3;
                     this.uvRotateSouth = 3;
                     this.uvRotateNorth = 3;
-
                 case 1:
                 default:
                     break;
-
                 case 2:
                     this.uvRotateSouth = 1;
                     this.uvRotateNorth = 2;
                     break;
-
                 case 3:
                     this.uvRotateSouth = 2;
                     this.uvRotateNorth = 1;
                     this.uvRotateTop = 3;
                     this.uvRotateBottom = 3;
                     break;
-
                 case 4:
                     this.uvRotateEast = 1;
                     this.uvRotateWest = 2;
                     this.uvRotateTop = 2;
                     this.uvRotateBottom = 1;
                     break;
-
                 case 5:
                     this.uvRotateEast = 2;
                     this.uvRotateWest = 1;
@@ -1629,7 +1565,6 @@
                 this.renderPistonRodUD((double)((float)par2 + 0.375F), (double)((float)par2 + 0.375F), (double)((float)par3 + 0.25F), (double)((float)par3 + 0.25F + var12), (double)((float)par4 + 0.375F), (double)((float)par4 + 0.625F), var11 * 0.6F, var13);
                 this.renderPistonRodUD((double)((float)par2 + 0.625F), (double)((float)par2 + 0.625F), (double)((float)par3 + 0.25F), (double)((float)par3 + 0.25F + var12), (double)((float)par4 + 0.625F), (double)((float)par4 + 0.375F), var11 * 0.6F, var13);
                 break;
-
             case 1:
                 this.setRenderBounds(0.0D, 0.75D, 0.0D, 1.0D, 1.0D, 1.0D);
                 this.renderStandardBlock(par1Block, par2, par3, par4);
@@ -1638,7 +1573,6 @@
                 this.renderPistonRodUD((double)((float)par2 + 0.375F), (double)((float)par2 + 0.375F), (double)((float)par3 - 0.25F + 1.0F - var12), (double)((float)par3 - 0.25F + 1.0F), (double)((float)par4 + 0.375F), (double)((float)par4 + 0.625F), var11 * 0.6F, var13);
                 this.renderPistonRodUD((double)((float)par2 + 0.625F), (double)((float)par2 + 0.625F), (double)((float)par3 - 0.25F + 1.0F - var12), (double)((float)par3 - 0.25F + 1.0F), (double)((float)par4 + 0.625F), (double)((float)par4 + 0.375F), var11 * 0.6F, var13);
                 break;
-
             case 2:
                 this.uvRotateSouth = 1;
                 this.uvRotateNorth = 2;
@@ -1649,7 +1583,6 @@
                 this.renderPistonRodSN((double)((float)par2 + 0.375F), (double)((float)par2 + 0.625F), (double)((float)par3 + 0.375F), (double)((float)par3 + 0.375F), (double)((float)par4 + 0.25F), (double)((float)par4 + 0.25F + var12), var11 * 0.5F, var13);
                 this.renderPistonRodSN((double)((float)par2 + 0.625F), (double)((float)par2 + 0.375F), (double)((float)par3 + 0.625F), (double)((float)par3 + 0.625F), (double)((float)par4 + 0.25F), (double)((float)par4 + 0.25F + var12), var11, var13);
                 break;
-
             case 3:
                 this.uvRotateSouth = 2;
                 this.uvRotateNorth = 1;
@@ -1662,7 +1595,6 @@
                 this.renderPistonRodSN((double)((float)par2 + 0.375F), (double)((float)par2 + 0.625F), (double)((float)par3 + 0.375F), (double)((float)par3 + 0.375F), (double)((float)par4 - 0.25F + 1.0F - var12), (double)((float)par4 - 0.25F + 1.0F), var11 * 0.5F, var13);
                 this.renderPistonRodSN((double)((float)par2 + 0.625F), (double)((float)par2 + 0.375F), (double)((float)par3 + 0.625F), (double)((float)par3 + 0.625F), (double)((float)par4 - 0.25F + 1.0F - var12), (double)((float)par4 - 0.25F + 1.0F), var11, var13);
                 break;
-
             case 4:
                 this.uvRotateEast = 1;
                 this.uvRotateWest = 2;
@@ -1675,7 +1607,6 @@
                 this.renderPistonRodEW((double)((float)par2 + 0.25F), (double)((float)par2 + 0.25F + var12), (double)((float)par3 + 0.375F), (double)((float)par3 + 0.625F), (double)((float)par4 + 0.375F), (double)((float)par4 + 0.375F), var11 * 0.6F, var13);
                 this.renderPistonRodEW((double)((float)par2 + 0.25F), (double)((float)par2 + 0.25F + var12), (double)((float)par3 + 0.625F), (double)((float)par3 + 0.375F), (double)((float)par4 + 0.625F), (double)((float)par4 + 0.625F), var11 * 0.6F, var13);
                 break;
-
             case 5:
                 this.uvRotateEast = 2;
                 this.uvRotateWest = 1;
@@ -3690,7 +3621,7 @@
 
         if (par1Block == Block.tallGrass)
         {
-            long var17 = (long)(par2 * 3129871) ^ (long)par4 * 116129781L ^ (long)par3;
+            long var17 = (long)(par2 * 3129871) ^(long)par4 * 116129781L ^(long)par3;
             var17 = var17 * var17 * 42317861L + var17 * 11L;
             var19 += ((double)((float)(var17 >> 16 & 15L) / 15.0F) - 0.5D) * 0.5D;
             var20 += ((double)((float)(var17 >> 20 & 15L) / 15.0F) - 1.0D) * 0.2D;
@@ -3926,7 +3857,7 @@
         double var10 = (double)var6.getMinV();
         double var12 = (double)var6.getMaxU();
         double var14 = (double)var6.getMaxV();
-        long var16 = (long)(par2 * 3129871) ^ (long)par4 * 116129781L ^ (long)par3;
+        long var16 = (long)(par2 * 3129871) ^(long)par4 * 116129781L ^(long)par3;
         var16 = var16 * var16 * 42317861L + var16 * 11L;
         int var18 = (int)(var16 >> 16 & 3L);
         var5.setBrightness(par1Block.getMixedBrightnessForBlock(this.blockAccess, par2, par3, par4));
@@ -6529,17 +6460,14 @@
                 var28 = 8.0D - (double)(var10 / 2);
                 var30 = 15.0D - (double)var10;
                 break;
-
             case 1:
                 var28 = 1.0D;
                 var30 = 8.0D - (double)(var10 / 2);
                 break;
-
             case 2:
                 var28 = 8.0D - (double)(var10 / 2);
                 var30 = 1.0D;
                 break;
-
             case 3:
                 var28 = 15.0D - (double)var10;
                 var30 = 8.0D - (double)(var10 / 2);
@@ -6603,17 +6531,14 @@
                 var20 = var22;
                 var22 = var45;
                 break;
-
             case 1:
                 var28 = 0.0D;
                 var30 = 8.0D;
                 break;
-
             case 2:
                 var28 = 8.0D;
                 var30 = 0.0D;
                 break;
-
             case 3:
                 var28 = 12.0D;
                 var30 = 8.0D;
@@ -9075,7 +9000,6 @@
             case 35:
             case 39:
                 return true;
-
             case 1:
             case 2:
             case 3:
