--- a/net/minecraft/src/RenderItem.java
+++ b/net/minecraft/src/RenderItem.java
@@ -333,10 +333,10 @@
         int var6 = par3ItemStack.itemID;
         int var7 = par3ItemStack.getItemDamage();
         Object var8 = par3ItemStack.getIconIndex();
-        float var17;
-        int var18;
         float var12;
         float var13;
+        float var17;
+        int var18;
 
         if (par3ItemStack.getItemSpriteNumber() == 0 && RenderBlocks.renderItemIn3d(Block.blocksList[var6].getRenderType()))
         {
