--- a/net/minecraft/src/RenderLightningBolt.java
+++ b/net/minecraft/src/RenderLightningBolt.java
@@ -131,10 +131,7 @@
         return null;
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
         return this.func_110805_a((EntityLightningBolt)par1Entity);
     }
