--- a/net/minecraft/src/RenderBoat.java
+++ b/net/minecraft/src/RenderBoat.java
@@ -4,7 +4,7 @@
 
 public class RenderBoat extends Render
 {
-    private static final ResourceLocation boatTextures = new ResourceLocation("textures/entity/boat.png");
+    private static final ResourceLocation field_110782_f = new ResourceLocation("textures/entity/boat.png");
 
     /** instance of ModelBoat for rendering */
     protected ModelBase modelBoat;
@@ -39,23 +39,20 @@
         float var12 = 0.75F;
         GL11.glScalef(var12, var12, var12);
         GL11.glScalef(1.0F / var12, 1.0F / var12, 1.0F / var12);
-        this.bindEntityTexture(par1EntityBoat);
+        this.func_110777_b(par1EntityBoat);
         GL11.glScalef(-1.0F, -1.0F, 1.0F);
         this.modelBoat.render(par1EntityBoat, 0.0F, 0.0F, -0.1F, 0.0F, 0.0F, 0.0625F);
         GL11.glPopMatrix();
     }
 
-    protected ResourceLocation getBoatTextures(EntityBoat par1EntityBoat)
+    protected ResourceLocation func_110781_a(EntityBoat par1EntityBoat)
     {
-        return boatTextures;
+        return field_110782_f;
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
-        return this.getBoatTextures((EntityBoat)par1Entity);
+        return this.func_110781_a((EntityBoat)par1Entity);
     }
 
     /**
