--- a/net/minecraft/src/RenderFireball.java
+++ b/net/minecraft/src/RenderFireball.java
@@ -15,7 +15,7 @@
     public void doRenderFireball(EntityFireball par1EntityFireball, double par2, double par4, double par6, float par8, float par9)
     {
         GL11.glPushMatrix();
-        this.bindEntityTexture(par1EntityFireball);
+        this.func_110777_b(par1EntityFireball);
         GL11.glTranslatef((float)par2, (float)par4, (float)par6);
         GL11.glEnable(GL12.GL_RESCALE_NORMAL);
         float var10 = this.field_77002_a;
@@ -42,17 +42,14 @@
         GL11.glPopMatrix();
     }
 
-    protected ResourceLocation getFireballTextures(EntityFireball par1EntityFireball)
+    protected ResourceLocation func_110790_a(EntityFireball par1EntityFireball)
     {
-        return TextureMap.locationItemsTexture;
+        return TextureMap.field_110576_c;
     }
 
-    /**
-     * Returns the location of an entity's texture. Doesn't seem to be called unless you call Render.bindEntityTexture.
-     */
-    protected ResourceLocation getEntityTexture(Entity par1Entity)
+    protected ResourceLocation func_110775_a(Entity par1Entity)
     {
-        return this.getFireballTextures((EntityFireball)par1Entity);
+        return this.func_110790_a((EntityFireball)par1Entity);
     }
 
     /**
