--- a/net/minecraft/src/Request.java
+++ b/net/minecraft/src/Request.java
@@ -16,7 +16,7 @@
         try
         {
             this.field_96365_b = par1Str;
-            this.field_96367_a = (HttpURLConnection)(new URL(par1Str)).openConnection(Minecraft.getMinecraft().getProxy());
+            this.field_96367_a = (HttpURLConnection)(new URL(par1Str)).openConnection(Minecraft.getMinecraft().func_110437_J());
             this.field_96367_a.setConnectTimeout(par2);
             this.field_96367_a.setReadTimeout(par3);
         }
