--- a/net/minecraft/src/ResourcePackRepositoryEntry.java
+++ b/net/minecraft/src/ResourcePackRepositoryEntry.java
@@ -8,73 +8,73 @@
 
 public class ResourcePackRepositoryEntry
 {
-    private final File resourcePackFile;
-    private ResourcePack reResourcePack;
-    private PackMetadataSection rePackMetadataSection;
-    private BufferedImage texturePackIcon;
-    private ResourceLocation locationTexturePackIcon;
+    private final File field_110523_b;
+    private ResourcePack field_110524_c;
+    private PackMetadataSection field_110521_d;
+    private BufferedImage field_110522_e;
+    private ResourceLocation field_110520_f;
 
-    final ResourcePackRepository reResourcePackRepository;
+    final ResourcePackRepository field_110525_a;
 
     private ResourcePackRepositoryEntry(ResourcePackRepository par1ResourcePackRepository, File par2File)
     {
-        this.reResourcePackRepository = par1ResourcePackRepository;
-        this.resourcePackFile = par2File;
+        this.field_110525_a = par1ResourcePackRepository;
+        this.field_110523_b = par2File;
     }
 
-    public void updateResourcePack() throws IOException
+    public void func_110516_a() throws IOException
     {
-        this.reResourcePack = (ResourcePack)(this.resourcePackFile.isDirectory() ? new FolderResourcePack(this.resourcePackFile) : new FileResourcePack(this.resourcePackFile));
-        this.rePackMetadataSection = (PackMetadataSection)this.reResourcePack.getPackMetadata(this.reResourcePackRepository.rprMetadataSerializer, "pack");
+        this.field_110524_c = (ResourcePack)(this.field_110523_b.isDirectory() ? new FolderResourcePack(this.field_110523_b) : new FileResourcePack(this.field_110523_b));
+        this.field_110521_d = (PackMetadataSection)this.field_110524_c.func_135058_a(this.field_110525_a.field_110621_c, "pack");
 
         try
         {
-            this.texturePackIcon = this.reResourcePack.getPackImage();
+            this.field_110522_e = this.field_110524_c.func_110586_a();
         }
         catch (IOException var2)
         {
             ;
         }
 
-        if (this.texturePackIcon == null)
+        if (this.field_110522_e == null)
         {
-            this.texturePackIcon = this.reResourcePackRepository.rprDefaultResourcePack.getPackImage();
+            this.field_110522_e = this.field_110525_a.field_110620_b.func_110586_a();
         }
 
-        this.closeResourcePack();
+        this.func_110517_b();
     }
 
-    public void bindTexturePackIcon(TextureManager par1TextureManager)
+    public void func_110518_a(TextureManager par1TextureManager)
     {
-        if (this.locationTexturePackIcon == null)
+        if (this.field_110520_f == null)
         {
-            this.locationTexturePackIcon = par1TextureManager.getDynamicTextureLocation("texturepackicon", new DynamicTexture(this.texturePackIcon));
+            this.field_110520_f = par1TextureManager.func_110578_a("texturepackicon", new DynamicTexture(this.field_110522_e));
         }
 
-        par1TextureManager.bindTexture(this.locationTexturePackIcon);
+        par1TextureManager.func_110577_a(this.field_110520_f);
     }
 
-    public void closeResourcePack()
+    public void func_110517_b()
     {
-        if (this.reResourcePack instanceof Closeable)
+        if (this.field_110524_c instanceof Closeable)
         {
-            IOUtils.closeQuietly((Closeable)this.reResourcePack);
+            IOUtils.closeQuietly((Closeable)this.field_110524_c);
         }
     }
 
-    public ResourcePack getResourcePack()
+    public ResourcePack func_110514_c()
     {
-        return this.reResourcePack;
+        return this.field_110524_c;
     }
 
-    public String getResourcePackName()
+    public String func_110515_d()
     {
-        return this.reResourcePack.getPackName();
+        return this.field_110524_c.func_130077_b();
     }
 
-    public String getTexturePackDescription()
+    public String func_110519_e()
     {
-        return this.rePackMetadataSection == null ? EnumChatFormatting.RED + "Invalid pack.mcmeta (or missing \'pack\' section)" : this.rePackMetadataSection.getPackDescription();
+        return this.field_110521_d == null ? EnumChatFormatting.RED + "Invalid pack.mcmeta (or missing \'pack\' section)" : this.field_110521_d.func_110461_a();
     }
 
     public boolean equals(Object par1Obj)
@@ -89,7 +89,7 @@
 
     public String toString()
     {
-        return String.format("%s:%s:%d", new Object[] {this.resourcePackFile.getName(), this.resourcePackFile.isDirectory() ? "folder" : "zip", Long.valueOf(this.resourcePackFile.lastModified())});
+        return String.format("%s:%s:%d", new Object[] {this.field_110523_b.getName(), this.field_110523_b.isDirectory() ? "folder" : "zip", Long.valueOf(this.field_110523_b.lastModified())});
     }
 
     ResourcePackRepositoryEntry(ResourcePackRepository par1ResourcePackRepository, File par2File, ResourcePackRepositoryFilter par3ResourcePackRepositoryFilter)
