--- a/net/minecraft/src/Score.java
+++ b/net/minecraft/src/Score.java
@@ -7,20 +7,20 @@
 {
     public static final Comparator field_96658_a = new ScoreComparator();
     private final Scoreboard theScoreboard;
-    private final ScoreObjective theScoreObjective;
+    private final ScoreObjective field_96657_c;
     private final String field_96654_d;
     private int field_96655_e;
 
     public Score(Scoreboard par1Scoreboard, ScoreObjective par2ScoreObjective, String par3Str)
     {
         this.theScoreboard = par1Scoreboard;
-        this.theScoreObjective = par2ScoreObjective;
+        this.field_96657_c = par2ScoreObjective;
         this.field_96654_d = par3Str;
     }
 
     public void func_96649_a(int par1)
     {
-        if (this.theScoreObjective.getCriteria().isReadOnly())
+        if (this.field_96657_c.getCriteria().isReadOnly())
         {
             throw new IllegalStateException("Cannot modify read-only score");
         }
@@ -32,7 +32,7 @@
 
     public void func_96646_b(int par1)
     {
-        if (this.theScoreObjective.getCriteria().isReadOnly())
+        if (this.field_96657_c.getCriteria().isReadOnly())
         {
             throw new IllegalStateException("Cannot modify read-only score");
         }
@@ -44,7 +44,7 @@
 
     public void func_96648_a()
     {
-        if (this.theScoreObjective.getCriteria().isReadOnly())
+        if (this.field_96657_c.getCriteria().isReadOnly())
         {
             throw new IllegalStateException("Cannot modify read-only score");
         }
@@ -72,7 +72,7 @@
 
     public ScoreObjective func_96645_d()
     {
-        return this.theScoreObjective;
+        return this.field_96657_c;
     }
 
     public String getPlayerName()
@@ -87,6 +87,6 @@
 
     public void func_96651_a(List par1List)
     {
-        this.func_96647_c(this.theScoreObjective.getCriteria().func_96635_a(par1List));
+        this.func_96647_c(this.field_96657_c.getCriteria().func_96635_a(par1List));
     }
 }
