--- a/net/minecraft/src/ScoreHealthCriteria.java
+++ b/net/minecraft/src/ScoreHealthCriteria.java
@@ -15,7 +15,7 @@
         float var2 = 0.0F;
         EntityPlayer var4;
 
-        for (Iterator var3 = par1List.iterator(); var3.hasNext(); var2 += var4.getHealth() + var4.getAbsorptionAmount())
+        for (Iterator var3 = par1List.iterator(); var3.hasNext(); var2 += var4.func_110143_aJ() + var4.func_110139_bj())
         {
             var4 = (EntityPlayer)var3.next();
         }
