--- a/net/minecraft/src/ServerCommandManager.java
+++ b/net/minecraft/src/ServerCommandManager.java
@@ -49,7 +49,6 @@
             this.registerCommand(new CommandServerKick());
             this.registerCommand(new CommandServerList());
             this.registerCommand(new CommandServerWhitelist());
-            this.registerCommand(new CommandSetPlayerTimeout());
         }
         else
         {
@@ -72,9 +71,9 @@
             var5 = false;
         }
 
-        ChatMessageComponent var6 = ChatMessageComponent.createFromTranslationWithSubstitutions("chat.type.admin", new Object[] {par1ICommandSender.getCommandSenderName(), ChatMessageComponent.createFromTranslationWithSubstitutions(par3Str, par4ArrayOfObj)});
-        var6.setColor(EnumChatFormatting.GRAY);
-        var6.setItalic(Boolean.valueOf(true));
+        ChatMessageComponent var6 = ChatMessageComponent.func_111082_b("chat.type.admin", new Object[] {par1ICommandSender.getCommandSenderName(), ChatMessageComponent.func_111082_b(par3Str, par4ArrayOfObj)});
+        var6.func_111059_a(EnumChatFormatting.GRAY);
+        var6.func_111063_b(Boolean.valueOf(true));
 
         if (var5)
         {
@@ -84,7 +83,7 @@
             {
                 EntityPlayerMP var8 = (EntityPlayerMP)var7.next();
 
-                if (var8 != par1ICommandSender && MinecraftServer.getServer().getConfigurationManager().isPlayerOpped(var8.getCommandSenderName()))
+                if (var8 != par1ICommandSender && MinecraftServer.getServer().getConfigurationManager().areCommandsAllowed(var8.getCommandSenderName()))
                 {
                     var8.sendChatToPlayer(var6);
                 }
@@ -98,7 +97,7 @@
 
         if ((par2 & 1) != 1)
         {
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions(par3Str, par4ArrayOfObj));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b(par3Str, par4ArrayOfObj));
         }
     }
 }
