--- a/net/minecraft/src/ServerCommandScoreboard.java
+++ b/net/minecraft/src/ServerCommandScoreboard.java
@@ -426,7 +426,7 @@
                     throw new WrongUsageException("commands.scoreboard.teams.option.noValue", new Object[] {var5, func_96333_a(Arrays.asList(new String[]{"true", "false"}))});
                 }
 
-                var4.setAllowFriendlyFire(var6.equalsIgnoreCase("true"));
+                var4.func_96660_a(var6.equalsIgnoreCase("true"));
             }
             else if (var5.equalsIgnoreCase("seeFriendlyInvisibles"))
             {
@@ -470,8 +470,8 @@
                 throw new CommandException("commands.scoreboard.teams.list.player.empty", new Object[] {var5.func_96661_b()});
             }
 
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.scoreboard.teams.list.player.count", new Object[] {Integer.valueOf(var6.size()), var5.func_96661_b()}).setColor(EnumChatFormatting.DARK_GREEN));
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromText(joinNiceString(var6.toArray())));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.scoreboard.teams.list.player.count", new Object[] {Integer.valueOf(var6.size()), var5.func_96661_b()}).func_111059_a(EnumChatFormatting.DARK_GREEN));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111066_d(joinNiceString(var6.toArray())));
         }
         else
         {
@@ -482,13 +482,13 @@
                 throw new CommandException("commands.scoreboard.teams.list.empty", new Object[0]);
             }
 
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.scoreboard.teams.list.count", new Object[] {Integer.valueOf(var8.size())}).setColor(EnumChatFormatting.DARK_GREEN));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.scoreboard.teams.list.count", new Object[] {Integer.valueOf(var8.size())}).func_111059_a(EnumChatFormatting.DARK_GREEN));
             Iterator var9 = var8.iterator();
 
             while (var9.hasNext())
             {
                 ScorePlayerTeam var7 = (ScorePlayerTeam)var9.next();
-                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.scoreboard.teams.list.entry", new Object[] {var7.func_96661_b(), var7.func_96669_c(), Integer.valueOf(var7.getMembershipCollection().size())}));
+                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.scoreboard.teams.list.entry", new Object[] {var7.func_96661_b(), var7.func_96669_c(), Integer.valueOf(var7.getMembershipCollection().size())}));
             }
         }
     }
@@ -628,13 +628,13 @@
         }
         else
         {
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.scoreboard.objectives.list.count", new Object[] {Integer.valueOf(var3.size())}).setColor(EnumChatFormatting.DARK_GREEN));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.scoreboard.objectives.list.count", new Object[] {Integer.valueOf(var3.size())}).func_111059_a(EnumChatFormatting.DARK_GREEN));
             Iterator var4 = var3.iterator();
 
             while (var4.hasNext())
             {
                 ScoreObjective var5 = (ScoreObjective)var4.next();
-                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.scoreboard.objectives.list.entry", new Object[] {var5.getName(), var5.getDisplayName(), var5.getCriteria().func_96636_a()}));
+                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.scoreboard.objectives.list.entry", new Object[] {var5.getName(), var5.getDisplayName(), var5.getCriteria().func_96636_a()}));
             }
         }
     }
@@ -690,13 +690,13 @@
                 throw new CommandException("commands.scoreboard.players.list.player.empty", new Object[] {var5});
             }
 
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.scoreboard.players.list.player.count", new Object[] {Integer.valueOf(var6.size()), var5}).setColor(EnumChatFormatting.DARK_GREEN));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.scoreboard.players.list.player.count", new Object[] {Integer.valueOf(var6.size()), var5}).func_111059_a(EnumChatFormatting.DARK_GREEN));
             Iterator var7 = var6.values().iterator();
 
             while (var7.hasNext())
             {
                 Score var8 = (Score)var7.next();
-                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.scoreboard.players.list.player.entry", new Object[] {Integer.valueOf(var8.getScorePoints()), var8.func_96645_d().getDisplayName(), var8.func_96645_d().getName()}));
+                par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.scoreboard.players.list.player.entry", new Object[] {Integer.valueOf(var8.getScorePoints()), var8.func_96645_d().getDisplayName(), var8.func_96645_d().getName()}));
             }
         }
         else
@@ -708,8 +708,8 @@
                 throw new CommandException("commands.scoreboard.players.list.empty", new Object[0]);
             }
 
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromTranslationWithSubstitutions("commands.scoreboard.players.list.count", new Object[] {Integer.valueOf(var9.size())}).setColor(EnumChatFormatting.DARK_GREEN));
-            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.createFromText(joinNiceString(var9.toArray())));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111082_b("commands.scoreboard.players.list.count", new Object[] {Integer.valueOf(var9.size())}).func_111059_a(EnumChatFormatting.DARK_GREEN));
+            par1ICommandSender.sendChatToPlayer(ChatMessageComponent.func_111066_d(joinNiceString(var9.toArray())));
         }
     }
 
