--- a/net/minecraft/src/SoundManager.java
+++ b/net/minecraft/src/SoundManager.java
@@ -21,7 +21,7 @@
     private static final String[] field_130084_a = new String[] {"ogg"};
 
     /** A reference to the sound system. */
-    private SoundSystem sndSystem;
+    public static SoundSystem sndSystem;
 
     /** Set to true when the SoundManager has been initialised. */
     private boolean loaded;
