--- a/net/minecraft/src/SpawnerAnimals.java
+++ b/net/minecraft/src/SpawnerAnimals.java
@@ -157,7 +157,7 @@
                                                             {
                                                                 ++var17;
                                                                 par1WorldServer.spawnEntityInWorld(var41);
-                                                                var24 = var41.onSpawnWithEgg(var24);
+                                                                var24 = var41.func_110161_a(var24);
 
                                                                 if (var17 >= var41.getMaxSpawnedInChunk())
                                                                 {
@@ -255,7 +255,7 @@
 
                             var22.setLocationAndAngles((double)var19, (double)var20, (double)var21, par6Random.nextFloat() * 360.0F, 0.0F);
                             par0World.spawnEntityInWorld(var22);
-                            var9 = var22.onSpawnWithEgg(var9);
+                            var9 = var22.func_110161_a(var9);
                             var16 = true;
                         }
 
