--- a/net/minecraft/src/StitchSlot.java
+++ b/net/minecraft/src/StitchSlot.java
@@ -36,7 +36,7 @@
         return this.originY;
     }
 
-    public boolean addSlot(StitchHolder par1StitchHolder)
+    public boolean func_94182_a(StitchHolder par1StitchHolder)
     {
         if (this.holder != null)
         {
@@ -101,7 +101,7 @@
 
                         var9 = (StitchSlot)var8.next();
                     }
-                    while (!var9.addSlot(par1StitchHolder));
+                    while (!var9.func_94182_a(par1StitchHolder));
 
                     return true;
                 }
