--- a/net/minecraft/src/Stitcher.java
+++ b/net/minecraft/src/Stitcher.java
@@ -37,17 +37,17 @@
         this.maxTileDimension = par4;
     }
 
-    public int getCurrentWidth()
+    public int func_110935_a()
     {
         return this.currentWidth;
     }
 
-    public int getCurrentHeight()
+    public int func_110936_b()
     {
         return this.currentHeight;
     }
 
-    public void addSprite(TextureAtlasSprite par1TextureAtlasSprite)
+    public void func_110934_a(TextureAtlasSprite par1TextureAtlasSprite)
     {
         StitchHolder var2 = new StitchHolder(par1TextureAtlasSprite);
 
@@ -77,7 +77,7 @@
 
             if (!this.allocateSlot(var5))
             {
-                String var6 = String.format("Unable to fit: %s - size: %dx%d - Maybe try a lowerresolution texturepack?", new Object[] {var5.getAtlasSprite().getIconName(), Integer.valueOf(var5.getAtlasSprite().getIconWidth()), Integer.valueOf(var5.getAtlasSprite().getIconHeight())});
+                String var6 = String.format("Unable to fit: %s - size: %dx%d - Maybe try a lowerresolution texturepack?", new Object[] {var5.func_98150_a().getIconName(), Integer.valueOf(var5.func_98150_a().getOriginX()), Integer.valueOf(var5.func_98150_a().getOriginY())});
                 throw new StitcherException(var5, var6);
             }
         }
@@ -107,8 +107,8 @@
         {
             StitchSlot var5 = (StitchSlot)var4.next();
             StitchHolder var6 = var5.getStitchHolder();
-            TextureAtlasSprite var7 = var6.getAtlasSprite();
-            var7.initSprite(this.currentWidth, this.currentHeight, var5.getOriginX(), var5.getOriginY(), var6.isRotated());
+            TextureAtlasSprite var7 = var6.func_98150_a();
+            var7.func_110971_a(this.currentWidth, this.currentHeight, var5.getOriginX(), var5.getOriginY(), var6.isRotated());
             var8.add(var7);
         }
 
@@ -136,14 +136,14 @@
     {
         for (int var2 = 0; var2 < this.stitchSlots.size(); ++var2)
         {
-            if (((StitchSlot)this.stitchSlots.get(var2)).addSlot(par1StitchHolder))
+            if (((StitchSlot)this.stitchSlots.get(var2)).func_94182_a(par1StitchHolder))
             {
                 return true;
             }
 
             par1StitchHolder.rotate();
 
-            if (((StitchSlot)this.stitchSlots.get(var2)).addSlot(par1StitchHolder))
+            if (((StitchSlot)this.stitchSlots.get(var2)).func_94182_a(par1StitchHolder))
             {
                 return true;
             }
@@ -248,7 +248,7 @@
             this.currentHeight += par1StitchHolder.getHeight();
         }
 
-        var17.addSlot(par1StitchHolder);
+        var17.func_94182_a(par1StitchHolder);
         this.stitchSlots.add(var17);
         return true;
     }
