--- a/net/minecraft/src/StitchHolder.java
+++ b/net/minecraft/src/StitchHolder.java
@@ -11,12 +11,12 @@
     public StitchHolder(TextureAtlasSprite par1TextureAtlasSprite)
     {
         this.theTexture = par1TextureAtlasSprite;
-        this.width = par1TextureAtlasSprite.getIconWidth();
-        this.height = par1TextureAtlasSprite.getIconHeight();
+        this.width = par1TextureAtlasSprite.getOriginX();
+        this.height = par1TextureAtlasSprite.getOriginY();
         this.rotated = this.ceil16(this.height) > this.ceil16(this.width);
     }
 
-    public TextureAtlasSprite getAtlasSprite()
+    public TextureAtlasSprite func_98150_a()
     {
         return this.theTexture;
     }
