--- a/net/minecraft/src/StructureStrongholdPieces.java
+++ b/net/minecraft/src/StructureStrongholdPieces.java
@@ -13,23 +13,6 @@
     static int totalWeight;
     private static final StructureStrongholdStones strongholdStones = new StructureStrongholdStones((StructureStrongholdPieceWeight2)null);
 
-    public static void func_143046_a()
-    {
-        MapGenStructureIO.func_143031_a(ComponentStrongholdChestCorridor.class, "SHCC");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdCorridor.class, "SHFC");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdCrossing.class, "SH5C");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdLeftTurn.class, "SHLT");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdLibrary.class, "SHLi");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdPortalRoom.class, "SHPR");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdPrison.class, "SHPH");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdRightTurn.class, "SHRT");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdRoomCrossing.class, "SHRC");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdStairs.class, "SHSD");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdStairs2.class, "SHStart");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdStraight.class, "SHS");
-        MapGenStructureIO.func_143031_a(ComponentStrongholdStairsStraight.class, "SHSSD");
-    }
-
     /**
      * sets up Arrays with the Structure pieces and their weights
      */
