--- a/net/minecraft/src/StructureComponent.java
+++ b/net/minecraft/src/StructureComponent.java
@@ -156,13 +156,10 @@
             case 0:
             case 2:
                 return this.boundingBox.minX + par1;
-
             case 1:
                 return this.boundingBox.maxX - par2;
-
             case 3:
                 return this.boundingBox.minX + par2;
-
             default:
                 return par1;
         }
@@ -179,14 +176,11 @@
         {
             case 0:
                 return this.boundingBox.minZ + par2;
-
             case 1:
             case 3:
                 return this.boundingBox.minZ + par1;
-
             case 2:
                 return this.boundingBox.maxZ - par2;
-
             default:
                 return par2;
         }
