--- a/net/minecraft/src/StructureBoundingBox.java
+++ b/net/minecraft/src/StructureBoundingBox.java
@@ -22,19 +22,6 @@
 
     public StructureBoundingBox() {}
 
-    public StructureBoundingBox(int[] par1ArrayOfInteger)
-    {
-        if (par1ArrayOfInteger.length == 6)
-        {
-            this.minX = par1ArrayOfInteger[0];
-            this.minY = par1ArrayOfInteger[1];
-            this.minZ = par1ArrayOfInteger[2];
-            this.maxX = par1ArrayOfInteger[3];
-            this.maxY = par1ArrayOfInteger[4];
-            this.maxZ = par1ArrayOfInteger[5];
-        }
-    }
-
     /**
      * returns a new StructureBoundingBox with MAX values
      */
@@ -52,16 +39,12 @@
         {
             case 0:
                 return new StructureBoundingBox(par0 + par3, par1 + par4, par2 + par5, par0 + par6 - 1 + par3, par1 + par7 - 1 + par4, par2 + par8 - 1 + par5);
-
             case 1:
                 return new StructureBoundingBox(par0 - par8 + 1 + par5, par1 + par4, par2 + par3, par0 + par5, par1 + par7 - 1 + par4, par2 + par6 - 1 + par3);
-
             case 2:
                 return new StructureBoundingBox(par0 + par3, par1 + par4, par2 - par8 + 1 + par5, par0 + par6 - 1 + par3, par1 + par7 - 1 + par4, par2 + par5);
-
             case 3:
                 return new StructureBoundingBox(par0 + par5, par1 + par4, par2 + par3, par0 + par8 - 1 + par5, par1 + par7 - 1 + par4, par2 + par6 - 1 + par3);
-
             default:
                 return new StructureBoundingBox(par0 + par3, par1 + par4, par2 + par5, par0 + par6 - 1 + par3, par1 + par7 - 1 + par4, par2 + par8 - 1 + par5);
         }
@@ -190,9 +173,4 @@
     {
         return "(" + this.minX + ", " + this.minY + ", " + this.minZ + "; " + this.maxX + ", " + this.maxY + ", " + this.maxZ + ")";
     }
-
-    public NBTTagIntArray func_143047_a(String par1Str)
-    {
-        return new NBTTagIntArray(par1Str, new int[] {this.minX, this.minY, this.minZ, this.maxX, this.maxY, this.maxZ});
-    }
 }
