--- a/net/minecraft/src/TaskOnlineConnect.java
+++ b/net/minecraft/src/TaskOnlineConnect.java
@@ -16,14 +16,14 @@
 
     public void run()
     {
-        this.setMessage(I18n.getString("mco.connect.connecting"));
-        McoClient var1 = new McoClient(this.getMinecraft().getSession());
+        this.func_96576_b(I18n.func_135053_a("mco.connect.connecting"));
+        McoClient var1 = new McoClient(this.func_96578_b().func_110432_I());
         boolean var2 = false;
         boolean var3 = false;
         int var4 = 5;
         McoServerAddress var5 = null;
 
-        for (int var6 = 0; var6 < 10 && !this.wasScreenClosed(); ++var6)
+        for (int var6 = 0; var6 < 10 && !this.func_96577_c(); ++var6)
         {
             try
             {
@@ -37,7 +37,7 @@
             catch (ExceptionMcoService var9)
             {
                 var3 = true;
-                this.setFailedMessage(var9.toString());
+                this.func_96575_a(var9.toString());
                 Minecraft.getMinecraft().getLogAgent().logSevere(var9.toString());
                 break;
             }
@@ -48,7 +48,7 @@
             catch (Exception var11)
             {
                 var3 = true;
-                this.setFailedMessage(var11.getLocalizedMessage());
+                this.func_96575_a(var11.getLocalizedMessage());
             }
 
             if (var2)
@@ -59,7 +59,7 @@
             this.func_111251_a(var4);
         }
 
-        if (!this.wasScreenClosed() && !var3)
+        if (!this.func_96577_c() && !var3)
         {
             if (var2)
             {
@@ -68,7 +68,7 @@
             }
             else
             {
-                this.getMinecraft().displayGuiScreen(this.field_96584_d);
+                this.func_96578_b().displayGuiScreen(this.field_96584_d);
             }
         }
     }
@@ -90,7 +90,7 @@
         (new ThreadOnlineConnect(this, par1Str, par2)).start();
     }
 
-    public void updateScreen()
+    public void func_96573_a()
     {
         if (this.field_96586_a != null)
         {
