--- a/net/minecraft/src/TcpConnection.java
+++ b/net/minecraft/src/TcpConnection.java
@@ -26,9 +26,7 @@
 
     /** The object used for synchronization on the send queue. */
     private final Object sendQueueLock;
-
-    /** Log agent for TCP connection */
-    private final ILogAgent tcpConLogAgent;
+    private final ILogAgent field_98215_i;
 
     /** The socket used by this network manager. */
     private Socket networkSocket;
@@ -77,9 +75,7 @@
 
     /** A String indicating why the network has shutdown. */
     private String terminationReason;
-
-    /** Contains shutdown description (internal error, etc.) as string array */
-    private Object[] shutdownDescription;
+    private Object[] field_74480_w;
     private int field_74490_x;
 
     /**
@@ -115,7 +111,7 @@
         this.chunkDataPacketsDelay = 50;
         this.field_74463_A = par5PrivateKey;
         this.networkSocket = par2Socket;
-        this.tcpConLogAgent = par1ILogAgent;
+        this.field_98215_i = par1ILogAgent;
         this.remoteSocketAddress = par2Socket.getRemoteSocketAddress();
         this.theNetHandler = par4NetHandler;
 
@@ -183,7 +179,7 @@
             int var10001;
             int[] var10000;
 
-            if (this.field_74468_e == 0 || !this.dataPackets.isEmpty() && MinecraftServer.getSystemTimeMillis() - ((Packet)this.dataPackets.get(0)).creationTimeMillis >= (long)this.field_74468_e)
+            if (this.field_74468_e == 0 || !this.dataPackets.isEmpty() && MinecraftServer.func_130071_aq() - ((Packet)this.dataPackets.get(0)).creationTimeMillis >= (long)this.field_74468_e)
             {
                 var2 = this.func_74460_a(false);
 
@@ -208,7 +204,7 @@
                 }
             }
 
-            if (this.chunkDataPacketsDelay-- <= 0 && (this.field_74468_e == 0 || !this.chunkDataPackets.isEmpty() && MinecraftServer.getSystemTimeMillis() - ((Packet)this.chunkDataPackets.get(0)).creationTimeMillis >= (long)this.field_74468_e))
+            if (this.chunkDataPacketsDelay-- <= 0 && (this.field_74468_e == 0 || !this.chunkDataPackets.isEmpty() && MinecraftServer.func_130071_aq() - ((Packet)this.chunkDataPackets.get(0)).creationTimeMillis >= (long)this.field_74468_e))
             {
                 var2 = this.func_74460_a(true);
 
@@ -312,7 +308,7 @@
 
         try
         {
-            Packet var2 = Packet.readPacket(this.tcpConLogAgent, this.socketInputStream, this.theNetHandler.isServerHandler(), this.networkSocket);
+            Packet var2 = Packet.readPacket(this.field_98215_i, this.socketInputStream, this.theNetHandler.isServerHandler(), this.networkSocket);
 
             if (var2 != null)
             {
@@ -382,7 +378,7 @@
         {
             this.isTerminating = true;
             this.terminationReason = par1Str;
-            this.shutdownDescription = par2ArrayOfObj;
+            this.field_74480_w = par2ArrayOfObj;
             this.isRunning = false;
             (new TcpMasterThread(this)).start();
 
@@ -447,7 +443,7 @@
         {
             Packet var2 = (Packet)this.readPackets.poll();
 
-            if (var2 != null && !this.theNetHandler.isConnectionClosed())
+            if (var2 != null && !this.theNetHandler.func_142032_c())
             {
                 var2.processPacket(this.theNetHandler);
             }
@@ -457,7 +453,7 @@
 
         if (this.isTerminating && this.readPackets.isEmpty())
         {
-            this.theNetHandler.handleErrorMessage(this.terminationReason, this.shutdownDescription);
+            this.theNetHandler.handleErrorMessage(this.terminationReason, this.field_74480_w);
         }
     }
 
