--- a/net/minecraft/src/Tessellator.java
+++ b/net/minecraft/src/Tessellator.java
@@ -657,8 +657,8 @@
 
                 int var22 = this.addedVertices;
                 this.vertexQuadIcons[var22 / 4] = var17;
-                int var23 = var17.sheetWidth / var17.getIconWidth();
-                int var24 = var17.sheetHeight / var17.getIconHeight();
+                int var23 = var17.sheetWidth / var17.width;
+                int var24 = var17.sheetHeight / var17.height;
                 int var25 = this.color;
                 int var26 = this.brightness;
 
