--- a/net/minecraft/src/TextureUtils$2.java
+++ b/net/minecraft/src/TextureUtils$2.java
@@ -4,14 +4,14 @@
 
 final class TextureUtils$2 implements TickableTextureObject
 {
-    public void tick()
+    public void func_110550_d()
     {
         TextureAnimations.updateCustomAnimations();
     }
 
-    public void loadTexture(ResourceManager var1) throws IOException {}
+    public void func_110551_a(ResourceManager var1) throws IOException {}
 
-    public int getGlTextureId()
+    public int func_110552_b()
     {
         return 0;
     }
