--- a/net/minecraft/src/TextureMetadataSection.java
+++ b/net/minecraft/src/TextureMetadataSection.java
@@ -2,22 +2,22 @@
 
 public class TextureMetadataSection implements MetadataSection
 {
-    private final boolean textureBlur;
-    private final boolean textureClamp;
+    private final boolean field_110482_a;
+    private final boolean field_110481_b;
 
     public TextureMetadataSection(boolean par1, boolean par2)
     {
-        this.textureBlur = par1;
-        this.textureClamp = par2;
+        this.field_110482_a = par1;
+        this.field_110481_b = par2;
     }
 
-    public boolean getTextureBlur()
+    public boolean func_110479_a()
     {
-        return this.textureBlur;
+        return this.field_110482_a;
     }
 
-    public boolean getTextureClamp()
+    public boolean func_110480_b()
     {
-        return this.textureClamp;
+        return this.field_110481_b;
     }
 }
