--- a/net/minecraft/src/TextureManagerINNER1.java
+++ b/net/minecraft/src/TextureManagerINNER1.java
@@ -4,19 +4,19 @@
 
 class TextureManagerINNER1 implements Callable
 {
-    final TextureObject theTextureObject;
+    final TextureObject field_135062_a;
 
-    final TextureManager theTextureManager;
+    final TextureManager field_135061_b;
 
     TextureManagerINNER1(TextureManager par1TextureManager, TextureObject par2TextureObject)
     {
-        this.theTextureManager = par1TextureManager;
-        this.theTextureObject = par2TextureObject;
+        this.field_135061_b = par1TextureManager;
+        this.field_135062_a = par2TextureObject;
     }
 
     public String func_135060_a()
     {
-        return this.theTextureObject.getClass().getName();
+        return this.field_135062_a.getClass().getName();
     }
 
     public Object call()
