--- a/net/minecraft/src/TextureAtlasSprite.java
+++ b/net/minecraft/src/TextureAtlasSprite.java
@@ -206,7 +206,6 @@
         this.field_130223_c = var4.getWidth();
         int[] var5 = new int[this.field_130224_d * this.field_130223_c];
         var4.getRGB(0, 0, this.field_130223_c, this.field_130224_d, var5, 0, this.field_130223_c);
-        int var6;
 
         if (var3 == null)
         {
@@ -219,7 +218,7 @@
         }
         else
         {
-            var6 = this.field_130224_d / this.field_130223_c;
+            int var6 = this.field_130224_d / this.field_130223_c;
             int var7 = this.field_130223_c;
             int var8 = this.field_130223_c;
             this.field_130224_d = this.field_130223_c;
@@ -257,14 +256,6 @@
                 this.field_110982_k = new AnimationMetadataSection(var11, this.field_130223_c, this.field_130224_d, var3.func_110469_d());
             }
         }
-
-        for (var6 = 0; var6 < this.field_110976_a.size(); ++var6)
-        {
-            if (this.field_110976_a.get(var6) == null)
-            {
-                this.field_110976_a.set(var6, new int[this.field_130223_c * this.field_130224_d]);
-            }
-        }
     }
 
     private void func_130099_d(int par1)
@@ -298,14 +289,6 @@
     public void func_110968_a(List par1List)
     {
         this.field_110976_a = par1List;
-
-        for (int var2 = 0; var2 < this.field_110976_a.size(); ++var2)
-        {
-            if (this.field_110976_a.get(var2) == null)
-            {
-                this.field_110976_a.set(var2, new int[this.field_130223_c * this.field_130224_d]);
-            }
-        }
     }
 
     private void func_130102_n()
