--- a/net/minecraft/src/TextureMetadataSectionSerializer.java
+++ b/net/minecraft/src/TextureMetadataSectionSerializer.java
@@ -15,10 +15,7 @@
         return new TextureMetadataSection(var5, var6);
     }
 
-    /**
-     * The name of this section type as it appears in JSON.
-     */
-    public String getSectionName()
+    public String func_110483_a()
     {
         return "texture";
     }
