--- a/net/minecraft/src/TextureMap.java
+++ b/net/minecraft/src/TextureMap.java
@@ -24,8 +24,8 @@
     private final Map mapUploadedSprites = Maps.newHashMap();
 
     /** 0 = terrain.png, 1 = items.png */
-    public final int textureType;
-    public final String basePath;
+    private final int textureType;
+    private final String basePath;
     private final TextureAtlasSprite missingImage = new TextureAtlasSprite("missingno");
     public static TextureMap textureMapBlocks = null;
     public static TextureMap textureMapItems = null;
@@ -556,7 +556,7 @@
         else
         {
             int[] data = ts.getFrameTextureData(0);
-            image.setRGB(ts.getOriginX(), ts.getOriginY(), ts.getIconWidth(), ts.getIconHeight(), data, 0, ts.getIconWidth());
+            image.setRGB(ts.originX, ts.originY, ts.width, ts.height, data, 0, ts.width);
         }
     }
 
