--- a/net/minecraft/src/ThreadOnlineConnect.java
+++ b/net/minecraft/src/ThreadOnlineConnect.java
@@ -22,43 +22,43 @@
     {
         try
         {
-            TaskOnlineConnect.func_96583_a(this.field_96594_c, new NetClientHandler(this.field_96594_c.getMinecraft(), this.field_96595_a, this.field_96593_b, TaskOnlineConnect.func_98172_a(this.field_96594_c)));
+            TaskOnlineConnect.func_96583_a(this.field_96594_c, new NetClientHandler(this.field_96594_c.func_96578_b(), this.field_96595_a, this.field_96593_b, TaskOnlineConnect.func_98172_a(this.field_96594_c)));
 
-            if (this.field_96594_c.wasScreenClosed())
+            if (this.field_96594_c.func_96577_c())
             {
                 return;
             }
 
-            this.field_96594_c.setMessage(I18n.getString("mco.connect.authorizing"));
-            TaskOnlineConnect.func_96580_a(this.field_96594_c).addToSendQueue(new Packet2ClientProtocol(78, this.field_96594_c.getMinecraft().getSession().getUsername(), this.field_96595_a, this.field_96593_b));
+            this.field_96594_c.func_96576_b(I18n.func_135053_a("mco.connect.authorizing"));
+            TaskOnlineConnect.func_96580_a(this.field_96594_c).addToSendQueue(new Packet2ClientProtocol(74, this.field_96594_c.func_96578_b().func_110432_I().func_111285_a(), this.field_96595_a, this.field_96593_b));
         }
         catch (UnknownHostException var2)
         {
-            if (this.field_96594_c.wasScreenClosed())
+            if (this.field_96594_c.func_96577_c())
             {
                 return;
             }
 
-            this.field_96594_c.getMinecraft().displayGuiScreen(new GuiScreenDisconnectedOnline(TaskOnlineConnect.func_98172_a(this.field_96594_c), "connect.failed", "disconnect.genericReason", new Object[] {"Unknown host \'" + this.field_96595_a + "\'"}));
+            this.field_96594_c.func_96578_b().displayGuiScreen(new GuiScreenDisconnectedOnline(TaskOnlineConnect.func_98172_a(this.field_96594_c), "connect.failed", "disconnect.genericReason", new Object[] {"Unknown host \'" + this.field_96595_a + "\'"}));
         }
         catch (ConnectException var3)
         {
-            if (this.field_96594_c.wasScreenClosed())
+            if (this.field_96594_c.func_96577_c())
             {
                 return;
             }
 
-            this.field_96594_c.getMinecraft().displayGuiScreen(new GuiScreenDisconnectedOnline(TaskOnlineConnect.func_98172_a(this.field_96594_c), "connect.failed", "disconnect.genericReason", new Object[] {var3.getMessage()}));
+            this.field_96594_c.func_96578_b().displayGuiScreen(new GuiScreenDisconnectedOnline(TaskOnlineConnect.func_98172_a(this.field_96594_c), "connect.failed", "disconnect.genericReason", new Object[] {var3.getMessage()}));
         }
         catch (Exception var4)
         {
-            if (this.field_96594_c.wasScreenClosed())
+            if (this.field_96594_c.func_96577_c())
             {
                 return;
             }
 
             var4.printStackTrace();
-            this.field_96594_c.getMinecraft().displayGuiScreen(new GuiScreenDisconnectedOnline(TaskOnlineConnect.func_98172_a(this.field_96594_c), "connect.failed", "disconnect.genericReason", new Object[] {var4.toString()}));
+            this.field_96594_c.func_96578_b().displayGuiScreen(new GuiScreenDisconnectedOnline(TaskOnlineConnect.func_98172_a(this.field_96594_c), "connect.failed", "disconnect.genericReason", new Object[] {var4.toString()}));
         }
     }
 }
