--- a/net/minecraft/src/ThreadConnectToServer.java
+++ b/net/minecraft/src/ThreadConnectToServer.java
@@ -32,7 +32,7 @@
                 return;
             }
 
-            GuiConnecting.getNetClientHandler(this.connectingGui).addToSendQueue(new Packet2ClientProtocol(78, GuiConnecting.func_74254_c(this.connectingGui).getSession().getUsername(), this.ip, this.port));
+            GuiConnecting.getNetClientHandler(this.connectingGui).addToSendQueue(new Packet2ClientProtocol(74, GuiConnecting.func_74254_c(this.connectingGui).func_110432_I().func_111285_a(), this.ip, this.port));
         }
         catch (UnknownHostException var2)
         {
