--- a/net/minecraft/src/ThreadOnlineScreen.java
+++ b/net/minecraft/src/ThreadOnlineScreen.java
@@ -17,7 +17,7 @@
 
             if (var1 != null)
             {
-                McoClient var2 = new McoClient(GuiScreenOnlineServers.func_98075_b(this.field_98173_a).getSession());
+                McoClient var2 = new McoClient(GuiScreenOnlineServers.func_98075_b(this.field_98173_a).func_110432_I());
                 GuiScreenOnlineServers.func_140040_h().func_140058_a(var1);
                 GuiScreenOnlineServers.func_140013_c(this.field_98173_a).remove(var1);
                 var2.func_140055_c(var1.field_96408_a);
