--- a/net/minecraft/src/ThreadDownloadImageData.java
+++ b/net/minecraft/src/ThreadDownloadImageData.java
@@ -2,7 +2,6 @@
 
 import java.awt.image.BufferedImage;
 import java.io.IOException;
-import java.net.URL;
 
 public class ThreadDownloadImageData extends AbstractTexture
 {
@@ -12,7 +11,6 @@
     private Thread imageThread;
     private SimpleTexture imageLocation;
     private boolean textureUploaded;
-    public boolean enabled = true;
 
     public ThreadDownloadImageData(String par1Str, ResourceLocation par2ResourceLocation, IImageBuffer par3IImageBuffer)
     {
@@ -60,42 +58,13 @@
             this.imageThread.setDaemon(true);
             this.imageThread.setName("Skin downloader: " + this.imageUrl);
             this.imageThread.start();
-
-            try
-            {
-                URL var2 = new URL(this.imageUrl);
-                String var3 = var2.getPath();
-                String var4 = "/MinecraftSkins/";
-                String var5 = "/MinecraftCloaks/";
-
-                if (var3.startsWith(var5))
-                {
-                    String var6 = var3.substring(var5.length());
-                    String var7 = "http://s.optifine.net/capes/" + var6;
-                    ThreadDownloadImage var8 = new ThreadDownloadImage(this, var7, new ImageBufferDownload());
-                    var8.setDaemon(true);
-                    var8.setName("Cape downloader: " + this.imageUrl);
-                    var8.start();
-                }
-            }
-            catch (Exception var9)
-            {
-                ;
-            }
         }
     }
 
     public boolean isTextureUploaded()
     {
-        if (!this.enabled)
-        {
-            return false;
-        }
-        else
-        {
-            this.getGlTextureId();
-            return this.textureUploaded;
-        }
+        this.getGlTextureId();
+        return this.textureUploaded;
     }
 
     static String getImageUrl(ThreadDownloadImageData par0ThreadDownloadImageData)
