--- a/net/minecraft/src/ThreadDownloadImageDataINNER1.java
+++ b/net/minecraft/src/ThreadDownloadImageDataINNER1.java
@@ -25,23 +25,23 @@
             var1.setDoOutput(false);
             var1.connect();
 
-            if (var1.getResponseCode() / 100 == 2)
+            if (var1.getResponseCode() / 100 != 2)
             {
-                BufferedImage var2 = ImageIO.read(var1.getInputStream());
+                return;
+            }
 
-                if (ThreadDownloadImageData.getImageBuffer(this.theThreadDownloadImageData) != null)
-                {
-                    var2 = ThreadDownloadImageData.getImageBuffer(this.theThreadDownloadImageData).parseUserSkin(var2);
-                }
+            BufferedImage var2 = ImageIO.read(var1.getInputStream());
 
-                this.theThreadDownloadImageData.getBufferedImage(var2);
-                return;
+            if (ThreadDownloadImageData.getImageBuffer(this.theThreadDownloadImageData) != null)
+            {
+                var2 = ThreadDownloadImageData.getImageBuffer(this.theThreadDownloadImageData).parseUserSkin(var2);
             }
+
+            this.theThreadDownloadImageData.getBufferedImage(var2);
         }
         catch (Exception var6)
         {
             var6.printStackTrace();
-            return;
         }
         finally
         {
