--- a/net/minecraft/src/TileEntitySkullRenderer.java
+++ b/net/minecraft/src/TileEntitySkullRenderer.java
@@ -38,32 +38,28 @@
         {
             case 0:
             default:
-                this.bindTexture(field_110642_c);
+                this.func_110628_a(field_110642_c);
                 break;
-
             case 1:
-                this.bindTexture(field_110640_d);
+                this.func_110628_a(field_110640_d);
                 break;
-
             case 2:
-                this.bindTexture(field_110641_e);
+                this.func_110628_a(field_110641_e);
                 var8 = this.field_82395_d;
                 break;
-
             case 3:
-                ResourceLocation var9 = AbstractClientPlayer.locationStevePng;
+                ResourceLocation var9 = AbstractClientPlayer.field_110314_b;
 
                 if (par7Str != null && par7Str.length() > 0)
                 {
-                    var9 = AbstractClientPlayer.getLocationSkull(par7Str);
-                    AbstractClientPlayer.getDownloadImageSkin(var9, par7Str);
+                    var9 = AbstractClientPlayer.func_110305_h(par7Str);
+                    AbstractClientPlayer.func_110304_a(var9, par7Str);
                 }
 
-                this.bindTexture(var9);
+                this.func_110628_a(var9);
                 break;
-
             case 4:
-                this.bindTexture(field_110639_f);
+                this.func_110628_a(field_110639_f);
         }
 
         GL11.glPushMatrix();
@@ -76,17 +72,14 @@
                 case 2:
                     GL11.glTranslatef(par1 + 0.5F, par2 + 0.25F, par3 + 0.74F);
                     break;
-
                 case 3:
                     GL11.glTranslatef(par1 + 0.5F, par2 + 0.25F, par3 + 0.26F);
                     par5 = 180.0F;
                     break;
-
                 case 4:
                     GL11.glTranslatef(par1 + 0.74F, par2 + 0.25F, par3 + 0.5F);
                     par5 = 270.0F;
                     break;
-
                 case 5:
                 default:
                     GL11.glTranslatef(par1 + 0.26F, par2 + 0.25F, par3 + 0.5F);
