--- a/net/minecraft/src/TileEntityEnderChestRenderer.java
+++ b/net/minecraft/src/TileEntityEnderChestRenderer.java
@@ -22,7 +22,7 @@
             var9 = par1TileEntityEnderChest.getBlockMetadata();
         }
 
-        this.bindTexture(field_110637_a);
+        this.func_110628_a(field_110637_a);
         GL11.glPushMatrix();
         GL11.glEnable(GL12.GL_RESCALE_NORMAL);
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
