--- a/net/minecraft/src/TileEntitySkullRenderer.java
+++ b/net/minecraft/src/TileEntitySkullRenderer.java
@@ -40,16 +40,13 @@
             default:
                 this.func_110628_a(field_110642_c);
                 break;
-
             case 1:
                 this.func_110628_a(field_110640_d);
                 break;
-
             case 2:
                 this.func_110628_a(field_110641_e);
                 var8 = this.field_82395_d;
                 break;
-
             case 3:
                 ResourceLocation var9 = AbstractClientPlayer.field_110314_b;
 
@@ -61,7 +58,6 @@
 
                 this.func_110628_a(var9);
                 break;
-
             case 4:
                 this.func_110628_a(field_110639_f);
         }
@@ -76,17 +72,14 @@
                 case 2:
                     GL11.glTranslatef(par1 + 0.5F, par2 + 0.25F, par3 + 0.74F);
                     break;
-
                 case 3:
                     GL11.glTranslatef(par1 + 0.5F, par2 + 0.25F, par3 + 0.26F);
                     par5 = 180.0F;
                     break;
-
                 case 4:
                     GL11.glTranslatef(par1 + 0.74F, par2 + 0.25F, par3 + 0.5F);
                     par5 = 270.0F;
                     break;
-
                 case 5:
                 default:
                     GL11.glTranslatef(par1 + 0.26F, par2 + 0.25F, par3 + 0.5F);
