--- a/net/minecraft/src/TileEntityHopper.java
+++ b/net/minecraft/src/TileEntityHopper.java
@@ -219,12 +219,12 @@
             if (!this.isCoolingDown())
             {
                 this.setTransferCooldown(0);
-                this.updateHopper();
+                this.func_98045_j();
             }
         }
     }
 
-    public boolean updateHopper()
+    public boolean func_98045_j()
     {
         if (this.worldObj != null && !this.worldObj.isRemote)
         {
@@ -301,7 +301,7 @@
 
                 for (int var5 = 0; var5 < var8.length; ++var5)
                 {
-                    if (insertStackFromInventory(par0Hopper, var1, var8[var5], var2))
+                    if (func_102012_a(par0Hopper, var1, var8[var5], var2))
                     {
                         return true;
                     }
@@ -313,7 +313,7 @@
 
                 for (int var4 = 0; var4 < var3; ++var4)
                 {
-                    if (insertStackFromInventory(par0Hopper, var1, var4, var2))
+                    if (func_102012_a(par0Hopper, var1, var4, var2))
                     {
                         return true;
                     }
@@ -322,18 +322,18 @@
         }
         else
         {
-            EntityItem var6 = getEntityAbove(par0Hopper.getWorldObj(), par0Hopper.getXPos(), par0Hopper.getYPos() + 1.0D, par0Hopper.getZPos());
+            EntityItem var6 = func_96119_a(par0Hopper.getWorldObj(), par0Hopper.getXPos(), par0Hopper.getYPos() + 1.0D, par0Hopper.getZPos());
 
             if (var6 != null)
             {
-                return insertStackFromEntity(par0Hopper, var6);
+                return func_96114_a(par0Hopper, var6);
             }
         }
 
         return false;
     }
 
-    private static boolean insertStackFromInventory(Hopper par0Hopper, IInventory par1IInventory, int par2, int par3)
+    private static boolean func_102012_a(Hopper par0Hopper, IInventory par1IInventory, int par2, int par3)
     {
         ItemStack var4 = par1IInventory.getStackInSlot(par2);
 
@@ -354,7 +354,7 @@
         return false;
     }
 
-    public static boolean insertStackFromEntity(IInventory par0IInventory, EntityItem par1EntityItem)
+    public static boolean func_96114_a(IInventory par0IInventory, EntityItem par1EntityItem)
     {
         boolean var2 = false;
 
@@ -482,7 +482,7 @@
         return getInventoryAtLocation(par0Hopper.getWorldObj(), par0Hopper.getXPos(), par0Hopper.getYPos() + 1.0D, par0Hopper.getZPos());
     }
 
-    public static EntityItem getEntityAbove(World par0World, double par1, double par3, double par5)
+    public static EntityItem func_96119_a(World par0World, double par1, double par3, double par5)
     {
         List var7 = par0World.selectEntitiesWithinAABB(EntityItem.class, AxisAlignedBB.getAABBPool().getAABB(par1, par3, par5, par1 + 1.0D, par3 + 1.0D, par5 + 1.0D), IEntitySelector.selectAnything);
         return var7.size() > 0 ? (EntityItem)var7.get(0) : null;
