--- a/net/minecraft/src/TileEntityChestRenderer.java
+++ b/net/minecraft/src/TileEntityChestRenderer.java
@@ -6,12 +6,12 @@
 
 public class TileEntityChestRenderer extends TileEntitySpecialRenderer
 {
-    private static final ResourceLocation RES_TRAPPED_DOUBLE = new ResourceLocation("textures/entity/chest/trapped_double.png");
-    private static final ResourceLocation RES_CHRISTMAS_DOUBLE = new ResourceLocation("textures/entity/chest/christmas_double.png");
-    private static final ResourceLocation RES_NORMAL_DOUBLE = new ResourceLocation("textures/entity/chest/normal_double.png");
-    private static final ResourceLocation RES_TRAPPED_SINGLE = new ResourceLocation("textures/entity/chest/trapped.png");
-    private static final ResourceLocation RES_CHRISTMAS_SINGLE = new ResourceLocation("textures/entity/chest/christmas.png");
-    private static final ResourceLocation RES_NORMAL_SINGLE = new ResourceLocation("textures/entity/chest/normal.png");
+    private static final ResourceLocation field_110635_a = new ResourceLocation("textures/entity/chest/trapped_double.png");
+    private static final ResourceLocation field_110634_c = new ResourceLocation("textures/entity/chest/christmas_double.png");
+    private static final ResourceLocation field_110632_d = new ResourceLocation("textures/entity/chest/normal_double.png");
+    private static final ResourceLocation field_110633_e = new ResourceLocation("textures/entity/chest/trapped.png");
+    private static final ResourceLocation field_110630_f = new ResourceLocation("textures/entity/chest/christmas.png");
+    private static final ResourceLocation field_110631_g = new ResourceLocation("textures/entity/chest/normal.png");
 
     /** The normal small chest model. */
     private ModelChest chestModel = new ModelChest();
@@ -65,34 +65,34 @@
             {
                 var14 = this.chestModel;
 
-                if (par1TileEntityChest.getChestType() == 1)
+                if (par1TileEntityChest.func_98041_l() == 1)
                 {
-                    this.bindTexture(RES_TRAPPED_SINGLE);
+                    this.func_110628_a(field_110633_e);
                 }
                 else if (this.isChristmas)
                 {
-                    this.bindTexture(RES_CHRISTMAS_SINGLE);
+                    this.func_110628_a(field_110630_f);
                 }
                 else
                 {
-                    this.bindTexture(RES_NORMAL_SINGLE);
+                    this.func_110628_a(field_110631_g);
                 }
             }
             else
             {
                 var14 = this.largeChestModel;
 
-                if (par1TileEntityChest.getChestType() == 1)
+                if (par1TileEntityChest.func_98041_l() == 1)
                 {
-                    this.bindTexture(RES_TRAPPED_DOUBLE);
+                    this.func_110628_a(field_110635_a);
                 }
                 else if (this.isChristmas)
                 {
-                    this.bindTexture(RES_CHRISTMAS_DOUBLE);
+                    this.func_110628_a(field_110634_c);
                 }
                 else
                 {
-                    this.bindTexture(RES_NORMAL_DOUBLE);
+                    this.func_110628_a(field_110632_d);
                 }
             }
 
