--- a/net/minecraft/src/TileEntityRendererPiston.java
+++ b/net/minecraft/src/TileEntityRendererPiston.java
@@ -14,7 +14,7 @@
         if (var9 != null && par1TileEntityPiston.getProgress(par8) < 1.0F)
         {
             Tessellator var10 = Tessellator.instance;
-            this.bindTexture(TextureMap.locationBlocksTexture);
+            this.func_110628_a(TextureMap.field_110575_b);
             RenderHelper.disableStandardItemLighting();
             GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
             GL11.glEnable(GL11.GL_BLEND);
