--- a/net/minecraft/src/TileEntitySignRenderer.java
+++ b/net/minecraft/src/TileEntitySignRenderer.java
@@ -49,7 +49,7 @@
             this.modelSign.signStick.showModel = false;
         }
 
-        this.bindTexture(field_110638_a);
+        this.func_110628_a(field_110638_a);
         GL11.glPushMatrix();
         GL11.glScalef(var10, -var10, -var10);
         this.modelSign.renderSign();
