--- a/net/minecraft/src/VersionCheckThread.java
+++ b/net/minecraft/src/VersionCheckThread.java
@@ -13,7 +13,7 @@
         try
         {
             Config.dbg("Checking for new version");
-            URL e = new URL("http://optifine.net/version/1.6.4/HD_U.txt");
+            URL e = new URL("http://optifine.net/version/1.6.2/HD_U.txt");
             conn = (HttpURLConnection)e.openConnection();
             conn.setDoInput(true);
             conn.setDoOutput(false);
@@ -34,7 +34,7 @@
                 String newVer = verLines[0];
                 Config.dbg("Version found: " + newVer);
 
-                if (Config.compareRelease(newVer, "C6") > 0)
+                if (Config.compareRelease(newVer, "B3") > 0)
                 {
                     Config.setNewRelease(newVer);
                     return;
