--- a/net/minecraft/src/VersionCheckThread.java
+++ b/net/minecraft/src/VersionCheckThread.java
@@ -26,16 +26,16 @@
                 in.close();
                 String[] verLines = Config.tokenize(verStr, "\n\r");
 
-                if (verLines.length >= 1)
+                if (verLines.length < 1)
                 {
-                    String newVer = verLines[0];
-                    Config.dbg("Version found: " + newVer);
+                    return;
+                }
 
-                    if (Config.compareRelease(newVer, "B2") <= 0)
-                    {
-                        return;
-                    }
+                String newVer = verLines[0];
+                Config.dbg("Version found: " + newVer);
 
+                if (Config.compareRelease(newVer, "B1") > 0)
+                {
                     Config.setNewRelease(newVer);
                     return;
                 }
