--- a/net/minecraft/src/VillageSiege.java
+++ b/net/minecraft/src/VillageSiege.java
@@ -181,7 +181,7 @@
             try
             {
                 var2 = new EntityZombie(this.worldObj);
-                var2.onSpawnWithEgg((EntityLivingData)null);
+                var2.func_110161_a((EntityLivingData)null);
                 var2.setVillager(false);
             }
             catch (Exception var4)
@@ -193,7 +193,7 @@
             var2.setLocationAndAngles(var1.xCoord, var1.yCoord, var1.zCoord, this.worldObj.rand.nextFloat() * 360.0F, 0.0F);
             this.worldObj.spawnEntityInWorld(var2);
             ChunkCoordinates var3 = this.theVillage.getCenter();
-            var2.setHomeArea(var3.posX, var3.posY, var3.posZ, this.theVillage.getVillageRadius());
+            var2.func_110171_b(var3.posX, var3.posY, var3.posZ, this.theVillage.getVillageRadius());
             return true;
         }
     }
