--- a/net/minecraft/src/WeightedRandomMinecart.java
+++ b/net/minecraft/src/WeightedRandomMinecart.java
@@ -23,11 +23,9 @@
                     case 0:
                         var4 = "MinecartRideable";
                         break;
-
                     case 1:
                         var4 = "MinecartChest";
                         break;
-
                     case 2:
                         var4 = "MinecartFurnace";
                 }
@@ -56,11 +54,9 @@
                     case 0:
                         par3Str = "MinecartRideable";
                         break;
-
                     case 1:
                         par3Str = "MinecartChest";
                         break;
-
                     case 2:
                         par3Str = "MinecartFurnace";
                 }
