--- a/net/minecraft/src/WorldClient.java
+++ b/net/minecraft/src/WorldClient.java
@@ -255,7 +255,7 @@
         this.sendQueue.quitWithPacket(new Packet255KickDisconnect("Quitting"));
     }
 
-    public IUpdatePlayerListBox getMinecartSoundUpdater(EntityMinecart par1EntityMinecart)
+    public IUpdatePlayerListBox func_82735_a(EntityMinecart par1EntityMinecart)
     {
         return new SoundUpdaterMinecart(this.mc.sndManager, par1EntityMinecart, this.mc.thePlayer);
     }
