--- a/net/minecraft/src/WorldServer.java
+++ b/net/minecraft/src/WorldServer.java
@@ -26,12 +26,8 @@
     /** is false if there are no players */
     private boolean allPlayersSleeping;
     private int updateEntityTick;
-
-    /**
-     * the teleporter to use when the entity is being transferred into the dimension
-     */
-    private final Teleporter worldTeleporter;
-    private final SpawnerAnimals animalSpawner = new SpawnerAnimals();
+    private final Teleporter field_85177_Q;
+    private final SpawnerAnimals field_135059_Q = new SpawnerAnimals();
 
     /**
      * Double buffer of ServerBlockEventList[] for holding pending BlockEventData's
@@ -71,7 +67,7 @@
             this.pendingTickListEntriesTreeSet = new TreeSet();
         }
 
-        this.worldTeleporter = new Teleporter(this);
+        this.field_85177_Q = new Teleporter(this);
         this.worldScoreboard = new ServerScoreboard(par1MinecraftServer);
         ScoreboardSaveData var8 = (ScoreboardSaveData)this.mapStorage.loadData(ScoreboardSaveData.class, "scoreboard");
 
@@ -114,7 +110,7 @@
 
         if (this.getGameRules().getGameRuleBooleanValue("doMobSpawning"))
         {
-            this.animalSpawner.findChunksForSpawning(this, this.spawnHostileMobs, this.spawnPeacefulMobs, this.worldInfo.getWorldTotalTime() % 400L == 0L);
+            this.field_135059_Q.findChunksForSpawning(this, this.spawnHostileMobs, this.spawnPeacefulMobs, this.worldInfo.getWorldTotalTime() % 400L == 0L);
         }
 
         this.theProfiler.endStartSection("chunkSource");
@@ -143,7 +139,7 @@
         this.villageCollectionObj.tick();
         this.villageSiegeObj.tick();
         this.theProfiler.endStartSection("portalForcer");
-        this.worldTeleporter.removeStalePortalLocations(this.getTotalWorldTime());
+        this.field_85177_Q.removeStalePortalLocations(this.getTotalWorldTime());
         this.theProfiler.endSection();
         this.sendAndApplyBlockEvents();
     }
@@ -549,7 +545,7 @@
                                 var10 = -1;
                             }
 
-                            CrashReportCategory.addBlockCrashInfo(var9, var4.xCoord, var4.yCoord, var4.zCoord, var6, var10);
+                            CrashReportCategory.func_85068_a(var9, var4.xCoord, var4.yCoord, var4.zCoord, var6, var10);
                             throw new ReportedException(var8);
                         }
                     }
@@ -672,7 +668,7 @@
      */
     public boolean canMineBlock(EntityPlayer par1EntityPlayer, int par2, int par3, int par4)
     {
-        return !this.mcServer.isBlockProtected(this, par2, par3, par4, par1EntityPlayer);
+        return !this.mcServer.func_96290_a(this, par2, par3, par4, par1EntityPlayer);
     }
 
     protected void initialize(WorldSettings par1WorldSettings)
@@ -801,14 +797,11 @@
         }
     }
 
-    /**
-     * saves chunk data - currently only called during execution of the Save All command
-     */
-    public void saveChunkData()
+    public void func_104140_m()
     {
         if (this.chunkProvider.canSave())
         {
-            this.chunkProvider.saveExtraData();
+            this.chunkProvider.func_104112_b();
         }
     }
 
@@ -1025,6 +1018,6 @@
 
     public Teleporter getDefaultTeleporter()
     {
-        return this.worldTeleporter;
+        return this.field_85177_Q;
     }
 }
