--- a/net/minecraft/server/management/PlayerManager.java
+++ b/net/minecraft/server/management/PlayerManager.java
@@ -57,7 +57,6 @@
     /** x, z direction vectors: east, south, west, north */
     private final int[][] xzDirectionsConst = new int[][] {{1, 0}, {0, 1}, { -1, 0}, {0, -1}};
     private static final String __OBFID = "CL_00001434";
-    private static final String __OBFID = "CL_00001434";
 
     public PlayerManager(WorldServer par1Minecraft)
     {
@@ -469,7 +468,6 @@
         private long previousWorldTime;
         private static final String __OBFID = "CL_00001435";
         public boolean chunkLoaded;
-        private static final String __OBFID = "CL_00001435";
 
         public PlayerInstance(int par2, int par3)
         {
