--- a/net/minecraft/client/renderer/culling/ClippingHelper.java
+++ b/net/minecraft/client/renderer/culling/ClippingHelper.java
@@ -7,7 +7,6 @@
     public float[] modelviewMatrix = new float[16];
     public float[] clippingMatrix = new float[16];
     private static final String __OBFID = "CL_00000977";
-    private static final String __OBFID = "CL_00000977";
 
     /**
      * Returns true if the box is inside all 6 clipping planes, otherwise returns false.
