--- a/net/minecraft/client/settings/KeyBinding.java
+++ b/net/minecraft/client/settings/KeyBinding.java
@@ -10,17 +10,18 @@
 
 public class KeyBinding implements Comparable
 {
-    private static final List keybindArray = new ArrayList();
-    private static final IntHashMap hash = new IntHashMap();
-    private static final Set keybindSet = new HashSet();
-    private final String keyDescription;
-    private final int keyCodeDefault;
-    private final String keyCategory;
-    private int keyCode;
+    // Minecrift - need to access these via reflection
+    public static final List keybindArray = new ArrayList();
+    public static final IntHashMap hash = new IntHashMap();
+    public static final Set keybindSet = new HashSet();
+    public final String keyDescription;
+    public final int keyCodeDefault;
+    public final String keyCategory;
+    public int keyCode;
 
     /** because _303 wanted me to call it that(Caironater) */
-    private boolean pressed;
-    private int presses;
+    public boolean pressed;
+    public int presses;
     private static final String __OBFID = "CL_00000628";
 
     public static void onTick(int p_74507_0_)
