--- a/net/minecraft/client/renderer/OpenGlHelper.java
+++ b/net/minecraft/client/renderer/OpenGlHelper.java
@@ -63,7 +63,6 @@
     private static final String __OBFID = "CL_00001179";
     public static float lastBrightnessX = 0.0F;
     public static float lastBrightnessY = 0.0F;
-    private static final String __OBFID = "CL_00001179";
 
     /**
      * Initializes the texture constants to be used when rendering lightmap values
