--- a/net/minecraft/client/renderer/RenderGlobal.java
+++ b/net/minecraft/client/renderer/RenderGlobal.java
@@ -10,6 +10,8 @@
 import java.util.Map;
 import java.util.Random;
 import java.util.concurrent.Callable;
+
+import com.mtbs3d.minecrift.settings.VRSettings;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockChest;
 import net.minecraft.block.material.Material;
@@ -48,6 +50,7 @@
 import net.minecraft.client.particle.EntitySuspendFX;
 import net.minecraft.client.renderer.culling.ICamera;
 import net.minecraft.client.renderer.entity.RenderManager;
+import net.minecraft.client.renderer.entity.RenderPlayer;
 import net.minecraft.client.renderer.texture.IIconRegister;
 import net.minecraft.client.renderer.texture.TextureManager;
 import net.minecraft.client.renderer.texture.TextureMap;
@@ -262,7 +265,6 @@
     private long lastMovedTime = System.currentTimeMillis();
     private long lastActionTime = System.currentTimeMillis();
     private static AxisAlignedBB AABB_INFINITE = AxisAlignedBB.getBoundingBox(Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY);
-    private static final String __OBFID = "CL_00000954";
 
     public RenderGlobal(Minecraft par1Minecraft)
     {
@@ -662,7 +664,20 @@
                         }
                     }
 
-                    if (te && (var19 != this.mc.renderViewEntity || this.mc.gameSettings.thirdPersonView != 0 || this.mc.renderViewEntity.isPlayerSleeping()) && this.theWorld.blockExists(MathHelper.floor_double(var19.posX), 0, MathHelper.floor_double(var19.posZ)))
+                    // Minecrift - logic to determine whether to render head, headwear etc...
+                    RenderPlayer renderPlayer = (RenderPlayer)RenderManager.instance.getEntityRenderObject(this.mc.thePlayer);
+
+                    if (te && (var19 == this.mc.renderViewEntity && this.mc.gameSettings.thirdPersonView == 0 && !mc.renderViewEntity.isPlayerSleeping()) && this.theWorld.blockExists(MathHelper.floor_double(var19.posX), 0, MathHelper.floor_double(var19.posZ)))
+                    {
+                        renderPlayer.setVR(true,this.mc.vrSettings.renderHeadWear, this.mc.vrSettings.renderPlayerOffset);
+                        if (this.mc.vrSettings.renderFullFirstPersonModelMode == VRSettings.RENDER_FIRST_PERSON_HAND ||
+                            this.mc.vrSettings.renderFullFirstPersonModelMode == VRSettings.RENDER_FIRST_PERSON_NONE)
+                        {
+                            te = false;
+                        }
+                    }
+
+                    if (te)
                     {
                         ++this.countEntitiesRendered;
 
@@ -675,6 +690,12 @@
                         RenderManager.instance.func_147937_a(var19, p_147589_3_);
                         this.renderedEntity = null;
                     }
+
+                    if (var19 == this.mc.renderViewEntity && this.mc.gameSettings.thirdPersonView == 0 && !mc.renderViewEntity.isPlayerSleeping() )
+                    {
+                        renderPlayer.setVR(false,true,0.0F);
+                    }
+                    // end Minecrift
                 }
             }
 
@@ -1765,7 +1786,7 @@
                     float partialTicks1 = par1;
                     par1 = 0.0F;
                     GL11.glDisable(GL11.GL_CULL_FACE);
-                    float var21 = (float)(this.mc.renderViewEntity.lastTickPosY + (this.mc.renderViewEntity.posY - this.mc.renderViewEntity.lastTickPosY) * (double)par1);
+                    float var2 = (float)(this.mc.renderViewEntity.lastTickPosY + (this.mc.renderViewEntity.posY - this.mc.renderViewEntity.lastTickPosY) * (double)par1);
                     byte var3 = 32;
                     int var4 = 256 / var3;
                     Tessellator var5 = Tessellator.instance;
@@ -1802,7 +1823,7 @@
                         int var18 = MathHelper.floor_double(cdx / 2048.0D);
                         dc -= (double)(cdz * 2048);
                         cdx -= (double)(var18 * 2048);
-                        float var19 = this.theWorld.provider.getCloudHeight() - var21 + 0.33F;
+                        float var19 = this.theWorld.provider.getCloudHeight() - var2 + 0.33F;
                         var19 += this.mc.gameSettings.ofCloudsHeight * 128.0F;
                         float var20 = (float)(dc * (double)var10);
                         float var21 = (float)(cdx * (double)var10);
@@ -2471,15 +2492,10 @@
             var16.addCrashSectionCallable("Position", new Callable()
             {
                 private static final String __OBFID = "CL_00000955";
-                private static final String __OBFID = "CL_00000955";
                 public String call()
                 {
                     return CrashReportCategory.func_85074_a(par2, par4, par6);
                 }
-                public Object call() throws Exception
-                {
-                    return this.call();
-                }
             });
             throw new ReportedException(var15);
         }
@@ -2502,7 +2518,7 @@
             double var15 = this.mc.renderViewEntity.posX - par2;
             double var17 = this.mc.renderViewEntity.posY - par4;
             double var19 = this.mc.renderViewEntity.posZ - par6;
-            Object var21 = null;
+            EntityFX var21 = null;
 
             if (par1Str.equals("hugeexplosion"))
             {
