--- a/net/minecraft/client/renderer/texture/TextureAtlasSprite.java
+++ b/net/minecraft/client/renderer/texture/TextureAtlasSprite.java
@@ -47,7 +47,6 @@
     private int uploadedOwnFrameIndex = -1;
     public IntBuffer[] frameBuffers;
     public Mipmaps[] frameMipmaps;
-    private static final String __OBFID = "CL_00001062";
 
     protected TextureAtlasSprite(String par1Str)
     {
@@ -330,7 +329,6 @@
                     var7.addCrashSectionCallable("Frame sizes", new Callable()
                     {
                         private static final String __OBFID = "CL_00001063";
-                        private static final String __OBFID = "CL_00001063";
                         public String call()
                         {
                             StringBuilder var1 = new StringBuilder();
@@ -351,10 +349,6 @@
 
                             return var1.toString();
                         }
-                        public Object call() throws Exception
-                        {
-                            return this.call();
-                        }
                     });
                     throw new ReportedException(var6);
                 }
