--- a/net/minecraft/client/renderer/texture/TextureMap.java
+++ b/net/minecraft/client/renderer/texture/TextureMap.java
@@ -63,7 +63,6 @@
     private int iconGridCountY = -1;
     private double iconGridSizeU = -1.0D;
     private double iconGridSizeV = -1.0D;
-    private static final String __OBFID = "CL_00001058";
 
     public TextureMap(int par1, String par2Str)
     {
@@ -243,41 +242,26 @@
                     var261.addCrashSectionCallable("Sprite name", new Callable()
                     {
                         private static final String __OBFID = "CL_00001059";
-                        private static final String __OBFID = "CL_00001059";
                         public String call()
                         {
                             return sheetWidth1.getIconName();
                         }
-                        public Object call() throws Exception
-                        {
-                            return this.call();
-                        }
                     });
                     var261.addCrashSectionCallable("Sprite size", new Callable()
                     {
                         private static final String __OBFID = "CL_00001060";
-                        private static final String __OBFID = "CL_00001060";
                         public String call()
                         {
                             return sheetWidth1.getIconWidth() + " x " + sheetWidth1.getIconHeight();
                         }
-                        public Object call() throws Exception
-                        {
-                            return this.call();
-                        }
                     });
                     var261.addCrashSectionCallable("Sprite frames", new Callable()
                     {
                         private static final String __OBFID = "CL_00001061";
-                        private static final String __OBFID = "CL_00001061";
                         public String call()
                         {
                             return sheetWidth1.getFrameCount() + " frames";
                         }
-                        public Object call() throws Exception
-                        {
-                            return this.call();
-                        }
                     });
                     var261.addCrashSection("Mipmap levels", Integer.valueOf(this.field_147636_j));
                     throw new ReportedException(debugImage1);
