--- a/net/minecraft/client/renderer/texture/TextureManager.java
+++ b/net/minecraft/client/renderer/texture/TextureManager.java
@@ -29,7 +29,6 @@
     private final Map mapTextureCounters = Maps.newHashMap();
     private IResourceManager theResourceManager;
     private static final String __OBFID = "CL_00001064";
-    private static final String __OBFID = "CL_00001064";
 
     public TextureManager(IResourceManager par1ResourceManager)
     {
@@ -109,15 +108,10 @@
             var6.addCrashSectionCallable("Texture object class", new Callable()
             {
                 private static final String __OBFID = "CL_00001065";
-                private static final String __OBFID = "CL_00001065";
                 public String call()
                 {
                     return par2TextureObject.getClass().getName();
                 }
-                public Object call() throws Exception
-                {
-                    return this.call();
-                }
             });
             throw new ReportedException(var5);
         }
