--- a/net/minecraft/server/integrated/IntegratedServer.java
+++ b/net/minecraft/server/integrated/IntegratedServer.java
@@ -36,7 +36,6 @@
     private boolean isPublic;
     private ThreadLanServerPing lanServerPing;
     private static final String __OBFID = "CL_00001129";
-    private static final String __OBFID = "CL_00001129";
 
     public IntegratedServer(Minecraft par1Minecraft, String par2Str, String par3Str, WorldSettings par4WorldSettings)
     {
@@ -253,20 +252,14 @@
         par1CrashReport.getCategory().addCrashSectionCallable("Type", new Callable()
         {
             private static final String __OBFID = "CL_00001130";
-            private static final String __OBFID = "CL_00001130";
             public String call()
             {
                 return "Integrated Server (map_client.txt)";
             }
-            public Object call() throws Exception
-            {
-                return this.call();
-            }
         });
         par1CrashReport.getCategory().addCrashSectionCallable("Is Modded", new Callable()
         {
             private static final String __OBFID = "CL_00001131";
-            private static final String __OBFID = "CL_00001131";
             public String call()
             {
                 String var1 = ClientBrandRetriever.getClientModName();
@@ -281,10 +274,6 @@
                     return !var1.equals("vanilla") ? "Definitely; Server brand changed to \'" + var1 + "\'" : (Minecraft.class.getSigners() == null ? "Very likely; Jar signature invalidated" : "Probably not. Jar signature remains and both client + server brands are untouched.");
                 }
             }
-            public Object call() throws Exception
-            {
-                return this.call();
-            }
         });
         return par1CrashReport;
     }
