--- a/net/minecraft/block/Block.java
+++ b/net/minecraft/block/Block.java
@@ -439,7 +439,7 @@
     /**
      * Sets how much light is blocked going through this block. Returns the object for convenience in constructing.
      */
-    protected Block setLightOpacity(int p_149713_1_)
+    public Block setLightOpacity(int p_149713_1_)
     {
         this.lightOpacity = p_149713_1_;
         return this;
