--- a/net/minecraft/util/LongHashMap.java
+++ b/net/minecraft/util/LongHashMap.java
@@ -21,7 +21,6 @@
     /** count of times elements have been added/removed */
     private transient volatile int modCount;
     private static final String __OBFID = "CL_00001492";
-    private static final String __OBFID = "CL_00001492";
 
     public LongHashMap()
     {
@@ -254,7 +253,6 @@
         LongHashMap.Entry nextEntry;
         final int hash;
         private static final String __OBFID = "CL_00001493";
-        private static final String __OBFID = "CL_00001493";
 
         Entry(int par1, long par2, Object par4Obj, LongHashMap.Entry par5LongHashMapEntry)
         {
