--- a/net/minecraft/util/MessageDeserializer2.java
+++ b/net/minecraft/util/MessageDeserializer2.java
@@ -35,13 +35,13 @@
                 {
                     int var7 = var6.readVarIntFromBuffer();
 
-                    if (p_decode_2_.readableBytes() < var7)
+                    if (p_decode_2_.readableBytes() >= var7)
                     {
-                        p_decode_2_.resetReaderIndex();
+                        p_decode_3_.add(p_decode_2_.readBytes(var7));
                         return;
                     }
 
-                    p_decode_3_.add(p_decode_2_.readBytes(var7));
+                    p_decode_2_.resetReaderIndex();
                 }
                 finally
                 {
