--- a/net/minecraft/profiler/Profiler.java
+++ b/net/minecraft/profiler/Profiler.java
@@ -36,7 +36,6 @@
     public long timeTickNano;
     private long startUpdateChunksNano;
     public long timeUpdateChunksNano;
-    private static final String __OBFID = "CL_00001497";
 
     public Profiler()
     {
@@ -247,7 +246,6 @@
         public double field_76330_b;
         public String field_76331_c;
         private static final String __OBFID = "CL_00001498";
-        private static final String __OBFID = "CL_00001498";
 
         public Result(String par1Str, double par2, double par4)
         {
