--- a/net/minecraft/client/gui/GuiButton.java
+++ b/net/minecraft/client/gui/GuiButton.java
@@ -4,6 +4,7 @@
 import net.minecraft.client.audio.PositionedSoundRecord;
 import net.minecraft.client.audio.SoundHandler;
 import net.minecraft.client.renderer.OpenGlHelper;
+import net.minecraft.src.Reflector;
 import net.minecraft.util.ResourceLocation;
 import org.lwjgl.opengl.GL11;
 
@@ -22,6 +23,7 @@
     public boolean field_146125_m;
     protected boolean field_146123_n;
     private static final String __OBFID = "CL_00000668";
+    public int packedFGColour;
 
     public GuiButton(int p_i1020_1_, int p_i1020_2_, int p_i1020_3_, String p_i1020_4_)
     {
@@ -68,6 +70,11 @@
             FontRenderer var4 = p_146112_1_.fontRenderer;
             p_146112_1_.getTextureManager().bindTexture(field_146122_a);
             GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
+
+            /** Minecrift **/
+            if (p_146112_2_ > Minecraft.getMinecraft().displayWidth)
+                p_146112_2_ -= Minecraft.getMinecraft().displayWidth;
+
             this.field_146123_n = p_146112_2_ >= this.field_146128_h && p_146112_3_ >= this.field_146129_i && p_146112_2_ < this.field_146128_h + this.field_146120_f && p_146112_3_ < this.field_146129_i + this.field_146121_g;
             int var5 = this.getHoverState(this.field_146123_n);
             GL11.glEnable(GL11.GL_BLEND);
@@ -78,7 +85,11 @@
             this.mouseDragged(p_146112_1_, p_146112_2_, p_146112_3_);
             int var6 = 14737632;
 
-            if (!this.enabled)
+            if (Reflector.forgeExists() && packedFGColour != 0)
+            {
+                var6 = packedFGColour;
+            }
+            else if (!this.enabled)
             {
                 var6 = 10526880;
             }
