--- a/net/minecraft/client/gui/GuiMainMenu.java
+++ b/net/minecraft/client/gui/GuiMainMenu.java
@@ -4,16 +4,18 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.net.URI;
-import java.util.ArrayList;
-import java.util.Calendar;
-import java.util.Date;
-import java.util.Random;
+import java.util.*;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import de.fruitfly.ovr.OculusRift;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.OpenGlHelper;
 import net.minecraft.client.renderer.Tessellator;
 import net.minecraft.client.renderer.texture.DynamicTexture;
 import net.minecraft.client.resources.I18n;
 import net.minecraft.realms.RealmsBridge;
+import net.minecraft.src.Reflector;
 import net.minecraft.util.EnumChatFormatting;
 import net.minecraft.util.MathHelper;
 import net.minecraft.util.ResourceLocation;
@@ -214,9 +216,24 @@
      */
     private void addSingleplayerMultiplayerButtons(int p_73969_1_, int p_73969_2_)
     {
-        this.buttonList.add(new GuiButton(1, this.width / 2 - 100, p_73969_1_, I18n.format("menu.singleplayer", new Object[0])));
-        this.buttonList.add(new GuiButton(2, this.width / 2 - 100, p_73969_1_ + p_73969_2_ * 1, I18n.format("menu.multiplayer", new Object[0])));
-        this.buttonList.add(new GuiButton(14, this.width / 2 - 100, p_73969_1_ + p_73969_2_ * 2, I18n.format("menu.online", new Object[0])));
+        if (!Reflector.forgeExists())
+		{
+            this.buttonList.add(new GuiButton(1, this.width / 2 - 100, p_73969_1_, I18n.format("menu.singleplayer", new Object[0])));
+            this.buttonList.add(new GuiButton(2, this.width / 2 - 100, p_73969_1_ + p_73969_2_ * 1, I18n.format("menu.multiplayer", new Object[0])));
+            this.buttonList.add(new GuiButton(14, this.width / 2 - 100, p_73969_1_ + p_73969_2_ * 2, I18n.format("menu.online", new Object[0])));
+		}
+		else
+		{
+            this.buttonList.add(new GuiButton(1, this.width / 2 - 100, p_73969_1_, I18n.format("menu.singleplayer", new Object[0])));
+            this.buttonList.add(new GuiButton(2, this.width / 2 - 100, p_73969_1_ + p_73969_2_ * 1, I18n.format("menu.multiplayer", new Object[0])));
+            GuiButton realmsButton = new GuiButton(14, this.width / 2 - 100, p_73969_1_ + p_73969_2_ * 2, I18n.format("menu.online", new Object[0]));
+            GuiButton fmlModButton = new GuiButton(6, this.width / 2 - 100, p_73969_1_ + p_73969_2_ * 2, "Mods");
+            fmlModButton.field_146128_h = this.width / 2 + 2;
+            realmsButton.field_146120_f = 98;
+            fmlModButton.field_146120_f = 98;
+            this.buttonList.add(realmsButton);
+            this.buttonList.add(fmlModButton);
+		}	
     }
 
     /**
@@ -267,6 +284,11 @@
             this.mc.shutdown();
         }
 
+        if (Reflector.FMLGuiModList_Constructor.exists() && p_146284_1_.id == 6)
+        {
+            this.mc.displayGuiScreen((GuiScreen)Reflector.newInstance(Reflector.FMLGuiModList_Constructor, new Object[] {this}));
+        }
+
         if (p_146284_1_.id == 11)
         {
             this.mc.launchIntegratedServer("Demo_World", "Demo_World", DemoWorldServer.demoWorldSettings);
@@ -448,7 +470,7 @@
      */
     private void renderSkybox(int p_73971_1_, int p_73971_2_, float p_73971_3_)
     {
-        this.mc.getFramebuffer().unbindFramebuffer();
+        //this.mc.getFramebuffer().unbindFramebuffer();  /** Minecrift **/
         GL11.glViewport(0, 0, 256, 256);
         this.drawPanorama(p_73971_1_, p_73971_2_, p_73971_3_);
         this.rotateAndBlurSkybox(p_73971_3_);
@@ -458,7 +480,7 @@
         this.rotateAndBlurSkybox(p_73971_3_);
         this.rotateAndBlurSkybox(p_73971_3_);
         this.rotateAndBlurSkybox(p_73971_3_);
-        this.mc.getFramebuffer().bindFramebuffer(true);
+        //this.mc.getFramebuffer().bindFramebuffer(true);  /** Minecrift **/
         GL11.glViewport(0, 0, this.mc.displayWidth, this.mc.displayHeight);
         Tessellator var4 = Tessellator.instance;
         var4.startDrawingQuads();
@@ -515,13 +537,29 @@
         GL11.glScalef(var8, var8, var8);
         this.drawCenteredString(this.fontRendererObj, this.splashText, 0, -8, -256);
         GL11.glPopMatrix();
-        String var9 = "Minecraft 1.7.10";
+        String var9 = "Minecrift 1.7.10, " + OculusRift.getVersionString();
 
         if (this.mc.isDemo())
         {
             var9 = var9 + " Demo";
         }
 
+        Object fmlCommonHandler = null;
+        if( Reflector.FMLCommonHandler_instance.exists()) {
+            fmlCommonHandler = Reflector.call(Reflector.FMLCommonHandler_instance, new Object[0]);
+        }
+        if( fmlCommonHandler != null ) {
+            List<String> brandings = Lists.reverse((List<String>)Reflector.call(fmlCommonHandler, Reflector.FMLCommonHandler_getBrandings, new Object[]{true}));
+            for (int i = 0; i < brandings.size(); i++)
+            {
+                String brd = brandings.get(i);
+                if (!Strings.isNullOrEmpty(brd))
+                {
+                    this.drawString(this.fontRendererObj, brd, 2, this.height - ( 10 + i * (this.fontRendererObj.FONT_HEIGHT + 1)), 16777215);
+                }
+            }
+            Reflector.callVoid(Reflector.ForgeHooksClient_renderMainMenu, new Object[] {this, fontRendererObj, width, height});
+		}
         this.drawString(this.fontRendererObj, var9, 2, this.height - 10, -1);
         String var10 = "Copyright Mojang AB. Do not distribute!";
         this.drawString(this.fontRendererObj, var10, this.width - this.fontRendererObj.getStringWidth(var10) - 2, this.height - 10, -1);
