--- a/net/minecraft/client/gui/GuiNewChat.java
+++ b/net/minecraft/client/gui/GuiNewChat.java
@@ -4,6 +4,8 @@
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
+
+import com.mtbs3d.minecrift.settings.VRSettings;
 import net.minecraft.client.Minecraft;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.ChatComponentText;
@@ -47,10 +49,17 @@
                     var3 = true;
                 }
 
-                float var7 = this.func_146244_h();
-                int var8 = MathHelper.ceiling_float_int((float)this.func_146228_f() / var7);
+                /* Minecrift */
+                float var7 = this.func_146244_h();//chat scale
+                int chatWidthPixels = this.func_146228_f() ;
+                int chatFullHeightPixels = func_146243_b(this.field_146247_f.gameSettings.chatHeightFocused );
+                int var8 = MathHelper.ceiling_float_int((float)chatWidthPixels/ var7);
+                ScaledResolution res = new ScaledResolution(this.field_146247_f, this.field_146247_f.displayWidth, this.field_146247_f.displayHeight);
+                int offsetX =  (int)(VRSettings.inst.chatOffsetX*(res.getScaledWidth()  - chatWidthPixels -2 ));
+                int offsetY = -(int)(VRSettings.inst.chatOffsetY*(res.getScaledHeight() - chatFullHeightPixels -20 ));
                 GL11.glPushMatrix();
-                GL11.glTranslatef(2.0F, 20.0F, 0.0F);
+                GL11.glTranslatef(2.0F + offsetX, 20 + offsetY, 0.0F);
+                /* end Minecrift */
                 GL11.glScalef(var7, var7, 1.0F);
                 int var9;
                 int var11;
