--- a/net/minecraft/client/gui/GuiSlot.java
+++ b/net/minecraft/client/gui/GuiSlot.java
@@ -9,26 +9,26 @@
 public abstract class GuiSlot
 {
     private final Minecraft field_148161_k;
-    protected int field_148155_a;
-    private int field_148158_l;
-    protected int field_148153_b;
-    protected int field_148154_c;
-    protected int field_148151_d;
-    protected int field_148152_e;
-    protected final int field_148149_f;
-    private int field_148159_m;
-    private int field_148156_n;
-    protected int field_148150_g;
-    protected int field_148162_h;
+    protected int field_148155_a;  // width
+    private int field_148158_l;    // height
+    protected int field_148153_b;  // top
+    protected int field_148154_c;  // bottom
+    protected int field_148151_d;  // right
+    protected int field_148152_e;  // left
+    protected final int field_148149_f;  // slotHeight
+    private int field_148159_m;    // scrollUpButtonID
+    private int field_148156_n;    // scrollDownButtonID
+    protected int field_148150_g;  // mouseX
+    protected int field_148162_h;  // mouseY
     protected boolean field_148163_i = true;
-    private float field_148157_o = -2.0F;
-    private float field_148170_p;
-    private float field_148169_q;
-    private int field_148168_r = -1;
-    private long field_148167_s;
-    private boolean field_148166_t = true;
-    private boolean field_148165_u;
-    protected int field_148160_j;
+    private float field_148157_o = -2.0F;  // initialClickY
+    private float field_148170_p;  // scrollMultiplier
+    private float field_148169_q;  // amountScrolled
+    private int field_148168_r = -1;  // selectedElement
+    private long field_148167_s;  // lastClicked
+    private boolean field_148166_t = true;  // showSelectionBox
+    private boolean field_148165_u;   // hasListHeader
+    protected int field_148160_j;     // headerPadding
     private boolean field_148164_v = true;
     private static final String __OBFID = "CL_00000679";
 
@@ -106,7 +106,7 @@
         this.field_148156_n = p_148134_2_;
     }
 
-    private void func_148121_k()
+    private void func_148121_k()     // bindAmountScrolled
     {
         int var1 = this.func_148135_f();
 
@@ -291,16 +291,7 @@
         GL11.glDisable(GL11.GL_LIGHTING);
         GL11.glDisable(GL11.GL_FOG);
         Tessellator var16 = Tessellator.instance;
-        this.field_148161_k.getTextureManager().bindTexture(Gui.optionsBackground);
-        GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-        float var17 = 32.0F;
-        var16.startDrawingQuads();
-        var16.setColorOpaque_I(2105376);
-        var16.addVertexWithUV((double)this.field_148152_e, (double)this.field_148154_c, 0.0D, (double)((float)this.field_148152_e / var17), (double)((float)(this.field_148154_c + (int)this.field_148169_q) / var17));
-        var16.addVertexWithUV((double)this.field_148151_d, (double)this.field_148154_c, 0.0D, (double)((float)this.field_148151_d / var17), (double)((float)(this.field_148154_c + (int)this.field_148169_q) / var17));
-        var16.addVertexWithUV((double)this.field_148151_d, (double)this.field_148153_b, 0.0D, (double)((float)this.field_148151_d / var17), (double)((float)(this.field_148153_b + (int)this.field_148169_q) / var17));
-        var16.addVertexWithUV((double)this.field_148152_e, (double)this.field_148153_b, 0.0D, (double)((float)this.field_148152_e / var17), (double)((float)(this.field_148153_b + (int)this.field_148169_q) / var17));
-        var16.draw();
+		drawContainerBackground(var16);
         var9 = this.field_148152_e + this.field_148155_a / 2 - this.func_148139_c() / 2 + 2;
         var10 = this.field_148153_b + 4 - (int)this.field_148169_q;
 
@@ -472,4 +463,31 @@
     {
         return this.field_148149_f;
     }
+
+    protected void drawContainerBackground(Tessellator var16)
+    {
+        this.field_148161_k.getTextureManager().bindTexture(Gui.optionsBackground);
+        GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
+        float var17 = 32.0F;
+        var16.startDrawingQuads();
+        var16.setColorOpaque_I(2105376);
+        var16.addVertexWithUV((double)this.field_148152_e, (double)this.field_148154_c, 0.0D, (double)((float)this.field_148152_e / var17), (double)((float)(this.field_148154_c + (int)this.field_148169_q) / var17));
+        var16.addVertexWithUV((double)this.field_148151_d, (double)this.field_148154_c, 0.0D, (double)((float)this.field_148151_d / var17), (double)((float)(this.field_148154_c + (int)this.field_148169_q) / var17));
+        var16.addVertexWithUV((double)this.field_148151_d, (double)this.field_148153_b, 0.0D, (double)((float)this.field_148151_d / var17), (double)((float)(this.field_148153_b + (int)this.field_148169_q) / var17));
+        var16.addVertexWithUV((double)this.field_148152_e, (double)this.field_148153_b, 0.0D, (double)((float)this.field_148152_e / var17), (double)((float)(this.field_148153_b + (int)this.field_148169_q) / var17));
+        var16.draw();
+    }
+	
+	/** Minecrift **/
+    public int publicGetSize() { return getSize(); }
+    public int select( int item, boolean dblClick) {
+        field_148169_q = item * field_148149_f;
+        func_148121_k();
+    	elementClicked(item, dblClick, field_148150_g, field_148162_h);
+    	
+    	if( !dblClick ) {
+    		return (int)(this.field_148153_b + 4 - this.field_148169_q + (item+0.5) * this.field_148149_f + this.field_148160_j);
+    	}
+    	return 0;
+	};	
 }
