--- a/net/minecraft/client/gui/GuiVideoSettings.java
+++ b/net/minecraft/client/gui/GuiVideoSettings.java
@@ -19,7 +19,6 @@
     private int lastMouseX = 0;
     private int lastMouseY = 0;
     private long mouseStillTime = 0L;
-    private static final String __OBFID = "CL_00000718";
 
     public GuiVideoSettings(GuiScreen par1GuiScreen, GameSettings par2GameSettings)
     {
@@ -67,7 +66,11 @@
 
             if (y.getEnumFloat())
             {
-                this.buttonList.add(new GuiOptionSlider(y.returnEnumOrdinal(), x, y1, y));
+                /* Minecrift - disable frame rate settings (always VSync) */
+                GuiOptionSlider slider = new GuiOptionSlider(y.returnEnumOrdinal(), x, y1, y);
+                if (y == GameSettings.Options.FRAMERATE_LIMIT)
+                    slider.enabled = false;
+                this.buttonList.add(slider);
             }
             else
             {
